{
 "cells": [
  {
   "cell_type": "raw",
   "id": "d871ba1c-8d1d-4e76-9f6c-82e2a85b8ea5",
   "metadata": {},
   "source": [
    "---\n",
    "title: Prepare ENCODE ChIP-seq 11 (TF)\n",
    "subtitle: Generate download script to download the data\n",
    "format:\n",
    "    html:\n",
    "        code-fold:  true\n",
    "        code-tools: true\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28c83ceb-af27-4f51-9862-202422b09294",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "34853b16-1190-4461-acae-162a8949328b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are working on        Singularity: singularity_proj_encode_fcc \n",
      "BASE DIRECTORY (FD_BASE): /data/reddylab/Kuei \n",
      "REPO DIRECTORY (FD_REPO): /data/reddylab/Kuei/repo \n",
      "WORK DIRECTORY (FD_WORK): /data/reddylab/Kuei/work \n",
      "DATA DIRECTORY (FD_DATA): /data/reddylab/Kuei/data \n",
      "\n",
      "You are working with      ENCODE FCC \n",
      "PATH OF PROJECT (FD_PRJ): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC \n",
      "PROJECT RESULTS (FD_RES): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results \n",
      "PROJECT SCRIPTS (FD_EXE): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/scripts \n",
      "PROJECT DATA    (FD_DAT): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/data \n",
      "PROJECT NOTE    (FD_NBK): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/notebooks \n",
      "PROJECT DOCS    (FD_DOC): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/docs \n",
      "PROJECT LOG     (FD_LOG): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/log \n",
      "PROJECT REF     (FD_REF): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/references \n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(source(\"../run_config_project_sing.R\")))\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "925bb3c4-9592-4c3d-82f7-eafb58d81d0c",
   "metadata": {},
   "source": [
    "## Prepare"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4b61fb81-e489-4a6a-8ed6-f2842d8e590c",
   "metadata": {},
   "outputs": [],
   "source": [
    "TXT_FOLDER_INP = \"encode_chipseq_tf\"\n",
    "TXT_FOLDER_OUT = \"encode_chipseq_tf_250120\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4a9e5915-a7e3-4253-89c1-ab861f3120d9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "files.chipseq_tf.default_analysis.250109.txt \n",
      "files.chipseq_tf.default_analysis.250120.txt \n",
      "files.chipseq_tf.default_files.241223.txt \n",
      "files.chipseq_tf.default_files.250108.txt \n",
      "files.chipseq_tf.default_files.250120.txt \n",
      "metadata.chipseq_tf.default_analysis.250109.tsv \n",
      "metadata.chipseq_tf.default_analysis.250120.tsv \n",
      "metadata.chipseq_tf.default_files.241223.tsv \n",
      "metadata.chipseq_tf.default_files.250108.tsv \n",
      "metadata.chipseq_tf.default_files.250120.tsv \n",
      "Readme.md \n"
     ]
    }
   ],
   "source": [
    "txt_folder = TXT_FOLDER_INP\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "\n",
    "vec = dir(txt_fdiry)\n",
    "for (txt in vec){cat(txt, \"\\n\")}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f6dbdc9f-10b9-4bdb-9fed-9b2a51b298bd",
   "metadata": {},
   "source": [
    "## Import metadata from reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bec00756-d133-40d9-a0d9-ca1bff6689ee",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 2406   59\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF447FCS </td>\n",
       "   <td style=\"text-align:left;\"> bigBed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bigBed </td>\n",
       "   <td style=\"text-align:left;\"> narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> CRISPR </td>\n",
       "   <td style=\"text-align:left;\"> insertion </td>\n",
       "   <td style=\"text-align:left;\"> {'schema_version': '14', 'aliases': ['michael-snyder:TS-985'], 'organism': {'schema_version': '6', '@type': ['Organism', 'Item'], 'name': 'human', 'taxon_id': '9606', 'scientific_name': 'Homo sapiens', '@id': '/organisms/human/', 'uuid': '7745b647-ff15-4ff3-9ced-b897d4e2983c', 'status': 'released'}, 'genes': ['/genes/84671/'], '@type': ['Target', 'Item'], 'name': 'ZNF347-human', 'label': 'ZNF347', '@id': '/targets/ZNF347-human/', 'title': 'ZNF347 (Homo sapiens)', 'investigated_as': ['transcription factor'], 'uuid': '7edf642a-7dbc-4d18-b8a7-a36fee801dda', 'status': 'released'} </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> formaldehyde </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2020-08-03 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 450-650 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1, 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF843NBV/ </td>\n",
       "   <td style=\"text-align:right;\"> 83030 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> 76853aa420fdb949461f8032db9e949c </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF447FCS/@@download/ENCFF447FCS.bigBed </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/07/03/352f11df-3c62-4954-8e58-8c0e14adda5e/ENCFF447FCS.bigBed </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/07/03/352f11df-3c62-4954-8e58-8c0e14adda5e/ENCFF447FCS.bigBed?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> moderate library complexity, low read depth, mild to moderate bottlenecking </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_folder = TXT_FOLDER_INP\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "txt_fname  = \"metadata.chipseq_tf.default_files.250120.tsv\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### show and assign\n",
    "dat_metadata_chipseq_v1 = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c29887af-e32f-409d-8aeb-ccc7f64958c9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 17353    59\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:right;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF521GKP </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> CRISPR </td>\n",
       "   <td style=\"text-align:left;\"> insertion </td>\n",
       "   <td style=\"text-align:left;\"> {'schema_version': '14', 'aliases': ['michael-snyder:TS-985'], 'organism': {'schema_version': '6', '@type': ['Organism', 'Item'], 'name': 'human', 'taxon_id': '9606', 'scientific_name': 'Homo sapiens', '@id': '/organisms/human/', 'uuid': '7745b647-ff15-4ff3-9ced-b897d4e2983c', 'status': 'released'}, 'genes': ['/genes/84671/'], '@type': ['Target', 'Item'], 'name': 'ZNF347-human', 'label': 'ZNF347', '@id': '/targets/ZNF347-human/', 'title': 'ZNF347 (Homo sapiens)', 'investigated_as': ['transcription factor'], 'uuid': '7edf642a-7dbc-4d18-b8a7-a36fee801dda', 'status': 'released'} </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> formaldehyde </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2020-08-03 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 450-650 </td>\n",
       "   <td style=\"text-align:left;\"> 2 </td>\n",
       "   <td style=\"text-align:left;\"> 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:right;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF022VGJ/, /files/ENCFF174OKT/ </td>\n",
       "   <td style=\"text-align:right;\"> 921948703 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> afaa174c390f22a9320ec3beeab8e490 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF521GKP/@@download/ENCFF521GKP.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/07/03/a543c551-9440-4331-99de-94621c7a6ee2/ENCFF521GKP.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/07/03/a543c551-9440-4331-99de-94621c7a6ee2/ENCFF521GKP.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> low read depth, mild to moderate bottlenecking, moderate library complexity </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_folder = TXT_FOLDER_INP\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "txt_fname  = \"metadata.chipseq_tf.default_analysis.250120.tsv\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### show and assign\n",
    "dat_metadata_chipseq_v2 = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8aac697b-f35e-4c49-b1a4-0ecf031df9fa",
   "metadata": {},
   "source": [
    "## Explore"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8df8ceda-e355-4db9-bfc3-1f3ffd570c19",
   "metadata": {},
   "source": [
    "**File type**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "52306484-42c4-4e03-b5e4-edb71620a12a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             \n",
       "              bed bigBed bigWig\n",
       "  TF ChIP-seq 802    802    802"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "table(dat$Assay, dat$`File type`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "336311b3-d8e9-4ffd-a4ee-7ddec6b98f58",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             \n",
       "               bam  bed bigBed bigWig\n",
       "  TF ChIP-seq 3208 6125   3260   4760"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "table(dat$Assay, dat$`File type`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8810a2b6-5119-449e-b75c-d54bcb73eddd",
   "metadata": {},
   "source": [
    "**Output type**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6e03305b-628d-41f9-b219-c61a4902def5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        \n",
       "         conservative IDR thresholded peaks IDR thresholded peaks\n",
       "  bed                                    66                   654\n",
       "  bigBed                                 66                   654\n",
       "  bigWig                                  0                     0\n",
       "        \n",
       "         optimal IDR thresholded peaks pseudoreplicated IDR thresholded peaks\n",
       "  bed                               78                                      4\n",
       "  bigBed                            78                                      4\n",
       "  bigWig                             0                                      0\n",
       "        \n",
       "         signal p-value\n",
       "  bed                 0\n",
       "  bigBed              0\n",
       "  bigWig            802"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "table(dat$`File type`, dat$`Output type`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "56f1ded8-2b64-4a60-98a1-72e7d4204440",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        \n",
       "         alignments conservative IDR thresholded peaks fold change over control\n",
       "  bam          1604                                  0                        0\n",
       "  bed             0                                782                        0\n",
       "  bigBed          0                                782                        0\n",
       "  bigWig          0                                  0                     2380\n",
       "        \n",
       "         IDR ranked peaks IDR thresholded peaks optimal IDR thresholded peaks\n",
       "  bam                   0                     0                             0\n",
       "  bed                2865                  2158                            78\n",
       "  bigBed                0                  2158                            78\n",
       "  bigWig                0                     0                             0\n",
       "        \n",
       "         peaks and background as input for IDR\n",
       "  bam                                        0\n",
       "  bed                                      238\n",
       "  bigBed                                   238\n",
       "  bigWig                                     0\n",
       "        \n",
       "         pseudoreplicated IDR thresholded peaks signal p-value\n",
       "  bam                                         0              0\n",
       "  bed                                         4              0\n",
       "  bigBed                                      4              0\n",
       "  bigWig                                      0           2380\n",
       "        \n",
       "         unfiltered alignments\n",
       "  bam                     1604\n",
       "  bed                        0\n",
       "  bigBed                     0\n",
       "  bigWig                     0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "table(dat$`File type`, dat$`Output type`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d438e08-792d-4907-afc4-98cfcae5a9f6",
   "metadata": {},
   "source": [
    "**Genome assembly**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "7b7f0a99-1033-4f6e-8728-84f56e84132a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "GRCh38   hg19 \n",
       "  2283    123 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "table(dat$`File assembly`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6b900c62-6c66-4a22-88d9-78d3594f8905",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "GRCh38   hg19 \n",
       " 16557    796 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "table(dat$`File assembly`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3fbbe4b-4c20-429d-b13f-68b2263a7d82",
   "metadata": {},
   "source": [
    "**Biosample**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "577036df-f576-459f-b03f-e7ded90b92d6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "K562 \n",
       "2406 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "table(dat$`Biosample term name`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "fe5f9632-6859-4197-8b77-d3fb002e6f78",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       " K562 \n",
       "17353 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "table(dat$`Biosample term name`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5243c14f-76cf-4fc1-ae66-341badae3f07",
   "metadata": {},
   "source": [
    "**Target**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "5ec5deb6-4caf-4613-b643-12bb6b1e2552",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Var1</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ADNP-human    </td><td>3</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AFF1-human    </td><td>6</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>AFF4-human    </td><td>3</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AGO1-human    </td><td>9</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ARHGAP35-human</td><td>3</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ARID1B-human  </td><td>3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Var1 & Freq\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & ADNP-human     & 3\\\\\n",
       "\t2 & AFF1-human     & 6\\\\\n",
       "\t3 & AFF4-human     & 3\\\\\n",
       "\t4 & AGO1-human     & 9\\\\\n",
       "\t5 & ARHGAP35-human & 3\\\\\n",
       "\t6 & ARID1B-human   & 3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| <!--/--> | Var1 &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | ADNP-human     | 3 |\n",
       "| 2 | AFF1-human     | 6 |\n",
       "| 3 | AFF4-human     | 3 |\n",
       "| 4 | AGO1-human     | 9 |\n",
       "| 5 | ARHGAP35-human | 3 |\n",
       "| 6 | ARID1B-human   | 3 |\n",
       "\n"
      ],
      "text/plain": [
       "  Var1           Freq\n",
       "1 ADNP-human     3   \n",
       "2 AFF1-human     6   \n",
       "3 AFF4-human     3   \n",
       "4 AGO1-human     9   \n",
       "5 ARHGAP35-human 3   \n",
       "6 ARID1B-human   3   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "dat = as.data.frame(table(dat$`Experiment target`))\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "5818a5a9-743a-4bc5-85fb-07790f073821",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Var1</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ADNP-human    </td><td>22</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AFF1-human    </td><td>44</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>AFF4-human    </td><td>22</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AGO1-human    </td><td>62</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ARHGAP35-human</td><td>22</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ARID1B-human  </td><td>22</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Var1 & Freq\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & ADNP-human     & 22\\\\\n",
       "\t2 & AFF1-human     & 44\\\\\n",
       "\t3 & AFF4-human     & 22\\\\\n",
       "\t4 & AGO1-human     & 62\\\\\n",
       "\t5 & ARHGAP35-human & 22\\\\\n",
       "\t6 & ARID1B-human   & 22\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| <!--/--> | Var1 &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | ADNP-human     | 22 |\n",
       "| 2 | AFF1-human     | 44 |\n",
       "| 3 | AFF4-human     | 22 |\n",
       "| 4 | AGO1-human     | 62 |\n",
       "| 5 | ARHGAP35-human | 22 |\n",
       "| 6 | ARID1B-human   | 22 |\n",
       "\n"
      ],
      "text/plain": [
       "  Var1           Freq\n",
       "1 ADNP-human     22  \n",
       "2 AFF1-human     44  \n",
       "3 AFF4-human     22  \n",
       "4 AGO1-human     62  \n",
       "5 ARHGAP35-human 22  \n",
       "6 ARID1B-human   22  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "dat = as.data.frame(table(dat$`Experiment target`))\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4ac99b4-3e67-4a42-ad39-f73f7269aef5",
   "metadata": {},
   "source": [
    "## Arrange metadata tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "b1f205a2-9901-442f-b7b3-e24b7377989b",
   "metadata": {},
   "outputs": [],
   "source": [
    "fun = function(dat_input){\n",
    "    ### init\n",
    "    dat = dat_input\n",
    "    \n",
    "    ### get the analysis version\n",
    "    dat = dat %>% \n",
    "        dplyr::rowwise() %>%\n",
    "        dplyr::mutate(Analysis_Version = str_extract(`File analysis title`, \"ENCODE\\\\d\")) %>%\n",
    "        dplyr::ungroup()\n",
    "    \n",
    "    ### rename some columns\n",
    "    dat = dat %>% \n",
    "        dplyr::mutate(\n",
    "            Index_Experiment = `Experiment accession`,\n",
    "            Index_File       = `File accession`,\n",
    "            File_Format      = `File format`,\n",
    "            File_Type        = `File type`,\n",
    "            Output_Type      = `Output type`,\n",
    "            Genome           = `File assembly`,\n",
    "            Target           = str_remove(`Experiment target`, \"-human\"),\n",
    "            Bio_Replicates   = `Biological replicate(s)`,\n",
    "            Analysis         = `File analysis title`,\n",
    "            File_Name\t     = basename(`File download URL`),\n",
    "            File_URL         = `File download URL`\n",
    "        )\n",
    "    \n",
    "    ### select the needed columns\n",
    "    dat = dat %>%\n",
    "        dplyr::select(\n",
    "            Assay,\n",
    "            Index_Experiment,\n",
    "            Index_File,\n",
    "            File_Format,\n",
    "            File_Type,\n",
    "            Output_Type,\n",
    "            Genome,\n",
    "            Target,\n",
    "            Bio_Replicates,\n",
    "            Analysis,\n",
    "            Analysis_Version,\n",
    "            md5sum,\n",
    "            File_Name,\n",
    "            File_URL\n",
    "        )\n",
    "    \n",
    "    ### filter only the hg38 and ENCODE4 analysis pipeline\n",
    "    dat = dat %>% dplyr::filter(\n",
    "        Genome == \"GRCh38\",\n",
    "        Analysis_Version == \"ENCODE4\"\n",
    "    )\n",
    "    \n",
    "    ### return the simplified table\n",
    "    return(dat)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "aa30385d-3277-434c-8b3a-50ad22a1e322",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 2160   14\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> Analysis_Version </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF447FCS </td>\n",
       "   <td style=\"text-align:left;\"> bigBed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bigBed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> 76853aa420fdb949461f8032db9e949c </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF447FCS.bigBed </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF447FCS/@@download/ENCFF447FCS.bigBed </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF563XMY </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> d01e3aa935f311f32991f796582632a6 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF563XMY.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF563XMY/@@download/ENCFF563XMY.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF843NBV </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> cab83f2be2220d6d328bd6566fc54d85 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF843NBV.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF843NBV/@@download/ENCFF843NBV.bed.gz </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "dat = fun(dat)\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_simplify_v1 = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "0ffa51f4-f622-40af-a3a3-af74f8c67352",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 15749    14\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> Analysis_Version </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF521GKP </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> afaa174c390f22a9320ec3beeab8e490 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF521GKP.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF521GKP/@@download/ENCFF521GKP.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF194RBI </td>\n",
       "   <td style=\"text-align:left;\"> bed idr_ranked_peak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> IDR ranked peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> 319a983b0063a6c88ba1b56e20d5238a </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF194RBI.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF194RBI/@@download/ENCFF194RBI.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF914ZHO </td>\n",
       "   <td style=\"text-align:left;\"> bigBed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bigBed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> 8355b8626f20355b3d7efe40259c0b0b </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF914ZHO.bigBed </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF914ZHO/@@download/ENCFF914ZHO.bigBed </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "dat = fun(dat)\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_simplify_v2 = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "00f09011-9dcb-47dd-8989-0fcf67e54ba7",
   "metadata": {},
   "source": [
    "## Prepare download files"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd6696c6-5d5e-4cd9-aa39-8b9121c5f30d",
   "metadata": {},
   "source": [
    "**NarrowPeak files**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e095f62e-5f4e-4950-81b0-5cda43e1a004",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 720  14\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> Analysis_Version </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF843NBV </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> cab83f2be2220d6d328bd6566fc54d85 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF843NBV.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF843NBV/@@download/ENCFF843NBV.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR536CBU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF588CXX </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB9 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> c4faaf3e805d2fcfd7c9037fdb768174 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF588CXX.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF588CXX/@@download/ENCFF588CXX.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR436CZV </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF142MJD </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> AFF4 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> c8f76e772192439a7847dbbea9f73f40 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF142MJD.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF142MJD/@@download/ENCFF142MJD.bed.gz </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "###\n",
    "dat = dat_metadata_chipseq_simplify_v1\n",
    "dat = dat %>% dplyr::filter(File_Type == \"bed\", File_Format == \"bed narrowPeak\")\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_bed_narrowpeak = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9be8a33c-d255-416a-af98-979a1a8b5a26",
   "metadata": {},
   "source": [
    "**Signal p-value**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "9a7f50df-de70-4560-a873-d988dc082e1a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 720  14\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> Analysis_Version </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF563XMY </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> d01e3aa935f311f32991f796582632a6 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF563XMY.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF563XMY/@@download/ENCFF563XMY.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR536CBU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF547AGX </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB9 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> c13833244163ed3ab3f74b13924002eb </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF547AGX.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF547AGX/@@download/ENCFF547AGX.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR436CZV </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF798FSS </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> AFF4 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> def72a79a02e4f8ebf0dcd97572eaeb0 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF798FSS.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF798FSS/@@download/ENCFF798FSS.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "###\n",
    "dat = dat_metadata_chipseq_simplify_v1\n",
    "dat = dat %>% dplyr::filter(File_Type == \"bigWig\", Output_Type == \"signal p-value\")\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_bigwig_pvalue = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a0d1358-2d56-4a00-919a-141eb89d6baf",
   "metadata": {},
   "source": [
    "**Fold change**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "7478d560-c07d-4926-ab1e-e934ea46b6ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 2142    4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:right;\"> Count_Replicates </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF521GKP </td>\n",
       "   <td style=\"text-align:left;\"> 2 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF876INH </td>\n",
       "   <td style=\"text-align:left;\"> 1 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF465ZAR </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:right;\"> 2 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR536CBU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF355MVO </td>\n",
       "   <td style=\"text-align:left;\"> 2 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR536CBU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF178FEN </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:right;\"> 2 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR536CBU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF760YQX </td>\n",
       "   <td style=\"text-align:left;\"> 1 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### helper function\n",
    "fun = function(txt){\n",
    "    lst = str_split(txt, \",\")\n",
    "    lst = lapply(lst, length)\n",
    "    vec = unlist(lst)\n",
    "    return(vec)\n",
    "}\n",
    "\n",
    "###\n",
    "dat = dat_metadata_chipseq_simplify_v2\n",
    "dat = dat %>% dplyr::filter(File_Type == \"bigWig\", Output_Type == \"fold change over control\")\n",
    "dat = dat %>% dplyr::mutate(Count_Replicates = fun(Bio_Replicates))\n",
    "dat = dat %>% dplyr::select(\n",
    "    Index_Experiment,\n",
    "    Index_File,\n",
    "    Bio_Replicates,\n",
    "    Count_Replicates)\n",
    "\n",
    "### assign and show\n",
    "dat_count_replicates = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "1e487ad8-9ffd-42d1-8174-9635441333ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 720  14\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> Analysis_Version </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF465ZAR </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> 9129c700630cd16e31a925d351562f14 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF465ZAR.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF465ZAR/@@download/ENCFF465ZAR.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR536CBU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF178FEN </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB9 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> c3b03e05c986366373390283fe28381f </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF178FEN.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF178FEN/@@download/ENCFF178FEN.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR436CZV </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF040PTP </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> AFF4 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> ec5d483f39e417045f201e447c62bdca </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF040PTP.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF040PTP/@@download/ENCFF040PTP.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "###\n",
    "dat = dat_count_replicates\n",
    "dat = dat %>% \n",
    "    dplyr::group_by(Index_Experiment) %>% \n",
    "    dplyr::slice_max(Count_Replicates)\n",
    "vec = dat$Index_File\n",
    "\n",
    "###\n",
    "dat = dat_metadata_chipseq_simplify_v2\n",
    "dat = dat %>% dplyr::filter(Index_File %in% vec)\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_bigwig_fold_change = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "b33b88d3-dae1-4586-a476-f78afc157e87",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "fold change over control \n",
       "                     720 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_bigwig_fold_change\n",
    "table(dat$Output_Type)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7822dba6-9173-48b4-86cb-52bbb467feaf",
   "metadata": {},
   "source": [
    "## Save results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08d5c8a6-6b1c-4321-b7a8-22f037ba5c4d",
   "metadata": {},
   "source": [
    "**Check equal size**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "5994636b-1000-4005-9fa6-dafc7a3ab767",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "region_narrowPeak \n",
      "[1] 720  14\n",
      "signal_fold_change \n",
      "[1] 720  14\n",
      "signal_pvalue \n",
      "[1] 720  14\n"
     ]
    }
   ],
   "source": [
    "lst = list(\n",
    "    \"region_narrowPeak\"  = dat_metadata_chipseq_bed_narrowpeak,\n",
    "    \"signal_fold_change\" = dat_metadata_chipseq_bigwig_fold_change,\n",
    "    \"signal_pvalue\"      = dat_metadata_chipseq_bigwig_pvalue\n",
    ")\n",
    "\n",
    "for (idx in names(lst)){\n",
    "    ### show\n",
    "    cat(idx, \"\\n\")\n",
    "\n",
    "    ### extract data\n",
    "    dat_metadata = lst[[idx]]\n",
    "\n",
    "    dat = dat_metadata\n",
    "    print(dim(dat))\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "712b8616-fb2f-4bd0-901c-d87ff108092f",
   "metadata": {},
   "source": [
    "**Check equal experiment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "c6048569-2002-43b9-b093-8c61405e92c2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] TRUE\n",
      "[1] TRUE\n"
     ]
    }
   ],
   "source": [
    "lst = list(\n",
    "    \"region_narrowPeak\"  = dat_metadata_chipseq_bed_narrowpeak,\n",
    "    \"signal_fold_change\" = dat_metadata_chipseq_bigwig_fold_change,\n",
    "    \"signal_pvalue\"      = dat_metadata_chipseq_bigwig_pvalue\n",
    ")\n",
    "\n",
    "lst = lapply(lst, function(dat){\n",
    "    vec = dat$Index_Experiment\n",
    "    vec = sort(vec)\n",
    "})\n",
    "\n",
    "print(all(lst[[1]] == lst[[2]]))\n",
    "print(all(lst[[1]] == lst[[3]]))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc5c9c84-ad5d-46f3-901d-52b67514f421",
   "metadata": {},
   "source": [
    "**export results: metadata table, file list, and checksum: split for parallel execution**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "e5f77501-5c45-48fc-b6f7-fd64b982da39",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "region_narrowPeak \n",
      "Split: 01 ; Shape: 36 14 \n",
      "Split: 02 ; Shape: 36 14 \n",
      "Split: 03 ; Shape: 36 14 \n",
      "Split: 04 ; Shape: 36 14 \n",
      "Split: 05 ; Shape: 36 14 \n",
      "Split: 06 ; Shape: 36 14 \n",
      "Split: 07 ; Shape: 36 14 \n",
      "Split: 08 ; Shape: 36 14 \n",
      "Split: 09 ; Shape: 36 14 \n",
      "Split: 10 ; Shape: 36 14 \n",
      "Split: 11 ; Shape: 36 14 \n",
      "Split: 12 ; Shape: 36 14 \n",
      "Split: 13 ; Shape: 36 14 \n",
      "Split: 14 ; Shape: 36 14 \n",
      "Split: 15 ; Shape: 36 14 \n",
      "Split: 16 ; Shape: 36 14 \n",
      "Split: 17 ; Shape: 36 14 \n",
      "Split: 18 ; Shape: 36 14 \n",
      "Split: 19 ; Shape: 36 14 \n",
      "Split: 20 ; Shape: 36 14 \n",
      "\n",
      "signal_fold_change \n",
      "Split: 01 ; Shape: 36 14 \n",
      "Split: 02 ; Shape: 36 14 \n",
      "Split: 03 ; Shape: 36 14 \n",
      "Split: 04 ; Shape: 36 14 \n",
      "Split: 05 ; Shape: 36 14 \n",
      "Split: 06 ; Shape: 36 14 \n",
      "Split: 07 ; Shape: 36 14 \n",
      "Split: 08 ; Shape: 36 14 \n",
      "Split: 09 ; Shape: 36 14 \n",
      "Split: 10 ; Shape: 36 14 \n",
      "Split: 11 ; Shape: 36 14 \n",
      "Split: 12 ; Shape: 36 14 \n",
      "Split: 13 ; Shape: 36 14 \n",
      "Split: 14 ; Shape: 36 14 \n",
      "Split: 15 ; Shape: 36 14 \n",
      "Split: 16 ; Shape: 36 14 \n",
      "Split: 17 ; Shape: 36 14 \n",
      "Split: 18 ; Shape: 36 14 \n",
      "Split: 19 ; Shape: 36 14 \n",
      "Split: 20 ; Shape: 36 14 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "lst_dat_metadata = list(\n",
    "    \"region_narrowPeak\"  = dat_metadata_chipseq_bed_narrowpeak,\n",
    "    \"signal_fold_change\" = dat_metadata_chipseq_bigwig_fold_change\n",
    ")\n",
    "\n",
    "NUM_SPLIT = 20\n",
    "for (txt_folder in names(lst_dat_metadata)){\n",
    "    ### show progress\n",
    "    cat(txt_folder, \"\\n\")\n",
    "    \n",
    "    ### extract data\n",
    "    dat_metadata = lst_dat_metadata[[txt_folder]]\n",
    "    \n",
    "    ### create folder\n",
    "    txt_fdiry = file.path(FD_DAT, \"external\", TXT_FOLDER_OUT, txt_folder)\n",
    "    txt_cmd   = paste(\"mkdir -p\", txt_fdiry)\n",
    "    system(txt_cmd)\n",
    "    \n",
    "    ### write metadata info\n",
    "    txt_fname = \"metadata.tsv\"\n",
    "    txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "    dat = dat_metadata\n",
    "    write_tsv(dat, txt_fpath)  \n",
    "\n",
    "    ### split the table into equal parts for parallel execution\n",
    "    dat = dat_metadata\n",
    "    vec = 1:nrow(dat)\n",
    "    vec = vec %% NUM_SPLIT\n",
    "    lst = split(dat, vec)\n",
    "    lst_dat_metadata_sub = lst\n",
    "    \n",
    "    for (idx in 1:length(lst_dat_metadata_sub)){\n",
    "        ### extract data\n",
    "        dat_metadata_sub = lst_dat_metadata_sub[[idx]]\n",
    "        \n",
    "        ### get file url\n",
    "        dat = dat_metadata_sub\n",
    "        dat = dat %>% dplyr::select(File_URL)\n",
    "        dat_download_furl = dat\n",
    "        \n",
    "        ### get md5sum for each file\n",
    "        dat = dat_metadata_sub\n",
    "        dat = dat %>% dplyr::select(md5sum, File_Name)\n",
    "        dat_download_md5sum = dat\n",
    "\n",
    "        ### set index for file name\n",
    "        txt_index = str_pad(idx, 2, pad=\"0\")\n",
    "        \n",
    "        ### write file list\n",
    "        txt_fname  = paste(\"files\", txt_index, \"txt\", sep=\".\")\n",
    "        txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "        \n",
    "        dat = dat_download_furl\n",
    "        write_tsv(dat, txt_fpath, col_names = FALSE)\n",
    "        \n",
    "        ### write checksum file\n",
    "        txt_fname  = paste(\"checksum_md5sum\", txt_index, \"txt\", sep=\".\")\n",
    "        txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "        \n",
    "        dat = dat_download_md5sum\n",
    "        write_tsv(dat, txt_fpath, col_names = FALSE)\n",
    "\n",
    "        ### show progress\n",
    "        txt = c(\"Split:\", txt_index, \"; Shape:\", dim(dat_metadata_sub))\n",
    "        cat(txt, \"\\n\")\n",
    "    }\n",
    "    cat(\"\\n\")\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f920610b-2fb9-49b7-ab7b-3916e4000382",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "27c61c10-ac92-475d-9380-a726262f9f22",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb0c8e0e-0c24-4724-8e60-bc9fb9cec74a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "24698a91-cd62-4b63-a92b-21760dcdf702",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07e9e6e8-6e90-40bb-b630-f87f755a9b2a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9dc76f1-362a-4cd9-ba46-f7f783a97ab6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "1e503737-9246-4dcb-adc1-09cde1c9a34c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'encode_chipseq_tf_250120'"
      ],
      "text/latex": [
       "'encode\\_chipseq\\_tf\\_250120'"
      ],
      "text/markdown": [
       "'encode_chipseq_tf_250120'"
      ],
      "text/plain": [
       "[1] \"encode_chipseq_tf_250120\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "TXT_FOLDER_OUT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "57e78d4a-336c-478d-b960-a9b6d6314003",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>⋯</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li><li>1</li><li>2</li><li>3</li><li>4</li><li>5</li><li>6</li><li>7</li><li>8</li><li>9</li><li>10</li><li>11</li><li>12</li><li>13</li><li>14</li><li>15</li><li>16</li><li>17</li><li>18</li><li>19</li><li>0</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item ⋯\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 10\n",
       "\\item 11\n",
       "\\item 12\n",
       "\\item 13\n",
       "\\item 14\n",
       "\\item 15\n",
       "\\item 16\n",
       "\\item 17\n",
       "\\item 18\n",
       "\\item 19\n",
       "\\item 0\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1\n",
       "2. 2\n",
       "3. 3\n",
       "4. 4\n",
       "5. 5\n",
       "6. 6\n",
       "7. 7\n",
       "8. 8\n",
       "9. 9\n",
       "10. 10\n",
       "11. 11\n",
       "12. 12\n",
       "13. 13\n",
       "14. 14\n",
       "15. 15\n",
       "16. 16\n",
       "17. 17\n",
       "18. 18\n",
       "19. 19\n",
       "20. 0\n",
       "21. 1\n",
       "22. 2\n",
       "23. 3\n",
       "24. 4\n",
       "25. 5\n",
       "26. 6\n",
       "27. 7\n",
       "28. 8\n",
       "29. 9\n",
       "30. 10\n",
       "31. 11\n",
       "32. 12\n",
       "33. 13\n",
       "34. 14\n",
       "35. 15\n",
       "36. 16\n",
       "37. 17\n",
       "38. 18\n",
       "39. 19\n",
       "40. 0\n",
       "41. 1\n",
       "42. 2\n",
       "43. 3\n",
       "44. 4\n",
       "45. 5\n",
       "46. 6\n",
       "47. 7\n",
       "48. 8\n",
       "49. 9\n",
       "50. 10\n",
       "51. 11\n",
       "52. 12\n",
       "53. 13\n",
       "54. 14\n",
       "55. 15\n",
       "56. 16\n",
       "57. 17\n",
       "58. 18\n",
       "59. 19\n",
       "60. 0\n",
       "61. 1\n",
       "62. 2\n",
       "63. 3\n",
       "64. 4\n",
       "65. 5\n",
       "66. 6\n",
       "67. 7\n",
       "68. 8\n",
       "69. 9\n",
       "70. 10\n",
       "71. 11\n",
       "72. 12\n",
       "73. 13\n",
       "74. 14\n",
       "75. 15\n",
       "76. 16\n",
       "77. 17\n",
       "78. 18\n",
       "79. 19\n",
       "80. 0\n",
       "81. 1\n",
       "82. 2\n",
       "83. 3\n",
       "84. 4\n",
       "85. 5\n",
       "86. 6\n",
       "87. 7\n",
       "88. 8\n",
       "89. 9\n",
       "90. 10\n",
       "91. 11\n",
       "92. 12\n",
       "93. 13\n",
       "94. 14\n",
       "95. 15\n",
       "96. 16\n",
       "97. 17\n",
       "98. 18\n",
       "99. 19\n",
       "100. 0\n",
       "101. 1\n",
       "102. 2\n",
       "103. 3\n",
       "104. 4\n",
       "105. 5\n",
       "106. 6\n",
       "107. 7\n",
       "108. 8\n",
       "109. 9\n",
       "110. 10\n",
       "111. 11\n",
       "112. 12\n",
       "113. 13\n",
       "114. 14\n",
       "115. 15\n",
       "116. 16\n",
       "117. 17\n",
       "118. 18\n",
       "119. 19\n",
       "120. 0\n",
       "121. 1\n",
       "122. 2\n",
       "123. 3\n",
       "124. 4\n",
       "125. 5\n",
       "126. 6\n",
       "127. 7\n",
       "128. 8\n",
       "129. 9\n",
       "130. 10\n",
       "131. 11\n",
       "132. 12\n",
       "133. 13\n",
       "134. 14\n",
       "135. 15\n",
       "136. 16\n",
       "137. 17\n",
       "138. 18\n",
       "139. 19\n",
       "140. 0\n",
       "141. 1\n",
       "142. 2\n",
       "143. 3\n",
       "144. 4\n",
       "145. 5\n",
       "146. 6\n",
       "147. 7\n",
       "148. 8\n",
       "149. 9\n",
       "150. 10\n",
       "151. 11\n",
       "152. 12\n",
       "153. 13\n",
       "154. 14\n",
       "155. 15\n",
       "156. 16\n",
       "157. 17\n",
       "158. 18\n",
       "159. 19\n",
       "160. 0\n",
       "161. 1\n",
       "162. 2\n",
       "163. 3\n",
       "164. 4\n",
       "165. 5\n",
       "166. 6\n",
       "167. 7\n",
       "168. 8\n",
       "169. 9\n",
       "170. 10\n",
       "171. 11\n",
       "172. 12\n",
       "173. 13\n",
       "174. 14\n",
       "175. 15\n",
       "176. 16\n",
       "177. 17\n",
       "178. 18\n",
       "179. 19\n",
       "180. 0\n",
       "181. 1\n",
       "182. 2\n",
       "183. 3\n",
       "184. 4\n",
       "185. 5\n",
       "186. 6\n",
       "187. 7\n",
       "188. 8\n",
       "189. 9\n",
       "190. 10\n",
       "191. 11\n",
       "192. 12\n",
       "193. 13\n",
       "194. 14\n",
       "195. 15\n",
       "196. 16\n",
       "197. 17\n",
       "198. 18\n",
       "199. 19\n",
       "200. 0\n",
       "201. ⋯\n",
       "202. 1\n",
       "203. 2\n",
       "204. 3\n",
       "205. 4\n",
       "206. 5\n",
       "207. 6\n",
       "208. 7\n",
       "209. 8\n",
       "210. 9\n",
       "211. 10\n",
       "212. 11\n",
       "213. 12\n",
       "214. 13\n",
       "215. 14\n",
       "216. 15\n",
       "217. 16\n",
       "218. 17\n",
       "219. 18\n",
       "220. 19\n",
       "221. 0\n",
       "222. 1\n",
       "223. 2\n",
       "224. 3\n",
       "225. 4\n",
       "226. 5\n",
       "227. 6\n",
       "228. 7\n",
       "229. 8\n",
       "230. 9\n",
       "231. 10\n",
       "232. 11\n",
       "233. 12\n",
       "234. 13\n",
       "235. 14\n",
       "236. 15\n",
       "237. 16\n",
       "238. 17\n",
       "239. 18\n",
       "240. 19\n",
       "241. 0\n",
       "242. 1\n",
       "243. 2\n",
       "244. 3\n",
       "245. 4\n",
       "246. 5\n",
       "247. 6\n",
       "248. 7\n",
       "249. 8\n",
       "250. 9\n",
       "251. 10\n",
       "252. 11\n",
       "253. 12\n",
       "254. 13\n",
       "255. 14\n",
       "256. 15\n",
       "257. 16\n",
       "258. 17\n",
       "259. 18\n",
       "260. 19\n",
       "261. 0\n",
       "262. 1\n",
       "263. 2\n",
       "264. 3\n",
       "265. 4\n",
       "266. 5\n",
       "267. 6\n",
       "268. 7\n",
       "269. 8\n",
       "270. 9\n",
       "271. 10\n",
       "272. 11\n",
       "273. 12\n",
       "274. 13\n",
       "275. 14\n",
       "276. 15\n",
       "277. 16\n",
       "278. 17\n",
       "279. 18\n",
       "280. 19\n",
       "281. 0\n",
       "282. 1\n",
       "283. 2\n",
       "284. 3\n",
       "285. 4\n",
       "286. 5\n",
       "287. 6\n",
       "288. 7\n",
       "289. 8\n",
       "290. 9\n",
       "291. 10\n",
       "292. 11\n",
       "293. 12\n",
       "294. 13\n",
       "295. 14\n",
       "296. 15\n",
       "297. 16\n",
       "298. 17\n",
       "299. 18\n",
       "300. 19\n",
       "301. 0\n",
       "302. 1\n",
       "303. 2\n",
       "304. 3\n",
       "305. 4\n",
       "306. 5\n",
       "307. 6\n",
       "308. 7\n",
       "309. 8\n",
       "310. 9\n",
       "311. 10\n",
       "312. 11\n",
       "313. 12\n",
       "314. 13\n",
       "315. 14\n",
       "316. 15\n",
       "317. 16\n",
       "318. 17\n",
       "319. 18\n",
       "320. 19\n",
       "321. 0\n",
       "322. 1\n",
       "323. 2\n",
       "324. 3\n",
       "325. 4\n",
       "326. 5\n",
       "327. 6\n",
       "328. 7\n",
       "329. 8\n",
       "330. 9\n",
       "331. 10\n",
       "332. 11\n",
       "333. 12\n",
       "334. 13\n",
       "335. 14\n",
       "336. 15\n",
       "337. 16\n",
       "338. 17\n",
       "339. 18\n",
       "340. 19\n",
       "341. 0\n",
       "342. 1\n",
       "343. 2\n",
       "344. 3\n",
       "345. 4\n",
       "346. 5\n",
       "347. 6\n",
       "348. 7\n",
       "349. 8\n",
       "350. 9\n",
       "351. 10\n",
       "352. 11\n",
       "353. 12\n",
       "354. 13\n",
       "355. 14\n",
       "356. 15\n",
       "357. 16\n",
       "358. 17\n",
       "359. 18\n",
       "360. 19\n",
       "361. 0\n",
       "362. 1\n",
       "363. 2\n",
       "364. 3\n",
       "365. 4\n",
       "366. 5\n",
       "367. 6\n",
       "368. 7\n",
       "369. 8\n",
       "370. 9\n",
       "371. 10\n",
       "372. 11\n",
       "373. 12\n",
       "374. 13\n",
       "375. 14\n",
       "376. 15\n",
       "377. 16\n",
       "378. 17\n",
       "379. 18\n",
       "380. 19\n",
       "381. 0\n",
       "382. 1\n",
       "383. 2\n",
       "384. 3\n",
       "385. 4\n",
       "386. 5\n",
       "387. 6\n",
       "388. 7\n",
       "389. 8\n",
       "390. 9\n",
       "391. 10\n",
       "392. 11\n",
       "393. 12\n",
       "394. 13\n",
       "395. 14\n",
       "396. 15\n",
       "397. 16\n",
       "398. 17\n",
       "399. 18\n",
       "400. 19\n",
       "401. 0\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5\n",
       " [26]  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10\n",
       " [51] 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15\n",
       " [76] 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0\n",
       "[101]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5\n",
       "[126]  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10\n",
       "[151] 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15\n",
       "[176] 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0\n",
       "[201]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5\n",
       "[226]  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10\n",
       "[251] 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15\n",
       "[276] 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0\n",
       "[301]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5\n",
       "[326]  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10\n",
       "[351] 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15\n",
       "[376] 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0\n",
       "[401]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5\n",
       "[426]  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10\n",
       "[451] 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15\n",
       "[476] 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0\n",
       "[501]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5\n",
       "[526]  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10\n",
       "[551] 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15\n",
       "[576] 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0\n",
       "[601]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5\n",
       "[626]  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10\n",
       "[651] 11 12 13 14 15 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15\n",
       "[676] 16 17 18 19  0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0\n",
       "[701]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19  0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "vec_num_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4d9b72a0-e11b-4fb2-a12d-2f7b74c6daf0",
   "metadata": {},
   "outputs": [],
   "source": [
    "dat = dat_metadata_files_bwg\n",
    "dat = dat %>% \n",
    "    dplyr::mutate(Index = 1:nrow(dat)) %>%\n",
    "    dplyr::mutate(Index = Index %% 20)\n",
    "\n",
    "lst = split(dat, dat$Index)\n",
    "for (idx in 1:length(lst)){\n",
    "    ### extract data\n",
    "    dat  = lst[[idx]]\n",
    "    dat1 = dat %>% dplyr::select(`File download URL`)\n",
    "    dat2 = dat %>% \n",
    "        dplyr::mutate(File_Name = basename(`File download URL`)) %>%\n",
    "        dplyr::select(md5sum, File_Name)\n",
    "\n",
    "    ### set index for file name\n",
    "    txt = str_pad(idx, 2, pad=\"0\")\n",
    "    \n",
    "    ### set directory\n",
    "    txt_folder = TXT_FOLDER_OUT\n",
    "    txt_fdiry  = file.path(FD_DAT, \"external\", txt_folder)\n",
    "\n",
    "    ### save download file list\n",
    "    txt_fname  = paste(\"files_bwg\", txt, \"txt\", sep=\".\")\n",
    "    txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "    print(txt_fname)\n",
    "   \n",
    "    #write_tsv(dat1, txt_fpath, col_names = FALSE)\n",
    "\n",
    "    ### save download file checksum\n",
    "    txt_folder = TXT_FOLDER_OUT\n",
    "    txt_fdiry  = file.path(FD_DAT, \"external\", txt_folder)\n",
    "    txt_fname  = paste(\"checksum_md5sum.bwg\", txt, \"txt\", sep=\".\")\n",
    "    txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "    print(txt_fname)\n",
    "\n",
    "    ### write table\n",
    "    #write_tsv(dat2, txt_fpath, col_names = FALSE)\n",
    "    cat(\"\\n\")\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ddd1707-de28-4ffc-9535-4375047c8e04",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "98646b9c-170a-4727-ad48-fee169b6ddba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "region_narrowPeak \n",
      "signal_fold_change \n",
      "signal_pvalue \n"
     ]
    }
   ],
   "source": [
    "lst = list(\n",
    "    \"region_narrowPeak\"  = dat_metadata_chipseq_bed_narrowpeak,\n",
    "    \"signal_fold_change\" = dat_metadata_chipseq_bigwig_pvalue,\n",
    "    \"signal_pvalue\"      = dat_metadata_chipseq_bigwig_fold_change\n",
    ")\n",
    "\n",
    "for (idx in names(lst)){\n",
    "    ### show\n",
    "    cat(idx, \"\\n\")\n",
    "\n",
    "    ### extract data\n",
    "    dat_metadata = lst[[idx]]\n",
    "\n",
    "    ### get file url\n",
    "    dat = dat_metadata\n",
    "    dat = dat %>% dplyr::select(File_URL)\n",
    "    dat_download_furl = dat\n",
    "    \n",
    "    ### get md5sum for each file\n",
    "    dat = dat_metadata\n",
    "    dat = dat %>% dplyr::select(md5sum, File_Name)\n",
    "    dat_download_md5sum = dat\n",
    "\n",
    "    ### create folder\n",
    "    txt_fdiry = file.path(FD_DAT, \"external\", TXT_FOLDER_OUT, idx)\n",
    "    txt_cmd   = paste(\"mkdir -p\", txt_fdiry)\n",
    "    system(txt_cmd)\n",
    "\n",
    "    ### write file list\n",
    "    txt_fname = \"files.txt\"\n",
    "    txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "    \n",
    "    dat = dat_download_furl\n",
    "    write_tsv(dat, txt_fpath, col_names = FALSE)\n",
    "\n",
    "    ### write checksum file\n",
    "    txt_fname = \"checksum_md5sum.txt\"\n",
    "    txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "    dat = dat_download_md5sum\n",
    "    write_tsv(dat, txt_fpath, col_names = FALSE)\n",
    "\n",
    "    ### write metadata info\n",
    "    txt_fname = \"metadata.tsv\"\n",
    "    txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "    dat = dat_metadata\n",
    "    write_tsv(dat, txt_fpath)  \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "765cd429-9afb-4005-8c83-21a5f2b87f6a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "872fccd8-5a4f-415a-b15b-e19ca3c82bb8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "de6a8cef-cc52-47c1-9447-b7435ec6c8be",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5ab4718b-e778-4592-98c5-98711b82009b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c3fa13f-ff6f-4e62-9b96-885abf0ce3c5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fb07e37e-52b8-44aa-ae4b-74a5893ecce8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ba3fb86-4f4f-4f86-aba1-d0943524576b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e618d907-9bba-45d3-9578-daa20dfc1ca5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "raw",
   "id": "813c503a-8690-4a40-8758-c70f1099ef09",
   "metadata": {},
   "source": [
    "---\n",
    "title: Prepare ENCODE ChIP-seq 11 (TF)\n",
    "subtitle: Generate download script to download the data\n",
    "format:\n",
    "    html:\n",
    "        code-fold:  true\n",
    "        code-tools: true\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01444860-b129-474d-8f8f-ccb2197a1ce9",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0d96b489-2446-427b-befd-3769dd05cf30",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are working on        Singularity: singularity_proj_encode_fcc \n",
      "BASE DIRECTORY (FD_BASE): /data/reddylab/Kuei \n",
      "REPO DIRECTORY (FD_REPO): /data/reddylab/Kuei/repo \n",
      "WORK DIRECTORY (FD_WORK): /data/reddylab/Kuei/work \n",
      "DATA DIRECTORY (FD_DATA): /data/reddylab/Kuei/data \n",
      "\n",
      "You are working with      ENCODE FCC \n",
      "PATH OF PROJECT (FD_PRJ): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC \n",
      "PROJECT RESULTS (FD_RES): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results \n",
      "PROJECT SCRIPTS (FD_EXE): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/scripts \n",
      "PROJECT DATA    (FD_DAT): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/data \n",
      "PROJECT NOTE    (FD_NBK): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/notebooks \n",
      "PROJECT DOCS    (FD_DOC): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/docs \n",
      "PROJECT LOG     (FD_LOG): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/log \n",
      "PROJECT REF     (FD_REF): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/references \n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(source(\"../run_config_project_sing.R\")))\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0b120069-8f65-41e1-b47e-54a761e47fbf",
   "metadata": {},
   "source": [
    "## Prepare"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3b3e787d-6167-4a2d-8858-3f84c67cb285",
   "metadata": {},
   "outputs": [],
   "source": [
    "TXT_FOLDER_INP = \"encode_chipseq_tf\"\n",
    "TXT_FOLDER_OUT = \"encode_chipseq_tf_250120\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1e72f6de-f993-4338-98c7-d763209b4c66",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "files.chipseq_tf.default_analysis.250109.txt \n",
      "files.chipseq_tf.default_analysis.250120.txt \n",
      "files.chipseq_tf.default_files.241223.txt \n",
      "files.chipseq_tf.default_files.250108.txt \n",
      "files.chipseq_tf.default_files.250120.txt \n",
      "metadata.chipseq_tf.default_analysis.250109.tsv \n",
      "metadata.chipseq_tf.default_analysis.250120.tsv \n",
      "metadata.chipseq_tf.default_files.241223.tsv \n",
      "metadata.chipseq_tf.default_files.250108.tsv \n",
      "metadata.chipseq_tf.default_files.250120.tsv \n",
      "Readme.md \n"
     ]
    }
   ],
   "source": [
    "txt_folder = TXT_FOLDER_INP\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "\n",
    "vec = dir(txt_fdiry)\n",
    "for (txt in vec){cat(txt, \"\\n\")}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9f052a15-beb4-4df9-b3fb-1622d192c82a",
   "metadata": {},
   "source": [
    "## Import metadata from reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e1213368-0d6d-4be3-90ed-4f0e6b7393d0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1604   60\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "   <th style=\"text-align:left;\"> File analysis version </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF982FCY </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR985OYK </td>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> CRISPR </td>\n",
       "   <td style=\"text-align:left;\"> insertion </td>\n",
       "   <td style=\"text-align:left;\"> {'dbxrefs': ['FactorBook:ZBTB11'], 'aliases': [], 'organism': {'schema_version': '6', '@type': ['Organism', 'Item'], 'name': 'human', 'taxon_id': '9606', 'scientific_name': 'Homo sapiens', '@id': '/organisms/human/', 'uuid': '7745b647-ff15-4ff3-9ced-b897d4e2983c', 'status': 'released'}, 'genes': ['/genes/27107/'], '@type': ['Target', 'Item'], 'label': 'ZBTB11', 'title': 'ZBTB11 (Homo sapiens)', 'investigated_as': ['transcription factor'], 'uuid': 'e6b7c1f8-62d9-48fb-a633-b01a6c77ac69', 'schema_version': '14', 'name': 'ZBTB11-human', '@id': '/targets/ZBTB11-human/', 'status': 'released'} </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB11-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2021-01-06 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> sonication (Bioruptor Pico) </td>\n",
       "   <td style=\"text-align:left;\"> 250-450 </td>\n",
       "   <td style=\"text-align:left;\"> 3, 4 </td>\n",
       "   <td style=\"text-align:left;\"> 3_1, 4_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF900MNG/, /files/ENCFF174OKT/, /files/ENCFF465SQQ/ </td>\n",
       "   <td style=\"text-align:right;\"> 1436439113 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> f54e9851ca83e2143a6712d8e99761c3 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF982FCY/@@download/ENCFF982FCY.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/11/04/8e7f5141-578f-43a7-8d29-9751e3f35df5/ENCFF982FCY.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/11/04/8e7f5141-578f-43a7-8d29-9751e3f35df5/ENCFF982FCY.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.6.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> mild to moderate bottlenecking </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_folder = TXT_FOLDER_INP\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "txt_fname  = \"metadata.chipseq_tf.default_files.250108.tsv\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "dat = dat %>% \n",
    "    dplyr::rowwise() %>%\n",
    "    dplyr::mutate(`File analysis version` = str_extract(`File analysis title`, \"ENCODE\\\\d\")) %>%\n",
    "    dplyr::ungroup()\n",
    "\n",
    "### show and assign\n",
    "dat_metadata_chipseq = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3c629a1-0db1-44f2-9272-08372a985476",
   "metadata": {},
   "source": [
    "## Explore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4d487008-0612-41f9-9a31-60a14f1b52d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             \n",
       "              bed bigWig\n",
       "  TF ChIP-seq 802    802"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq\n",
    "table(dat$Assay, dat$`File type`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "da3f03ac-3706-4a78-ac88-7c9a2e610b08",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        \n",
       "         conservative IDR thresholded peaks IDR thresholded peaks\n",
       "  bed                                    66                   654\n",
       "  bigWig                                  0                     0\n",
       "        \n",
       "         optimal IDR thresholded peaks pseudoreplicated IDR thresholded peaks\n",
       "  bed                               78                                      4\n",
       "  bigWig                             0                                      0\n",
       "        \n",
       "         signal p-value\n",
       "  bed                 0\n",
       "  bigWig            802"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq\n",
    "table(dat$`File type`, dat$`Output type`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "04953467-792e-4fef-9752-a84bdadeced3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "K562 \n",
       "1604 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq\n",
    "table(dat$`Biosample term name`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1753f460-db55-46f8-8605-c7161f39b167",
   "metadata": {},
   "source": [
    "**Question: why there is hg19? I have filtered out by setting `assembly=GRCh38`**\n",
    "\n",
    "**Results:**\n",
    "- Example: https://www.encodeproject.org/experiments/ENCSR690GUG/\n",
    "- Observe:\n",
    "    - there are multiple default files for each file type, including hg38 and hg19\n",
    "    - almost all of them have both ENCODE3 and ENCODE4 analysis title\n",
    "- Decision:\n",
    "    - filter out files with ENCODE3 analysis title"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "187bf25b-4b85-405b-8d30-0f334f565669",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        \n",
       "         GRCh38 hg19\n",
       "  bed       761   41\n",
       "  bigWig    761   41"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq\n",
    "table(dat$`File type`, dat$`File assembly`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a9c61186-433d-4f77-b94e-7acc973194bd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "         \n",
       "          GRCh38 hg19\n",
       "  ENCODE3     82   82\n",
       "  ENCODE4   1440    0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq\n",
    "table(dat$`File analysis version`, dat$`File assembly`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "07231004-af49-4839-8a19-55542e0dada5",
   "metadata": {},
   "source": [
    "## Prepare download file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "dd8bd333-581c-4e40-bffb-e13be9aa0488",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1440   60\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "   <th style=\"text-align:left;\"> File analysis version </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF982FCY </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR985OYK </td>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> CRISPR </td>\n",
       "   <td style=\"text-align:left;\"> insertion </td>\n",
       "   <td style=\"text-align:left;\"> {'dbxrefs': ['FactorBook:ZBTB11'], 'aliases': [], 'organism': {'schema_version': '6', '@type': ['Organism', 'Item'], 'name': 'human', 'taxon_id': '9606', 'scientific_name': 'Homo sapiens', '@id': '/organisms/human/', 'uuid': '7745b647-ff15-4ff3-9ced-b897d4e2983c', 'status': 'released'}, 'genes': ['/genes/27107/'], '@type': ['Target', 'Item'], 'label': 'ZBTB11', 'title': 'ZBTB11 (Homo sapiens)', 'investigated_as': ['transcription factor'], 'uuid': 'e6b7c1f8-62d9-48fb-a633-b01a6c77ac69', 'schema_version': '14', 'name': 'ZBTB11-human', '@id': '/targets/ZBTB11-human/', 'status': 'released'} </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB11-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2021-01-06 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> sonication (Bioruptor Pico) </td>\n",
       "   <td style=\"text-align:left;\"> 250-450 </td>\n",
       "   <td style=\"text-align:left;\"> 3, 4 </td>\n",
       "   <td style=\"text-align:left;\"> 3_1, 4_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF900MNG/, /files/ENCFF174OKT/, /files/ENCFF465SQQ/ </td>\n",
       "   <td style=\"text-align:right;\"> 1436439113 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> f54e9851ca83e2143a6712d8e99761c3 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF982FCY/@@download/ENCFF982FCY.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/11/04/8e7f5141-578f-43a7-8d29-9751e3f35df5/ENCFF982FCY.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/11/04/8e7f5141-578f-43a7-8d29-9751e3f35df5/ENCFF982FCY.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.6.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> mild to moderate bottlenecking </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq\n",
    "dat = dat %>% dplyr::filter(`File analysis version` == \"ENCODE4\")\n",
    "\n",
    "dat_metadata_chipseq_filter = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ce5647d4-a8c0-4eea-892c-ba05d70436b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "dat = dat_metadata_chipseq_filter\n",
    "lst = split(dat, dat$`File type`)\n",
    "\n",
    "dat_metadata_files_bed = lst[[\"bed\"]]\n",
    "dat_metadata_files_bwg = lst[[\"bigWig\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "0684dc5f-f1e9-445e-9408-af6caaa26a59",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "bed narrowPeak \n",
      "           720 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "   <th style=\"text-align:left;\"> File analysis version </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF227HVU </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR985OYK </td>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> CRISPR </td>\n",
       "   <td style=\"text-align:left;\"> insertion </td>\n",
       "   <td style=\"text-align:left;\"> {'dbxrefs': ['FactorBook:ZBTB11'], 'aliases': [], 'organism': {'schema_version': '6', '@type': ['Organism', 'Item'], 'name': 'human', 'taxon_id': '9606', 'scientific_name': 'Homo sapiens', '@id': '/organisms/human/', 'uuid': '7745b647-ff15-4ff3-9ced-b897d4e2983c', 'status': 'released'}, 'genes': ['/genes/27107/'], '@type': ['Target', 'Item'], 'label': 'ZBTB11', 'title': 'ZBTB11 (Homo sapiens)', 'investigated_as': ['transcription factor'], 'uuid': 'e6b7c1f8-62d9-48fb-a633-b01a6c77ac69', 'schema_version': '14', 'name': 'ZBTB11-human', '@id': '/targets/ZBTB11-human/', 'status': 'released'} </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB11-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2021-01-06 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> sonication (Bioruptor Pico) </td>\n",
       "   <td style=\"text-align:left;\"> 250-450 </td>\n",
       "   <td style=\"text-align:left;\"> 3, 4 </td>\n",
       "   <td style=\"text-align:left;\"> 3_1, 4_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF900MNG/, /files/ENCFF356LFX/, /files/ENCFF174OKT/, /files/ENCFF465SQQ/ </td>\n",
       "   <td style=\"text-align:right;\"> 76588 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> ed4dfdea1e3ee63fe51e82eea5399fc2 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF227HVU/@@download/ENCFF227HVU.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/11/04/4d549787-8bfc-4152-9d73-0858392df1be/ENCFF227HVU.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/11/04/4d549787-8bfc-4152-9d73-0858392df1be/ENCFF227HVU.bed.gz?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.6.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> mild to moderate bottlenecking </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_files_bed\n",
    "print(table(dat$`File format`))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "30ce7e74-2787-4a20-8fc7-9a110e3728e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "bigWig \n",
      "   720 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "   <th style=\"text-align:left;\"> File analysis version </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF982FCY </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR985OYK </td>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> CRISPR </td>\n",
       "   <td style=\"text-align:left;\"> insertion </td>\n",
       "   <td style=\"text-align:left;\"> {'dbxrefs': ['FactorBook:ZBTB11'], 'aliases': [], 'organism': {'schema_version': '6', '@type': ['Organism', 'Item'], 'name': 'human', 'taxon_id': '9606', 'scientific_name': 'Homo sapiens', '@id': '/organisms/human/', 'uuid': '7745b647-ff15-4ff3-9ced-b897d4e2983c', 'status': 'released'}, 'genes': ['/genes/27107/'], '@type': ['Target', 'Item'], 'label': 'ZBTB11', 'title': 'ZBTB11 (Homo sapiens)', 'investigated_as': ['transcription factor'], 'uuid': 'e6b7c1f8-62d9-48fb-a633-b01a6c77ac69', 'schema_version': '14', 'name': 'ZBTB11-human', '@id': '/targets/ZBTB11-human/', 'status': 'released'} </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB11-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2021-01-06 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> sonication (Bioruptor Pico) </td>\n",
       "   <td style=\"text-align:left;\"> 250-450 </td>\n",
       "   <td style=\"text-align:left;\"> 3, 4 </td>\n",
       "   <td style=\"text-align:left;\"> 3_1, 4_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF900MNG/, /files/ENCFF174OKT/, /files/ENCFF465SQQ/ </td>\n",
       "   <td style=\"text-align:right;\"> 1436439113 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> f54e9851ca83e2143a6712d8e99761c3 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF982FCY/@@download/ENCFF982FCY.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/11/04/8e7f5141-578f-43a7-8d29-9751e3f35df5/ENCFF982FCY.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/11/04/8e7f5141-578f-43a7-8d29-9751e3f35df5/ENCFF982FCY.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.6.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> mild to moderate bottlenecking </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_files_bwg\n",
    "print(table(dat$`File format`))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "37fd3d9f-17b6-4d73-ad54-57119ec44f5e",
   "metadata": {},
   "source": [
    "**Check results**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "99d8862e-a786-4eac-b609-7a2f05246961",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        \n",
       "         GRCh38\n",
       "  bed       720\n",
       "  bigWig    720"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_filter\n",
    "table(dat$`File type`, dat$`File assembly`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "05d645f1-4e77-4a35-a7af-e867c53a8259",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 511\n",
      "[1] 510\n"
     ]
    }
   ],
   "source": [
    "dat  = dat_metadata_chipseq\n",
    "vec1 = unique(dat$`Experiment target`)\n",
    "\n",
    "dat  = dat_metadata_chipseq_filter\n",
    "vec2 = unique(dat$`Experiment target`)\n",
    "\n",
    "print(length(vec1))\n",
    "print(length(vec2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "880f3b77-3431-4bdb-8f2e-cd390bb4c5fd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 721\n",
      "[1] 720\n",
      "[1] \"ENCSR000EHA\"\n"
     ]
    }
   ],
   "source": [
    "dat  = dat_metadata_chipseq\n",
    "vec1 = unique(dat$`Experiment accession`)\n",
    "\n",
    "dat  = dat_metadata_chipseq_filter\n",
    "vec2 = unique(dat$`Experiment accession`)\n",
    "\n",
    "print(length(vec1))\n",
    "print(length(vec2))\n",
    "print(setdiff(vec1, vec2))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "88590868-7ccd-4e80-8bc1-b156da030411",
   "metadata": {},
   "source": [
    "The only TF that are filtered out:  \n",
    "https://www.encodeproject.org/experiments/ENCSR000EHA/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b74ae601-46fb-4865-97eb-69a4c0e098ba",
   "metadata": {},
   "source": [
    "## Prepare download checksum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "8a27844c-1604-409d-8318-f9f424ab050f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1440    2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> f54e9851ca83e2143a6712d8e99761c3 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF982FCY.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ed4dfdea1e3ee63fe51e82eea5399fc2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF227HVU.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> c6da6bcb0fa81283ac19cd1c6ee3efd6 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF448BAC.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### get md5sum for each file\n",
    "dat = dat_metadata_chipseq_filter\n",
    "dat = dat %>% \n",
    "    dplyr::mutate(File_Name = basename(`File download URL`)) %>%\n",
    "    dplyr::select(md5sum, File_Name)\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_md5sum = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9f083041-2ef5-4ab4-bb4e-18ce5e803c1d",
   "metadata": {},
   "source": [
    "## Export results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "1e80434d-49ba-468c-84cd-3ca4b4c9e8fb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/data/external/encode_chipseq_tf_250108\"\n"
     ]
    }
   ],
   "source": [
    "### set directory\n",
    "txt_folder = TXT_FOLDER_OUT\n",
    "txt_fdiry  = file.path(FD_DAT, \"external\", txt_folder)\n",
    "\n",
    "### create folder for data\n",
    "txt_cmd = paste(\"mkdir -p\", txt_fdiry)\n",
    "system(txt_cmd)\n",
    "print(txt_fdiry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "bd723247-5ad6-413e-a122-eadfeb6cf840",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set directory\n",
    "txt_folder = TXT_FOLDER_OUT\n",
    "txt_fdiry  = file.path(FD_DAT, \"external\", txt_folder)\n",
    "txt_fname  = \"checksum_md5sum.txt\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### write table\n",
    "dat = dat_metadata_md5sum\n",
    "write_tsv(dat, txt_fpath, col_names = FALSE)  "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "472bb0c4-7469-4631-a944-2647c79d9c55",
   "metadata": {},
   "source": [
    "```\n",
    "### set directory\n",
    "txt_folder = TXT_FOLDER_OUT\n",
    "txt_fdiry  = file.path(FD_DAT, \"external\", txt_folder)\n",
    "txt_fname  = \"files.txt\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### write table\n",
    "dat = dat_metadata_files\n",
    "write_tsv(dat, txt_fpath, col_names = FALSE)\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "ad5c1c06-96cb-405d-95e9-559af68a9611",
   "metadata": {},
   "outputs": [],
   "source": [
    "### write table\n",
    "dat = dat_metadata_files_bed\n",
    "dat1 = dat %>% dplyr::select(`File download URL`)\n",
    "dat2 = dat %>% \n",
    "    dplyr::mutate(File_Name = basename(`File download URL`)) %>%\n",
    "    dplyr::select(md5sum, File_Name)\n",
    "\n",
    "### set directory\n",
    "txt_folder = TXT_FOLDER_OUT\n",
    "txt_fdiry  = file.path(FD_DAT, \"external\", txt_folder)\n",
    "\n",
    "### save download file list\n",
    "txt_fname  = \"files_bed.txt\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "write_tsv(dat1, txt_fpath, col_names = FALSE)\n",
    "\n",
    "### save download file checksum\n",
    "txt_fname  = \"checksum_md5sum.bed.txt\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "write_tsv(dat2, txt_fpath, col_names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "21be9962-2400-45e7-a4c2-e2d7a603787c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'001'"
      ],
      "text/latex": [
       "'001'"
      ],
      "text/markdown": [
       "'001'"
      ],
      "text/plain": [
       "[1] \"001\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "str_pad(\"1\", 3, pad=\"0\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "17c8a02f-95ee-42fb-a75f-a2f09c63e8e4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"files_bwg.01.txt\"\n",
      "[1] \"checksum_md5sum.bwg.01.txt\"\n",
      "\n",
      "[1] \"files_bwg.02.txt\"\n",
      "[1] \"checksum_md5sum.bwg.02.txt\"\n",
      "\n",
      "[1] \"files_bwg.03.txt\"\n",
      "[1] \"checksum_md5sum.bwg.03.txt\"\n",
      "\n",
      "[1] \"files_bwg.04.txt\"\n",
      "[1] \"checksum_md5sum.bwg.04.txt\"\n",
      "\n",
      "[1] \"files_bwg.05.txt\"\n",
      "[1] \"checksum_md5sum.bwg.05.txt\"\n",
      "\n",
      "[1] \"files_bwg.06.txt\"\n",
      "[1] \"checksum_md5sum.bwg.06.txt\"\n",
      "\n",
      "[1] \"files_bwg.07.txt\"\n",
      "[1] \"checksum_md5sum.bwg.07.txt\"\n",
      "\n",
      "[1] \"files_bwg.08.txt\"\n",
      "[1] \"checksum_md5sum.bwg.08.txt\"\n",
      "\n",
      "[1] \"files_bwg.09.txt\"\n",
      "[1] \"checksum_md5sum.bwg.09.txt\"\n",
      "\n",
      "[1] \"files_bwg.10.txt\"\n",
      "[1] \"checksum_md5sum.bwg.10.txt\"\n",
      "\n",
      "[1] \"files_bwg.11.txt\"\n",
      "[1] \"checksum_md5sum.bwg.11.txt\"\n",
      "\n",
      "[1] \"files_bwg.12.txt\"\n",
      "[1] \"checksum_md5sum.bwg.12.txt\"\n",
      "\n",
      "[1] \"files_bwg.13.txt\"\n",
      "[1] \"checksum_md5sum.bwg.13.txt\"\n",
      "\n",
      "[1] \"files_bwg.14.txt\"\n",
      "[1] \"checksum_md5sum.bwg.14.txt\"\n",
      "\n",
      "[1] \"files_bwg.15.txt\"\n",
      "[1] \"checksum_md5sum.bwg.15.txt\"\n",
      "\n",
      "[1] \"files_bwg.16.txt\"\n",
      "[1] \"checksum_md5sum.bwg.16.txt\"\n",
      "\n",
      "[1] \"files_bwg.17.txt\"\n",
      "[1] \"checksum_md5sum.bwg.17.txt\"\n",
      "\n",
      "[1] \"files_bwg.18.txt\"\n",
      "[1] \"checksum_md5sum.bwg.18.txt\"\n",
      "\n",
      "[1] \"files_bwg.19.txt\"\n",
      "[1] \"checksum_md5sum.bwg.19.txt\"\n",
      "\n",
      "[1] \"files_bwg.20.txt\"\n",
      "[1] \"checksum_md5sum.bwg.20.txt\"\n",
      "\n"
     ]
    }
   ],
   "source": [
    "dat = dat_metadata_files_bwg\n",
    "dat = dat %>% \n",
    "    dplyr::mutate(Index = 1:nrow(dat)) %>%\n",
    "    dplyr::mutate(Index = Index %% 20)\n",
    "\n",
    "lst = split(dat, dat$Index)\n",
    "for (idx in 1:length(lst)){\n",
    "    ### extract data\n",
    "    dat  = lst[[idx]]\n",
    "    dat1 = dat %>% dplyr::select(`File download URL`)\n",
    "    dat2 = dat %>% \n",
    "        dplyr::mutate(File_Name = basename(`File download URL`)) %>%\n",
    "        dplyr::select(md5sum, File_Name)\n",
    "\n",
    "    ### set index for file name\n",
    "    txt = str_pad(idx, 2, pad=\"0\")\n",
    "    \n",
    "    ### set directory\n",
    "    txt_folder = TXT_FOLDER_OUT\n",
    "    txt_fdiry  = file.path(FD_DAT, \"external\", txt_folder)\n",
    "\n",
    "    ### save download file list\n",
    "    txt_fname  = paste(\"files_bwg\", txt, \"txt\", sep=\".\")\n",
    "    txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "    print(txt_fname)\n",
    "   \n",
    "    #write_tsv(dat1, txt_fpath, col_names = FALSE)\n",
    "\n",
    "    ### save download file checksum\n",
    "    txt_folder = TXT_FOLDER_OUT\n",
    "    txt_fdiry  = file.path(FD_DAT, \"external\", txt_folder)\n",
    "    txt_fname  = paste(\"checksum_md5sum.bwg\", txt, \"txt\", sep=\".\")\n",
    "    txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "    print(txt_fname)\n",
    "\n",
    "    ### write table\n",
    "    #write_tsv(dat2, txt_fpath, col_names = FALSE)\n",
    "    cat(\"\\n\")\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58110182-d4d1-4323-a131-cc48355ff9ac",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
