{
 "cells": [
  {
   "cell_type": "raw",
   "id": "9fc0b8a7-4a37-4513-b605-d80ccb842df5",
   "metadata": {},
   "source": [
    "---\n",
    "title: Prepare CREs annotation 01\n",
    "subtitle: Prepare data\n",
    "format:\n",
    "    html:\n",
    "        code-fold:  true\n",
    "        code-tools: true\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "61db269b-929c-4249-bdf0-409781a0d953",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0d52eb64-6a73-4f88-9fa9-bbbf0c81a643",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are working on        Singularity: singularity_proj_encode_fcc \n",
      "BASE DIRECTORY (FD_BASE): /data/reddylab/Kuei \n",
      "REPO DIRECTORY (FD_REPO): /data/reddylab/Kuei/repo \n",
      "WORK DIRECTORY (FD_WORK): /data/reddylab/Kuei/work \n",
      "DATA DIRECTORY (FD_DATA): /data/reddylab/Kuei/data \n",
      "\n",
      "You are working with      ENCODE FCC \n",
      "PATH OF PROJECT (FD_PRJ): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC \n",
      "PROJECT RESULTS (FD_RES): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results \n",
      "PROJECT SCRIPTS (FD_EXE): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/scripts \n",
      "PROJECT DATA    (FD_DAT): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/data \n",
      "PROJECT NOTE    (FD_NBK): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/notebooks \n",
      "PROJECT DOCS    (FD_DOC): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/docs \n",
      "PROJECT LOG     (FD_LOG): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/log \n",
      "PROJECT REF     (FD_REF): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/references \n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(source(\"../run_config_project_sing.R\")))\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e17bb213-9671-4b3e-aac3-7735b0ea4918",
   "metadata": {},
   "source": [
    "**Set global variables**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "048625b0-f3ee-4628-8eb2-3b323082593d",
   "metadata": {},
   "outputs": [],
   "source": [
    "TXT_REGION_FOLDER = \"genome_cres\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1dafdee5-dcaf-4fac-bce5-f1b029bb8e3f",
   "metadata": {},
   "source": [
    "**Check input files**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "40126d42-76e3-4ec7-9286-4ef3e5c7985b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K562.hg38.label_cres.tsv \n"
     ]
    }
   ],
   "source": [
    "txt_folder = \"genome_cres\"\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "\n",
    "vec = dir(txt_fdiry)\n",
    "for (txt in vec){cat(txt, \"\\n\")}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "41d3bb35-de8a-4557-9c03-ca722c950ae1",
   "metadata": {},
   "source": [
    "## Import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9d53dba9-6f06-4f64-b94f-f1b066447c48",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 27 11\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Chrom </th>\n",
       "   <th style=\"text-align:right;\"> ChromStart </th>\n",
       "   <th style=\"text-align:right;\"> ChromEnd </th>\n",
       "   <th style=\"text-align:left;\"> Region </th>\n",
       "   <th style=\"text-align:left;\"> Name </th>\n",
       "   <th style=\"text-align:left;\"> Type </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Target_TSS </th>\n",
       "   <th style=\"text-align:left;\"> Description </th>\n",
       "   <th style=\"text-align:left;\"> Reference </th>\n",
       "   <th style=\"text-align:left;\"> Note </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chrX </td>\n",
       "   <td style=\"text-align:right;\"> 48785794 </td>\n",
       "   <td style=\"text-align:right;\"> 48786722 </td>\n",
       "   <td style=\"text-align:left;\"> chrX:48,785,794-48,786,722 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 promoter </td>\n",
       "   <td style=\"text-align:left;\"> promoter </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1|chrX:48786589-48786590 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 promoter </td>\n",
       "   <td style=\"text-align:left;\"> Fulco et al., 2016; Reilly et al., 2021 </td>\n",
       "   <td style=\"text-align:left;\"> known CREs </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chrX </td>\n",
       "   <td style=\"text-align:right;\"> 48782614 </td>\n",
       "   <td style=\"text-align:right;\"> 48783539 </td>\n",
       "   <td style=\"text-align:left;\"> chrX:48,782,614-48,783,539 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 enhancer </td>\n",
       "   <td style=\"text-align:left;\"> enhancer </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1|chrX:48786589-48786590 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 Hematopoietic Enhancer </td>\n",
       "   <td style=\"text-align:left;\"> Fulco et al., 2016; Reilly et al., 2021 </td>\n",
       "   <td style=\"text-align:left;\"> known CREs </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chrX </td>\n",
       "   <td style=\"text-align:right;\"> 48801698 </td>\n",
       "   <td style=\"text-align:right;\"> 48802997 </td>\n",
       "   <td style=\"text-align:left;\"> chrX:48,801,698-48,802,997 </td>\n",
       "   <td style=\"text-align:left;\"> HDAC6 promoter/enhancer </td>\n",
       "   <td style=\"text-align:left;\"> promoter,enhancer </td>\n",
       "   <td style=\"text-align:left;\"> HDAC6 </td>\n",
       "   <td style=\"text-align:left;\"> HDAC6|chrX:48802066-48802067 </td>\n",
       "   <td style=\"text-align:left;\"> HDAC6 promoter and enhancer </td>\n",
       "   <td style=\"text-align:left;\"> Fulco et al., 2016; Reilly et al., 2021 </td>\n",
       "   <td style=\"text-align:left;\"> known CREs </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr8 </td>\n",
       "   <td style=\"text-align:right;\"> 127735050 </td>\n",
       "   <td style=\"text-align:right;\"> 127736592 </td>\n",
       "   <td style=\"text-align:left;\"> chr8:127,735,050-127,736,592 </td>\n",
       "   <td style=\"text-align:left;\"> MYC promoter </td>\n",
       "   <td style=\"text-align:left;\"> promoter </td>\n",
       "   <td style=\"text-align:left;\"> MYC </td>\n",
       "   <td style=\"text-align:left;\"> MYC|chr8:127736230-127736231 </td>\n",
       "   <td style=\"text-align:left;\"> MYC promoter </td>\n",
       "   <td style=\"text-align:left;\"> Fulco et al., 2016; Reilly et al., 2021 </td>\n",
       "   <td style=\"text-align:left;\"> known CREs </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr8 </td>\n",
       "   <td style=\"text-align:right;\"> 127898376 </td>\n",
       "   <td style=\"text-align:right;\"> 127899697 </td>\n",
       "   <td style=\"text-align:left;\"> chr8:127,898,376-127,899,697 </td>\n",
       "   <td style=\"text-align:left;\"> MYC E1 </td>\n",
       "   <td style=\"text-align:left;\"> enhancer </td>\n",
       "   <td style=\"text-align:left;\"> MYC </td>\n",
       "   <td style=\"text-align:left;\"> MYC|chr8:127736230-127736231 </td>\n",
       "   <td style=\"text-align:left;\"> intragenic enhancer within PVT1 </td>\n",
       "   <td style=\"text-align:left;\"> Fulco et al., 2016; Reilly et al., 2021 </td>\n",
       "   <td style=\"text-align:left;\"> known CREs </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr8 </td>\n",
       "   <td style=\"text-align:right;\"> 127959692 </td>\n",
       "   <td style=\"text-align:right;\"> 127960934 </td>\n",
       "   <td style=\"text-align:left;\"> chr8:127,959,692-127,960,934 </td>\n",
       "   <td style=\"text-align:left;\"> MYC E2 </td>\n",
       "   <td style=\"text-align:left;\"> enhancer </td>\n",
       "   <td style=\"text-align:left;\"> MYC </td>\n",
       "   <td style=\"text-align:left;\"> MYC|chr8:127736230-127736231 </td>\n",
       "   <td style=\"text-align:left;\"> intragenic enhancer within PVT1 </td>\n",
       "   <td style=\"text-align:left;\"> Fulco et al., 2016; Reilly et al., 2021 </td>\n",
       "   <td style=\"text-align:left;\"> known CREs </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_folder = TXT_REGION_FOLDER\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "txt_fname = \"K562.hg38.label_cres.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### assign and show\n",
    "dat_region_cres_import = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "78d88017-b3e2-4d68-a0a7-958bf26a4651",
   "metadata": {},
   "source": [
    "## Arrange table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "619a77fa-a78d-405d-9866-c3ea295a46a8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 27 13\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Chrom </th>\n",
       "   <th style=\"text-align:right;\"> ChromStart </th>\n",
       "   <th style=\"text-align:right;\"> ChromEnd </th>\n",
       "   <th style=\"text-align:left;\"> Region </th>\n",
       "   <th style=\"text-align:left;\"> Name </th>\n",
       "   <th style=\"text-align:left;\"> Type </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Target_TSS </th>\n",
       "   <th style=\"text-align:left;\"> Description </th>\n",
       "   <th style=\"text-align:left;\"> Reference </th>\n",
       "   <th style=\"text-align:left;\"> Note </th>\n",
       "   <th style=\"text-align:left;\"> Group </th>\n",
       "   <th style=\"text-align:left;\"> Label </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chrX </td>\n",
       "   <td style=\"text-align:right;\"> 48785794 </td>\n",
       "   <td style=\"text-align:right;\"> 48786722 </td>\n",
       "   <td style=\"text-align:left;\"> chrX:48785794-48786722 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 promoter </td>\n",
       "   <td style=\"text-align:left;\"> promoter </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1|chrX:48786589-48786590 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 promoter </td>\n",
       "   <td style=\"text-align:left;\"> Fulco et al., 2016; Reilly et al., 2021 </td>\n",
       "   <td style=\"text-align:left;\"> known CREs </td>\n",
       "   <td style=\"text-align:left;\"> genome_cres </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 promoter </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chrX </td>\n",
       "   <td style=\"text-align:right;\"> 48782614 </td>\n",
       "   <td style=\"text-align:right;\"> 48783539 </td>\n",
       "   <td style=\"text-align:left;\"> chrX:48782614-48783539 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 enhancer </td>\n",
       "   <td style=\"text-align:left;\"> enhancer </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1|chrX:48786589-48786590 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 Hematopoietic Enhancer </td>\n",
       "   <td style=\"text-align:left;\"> Fulco et al., 2016; Reilly et al., 2021 </td>\n",
       "   <td style=\"text-align:left;\"> known CREs </td>\n",
       "   <td style=\"text-align:left;\"> genome_cres </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 enhancer </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chrX </td>\n",
       "   <td style=\"text-align:right;\"> 48801698 </td>\n",
       "   <td style=\"text-align:right;\"> 48802997 </td>\n",
       "   <td style=\"text-align:left;\"> chrX:48801698-48802997 </td>\n",
       "   <td style=\"text-align:left;\"> HDAC6 promoter/enhancer </td>\n",
       "   <td style=\"text-align:left;\"> promoter,enhancer </td>\n",
       "   <td style=\"text-align:left;\"> HDAC6 </td>\n",
       "   <td style=\"text-align:left;\"> HDAC6|chrX:48802066-48802067 </td>\n",
       "   <td style=\"text-align:left;\"> HDAC6 promoter and enhancer </td>\n",
       "   <td style=\"text-align:left;\"> Fulco et al., 2016; Reilly et al., 2021 </td>\n",
       "   <td style=\"text-align:left;\"> known CREs </td>\n",
       "   <td style=\"text-align:left;\"> genome_cres </td>\n",
       "   <td style=\"text-align:left;\"> HDAC6 promoter/enhancer </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr8 </td>\n",
       "   <td style=\"text-align:right;\"> 127735050 </td>\n",
       "   <td style=\"text-align:right;\"> 127736592 </td>\n",
       "   <td style=\"text-align:left;\"> chr8:127735050-127736592 </td>\n",
       "   <td style=\"text-align:left;\"> MYC promoter </td>\n",
       "   <td style=\"text-align:left;\"> promoter </td>\n",
       "   <td style=\"text-align:left;\"> MYC </td>\n",
       "   <td style=\"text-align:left;\"> MYC|chr8:127736230-127736231 </td>\n",
       "   <td style=\"text-align:left;\"> MYC promoter </td>\n",
       "   <td style=\"text-align:left;\"> Fulco et al., 2016; Reilly et al., 2021 </td>\n",
       "   <td style=\"text-align:left;\"> known CREs </td>\n",
       "   <td style=\"text-align:left;\"> genome_cres </td>\n",
       "   <td style=\"text-align:left;\"> MYC promoter </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr8 </td>\n",
       "   <td style=\"text-align:right;\"> 127898376 </td>\n",
       "   <td style=\"text-align:right;\"> 127899697 </td>\n",
       "   <td style=\"text-align:left;\"> chr8:127898376-127899697 </td>\n",
       "   <td style=\"text-align:left;\"> MYC E1 </td>\n",
       "   <td style=\"text-align:left;\"> enhancer </td>\n",
       "   <td style=\"text-align:left;\"> MYC </td>\n",
       "   <td style=\"text-align:left;\"> MYC|chr8:127736230-127736231 </td>\n",
       "   <td style=\"text-align:left;\"> intragenic enhancer within PVT1 </td>\n",
       "   <td style=\"text-align:left;\"> Fulco et al., 2016; Reilly et al., 2021 </td>\n",
       "   <td style=\"text-align:left;\"> known CREs </td>\n",
       "   <td style=\"text-align:left;\"> genome_cres </td>\n",
       "   <td style=\"text-align:left;\"> MYC E1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr8 </td>\n",
       "   <td style=\"text-align:right;\"> 127959692 </td>\n",
       "   <td style=\"text-align:right;\"> 127960934 </td>\n",
       "   <td style=\"text-align:left;\"> chr8:127959692-127960934 </td>\n",
       "   <td style=\"text-align:left;\"> MYC E2 </td>\n",
       "   <td style=\"text-align:left;\"> enhancer </td>\n",
       "   <td style=\"text-align:left;\"> MYC </td>\n",
       "   <td style=\"text-align:left;\"> MYC|chr8:127736230-127736231 </td>\n",
       "   <td style=\"text-align:left;\"> intragenic enhancer within PVT1 </td>\n",
       "   <td style=\"text-align:left;\"> Fulco et al., 2016; Reilly et al., 2021 </td>\n",
       "   <td style=\"text-align:left;\"> known CREs </td>\n",
       "   <td style=\"text-align:left;\"> genome_cres </td>\n",
       "   <td style=\"text-align:left;\"> MYC E2 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### init\n",
    "dat = dat_region_cres_import\n",
    "\n",
    "### arrange location column type and region column\n",
    "dat = dat %>%\n",
    "    dplyr::mutate(\n",
    "        Chrom      = as.character(Chrom),\n",
    "        ChromStart = as.integer(ChromStart),\n",
    "        ChromEnd   = as.integer(ChromEnd)\n",
    "    ) %>%\n",
    "    dplyr::mutate(Region = fun_gen_region(Chrom, ChromStart, ChromEnd))\n",
    "\n",
    "### add group and label\n",
    "dat = dat %>%\n",
    "    dplyr::mutate(\n",
    "        Group = \"genome_cres\",\n",
    "        Label = Name\n",
    "    )\n",
    "\n",
    "### assign and show\n",
    "dat_region_cres_arrange = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eb693105-8d24-4dcb-b389-77b95eb4b17c",
   "metadata": {},
   "source": [
    "## Define column description"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a9ad1f52-03b6-4917-b0c6-22cc45ac080b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Name </th>\n",
       "   <th style=\"text-align:left;\"> Note </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Chrom </td>\n",
       "   <td style=\"text-align:left;\"> Name of the chromosome </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ChromStart </td>\n",
       "   <td style=\"text-align:left;\"> The starting position of the feature in the chromosome </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ChromEnd </td>\n",
       "   <td style=\"text-align:left;\"> The ending position of the feature in the chromosome </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Region </td>\n",
       "   <td style=\"text-align:left;\"> Region location </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Name </td>\n",
       "   <td style=\"text-align:left;\"> Name given to a region; Use '.' if no name is assigned. </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Type </td>\n",
       "   <td style=\"text-align:left;\"> promoter, enhancer, or silencer </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Target </td>\n",
       "   <td style=\"text-align:left;\"> Gene </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Target_TSS </td>\n",
       "   <td style=\"text-align:left;\"> TSS of the Target used in this study </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Description </td>\n",
       "   <td style=\"text-align:left;\"> Summary of the CREs annotated </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Reference </td>\n",
       "   <td style=\"text-align:left;\"> Previous studies cited </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Note </td>\n",
       "   <td style=\"text-align:left;\"> Note </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Group </td>\n",
       "   <td style=\"text-align:left;\"> Region group </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Label </td>\n",
       "   <td style=\"text-align:left;\"> Region label </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### setup column description\n",
    "dat = tribble(\n",
    "    ~Name,              ~Note,\n",
    "    \"Chrom\",            \"Name of the chromosome\",\n",
    "    \"ChromStart\",       \"The starting position of the feature in the chromosome\",\n",
    "    \"ChromEnd\",         \"The ending position of the feature in the chromosome\",\n",
    "    \"Region\",           \"Region location\",\n",
    "    \"Name\",             \"Name given to a region; Use '.' if no name is assigned.\",\n",
    "    \"Type\",             \"promoter, enhancer, or silencer\",\n",
    "    \"Target\",           \"Gene\",\n",
    "    \"Target_TSS\",       \"TSS of the Target used in this study\",\n",
    "    \"Description\",      \"Summary of the CREs annotated\",\n",
    "    \"Reference\",        \"Previous studies cited\",\n",
    "    \"Note\",             \"Note\",\n",
    "    \"Group\",            \"Region group\",\n",
    "    \"Label\",            \"Region label\"\n",
    ")\n",
    "\n",
    "### assign and show\n",
    "dat_cname = dat\n",
    "fun_display_table(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "165c9653-d335-4d09-a367-bf0ab7f5a201",
   "metadata": {},
   "source": [
    "## Export"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7cf4b99b-476a-4bae-9880-62c58e42eb1e",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set directory\n",
    "txt_folder = TXT_REGION_FOLDER\n",
    "txt_fdiry  = file.path(FD_RES, \"region\", txt_folder)\n",
    "txt_cmd    = paste(\"mkdir -p\", txt_fdiry)\n",
    "system(txt_cmd)\n",
    "\n",
    "### write table\n",
    "txt_fname = \"K562.hg38.label_cres.bed.gz\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "dat = dat_region_cres_arrange\n",
    "write_tsv(dat, txt_fpath, col_names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4b72ac6a-6193-4308-9926-881d581e399c",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set directory\n",
    "txt_folder = TXT_REGION_FOLDER\n",
    "txt_fdiry  = file.path(FD_RES, \"region\", txt_folder, \"summary\")\n",
    "dir.create(txt_fdiry, showWarnings = FALSE)\n",
    "\n",
    "### write table\n",
    "txt_fname = \"K562.hg38.label_cres.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "dat = dat_region_cres_arrange\n",
    "write_tsv(dat, txt_fpath, col_names = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a6b9560c-fef6-401e-a8c0-0b0e562a0276",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set directory\n",
    "txt_folder = TXT_REGION_FOLDER\n",
    "txt_fdiry  = file.path(FD_RES, \"region\", txt_folder, \"summary\")\n",
    "dir.create(txt_fdiry, showWarnings = FALSE)\n",
    "\n",
    "### write table\n",
    "txt_fname  = \"description.tsv\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "dat = dat_cname\n",
    "write_tsv(dat, txt_fpath)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c2d2455-d3d9-4c38-99c0-00df35c10014",
   "metadata": {},
   "source": [
    "## Check: Target TSS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "21298f89-8cf9-4df9-8a49-8cb0d2c313c9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 29330     8\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Chrom </th>\n",
       "   <th style=\"text-align:right;\"> ChromStart </th>\n",
       "   <th style=\"text-align:right;\"> ChromEnd </th>\n",
       "   <th style=\"text-align:left;\"> Region </th>\n",
       "   <th style=\"text-align:left;\"> Gene </th>\n",
       "   <th style=\"text-align:right;\"> Score </th>\n",
       "   <th style=\"text-align:left;\"> Group </th>\n",
       "   <th style=\"text-align:left;\"> Label </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:right;\"> 11873 </td>\n",
       "   <td style=\"text-align:right;\"> 11874 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:11873-11874 </td>\n",
       "   <td style=\"text-align:left;\"> DDX11L1 </td>\n",
       "   <td style=\"text-align:right;\"> 0.00023 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> DDX11L1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:right;\"> 17436 </td>\n",
       "   <td style=\"text-align:right;\"> 17437 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:17436-17437 </td>\n",
       "   <td style=\"text-align:left;\"> MIR6859-1 </td>\n",
       "   <td style=\"text-align:right;\"> 9.43812 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> MIR6859-1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:right;\"> 17436 </td>\n",
       "   <td style=\"text-align:right;\"> 17437 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:17436-17437 </td>\n",
       "   <td style=\"text-align:left;\"> MIR6859-2 </td>\n",
       "   <td style=\"text-align:right;\"> 9.43812 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> MIR6859-2 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:right;\"> 17436 </td>\n",
       "   <td style=\"text-align:right;\"> 17437 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:17436-17437 </td>\n",
       "   <td style=\"text-align:left;\"> MIR6859-3 </td>\n",
       "   <td style=\"text-align:right;\"> 9.43812 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> MIR6859-3 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:right;\"> 17436 </td>\n",
       "   <td style=\"text-align:right;\"> 17437 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:17436-17437 </td>\n",
       "   <td style=\"text-align:left;\"> MIR6859-4 </td>\n",
       "   <td style=\"text-align:right;\"> 9.43812 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> MIR6859-4 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:right;\"> 29370 </td>\n",
       "   <td style=\"text-align:right;\"> 29371 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:29370-29371 </td>\n",
       "   <td style=\"text-align:left;\"> WASH7P </td>\n",
       "   <td style=\"text-align:right;\"> 0.00023 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> WASH7P </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_folder = \"genome_tss\"\n",
    "txt_fdiry  = file.path(FD_RES, \"region\", txt_folder, \"summary\")\n",
    "txt_fname = \"K562.hg38.TSS.selected_by_highest_Pol2_signal.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### assign and show\n",
    "dat_region_tss = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0fbe9fe9-588b-40be-ae53-0321bfbd697b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Chrom </th>\n",
       "   <th style=\"text-align:right;\"> ChromStart </th>\n",
       "   <th style=\"text-align:right;\"> ChromEnd </th>\n",
       "   <th style=\"text-align:left;\"> Region </th>\n",
       "   <th style=\"text-align:left;\"> Gene </th>\n",
       "   <th style=\"text-align:right;\"> Score </th>\n",
       "   <th style=\"text-align:left;\"> Group </th>\n",
       "   <th style=\"text-align:left;\"> Label </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chrX </td>\n",
       "   <td style=\"text-align:right;\"> 48786589 </td>\n",
       "   <td style=\"text-align:right;\"> 48786590 </td>\n",
       "   <td style=\"text-align:left;\"> chrX:48786589-48786590 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 </td>\n",
       "   <td style=\"text-align:right;\"> 70.420800 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> GATA1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chrX </td>\n",
       "   <td style=\"text-align:right;\"> 48802066 </td>\n",
       "   <td style=\"text-align:right;\"> 48802067 </td>\n",
       "   <td style=\"text-align:left;\"> chrX:48802066-48802067 </td>\n",
       "   <td style=\"text-align:left;\"> HDAC6 </td>\n",
       "   <td style=\"text-align:right;\"> 183.300000 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> HDAC6 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr8 </td>\n",
       "   <td style=\"text-align:right;\"> 127736230 </td>\n",
       "   <td style=\"text-align:right;\"> 127736231 </td>\n",
       "   <td style=\"text-align:left;\"> chr8:127736230-127736231 </td>\n",
       "   <td style=\"text-align:left;\"> MYC </td>\n",
       "   <td style=\"text-align:right;\"> 272.781000 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> MYC </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr11 </td>\n",
       "   <td style=\"text-align:right;\"> 5269945 </td>\n",
       "   <td style=\"text-align:right;\"> 5269946 </td>\n",
       "   <td style=\"text-align:left;\"> chr11:5269945-5269946 </td>\n",
       "   <td style=\"text-align:left;\"> HBE1 </td>\n",
       "   <td style=\"text-align:right;\"> 319.925000 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> HBE1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr11 </td>\n",
       "   <td style=\"text-align:right;\"> 61817003 </td>\n",
       "   <td style=\"text-align:right;\"> 61817004 </td>\n",
       "   <td style=\"text-align:left;\"> chr11:61817003-61817004 </td>\n",
       "   <td style=\"text-align:left;\"> FADS1 </td>\n",
       "   <td style=\"text-align:right;\"> 158.661000 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> FADS1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr11 </td>\n",
       "   <td style=\"text-align:right;\"> 61816202 </td>\n",
       "   <td style=\"text-align:right;\"> 61816203 </td>\n",
       "   <td style=\"text-align:left;\"> chr11:61816202-61816203 </td>\n",
       "   <td style=\"text-align:left;\"> FADS2 </td>\n",
       "   <td style=\"text-align:right;\"> 418.267000 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> FADS2 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr11 </td>\n",
       "   <td style=\"text-align:right;\"> 61891545 </td>\n",
       "   <td style=\"text-align:right;\"> 61891546 </td>\n",
       "   <td style=\"text-align:left;\"> chr11:61891545-61891546 </td>\n",
       "   <td style=\"text-align:left;\"> FADS3 </td>\n",
       "   <td style=\"text-align:right;\"> 8.249710 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> FADS3 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr11 </td>\n",
       "   <td style=\"text-align:right;\"> 33869878 </td>\n",
       "   <td style=\"text-align:right;\"> 33869879 </td>\n",
       "   <td style=\"text-align:left;\"> chr11:33869878-33869879 </td>\n",
       "   <td style=\"text-align:left;\"> LMO2 </td>\n",
       "   <td style=\"text-align:right;\"> 8.552780 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> LMO2 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr6 </td>\n",
       "   <td style=\"text-align:right;\"> 135181307 </td>\n",
       "   <td style=\"text-align:right;\"> 135181308 </td>\n",
       "   <td style=\"text-align:left;\"> chr6:135181307-135181308 </td>\n",
       "   <td style=\"text-align:left;\"> MYB </td>\n",
       "   <td style=\"text-align:right;\"> 15.101300 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> MYB </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chrX </td>\n",
       "   <td style=\"text-align:right;\"> 47619857 </td>\n",
       "   <td style=\"text-align:right;\"> 47619858 </td>\n",
       "   <td style=\"text-align:left;\"> chrX:47619857-47619858 </td>\n",
       "   <td style=\"text-align:left;\"> SYN1 </td>\n",
       "   <td style=\"text-align:right;\"> 0.627251 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> SYN1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr11 </td>\n",
       "   <td style=\"text-align:right;\"> 113475398 </td>\n",
       "   <td style=\"text-align:right;\"> 113475399 </td>\n",
       "   <td style=\"text-align:left;\"> chr11:113475398-113475399 </td>\n",
       "   <td style=\"text-align:left;\"> DRD2 </td>\n",
       "   <td style=\"text-align:right;\"> 0.024114 </td>\n",
       "   <td style=\"text-align:left;\"> TSS_Pol2 </td>\n",
       "   <td style=\"text-align:left;\"> DRD2 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "vec_txt_target = c(\n",
    "    \"GATA1\", \"HDAC6\", \n",
    "    \"MYC\",   \"HBE1\",\n",
    "    \"FADS1\", \"FADS2\", \"FADS3\", \n",
    "    \"LMO2\", \n",
    "    \"MYB\", \n",
    "    \"SYN1\", \"DRD2\"\n",
    ")\n",
    "dat = dat_region_tss\n",
    "dat = dat %>% \n",
    "    dplyr::filter(Gene %in% vec_txt_target) %>%\n",
    "    dplyr::mutate(Gene = factor(Gene, levels = vec_txt_target)) %>%\n",
    "    dplyr::arrange(Gene)\n",
    "\n",
    "fun_display_table(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f36fd9de-b199-4b07-bda5-1a9133d332fc",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.5.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
