{
 "cells": [
  {
   "cell_type": "raw",
   "id": "d9f2bb56-76ce-4e83-aa77-3c5b63b13f82",
   "metadata": {},
   "source": [
    "---\n",
    "title: Prepare ENCODE ChIP-seq 01 (Histone)\n",
    "subtitle: Generate download script to download the data\n",
    "format:\n",
    "    html:\n",
    "        code-fold:  true\n",
    "        code-tools: true\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "711b06d8-4794-4760-a8ad-9a659cf804c1",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f8f94dc3-1bd7-4908-af93-e161dc5c51fb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are working on        Singularity: singularity_proj_encode_fcc \n",
      "BASE DIRECTORY (FD_BASE): /data/reddylab/Kuei \n",
      "REPO DIRECTORY (FD_REPO): /data/reddylab/Kuei/repo \n",
      "WORK DIRECTORY (FD_WORK): /data/reddylab/Kuei/work \n",
      "DATA DIRECTORY (FD_DATA): /data/reddylab/Kuei/data \n",
      "\n",
      "You are working with      ENCODE FCC \n",
      "PATH OF PROJECT (FD_PRJ): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC \n",
      "PROJECT RESULTS (FD_RES): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results \n",
      "PROJECT SCRIPTS (FD_EXE): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/scripts \n",
      "PROJECT DATA    (FD_DAT): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/data \n",
      "PROJECT NOTE    (FD_NBK): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/notebooks \n",
      "PROJECT DOCS    (FD_DOC): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/docs \n",
      "PROJECT LOG     (FD_LOG): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/log \n",
      "PROJECT REF     (FD_REF): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/references \n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(source(\"../run_config_project_sing.R\")))\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c526c738-aac6-47b0-9c71-617fb40fe3a1",
   "metadata": {},
   "source": [
    "**Set global variables**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "935c1b13-61f2-4ecb-aadc-fcc00a8f3d18",
   "metadata": {},
   "outputs": [],
   "source": [
    "TXT_FOLDER_INP = \"encode_chipseq_histone\"\n",
    "TXT_FOLDER_OUT = \"encode_chipseq_histone_250120\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "80c86549-ffa9-46c8-a0bd-334e12972bdf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "files.chipseq_histone.default_analysis.250120.txt \n",
      "files.chipseq_histone.default_files.250120.txt \n",
      "metadata.chipseq_histone.default_analysis.250120.tsv \n",
      "metadata.chipseq_histone.default_files.250120.tsv \n"
     ]
    }
   ],
   "source": [
    "txt_folder = TXT_FOLDER_INP\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "\n",
    "vec = dir(txt_fdiry)\n",
    "for (txt in vec){cat(txt, \"\\n\")}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6810a004-26ef-4e3c-b757-cf6f63ec73c3",
   "metadata": {},
   "source": [
    "## Import metadata from reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ff23d7c3-53c9-4d5e-a137-9bc82ff8daee",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 57 59\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF767UON </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKU </td>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me3-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2011-02-10 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> see document </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1, 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF420ARY/, /files/ENCFF666YXZ/, /files/ENCFF564SVK/, /files/ENCFF685PPQ/, /files/ENCFF226FKB/ </td>\n",
       "   <td style=\"text-align:right;\"> 797629786 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> 4c102d45be8326062895ed0a03d4ded7 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF767UON/@@download/ENCFF767UON.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/09/30/ed53d64d-7a8c-4423-a9cd-7c5d80e80b20/ENCFF767UON.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/09/30/ed53d64d-7a8c-4423-a9cd-7c5d80e80b20/ENCFF767UON.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> low read depth, control low read depth, mild to moderate bottlenecking, inconsistent platforms </td>\n",
       "   <td style=\"text-align:left;\"> insufficient read depth </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF689QIJ </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> pseudoreplicated peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKU </td>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me3-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2011-02-10 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> see document </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1, 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF420ARY/, /files/ENCFF666YXZ/, /files/ENCFF564SVK/, /files/ENCFF685PPQ/, /files/ENCFF356LFX/, /files/ENCFF226FKB/ </td>\n",
       "   <td style=\"text-align:right;\"> 1111370 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> 5dea2993c0831ae344a989d601c09178 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/09/30/e8359d31-b2ab-44d2-8443-75a04a57723e/ENCFF689QIJ.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/09/30/e8359d31-b2ab-44d2-8443-75a04a57723e/ENCFF689QIJ.bed.gz?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> low read depth, control low read depth, mild to moderate bottlenecking, inconsistent platforms </td>\n",
       "   <td style=\"text-align:left;\"> insufficient read depth </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF217BRB </td>\n",
       "   <td style=\"text-align:left;\"> bigBed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bigBed </td>\n",
       "   <td style=\"text-align:left;\"> narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> pseudoreplicated peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKU </td>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me3-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2011-02-10 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> see document </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1, 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF689QIJ/ </td>\n",
       "   <td style=\"text-align:right;\"> 2416308 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> 9cea2c6388211c15a9dd800ed86b8bab </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF217BRB/@@download/ENCFF217BRB.bigBed </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/09/30/c480ed4a-14f0-4188-be39-adb818e5c5fa/ENCFF217BRB.bigBed </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/09/30/c480ed4a-14f0-4188-be39-adb818e5c5fa/ENCFF217BRB.bigBed?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> low read depth, control low read depth, mild to moderate bottlenecking, inconsistent platforms </td>\n",
       "   <td style=\"text-align:left;\"> insufficient read depth </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_folder = TXT_FOLDER_INP\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "txt_fname  = \"metadata.chipseq_histone.default_files.250120.tsv\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### show and assign\n",
    "dat_metadata_chipseq_v1 = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "11e6b061-22ac-4def-b721-c886a806e3ee",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 380  59\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:right;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF677CFG </td>\n",
       "   <td style=\"text-align:left;\"> bam </td>\n",
       "   <td style=\"text-align:left;\"> bam </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> unfiltered alignments </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000APD </td>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> H3K79me2-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2011-02-10 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> see document </td>\n",
       "   <td style=\"text-align:left;\"> 200-600 </td>\n",
       "   <td style=\"text-align:left;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:right;\"> 36 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF110MCL/, /files/ENCFF000BYO/ </td>\n",
       "   <td style=\"text-align:right;\"> 1454235591 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> 4376f4b6f7bc508b2359556756403670 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF677CFG/@@download/ENCFF677CFG.bam </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2022/06/16/827755a0-0780-44b9-b857-46fad0a4df36/ENCFF677CFG.bam </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2022/06/16/827755a0-0780-44b9-b857-46fad0a4df36/ENCFF677CFG.bam?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> low read length, inconsistent platforms </td>\n",
       "   <td style=\"text-align:left;\"> control insufficient read depth, insufficient read depth </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF112LWK </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000APD </td>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> H3K79me2-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2011-02-10 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> see document </td>\n",
       "   <td style=\"text-align:left;\"> 200-600 </td>\n",
       "   <td style=\"text-align:left;\"> 3 </td>\n",
       "   <td style=\"text-align:left;\"> 3_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:right;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF132YSI/, /files/ENCFF226FKB/, /files/ENCFF666YXZ/, /files/ENCFF420ARY/ </td>\n",
       "   <td style=\"text-align:right;\"> 484259878 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> 106649b0439fa81bd50cbddcbb0e4736 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF112LWK/@@download/ENCFF112LWK.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2022/06/16/e05ed78a-b5e7-490a-98ae-ba7da1de26b1/ENCFF112LWK.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2022/06/16/e05ed78a-b5e7-490a-98ae-ba7da1de26b1/ENCFF112LWK.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> low read length, inconsistent platforms </td>\n",
       "   <td style=\"text-align:left;\"> control insufficient read depth, insufficient read depth </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF544AVW </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000APD </td>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> H3K79me2-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2011-02-10 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> see document </td>\n",
       "   <td style=\"text-align:left;\"> 200-600 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2, 3 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1, 2_1, 3_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:right;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF711PLM/, /files/ENCFF132YSI/, /files/ENCFF649PBO/, /files/ENCFF226FKB/, /files/ENCFF666YXZ/, /files/ENCFF420ARY/ </td>\n",
       "   <td style=\"text-align:right;\"> 927611153 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> 61dc50179ae8d880b972c3697a6a2fc2 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF544AVW/@@download/ENCFF544AVW.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2022/06/16/b460dd47-f161-4ac0-9456-1c7dbd66ad54/ENCFF544AVW.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2022/06/16/b460dd47-f161-4ac0-9456-1c7dbd66ad54/ENCFF544AVW.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> low read length, inconsistent platforms </td>\n",
       "   <td style=\"text-align:left;\"> control insufficient read depth, insufficient read depth </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_folder = TXT_FOLDER_INP\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "txt_fname  = \"metadata.chipseq_histone.default_analysis.250120.tsv\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### show and assign\n",
    "dat_metadata_chipseq_v2 = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9f3c6a89-3979-4ee5-a5fb-9dd7fb2ec75b",
   "metadata": {},
   "source": [
    "## Explore"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edf5286d-4a43-4a86-a2e8-09d62033e5e0",
   "metadata": {},
   "source": [
    "**File type**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2f4edf37-9d89-46c8-a6c7-9e828d1a944f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                  \n",
       "                   bed bigBed bigWig\n",
       "  Histone ChIP-seq  19     19     19"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "table(dat$Assay, dat$`File type`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "92542ed8-a389-4c93-8467-6900b01e5bdf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                  \n",
       "                   bam bed bigBed bigWig\n",
       "  Histone ChIP-seq  84  88     88    120"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "table(dat$Assay, dat$`File type`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "982716f4-4bac-4878-9e7b-2f16f730dddb",
   "metadata": {},
   "source": [
    "**Output type**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6982729d-c02d-4858-8d64-78408a331bd4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        \n",
       "         pseudoreplicated peaks replicated peaks signal p-value\n",
       "  bed                        14                5              0\n",
       "  bigBed                     14                5              0\n",
       "  bigWig                      0                0             19"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "table(dat$`File type`, dat$`Output type`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "66ea695b-dfc6-41b1-b827-e7452057c170",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        \n",
       "         alignments fold change over control pseudoreplicated peaks\n",
       "  bam            42                        0                      0\n",
       "  bed             0                        0                     60\n",
       "  bigBed          0                        0                     60\n",
       "  bigWig          0                       60                      0\n",
       "        \n",
       "         replicated peaks signal p-value unfiltered alignments\n",
       "  bam                   0              0                    42\n",
       "  bed                  28              0                     0\n",
       "  bigBed               28              0                     0\n",
       "  bigWig                0             60                     0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "table(dat$`File type`, dat$`Output type`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f917a76-8aea-4b83-8192-f9365e9061ea",
   "metadata": {},
   "source": [
    "**Genome assembly**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "e0ae9b8e-2108-4f1e-8ab4-d9541f95d4d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "GRCh38 \n",
       "    57 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "table(dat$`File assembly`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0eb6037b-0de4-4500-95a6-439a4fcb3fae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "GRCh38 \n",
       "   380 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "table(dat$`File assembly`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce158742-e254-4499-8d1f-29667b766a61",
   "metadata": {},
   "source": [
    "**Biosample**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2e732e0d-2c03-438f-b364-23935ea9519f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "K562 \n",
       "  57 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "table(dat$`Biosample term name`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3d91472d-55d6-4c0f-a16d-a20f1265bb82",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "K562 \n",
       " 380 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "table(dat$`Biosample term name`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "042693bc-e98f-40ea-a75e-c742d84a52c8",
   "metadata": {},
   "source": [
    "**Target**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "c79e3b13-5816-49c6-9e4b-fa22013157e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "   H2AFZ-human  H3K27ac-human H3K27me3-human H3K36me3-human  H3K4me1-human \n",
       "             3              3              6              6              6 \n",
       " H3K4me2-human  H3K4me3-human H3K79me2-human   H3K9ac-human  H3K9me1-human \n",
       "             3             12              3              6              3 \n",
       " H3K9me3-human H4K20me1-human \n",
       "             3              3 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "table(dat$`Experiment target`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "3266016f-9101-45ce-b18e-ccfd538247e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "   H2AFZ-human  H3K27ac-human H3K27me3-human H3K36me3-human  H3K4me1-human \n",
       "            18             28             46             46             46 \n",
       " H3K4me2-human  H3K4me3-human H3K79me2-human   H3K9ac-human  H3K9me1-human \n",
       "            18             72             28             36              6 \n",
       " H3K9me3-human H4K20me1-human \n",
       "            18             18 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "table(dat$`Experiment target`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2e59d54e-9fbe-449e-a11e-4f59a04aeeff",
   "metadata": {},
   "source": [
    "## Arrange metadata tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "5c8a414d-70e3-4ea1-aeb7-1bc6f1ec7a0f",
   "metadata": {},
   "outputs": [],
   "source": [
    "fun = function(dat_input){\n",
    "    ### init\n",
    "    dat = dat_input\n",
    "    \n",
    "    ### rename some columns\n",
    "    dat = dat %>% \n",
    "        dplyr::mutate(\n",
    "            Index_Experiment = `Experiment accession`,\n",
    "            Index_File       = `File accession`,\n",
    "            File_Format      = `File format`,\n",
    "            File_Type        = `File type`,\n",
    "            Output_Type      = `Output type`,\n",
    "            Genome           = `File assembly`,\n",
    "            Target           = str_remove(`Experiment target`, \"-human\"),\n",
    "            Bio_Replicates   = `Biological replicate(s)`,\n",
    "            Analysis         = `File analysis title`,\n",
    "            File_Name\t     = basename(`File download URL`),\n",
    "            File_URL         = `File download URL`\n",
    "        )\n",
    "    \n",
    "    ### select the needed columns\n",
    "    dat = dat %>%\n",
    "        dplyr::select(\n",
    "            Assay,\n",
    "            Index_Experiment,\n",
    "            Index_File,\n",
    "            File_Format,\n",
    "            File_Type,\n",
    "            Output_Type,\n",
    "            Genome,\n",
    "            Target,\n",
    "            Bio_Replicates,\n",
    "            Analysis,\n",
    "            md5sum,\n",
    "            File_Name,\n",
    "            File_URL\n",
    "        )\n",
    "\n",
    "    ### return the simplified table\n",
    "    return(dat)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "40f13eb2-3a08-4a9e-b0a1-76891154cd83",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 57 13\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF767UON </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me3 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 4c102d45be8326062895ed0a03d4ded7 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF767UON.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF767UON/@@download/ENCFF767UON.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF689QIJ </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> pseudoreplicated peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me3 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 5dea2993c0831ae344a989d601c09178 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF689QIJ.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF217BRB </td>\n",
       "   <td style=\"text-align:left;\"> bigBed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bigBed </td>\n",
       "   <td style=\"text-align:left;\"> pseudoreplicated peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me3 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 9cea2c6388211c15a9dd800ed86b8bab </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF217BRB.bigBed </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF217BRB/@@download/ENCFF217BRB.bigBed </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### arrange the metadata of default files\n",
    "dat = dat_metadata_chipseq_v1\n",
    "dat = fun(dat)\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_simplify_v1 = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "75092ee7-1870-44fb-bb0b-0abbab4b188b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 380  13\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000APD </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF677CFG </td>\n",
       "   <td style=\"text-align:left;\"> bam </td>\n",
       "   <td style=\"text-align:left;\"> bam </td>\n",
       "   <td style=\"text-align:left;\"> unfiltered alignments </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K79me2 </td>\n",
       "   <td style=\"text-align:left;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 4376f4b6f7bc508b2359556756403670 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF677CFG.bam </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF677CFG/@@download/ENCFF677CFG.bam </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000APD </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF112LWK </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K79me2 </td>\n",
       "   <td style=\"text-align:left;\"> 3 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 106649b0439fa81bd50cbddcbb0e4736 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF112LWK.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF112LWK/@@download/ENCFF112LWK.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000APD </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF544AVW </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K79me2 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2, 3 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 61dc50179ae8d880b972c3697a6a2fc2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF544AVW.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF544AVW/@@download/ENCFF544AVW.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### arrange the metadata of default files\n",
    "dat = dat_metadata_chipseq_v2\n",
    "dat = fun(dat)\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_simplify_v2 = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "94537e35-898b-4b4e-9056-5181e6c1a8c4",
   "metadata": {},
   "source": [
    "## Prepare download files"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f3869501-1733-41dc-a487-4bfcf968519d",
   "metadata": {},
   "source": [
    "**NarrowPeak files**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "f7bca1ee-29fb-454b-aa39-b67e1e36a647",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 19 13\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF689QIJ </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> pseudoreplicated peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me3 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 5dea2993c0831ae344a989d601c09178 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF689QIJ.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKQ </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF323WOT </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> pseudoreplicated peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K27me3 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2, 3 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 4422969d0b63260e2fcb83e10fdcc02f </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF323WOT.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000EWC </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF540NGG </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> pseudoreplicated peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me1 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 63db47e5b9b98dbebff2ce20df066106 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF540NGG.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "###\n",
    "dat = dat_metadata_chipseq_simplify_v1\n",
    "dat = dat %>% dplyr::filter(File_Type == \"bed\", File_Format == \"bed narrowPeak\")\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_bed_narrowpeak = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8476235f-0be8-4bf4-b169-8a9479ea434c",
   "metadata": {},
   "source": [
    "**Signal p-value**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "069e1ddf-17f3-4145-b7f9-f5d4a524b2eb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 19 13\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF767UON </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me3 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 4c102d45be8326062895ed0a03d4ded7 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF767UON.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF767UON/@@download/ENCFF767UON.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKQ </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF582IMB </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K27me3 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2, 3 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 2ca48f44075eef7118a387260f2f95b9 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF582IMB.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF582IMB/@@download/ENCFF582IMB.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000EWC </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF287LBI </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me1 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 28df1a757a2e5517209c10d57f0ce03e </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF287LBI.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF287LBI/@@download/ENCFF287LBI.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "###\n",
    "dat = dat_metadata_chipseq_simplify_v1\n",
    "dat = dat %>% dplyr::filter(File_Type == \"bigWig\", Output_Type == \"signal p-value\")\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_bigwig_pvalue = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd003a17-fbf7-4d68-b62d-466b223d0ab8",
   "metadata": {},
   "source": [
    "**Fold change**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "46b379b1-5c4b-4e28-9a6a-ea20a50981ee",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 60  4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:right;\"> Count_Replicates </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR000APD </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF112LWK </td>\n",
       "   <td style=\"text-align:left;\"> 3 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR000APD </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF544AVW </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2, 3 </td>\n",
       "   <td style=\"text-align:right;\"> 3 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR000APD </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF767ECL </td>\n",
       "   <td style=\"text-align:left;\"> 1 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR000APD </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF457RYD </td>\n",
       "   <td style=\"text-align:left;\"> 2 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKV </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF286WRJ </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:right;\"> 2 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKV </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF236VCK </td>\n",
       "   <td style=\"text-align:left;\"> 1 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### helper function\n",
    "fun = function(txt){\n",
    "    lst = str_split(txt, \",\")\n",
    "    lst = lapply(lst, length)\n",
    "    vec = unlist(lst)\n",
    "    return(vec)\n",
    "}\n",
    "\n",
    "###\n",
    "dat = dat_metadata_chipseq_simplify_v2\n",
    "dat = dat %>% dplyr::filter(File_Type == \"bigWig\", Output_Type == \"fold change over control\")\n",
    "dat = dat %>% dplyr::mutate(Count_Replicates = fun(Bio_Replicates))\n",
    "dat = dat %>% dplyr::select(\n",
    "    Index_Experiment,\n",
    "    Index_File,\n",
    "    Bio_Replicates,\n",
    "    Count_Replicates)\n",
    "\n",
    "### assign and show\n",
    "dat_count_replicates = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "56ecd2cb-3f8f-41fb-abc9-302794c81fb3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 19 13\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000APD </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF544AVW </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K79me2 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2, 3 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 61dc50179ae8d880b972c3697a6a2fc2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF544AVW.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF544AVW/@@download/ENCFF544AVW.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKV </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF286WRJ </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K9ac </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.6.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ccd7b8c413fdb998ffd799ec52dd5098 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF286WRJ.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF286WRJ/@@download/ENCFF286WRJ.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000APC </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF621DJP </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.6.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 3492c0e4a64e29231558f9e1e2fe520e </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF621DJP.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF621DJP/@@download/ENCFF621DJP.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "###\n",
    "dat = dat_count_replicates\n",
    "dat = dat %>% \n",
    "    dplyr::group_by(Index_Experiment) %>% \n",
    "    dplyr::slice_max(Count_Replicates)\n",
    "vec = dat$Index_File\n",
    "\n",
    "###\n",
    "dat = dat_metadata_chipseq_simplify_v2\n",
    "dat = dat %>% dplyr::filter(Index_File %in% vec)\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_bigwig_fold_change = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "9e77ffb0-46fb-46b9-9e94-5551ed7dc1b8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "fold change over control \n",
       "                      19 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_bigwig_fold_change\n",
    "table(dat$Output_Type)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "985dfffa-721b-43ce-a93e-ce2b78e269f6",
   "metadata": {},
   "source": [
    "## Save results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "cb71b079-aa09-42d2-9da5-d6984810b55b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "region_narrowPeak \n",
      "signal_fold_change \n",
      "signal_pvalue \n"
     ]
    }
   ],
   "source": [
    "lst = list(\n",
    "    \"region_narrowPeak\"  = dat_metadata_chipseq_bed_narrowpeak,\n",
    "    \"signal_fold_change\" = dat_metadata_chipseq_bigwig_fold_change, \n",
    "    \"signal_pvalue\"      = dat_metadata_chipseq_bigwig_pvalue\n",
    ")\n",
    "\n",
    "for (idx in names(lst)){\n",
    "    ### show progress\n",
    "    cat(idx, \"\\n\")\n",
    "    \n",
    "    ### create folder\n",
    "    txt_fdiry = file.path(FD_DAT, \"external\", TXT_FOLDER_OUT, idx)\n",
    "    txt_cmd   = paste(\"mkdir -p\", txt_fdiry)\n",
    "    system(txt_cmd)\n",
    "    \n",
    "    ### extract data\n",
    "    dat_metadata = lst[[idx]]\n",
    "\n",
    "    ### get file url\n",
    "    dat = dat_metadata\n",
    "    dat = dat %>% dplyr::select(File_URL)\n",
    "    dat_download_furl = dat\n",
    "    \n",
    "    ### get md5sum for each file\n",
    "    dat = dat_metadata\n",
    "    dat = dat %>% dplyr::select(md5sum, File_Name)\n",
    "    dat_download_md5sum = dat\n",
    "\n",
    "    ### write file list\n",
    "    txt_fname = \"files.txt\"\n",
    "    txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "    \n",
    "    dat = dat_download_furl\n",
    "    write_tsv(dat, txt_fpath, col_names = FALSE)\n",
    "\n",
    "    ### write checksum file\n",
    "    txt_fname = \"checksum_md5sum.txt\"\n",
    "    txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "    dat = dat_download_md5sum\n",
    "    write_tsv(dat, txt_fpath, col_names = FALSE)\n",
    "\n",
    "    ### write metadata info\n",
    "    txt_fname = \"metadata.tsv\"\n",
    "    txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "    dat = dat_metadata\n",
    "    write_tsv(dat, txt_fpath)  \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8bef779b-865e-436e-aa86-23700e9d7698",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c9b8e5ed-694d-4f9b-ba23-9a8d56b6b47d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fadc7e36-2305-4564-a928-3fd73b28a062",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fcb2f70f-fe23-4020-8355-dd89d7f56a3e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8efeb775-4479-4f4a-9ff4-ee111bf1fd92",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
