---
title: "Other Annotations"
---

# Define transcription start sites (TSS) in K562

- [Select isoforms with the highest K562 Pol2 signals](https://github.com/Reilly-Lab-Yale/ENCODE-CRISPR/tree/main/Figure6/identifying_K562_TSS_TES_for_genebody_definition)
    - For each gene in NCBI Refgene, the isoform with highest level of pol2 chip seq (ENCFF914WIS.bigWig) at [TSS-500, TSS+500] were selected. 
    - If there are genes with multiple lines, it means they had isoforms with the same pol2 signal (they are mostly genes with very low pol2 signals, e,g., TP53TG3B) 
    - This was done by Jin-Woo.
- Filter based on RNA-seq to get K562 specific TSS

[Download K562 TSS used in this documentation](https://github.com/ReddyLab/Proj_ENCODE_FCC/tree/main/references/fcc_genome_tss)

# TF modules

- "K562.full.region.assignments.txt" has numbered columns referring to the Module numbers (1-77) 
    - Note that there are not 77 total modules because only modules with >500 regions were kept
    - followed by “chr”, “start”, “end”, and “TF” columns which refer to the genomic positions and all TFs bound at that region

- "K562.TFzscore.txt” contains the TFs that passed our thresholds to be considered in the module. 
    - Note that a TF may be listed as bound to a module region in the above file, but if it did not pass the threshold it was not considered a module member. 
    - Details on our thresholds are below: a TF is kept in that module if it passed the following two conditions:
        - `TF z-score > 0`. 
            - The `TF z-score` = (# of regions bound by TF in the module of interest – mean(# of regions bound per module))/SD(# regions bound per module). 
        - `module z-score > 0`. 
            - The `module z-score` = (# of regions bound within module by TF of interest) – mean (# of regions bound by each TF within module))/SD(# regions bound per TF within module)
