{
 "cells": [
  {
   "cell_type": "raw",
   "id": "31efdb40-d38b-4bcc-a1ad-c3e3ab2f08d4",
   "metadata": {},
   "source": [
    "---\n",
    "title: Prepare ENCODE ChIP-seq 34 (TF; ENCODE4)\n",
    "subtitle: Region metadata\n",
    "format:\n",
    "    html:\n",
    "        code-fold:  true\n",
    "        code-tools: true\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e36d76e-cf9f-4e9e-8644-66e22215b549",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "91f900a1-6f65-4bb3-bc6e-7971c6310bf5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are working on        Singularity \n",
      "BASE DIRECTORY (FD_BASE): /mount \n",
      "REPO DIRECTORY (FD_REPO): /mount/repo \n",
      "WORK DIRECTORY (FD_WORK): /mount/work \n",
      "DATA DIRECTORY (FD_DATA): /mount/data \n",
      "\n",
      "You are working with      ENCODE FCC \n",
      "PATH OF PROJECT (FD_PRJ): /mount/repo/Proj_ENCODE_FCC \n",
      "PROJECT RESULTS (FD_RES): /mount/repo/Proj_ENCODE_FCC/results \n",
      "PROJECT SCRIPTS (FD_EXE): /mount/repo/Proj_ENCODE_FCC/scripts \n",
      "PROJECT DATA    (FD_DAT): /mount/repo/Proj_ENCODE_FCC/data \n",
      "PROJECT NOTE    (FD_NBK): /mount/repo/Proj_ENCODE_FCC/notebooks \n",
      "PROJECT DOCS    (FD_DOC): /mount/repo/Proj_ENCODE_FCC/docs \n",
      "PROJECT LOG     (FD_LOG): /mount/repo/Proj_ENCODE_FCC/log \n",
      "PROJECT APP     (FD_APP): /mount/repo/Proj_ENCODE_FCC/app \n",
      "PROJECT REF     (FD_REF): /mount/repo/Proj_ENCODE_FCC/references \n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(source(\"../run_config_project_sing.R\")))\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c9fce1b-a43f-4e03-ae92-488292a491c7",
   "metadata": {},
   "source": [
    "**Set global variables**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "49683b05-381f-40f6-963f-4c72324beb60",
   "metadata": {},
   "outputs": [],
   "source": [
    "TXT_FOLDER_REGION = \"encode_chipseq_encode4\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "20e34c0a-e61f-4392-9141-744d012bfb11",
   "metadata": {},
   "source": [
    "## Import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "023e8c50-aa88-4045-a11b-6ef1c5564a11",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 711  59\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF124KCE </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR800KMQ </td>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> CRISPR </td>\n",
       "   <td style=\"text-align:left;\"> insertion </td>\n",
       "   <td style=\"text-align:left;\"> {'schema_version': '14', 'aliases': [], 'organism': {'schema_version': '6', '@type': ['Organism', 'Item'], 'name': 'human', 'taxon_id': '9606', 'scientific_name': 'Homo sapiens', '@id': '/organisms/human/', 'uuid': '7745b647-ff15-4ff3-9ced-b897d4e2983c', 'status': 'released'}, 'genes': ['/genes/23099/'], '@type': ['Target', 'Item'], 'name': 'ZBTB43-human', 'label': 'ZBTB43', '@id': '/targets/ZBTB43-human/', 'title': 'ZBTB43 (Homo sapiens)', 'investigated_as': ['transcription factor'], 'uuid': 'c4fd1273-5058-4d6b-90b2-4fa94d61d714', 'status': 'released'} </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB43-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> formaldehyde </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2023-01-10 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> sonication (generic) </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1, 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF441EKA/, /files/ENCFF356LFX/, /files/ENCFF837JNH/, /files/ENCFF624WBH/ </td>\n",
       "   <td style=\"text-align:right;\"> 96907 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> 3154aa211bac58ce311f643c7aa9023e </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF124KCE/@@download/ENCFF124KCE.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2022/07/23/a46bd98a-7686-4fa4-ae66-7e0a8e8b7748/ENCFF124KCE.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2022/07/23/a46bd98a-7686-4fa4-ae66-7e0a8e8b7748/ENCFF124KCE.bed.gz?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> missing genetic modification reagents, mild to moderate bottlenecking </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set directory\n",
    "txt_fdiry = file.path(FD_DAT, \"external\", TXT_FOLDER_REGION)\n",
    "txt_fname = \"metadata.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### assing and show\n",
    "dat_meta_import = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "4fd49e0d-e070-499c-884f-70395d8985df",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        \n",
       "         conservative IDR thresholded peaks IDR thresholded peaks\n",
       "  bed                                    24                   213\n",
       "  bigBed                                 24                   213\n",
       "  bigWig                                  0                     0\n",
       "        \n",
       "         signal p-value\n",
       "  bed                 0\n",
       "  bigBed              0\n",
       "  bigWig            237"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_meta_import\n",
    "table(dat$`File type`, dat$`Output type`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9efa3b11-a746-4504-a430-8c4eae354d3e",
   "metadata": {},
   "source": [
    "## Helper function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "335cd2fd-0809-4201-ae22-82ec9569d088",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$schema_version</dt>\n",
       "\t\t<dd>'14'</dd>\n",
       "\t<dt>$aliases</dt>\n",
       "\t\t<dd>'michael-snyder:TS-1589'</dd>\n",
       "\t<dt>$organism</dt>\n",
       "\t\t<dd><dl>\n",
       "\t<dt>$schema_version</dt>\n",
       "\t\t<dd>'6'</dd>\n",
       "\t<dt>$`@type`</dt>\n",
       "\t\t<dd><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Organism'</li><li>'Item'</li></ol>\n",
       "</dd>\n",
       "\t<dt>$name</dt>\n",
       "\t\t<dd>'human'</dd>\n",
       "\t<dt>$taxon_id</dt>\n",
       "\t\t<dd>'9606'</dd>\n",
       "\t<dt>$scientific_name</dt>\n",
       "\t\t<dd>'Homo sapiens'</dd>\n",
       "\t<dt>$`@id`</dt>\n",
       "\t\t<dd>'/organisms/human/'</dd>\n",
       "\t<dt>$uuid</dt>\n",
       "\t\t<dd>'7745b647-ff15-4ff3-9ced-b897d4e2983c'</dd>\n",
       "\t<dt>$status</dt>\n",
       "\t\t<dd>'released'</dd>\n",
       "</dl>\n",
       "</dd>\n",
       "\t<dt>$genes</dt>\n",
       "\t\t<dd>'/genes/7678/'</dd>\n",
       "\t<dt>$`@type`</dt>\n",
       "\t\t<dd><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Target'</li><li>'Item'</li></ol>\n",
       "</dd>\n",
       "\t<dt>$name</dt>\n",
       "\t\t<dd>'ZNF124-human'</dd>\n",
       "\t<dt>$label</dt>\n",
       "\t\t<dd>'ZNF124'</dd>\n",
       "\t<dt>$`@id`</dt>\n",
       "\t\t<dd>'/targets/ZNF124-human/'</dd>\n",
       "\t<dt>$title</dt>\n",
       "\t\t<dd>'ZNF124 (Homo sapiens)'</dd>\n",
       "\t<dt>$investigated_as</dt>\n",
       "\t\t<dd>'transcription factor'</dd>\n",
       "\t<dt>$uuid</dt>\n",
       "\t\t<dd>'1aeb7ae9-5865-4a6a-97c8-225196c163a0'</dd>\n",
       "\t<dt>$status</dt>\n",
       "\t\t<dd>'released'</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$schema\\_version] '14'\n",
       "\\item[\\$aliases] 'michael-snyder:TS-1589'\n",
       "\\item[\\$organism] \\begin{description}\n",
       "\\item[\\$schema\\_version] '6'\n",
       "\\item[\\$`@type`] \\begin{enumerate*}\n",
       "\\item 'Organism'\n",
       "\\item 'Item'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$name] 'human'\n",
       "\\item[\\$taxon\\_id] '9606'\n",
       "\\item[\\$scientific\\_name] 'Homo sapiens'\n",
       "\\item[\\$`@id`] '/organisms/human/'\n",
       "\\item[\\$uuid] '7745b647-ff15-4ff3-9ced-b897d4e2983c'\n",
       "\\item[\\$status] 'released'\n",
       "\\end{description}\n",
       "\n",
       "\\item[\\$genes] '/genes/7678/'\n",
       "\\item[\\$`@type`] \\begin{enumerate*}\n",
       "\\item 'Target'\n",
       "\\item 'Item'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$name] 'ZNF124-human'\n",
       "\\item[\\$label] 'ZNF124'\n",
       "\\item[\\$`@id`] '/targets/ZNF124-human/'\n",
       "\\item[\\$title] 'ZNF124 (Homo sapiens)'\n",
       "\\item[\\$investigated\\_as] 'transcription factor'\n",
       "\\item[\\$uuid] '1aeb7ae9-5865-4a6a-97c8-225196c163a0'\n",
       "\\item[\\$status] 'released'\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$schema_version\n",
       ":   '14'\n",
       "$aliases\n",
       ":   'michael-snyder:TS-1589'\n",
       "$organism\n",
       ":   $schema_version\n",
       ":   '6'\n",
       "$`@type`\n",
       ":   1. 'Organism'\n",
       "2. 'Item'\n",
       "\n",
       "\n",
       "\n",
       "$name\n",
       ":   'human'\n",
       "$taxon_id\n",
       ":   '9606'\n",
       "$scientific_name\n",
       ":   'Homo sapiens'\n",
       "$`@id`\n",
       ":   '/organisms/human/'\n",
       "$uuid\n",
       ":   '7745b647-ff15-4ff3-9ced-b897d4e2983c'\n",
       "$status\n",
       ":   'released'\n",
       "\n",
       "\n",
       "\n",
       "$genes\n",
       ":   '/genes/7678/'\n",
       "$`@type`\n",
       ":   1. 'Target'\n",
       "2. 'Item'\n",
       "\n",
       "\n",
       "\n",
       "$name\n",
       ":   'ZNF124-human'\n",
       "$label\n",
       ":   'ZNF124'\n",
       "$`@id`\n",
       ":   '/targets/ZNF124-human/'\n",
       "$title\n",
       ":   'ZNF124 (Homo sapiens)'\n",
       "$investigated_as\n",
       ":   'transcription factor'\n",
       "$uuid\n",
       ":   '1aeb7ae9-5865-4a6a-97c8-225196c163a0'\n",
       "$status\n",
       ":   'released'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$schema_version\n",
       "[1] \"14\"\n",
       "\n",
       "$aliases\n",
       "[1] \"michael-snyder:TS-1589\"\n",
       "\n",
       "$organism\n",
       "$organism$schema_version\n",
       "[1] \"6\"\n",
       "\n",
       "$organism$`@type`\n",
       "[1] \"Organism\" \"Item\"    \n",
       "\n",
       "$organism$name\n",
       "[1] \"human\"\n",
       "\n",
       "$organism$taxon_id\n",
       "[1] \"9606\"\n",
       "\n",
       "$organism$scientific_name\n",
       "[1] \"Homo sapiens\"\n",
       "\n",
       "$organism$`@id`\n",
       "[1] \"/organisms/human/\"\n",
       "\n",
       "$organism$uuid\n",
       "[1] \"7745b647-ff15-4ff3-9ced-b897d4e2983c\"\n",
       "\n",
       "$organism$status\n",
       "[1] \"released\"\n",
       "\n",
       "\n",
       "$genes\n",
       "[1] \"/genes/7678/\"\n",
       "\n",
       "$`@type`\n",
       "[1] \"Target\" \"Item\"  \n",
       "\n",
       "$name\n",
       "[1] \"ZNF124-human\"\n",
       "\n",
       "$label\n",
       "[1] \"ZNF124\"\n",
       "\n",
       "$`@id`\n",
       "[1] \"/targets/ZNF124-human/\"\n",
       "\n",
       "$title\n",
       "[1] \"ZNF124 (Homo sapiens)\"\n",
       "\n",
       "$investigated_as\n",
       "[1] \"transcription factor\"\n",
       "\n",
       "$uuid\n",
       "[1] \"1aeb7ae9-5865-4a6a-97c8-225196c163a0\"\n",
       "\n",
       "$status\n",
       "[1] \"released\"\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "txt = \"{'schema_version': '14', 'aliases': ['michael-snyder:TS-1589'], 'organism': {'schema_version': '6', '@type': ['Organism', 'Item'], 'name': 'human', 'taxon_id': '9606', 'scientific_name': 'Homo sapiens', '@id': '/organisms/human/', 'uuid': '7745b647-ff15-4ff3-9ced-b897d4e2983c', 'status': 'released'}, 'genes': ['/genes/7678/'], '@type': ['Target', 'Item'], 'name': 'ZNF124-human', 'label': 'ZNF124', '@id': '/targets/ZNF124-human/', 'title': 'ZNF124 (Homo sapiens)', 'investigated_as': ['transcription factor'], 'uuid': '1aeb7ae9-5865-4a6a-97c8-225196c163a0', 'status': 'released'}\"\n",
    "txt = stringr::str_replace_all(txt, pattern = \"'\", replacement = '\"')\n",
    "tmp = jsonlite::fromJSON(txt)\n",
    "tmp"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "651a657d-5078-41c6-81cf-65e22047ab1a",
   "metadata": {},
   "source": [
    "## Arrange table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "4be50449-22ad-4cd2-9339-502d77498536",
   "metadata": {},
   "outputs": [],
   "source": [
    "fun_txt2json = function(txt){\n",
    "    txt = stringr::str_replace_all(txt, pattern = \"'\", replacement = '\"')\n",
    "    lst = jsonlite::fromJSON(txt)\n",
    "    return(lst)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "6bdc220d-c98b-43c7-a5e1-ade4babb5837",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 237   1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ZBTB43 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> E2F5 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ZNF431 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ZNF512 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> IRF2 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ZNF124 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### init\n",
    "dat = dat_meta_import\n",
    "fun = function(vec){\n",
    "    lst = lapply(vec, fun_txt2json)\n",
    "    lst = lapply(lst, function(tmp){tmp$label})\n",
    "    res = unlist(lst)\n",
    "    return(res)\n",
    "}\n",
    "\n",
    "###\n",
    "dat = dat %>% \n",
    "    dplyr::filter(`File format` == \"bed narrowPeak\") %>%\n",
    "    dplyr::mutate(\n",
    "        Target = fun(`Biosample genetic modifications targets`)\n",
    "    ) %>%\n",
    "    dplyr::select(Target)\n",
    "\n",
    "### assing and show\n",
    "dat_label = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "2c9e754a-9ea1-4f70-afb6-92e3d216536b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 711   9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> FName </th>\n",
       "   <th style=\"text-align:left;\"> FUrl </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR800KMQ </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF124KCE </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB43 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF124KCE.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF124KCE/@@download/ENCFF124KCE.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR800KMQ </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF334ZCX </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB43 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF334ZCX.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF334ZCX/@@download/ENCFF334ZCX.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR800KMQ </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF722QWH </td>\n",
       "   <td style=\"text-align:left;\"> bigBed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB43 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF722QWH.bigBed </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF722QWH/@@download/ENCFF722QWH.bigBed </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR744GHR </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF173QUY </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> E2F5 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF173QUY.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF173QUY/@@download/ENCFF173QUY.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR744GHR </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF225ZXF </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> E2F5 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF225ZXF.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF225ZXF/@@download/ENCFF225ZXF.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR744GHR </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF470UPO </td>\n",
       "   <td style=\"text-align:left;\"> bigBed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> E2F5 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF470UPO.bigBed </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF470UPO/@@download/ENCFF470UPO.bigBed </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### init\n",
    "dat = dat_meta_import\n",
    "fun = function(vec){\n",
    "    lst = lapply(vec, fun_txt2json)\n",
    "    lst = lapply(lst, function(tmp){tmp$label})\n",
    "    res = unlist(lst)\n",
    "    return(res)\n",
    "}\n",
    "\n",
    "###\n",
    "dat = dat %>% \n",
    "    dplyr::mutate(\n",
    "        Biosample        = `Biosample term name`,\n",
    "        Index_Experiment = `Experiment accession`,\n",
    "        Index_File       = `File accession`,\n",
    "        File_Type        = `File format`,\n",
    "        Output_Type      = `Output type`,\n",
    "        Genome           = `File assembly`,\n",
    "        Target           = fun(`Biosample genetic modifications targets`),\n",
    "        FName            = basename(`File download URL`),\n",
    "        FUrl             = `File download URL`\n",
    "    ) %>%\n",
    "    dplyr::select(\n",
    "        Assay, \n",
    "        Index_Experiment, \n",
    "        Index_File,\n",
    "        File_Type,\n",
    "        Output_Type,\n",
    "        Genome,\n",
    "        Target,\n",
    "        #Lab,\n",
    "        FName,\n",
    "        FUrl)\n",
    "\n",
    "### assing and show\n",
    "dat_meta_total = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "0518c0c0-9247-4c84-94a6-9a8631d95cd0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 237   9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> FName </th>\n",
       "   <th style=\"text-align:left;\"> FUrl </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR800KMQ </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF124KCE </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB43 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF124KCE.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF124KCE/@@download/ENCFF124KCE.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR744GHR </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF173QUY </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> E2F5 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF173QUY.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF173QUY/@@download/ENCFF173QUY.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR609ZLF </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF442RYH </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF431 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF442RYH.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF442RYH/@@download/ENCFF442RYH.bed.gz </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_meta_total %>% dplyr::filter(File_Type == \"bed narrowPeak\")\n",
    "\n",
    "dat_meta_region = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "05166aa0-03f5-4aa5-b52a-d353ab6a7a20",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 237   9\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> FName </th>\n",
       "   <th style=\"text-align:left;\"> FUrl </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR800KMQ </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF334ZCX </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB43 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF334ZCX.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF334ZCX/@@download/ENCFF334ZCX.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR744GHR </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF225ZXF </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> E2F5 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF225ZXF.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF225ZXF/@@download/ENCFF225ZXF.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR609ZLF </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF664RMA </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF431 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF664RMA.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF664RMA/@@download/ENCFF664RMA.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_meta_total %>% dplyr::filter(File_Type == \"bigWig\")\n",
    "\n",
    "dat_meta_signal = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "127eb93a-9002-482f-814d-7b527e06b4fe",
   "metadata": {},
   "source": [
    "## Arrange label"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "cafff4d0-e937-47c3-aaff-1b2e50546e3b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 237\n",
      "[1] \"ENCFF009RFC.bed.gz\" \"ENCFF014HYS.bed.gz\" \"ENCFF018HWM.bed.gz\"\n",
      "[4] \"ENCFF018TNP.bed.gz\" \"ENCFF019NGO.bed.gz\" \"ENCFF021JCJ.bed.gz\"\n"
     ]
    }
   ],
   "source": [
    "### set directory\n",
    "txt_folder = TXT_FOLDER_REGION\n",
    "txt_fdiry  = file.path(FD_RES, \"region\", txt_folder)\n",
    "txt_fglob  = file.path(txt_fdiry, \"*bed*\")\n",
    "\n",
    "### get file names\n",
    "vec_txt_fpath = Sys.glob(txt_fglob)\n",
    "vec_txt_fname = basename(vec_txt_fpath)\n",
    "\n",
    "###\n",
    "vec = vec_txt_fname\n",
    "print(length(vec))\n",
    "print(head(vec))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "3fa5b0c0-bf9b-4f98-a084-0444bb3d5399",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 237   3\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Folder </th>\n",
       "   <th style=\"text-align:left;\"> FName </th>\n",
       "   <th style=\"text-align:left;\"> Label </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> encode_chipseq_encode4 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF124KCE.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> chipseq_ZBTB43_ENCFF124KCE </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> encode_chipseq_encode4 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF173QUY.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> chipseq_E2F5_ENCFF173QUY </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> encode_chipseq_encode4 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF442RYH.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> chipseq_ZNF431_ENCFF442RYH </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> encode_chipseq_encode4 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF817EJA.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> chipseq_ZNF512_ENCFF817EJA </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> encode_chipseq_encode4 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF430YXJ.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> chipseq_IRF2_ENCFF430YXJ </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> encode_chipseq_encode4 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF881MMD.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> chipseq_ZNF124_ENCFF881MMD </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "###\n",
    "dat = dat_meta_total %>% \n",
    "    dplyr::mutate(\n",
    "        Folder = TXT_FOLDER_REGION,\n",
    "        Label  = paste(\"chipseq\", Target, Index_File, sep = \"_\")) %>%\n",
    "    dplyr::select(Folder, FName, Label) %>%\n",
    "    dplyr::filter(FName %in% vec_txt_fname)\n",
    "\n",
    "###\n",
    "dat_region_label = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "66854104-f65f-4cc7-b738-4302125f354c",
   "metadata": {},
   "source": [
    "## Define column description\n",
    "The peak file is in narrowPeak format, which is a standard six field bed with four additional fields (BED6+4 format)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "1fe1ee53-d3a1-4bad-b31c-fdbdefbb420a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 237\n",
      "[1] \"ENCFF009RFC.bed.gz\" \"ENCFF014HYS.bed.gz\" \"ENCFF018HWM.bed.gz\"\n",
      "[4] \"ENCFF018TNP.bed.gz\" \"ENCFF019NGO.bed.gz\" \"ENCFF021JCJ.bed.gz\"\n"
     ]
    }
   ],
   "source": [
    "### set directory\n",
    "txt_folder = TXT_FOLDER_REGION\n",
    "txt_fdiry  = file.path(FD_RES, \"region\", txt_folder)\n",
    "txt_fglob  = file.path(txt_fdiry, \"*bed*\")\n",
    "\n",
    "### get file names\n",
    "vec_txt_fpath = Sys.glob(txt_fglob)\n",
    "vec_txt_fname = basename(vec_txt_fpath)\n",
    "\n",
    "###\n",
    "vec = vec_txt_fname\n",
    "print(length(vec))\n",
    "print(head(vec))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "bd646882-f33b-4c79-a795-494bc09c624b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>X1</th><th scope=col>X2</th><th scope=col>X3</th><th scope=col>X4</th><th scope=col>X5</th><th scope=col>X6</th><th scope=col>X7</th><th scope=col>X8</th><th scope=col>X9</th><th scope=col>X10</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>778490</td><td>779074</td><td>.</td><td>1000</td><td>.</td><td>102.00797</td><td>-1</td><td>4.23485</td><td>292</td></tr>\n",
       "\t<tr><td>chr1</td><td>827311</td><td>827661</td><td>.</td><td>1000</td><td>.</td><td>234.44607</td><td>-1</td><td>4.23485</td><td>169</td></tr>\n",
       "\t<tr><td>chr1</td><td>904451</td><td>905035</td><td>.</td><td>1000</td><td>.</td><td> 85.29658</td><td>-1</td><td>4.23485</td><td>292</td></tr>\n",
       "\t<tr><td>chr1</td><td>923229</td><td>923813</td><td>.</td><td> 641</td><td>.</td><td> 35.53121</td><td>-1</td><td>2.11833</td><td>292</td></tr>\n",
       "\t<tr><td>chr1</td><td>938076</td><td>938660</td><td>.</td><td> 858</td><td>.</td><td> 44.12972</td><td>-1</td><td>2.64528</td><td>292</td></tr>\n",
       "\t<tr><td>chr1</td><td>940680</td><td>941264</td><td>.</td><td>1000</td><td>.</td><td> 51.28175</td><td>-1</td><td>3.39127</td><td>292</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " X1 & X2 & X3 & X4 & X5 & X6 & X7 & X8 & X9 & X10\\\\\n",
       " <chr> & <dbl> & <dbl> & <chr> & <dbl> & <chr> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr1 & 778490 & 779074 & . & 1000 & . & 102.00797 & -1 & 4.23485 & 292\\\\\n",
       "\t chr1 & 827311 & 827661 & . & 1000 & . & 234.44607 & -1 & 4.23485 & 169\\\\\n",
       "\t chr1 & 904451 & 905035 & . & 1000 & . &  85.29658 & -1 & 4.23485 & 292\\\\\n",
       "\t chr1 & 923229 & 923813 & . &  641 & . &  35.53121 & -1 & 2.11833 & 292\\\\\n",
       "\t chr1 & 938076 & 938660 & . &  858 & . &  44.12972 & -1 & 2.64528 & 292\\\\\n",
       "\t chr1 & 940680 & 941264 & . & 1000 & . &  51.28175 & -1 & 3.39127 & 292\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 10\n",
       "\n",
       "| X1 &lt;chr&gt; | X2 &lt;dbl&gt; | X3 &lt;dbl&gt; | X4 &lt;chr&gt; | X5 &lt;dbl&gt; | X6 &lt;chr&gt; | X7 &lt;dbl&gt; | X8 &lt;dbl&gt; | X9 &lt;dbl&gt; | X10 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 778490 | 779074 | . | 1000 | . | 102.00797 | -1 | 4.23485 | 292 |\n",
       "| chr1 | 827311 | 827661 | . | 1000 | . | 234.44607 | -1 | 4.23485 | 169 |\n",
       "| chr1 | 904451 | 905035 | . | 1000 | . |  85.29658 | -1 | 4.23485 | 292 |\n",
       "| chr1 | 923229 | 923813 | . |  641 | . |  35.53121 | -1 | 2.11833 | 292 |\n",
       "| chr1 | 938076 | 938660 | . |  858 | . |  44.12972 | -1 | 2.64528 | 292 |\n",
       "| chr1 | 940680 | 941264 | . | 1000 | . |  51.28175 | -1 | 3.39127 | 292 |\n",
       "\n"
      ],
      "text/plain": [
       "  X1   X2     X3     X4 X5   X6 X7        X8 X9      X10\n",
       "1 chr1 778490 779074 .  1000 .  102.00797 -1 4.23485 292\n",
       "2 chr1 827311 827661 .  1000 .  234.44607 -1 4.23485 169\n",
       "3 chr1 904451 905035 .  1000 .   85.29658 -1 4.23485 292\n",
       "4 chr1 923229 923813 .   641 .   35.53121 -1 2.11833 292\n",
       "5 chr1 938076 938660 .   858 .   44.12972 -1 2.64528 292\n",
       "6 chr1 940680 941264 .  1000 .   51.28175 -1 3.39127 292"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "txt_fpath = vec_txt_fpath[1]\n",
    "dat = read_tsv(txt_fpath, col_names = FALSE, show_col_types = FALSE)\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "173ccdd2-14ae-43fd-9da1-e8ae71383511",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Name </th>\n",
       "   <th style=\"text-align:left;\"> Note </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Chrom </td>\n",
       "   <td style=\"text-align:left;\"> Name of the chromosome </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ChromStart </td>\n",
       "   <td style=\"text-align:left;\"> The starting position of the feature in the chromosome </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ChromEnd </td>\n",
       "   <td style=\"text-align:left;\"> The ending position of the feature in the chromosome </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Name </td>\n",
       "   <td style=\"text-align:left;\"> Name given to a region; Use '.' if no name is assigned. </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Score </td>\n",
       "   <td style=\"text-align:left;\"> Indicates how dark the peak will be displayed in the browser (0-1000). </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Strand </td>\n",
       "   <td style=\"text-align:left;\"> +/- to denote strand or orientation. Use '.' if no orientation is assigned. </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> SignalValue </td>\n",
       "   <td style=\"text-align:left;\"> Measurement of overall (usually, average) enrichment for the region. </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> PValue </td>\n",
       "   <td style=\"text-align:left;\"> Measurement of statistical significance (-log10). Use -1 if no pValue is assigned. </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> QValue </td>\n",
       "   <td style=\"text-align:left;\"> Measurement of statistical significance using false discovery rate (-log10). Use -1 if no qValue is assigned. </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Peak </td>\n",
       "   <td style=\"text-align:left;\"> Point-source called for this peak; 0-based offset from chromStart. Use -1 if no point-source called. </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### ENCODE narrowPeak: Narrow (or Point-Source) Peaks format\n",
    "dat = tribble(\n",
    "    ~Name,        ~Note,\n",
    "    \"Chrom\",      \"Name of the chromosome\",\n",
    "    \"ChromStart\", \"The starting position of the feature in the chromosome\",\n",
    "    \"ChromEnd\",   \"The ending position of the feature in the chromosome\",\n",
    "    \"Name\",       \"Name given to a region; Use '.' if no name is assigned.\",\n",
    "    \"Score\",      \"Indicates how dark the peak will be displayed in the browser (0-1000).\",\n",
    "    \"Strand\",     \"+/- to denote strand or orientation. Use '.' if no orientation is assigned.\",\n",
    "    \"SignalValue\",\"Measurement of overall (usually, average) enrichment for the region.\",\n",
    "    \"PValue\",     \"Measurement of statistical significance (-log10). Use -1 if no pValue is assigned.\",\n",
    "    \"QValue\",     \"Measurement of statistical significance using false discovery rate (-log10). Use -1 if no qValue is assigned.\",\n",
    "    \"Peak\",       \"Point-source called for this peak; 0-based offset from chromStart. Use -1 if no point-source called.\"\n",
    ")\n",
    "\n",
    "### assign and show\n",
    "dat_cname = dat\n",
    "fun_display_table(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51416e42-a93b-4e5e-9a23-3e0976d2120a",
   "metadata": {},
   "source": [
    "## Save results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "8d68b075-7149-47c7-9c95-12e20439328e",
   "metadata": {},
   "outputs": [],
   "source": [
    "txt_folder = TXT_FOLDER_REGION\n",
    "txt_fdiry  = file.path(FD_RES, \"region\", txt_folder, \"summary\")\n",
    "txt_fname  = \"metadata.tsv\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "dir.create(txt_fdiry, showWarnings = FALSE)\n",
    "dat = dat_meta_region\n",
    "write_tsv(dat, txt_fpath)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "7ac7675b-0b87-4766-a924-5a1af2d4be52",
   "metadata": {},
   "outputs": [],
   "source": [
    "txt_folder = TXT_FOLDER_REGION\n",
    "txt_fdiry  = file.path(FD_RES, \"region\", txt_folder, \"summary\")\n",
    "txt_fname  = \"metadata.label.tsv\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "dir.create(txt_fdiry, showWarnings = FALSE)\n",
    "dat = dat_region_label\n",
    "write_tsv(dat, txt_fpath)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "7dfce92d-35a7-48fd-9790-4867bb4cf21f",
   "metadata": {},
   "outputs": [],
   "source": [
    "txt_folder = TXT_FOLDER_REGION\n",
    "txt_fdiry  = file.path(FD_RES, \"region\", txt_folder, \"summary\")\n",
    "txt_fname  = \"description.tsv\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "dir.create(txt_fdiry, showWarnings = FALSE)\n",
    "dat = dat_cname\n",
    "write_tsv(dat, txt_fpath)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "81fdfe5f-0d73-46f9-ac49-d2430832f554",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
