{
 "cells": [
  {
   "cell_type": "raw",
   "id": "0ff26148-34b4-47ae-958b-86377f167ffc",
   "metadata": {},
   "source": [
    "---\n",
    "title: Prepare ENCODE ChIP-seq 11 (Histone)\n",
    "subtitle: Generate download script to download the data\n",
    "format:\n",
    "    html:\n",
    "        code-fold:  true\n",
    "        code-tools: true\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4ce3214-dfe4-4696-9a21-3c363e04216a",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c8b86e0a-604d-46d0-a295-b863844d5706",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are working on        Singularity \n",
      "BASE DIRECTORY (FD_BASE): /mount \n",
      "REPO DIRECTORY (FD_REPO): /mount/repo \n",
      "WORK DIRECTORY (FD_WORK): /mount/work \n",
      "DATA DIRECTORY (FD_DATA): /mount/data \n",
      "\n",
      "You are working with      ENCODE FCC \n",
      "PATH OF PROJECT (FD_PRJ): /mount/repo/Proj_ENCODE_FCC \n",
      "PROJECT RESULTS (FD_RES): /mount/repo/Proj_ENCODE_FCC/results \n",
      "PROJECT SCRIPTS (FD_EXE): /mount/repo/Proj_ENCODE_FCC/scripts \n",
      "PROJECT DATA    (FD_DAT): /mount/repo/Proj_ENCODE_FCC/data \n",
      "PROJECT NOTE    (FD_NBK): /mount/repo/Proj_ENCODE_FCC/notebooks \n",
      "PROJECT DOCS    (FD_DOC): /mount/repo/Proj_ENCODE_FCC/docs \n",
      "PROJECT LOG     (FD_LOG): /mount/repo/Proj_ENCODE_FCC/log \n",
      "PROJECT APP     (FD_APP): /mount/repo/Proj_ENCODE_FCC/app \n",
      "PROJECT REF     (FD_REF): /mount/repo/Proj_ENCODE_FCC/references \n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(source(\"../run_config_project_sing.R\")))\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83d64184-d069-4654-80d7-45d35dcf57d1",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Import metadata from reference file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "344a785d-f912-4bc3-886a-a11fce201a27",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 38 59\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF461RKK </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000EWA </td>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me3-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2011-05-19 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> see document </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1, 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF181ANT/, /files/ENCFF747HEB/, /files/ENCFF355LXQ/ </td>\n",
       "   <td style=\"text-align:right;\"> 510104216 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> 46d7c60097cc9a09d3c3bede8302f01b </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF461RKK/@@download/ENCFF461RKK.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/09/29/120fbb5a-db8e-464e-bdeb-9e72fdbb380b/ENCFF461RKK.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/09/29/120fbb5a-db8e-464e-bdeb-9e72fdbb380b/ENCFF461RKK.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> low read length, low read depth, mild to moderate bottlenecking, moderate library complexity, inconsistent platforms </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF122CSI </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> pseudoreplicated peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000EWA </td>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me3-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2011-05-19 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> see document </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1, 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF181ANT/, /files/ENCFF747HEB/, /files/ENCFF355LXQ/, /files/ENCFF356LFX/ </td>\n",
       "   <td style=\"text-align:right;\"> 663053 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> e5694050796ccb5eeace1e6a4efec34d </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/09/29/f70dc6ea-ac1b-4167-a054-e3dfb1f6684f/ENCFF122CSI.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/09/29/f70dc6ea-ac1b-4167-a054-e3dfb1f6684f/ENCFF122CSI.bed.gz?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> low read length, low read depth, mild to moderate bottlenecking, moderate library complexity, inconsistent platforms </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF801AHF </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> pseudoreplicated peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000EWB </td>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> H3K27me3-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2011-05-19 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> see document </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1, 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF356LFX/, /files/ENCFF652TXG/, /files/ENCFF355LXQ/, /files/ENCFF508LLH/ </td>\n",
       "   <td style=\"text-align:right;\"> 3187579 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> e2bef555c66ac7cb95dd56dc3339420f </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/11/01/3c1bac8c-3835-413b-a600-197421de4a1c/ENCFF801AHF.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/11/01/3c1bac8c-3835-413b-a600-197421de4a1c/ENCFF801AHF.bed.gz?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.6.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> low read length, inconsistent platforms </td>\n",
       "   <td style=\"text-align:left;\"> insufficient read depth, control insufficient read depth </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_fdiry = file.path(FD_REF, \"encode_chipseq_histone\")\n",
    "txt_fname = \"metadata.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### show and assign\n",
    "dat_metadata_import = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae52ff80-5a21-4503-8504-4cb6f22fd8f3",
   "metadata": {},
   "source": [
    "## Arrange columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "47d4f26a-699e-422c-9de9-5cafa4349699",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 38 10\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Biosample </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_Process </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> File_URL_Download </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000EWA </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF461RKK </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me3 </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF461RKK/@@download/ENCFF461RKK.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000EWA </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF122CSI </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> pseudoreplicated peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me3 </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000EWB </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.6.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF801AHF </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> pseudoreplicated peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K27me3 </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000EWB </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.6.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF847BFA </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K27me3 </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF847BFA/@@download/ENCFF847BFA.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKR </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF193ERO </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> pseudoreplicated peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K36me3 </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKR </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF605EVL </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K36me3 </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF605EVL/@@download/ENCFF605EVL.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_import\n",
    "dat = dat %>% \n",
    "    dplyr::filter(`File format` %in% c(\"bigWig\", \"bed narrowPeak\")) %>% \n",
    "    dplyr::mutate(`Experiment target` = str_remove(`Experiment target`, \"-human\")) %>%\n",
    "    dplyr::select(\n",
    "        `Assay`,\n",
    "        `Biosample term name`,\n",
    "        `Experiment accession`,\n",
    "        `File analysis title`,\n",
    "        `File accession`,\n",
    "        `File type`,\n",
    "        `Output type`,\n",
    "        `File assembly`,\n",
    "        `Experiment target`,\n",
    "        `File download URL`\n",
    "    ) \n",
    "colnames(dat) = c(\n",
    "    'Assay', 'Biosample', \n",
    "    'Index_Experiment', 'Index_Process', 'Index_File', \n",
    "    'File_Type', 'Output_Type',\n",
    "    'Genome', 'Target', \n",
    "    'File_URL_Download')\n",
    "\n",
    "### assign and show\n",
    "dat_metadata = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 6))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "645b1bed-6280-4c19-9b2c-c356231a3840",
   "metadata": {},
   "source": [
    "## Generate download commands\n",
    "```\n",
    "wget -O FILE URL\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1ece297e-2bb8-4d93-86db-a573431fa2b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> #!/bin/bash </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> echo -n &quot;&quot; &gt; run_download.log.txt </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000EWA.ENCFF461RKK.ChIPseq.H3K4me3.bw https://www.encodeproject.org/files/ENCFF461RKK/@@download/ENCFF461RKK.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000EWA.ENCFF122CSI.ChIPseq.H3K4me3.bed.gz https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000EWB.ENCFF801AHF.ChIPseq.H3K27me3.bed.gz https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000EWB.ENCFF847BFA.ChIPseq.H3K27me3.bw https://www.encodeproject.org/files/ENCFF847BFA/@@download/ENCFF847BFA.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKR.ENCFF193ERO.ChIPseq.H3K36me3.bed.gz https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKR.ENCFF605EVL.ChIPseq.H3K36me3.bw https://www.encodeproject.org/files/ENCFF605EVL/@@download/ENCFF605EVL.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000APE.ENCFF632NQA.ChIPseq.H3K9me3.bw https://www.encodeproject.org/files/ENCFF632NQA/@@download/ENCFF632NQA.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000APE.ENCFF963GZJ.ChIPseq.H3K9me3.bed.gz https://www.encodeproject.org/files/ENCFF963GZJ/@@download/ENCFF963GZJ.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000EWC.ENCFF287LBI.ChIPseq.H3K4me1.bw https://www.encodeproject.org/files/ENCFF287LBI/@@download/ENCFF287LBI.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000EWC.ENCFF540NGG.ChIPseq.H3K4me1.bed.gz https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000EVZ.ENCFF178QDA.ChIPseq.H3K9ac.bw https://www.encodeproject.org/files/ENCFF178QDA/@@download/ENCFF178QDA.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000EVZ.ENCFF148UQI.ChIPseq.H3K9ac.bed.gz https://www.encodeproject.org/files/ENCFF148UQI/@@download/ENCFF148UQI.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR668LDD.ENCFF405ZDL.ChIPseq.H3K4me3.bw https://www.encodeproject.org/files/ENCFF405ZDL/@@download/ENCFF405ZDL.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR668LDD.ENCFF885FQN.ChIPseq.H3K4me3.bed.gz https://www.encodeproject.org/files/ENCFF885FQN/@@download/ENCFF885FQN.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKU.ENCFF767UON.ChIPseq.H3K4me3.bw https://www.encodeproject.org/files/ENCFF767UON/@@download/ENCFF767UON.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKU.ENCFF689QIJ.ChIPseq.H3K4me3.bed.gz https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKX.ENCFF909RKY.ChIPseq.H4K20me1.bed.gz https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKX.ENCFF694ODT.ChIPseq.H4K20me1.bw https://www.encodeproject.org/files/ENCFF694ODT/@@download/ENCFF694ODT.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKQ.ENCFF323WOT.ChIPseq.H3K27me3.bed.gz https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKQ.ENCFF582IMB.ChIPseq.H3K27me3.bw https://www.encodeproject.org/files/ENCFF582IMB/@@download/ENCFF582IMB.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000DWB.ENCFF633OZC.ChIPseq.H3K36me3.bw https://www.encodeproject.org/files/ENCFF633OZC/@@download/ENCFF633OZC.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000DWB.ENCFF561OUZ.ChIPseq.H3K36me3.bed.gz https://www.encodeproject.org/files/ENCFF561OUZ/@@download/ENCFF561OUZ.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKT.ENCFF054RSU.ChIPseq.H3K4me2.bw https://www.encodeproject.org/files/ENCFF054RSU/@@download/ENCFF054RSU.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKT.ENCFF749KLQ.ChIPseq.H3K4me2.bed.gz https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKV.ENCFF239EBH.ChIPseq.H3K9ac.bw https://www.encodeproject.org/files/ENCFF239EBH/@@download/ENCFF239EBH.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKV.ENCFF891CHI.ChIPseq.H3K9ac.bed.gz https://www.encodeproject.org/files/ENCFF891CHI/@@download/ENCFF891CHI.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKS.ENCFF135ZLM.ChIPseq.H3K4me1.bed.gz https://www.encodeproject.org/files/ENCFF135ZLM/@@download/ENCFF135ZLM.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKS.ENCFF457URZ.ChIPseq.H3K4me1.bw https://www.encodeproject.org/files/ENCFF457URZ/@@download/ENCFF457URZ.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKW.ENCFF462AVD.ChIPseq.H3K9me1.bed.gz https://www.encodeproject.org/files/ENCFF462AVD/@@download/ENCFF462AVD.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKW.ENCFF220RGS.ChIPseq.H3K9me1.bw https://www.encodeproject.org/files/ENCFF220RGS/@@download/ENCFF220RGS.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000APC.ENCFF202EVH.ChIPseq.H2AFZ.bw https://www.encodeproject.org/files/ENCFF202EVH/@@download/ENCFF202EVH.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000APC.ENCFF213OTI.ChIPseq.H2AFZ.bed.gz https://www.encodeproject.org/files/ENCFF213OTI/@@download/ENCFF213OTI.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000APD.ENCFF334HSS.ChIPseq.H3K79me2.bw https://www.encodeproject.org/files/ENCFF334HSS/@@download/ENCFF334HSS.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000APD.ENCFF209OQD.ChIPseq.H3K79me2.bed.gz https://www.encodeproject.org/files/ENCFF209OQD/@@download/ENCFF209OQD.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKP.ENCFF544LXB.ChIPseq.H3K27ac.bed.gz https://www.encodeproject.org/files/ENCFF544LXB/@@download/ENCFF544LXB.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000AKP.ENCFF465GBD.ChIPseq.H3K27ac.bw https://www.encodeproject.org/files/ENCFF465GBD/@@download/ENCFF465GBD.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000DWD.ENCFF071GML.ChIPseq.H3K4me3.bw https://www.encodeproject.org/files/ENCFF071GML/@@download/ENCFF071GML.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR000DWD.ENCFF706WUF.ChIPseq.H3K4me3.bed.gz https://www.encodeproject.org/files/ENCFF706WUF/@@download/ENCFF706WUF.bed.gz </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### init\n",
    "dat = dat_metadata\n",
    "\n",
    "### setup download file name and wget command\n",
    "dat = dat %>%\n",
    "    dplyr::mutate(\n",
    "        File_Ext1 = ifelse(str_detect(File_Type, \"bed\"), \"bed.gz\", \"bw\"),\n",
    "        File_Ext2 = ifelse(str_detect(File_Type, \"bed\"), \"bed.gz\", \"bigWig\")\n",
    "    ) %>%\n",
    "    dplyr::mutate(\n",
    "        File_Name = paste(\n",
    "            Biosample, \n",
    "            \"hg38\", \n",
    "            Index_Experiment, \n",
    "            Index_File,\n",
    "            \"ChIPseq\",\n",
    "            Target,\n",
    "            File_Ext1, \n",
    "            sep=\".\")\n",
    "    ) %>%\n",
    "    dplyr::mutate(\n",
    "        CMD = paste(\"wget\", \"--append-output=run_download.log.txt\", \"-O\", File_Name, File_URL_Download)\n",
    "    )\n",
    "\n",
    "### add Shebang and initial commands\n",
    "#dat = dat %>% dplyr::select(Assay, Biosample, Index_Experiment, Index_File, File_Name, CMD)\n",
    "dat = dat %>% dplyr::select(CMD)\n",
    "dat = rbind('echo -n \"\" > run_download.log.txt', dat)\n",
    "colnames(dat) = \"#!/bin/bash\"\n",
    "\n",
    "### assign and show\n",
    "dat_cmd = dat\n",
    "fun_display_table(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa976699-de43-4fae-b233-f35ed0072460",
   "metadata": {},
   "source": [
    "## Save to script\n",
    "Save the command lines for each row into a bash script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "33b1014f-0f1b-4f9c-a5e9-b7ab80c7ca88",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set output path\n",
    "txt_fdiry = file.path(FD_DAT, \"external\", \"encode_chipseq_subset\")\n",
    "txt_fname = \"run_download.sh\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### save table\n",
    "dir.create(txt_fdiry, showWarnings = FALSE)\n",
    "dat = dat_cmd\n",
    "write_tsv(dat, txt_fpath)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e8f28e25-2ae6-41ac-877b-34b96d3c5bf6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R (alex_py3)",
   "language": "R",
   "name": "ir_alex_py3"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
