{
 "cells": [
  {
   "cell_type": "raw",
   "id": "e94e1f2a-986f-4d0a-a379-ca5d1b85e135",
   "metadata": {},
   "source": [
    "---\n",
    "title: Region annotation 11 (ASTARR MACS peaks)\n",
    "subtitle: Summarize annotations (Main)\n",
    "format:\n",
    "    html:\n",
    "        code-fold:  true\n",
    "        code-tools: true\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6dbb8538-ea31-4941-8c3b-236950cf2da3",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "56345603-c707-482c-bdec-f62a12dba9ef",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are working on        Singularity: singularity_proj_encode_fcc \n",
      "BASE DIRECTORY (FD_BASE): /data/reddylab/Kuei \n",
      "REPO DIRECTORY (FD_REPO): /data/reddylab/Kuei/repo \n",
      "WORK DIRECTORY (FD_WORK): /data/reddylab/Kuei/work \n",
      "DATA DIRECTORY (FD_DATA): /data/reddylab/Kuei/data \n",
      "\n",
      "You are working with      ENCODE FCC \n",
      "PATH OF PROJECT (FD_PRJ): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC \n",
      "PROJECT RESULTS (FD_RES): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results \n",
      "PROJECT SCRIPTS (FD_EXE): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/scripts \n",
      "PROJECT DATA    (FD_DAT): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/data \n",
      "PROJECT NOTE    (FD_NBK): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/notebooks \n",
      "PROJECT DOCS    (FD_DOC): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/docs \n",
      "PROJECT LOG     (FD_LOG): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/log \n",
      "PROJECT REF     (FD_REF): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/references \n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(source(\"../run_config_project_sing.R\")))\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92ed551b-5482-43ae-824d-247b4eebcb51",
   "metadata": {},
   "source": [
    "**Set global variables**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1e8434ca-6e72-49b4-8c3e-a28d70775e7b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metadata.label.astarr_macs_merge.tsv \n",
      "metadata.label.chipseq_histone.tsv \n",
      "metadata.label.chipseq_subset.tsv \n",
      "metadata.label.chipseq_tf_full.tsv \n",
      "metadata.label.hic.tsv \n",
      "metadata.label.main.tsv \n",
      "metadata.label.ocr.tsv \n"
     ]
    }
   ],
   "source": [
    "txt_fdiry = file.path(FD_RES, \"region\", \"summary\")\n",
    "vec = dir(txt_fdiry)\n",
    "for (txt in vec){cat(txt, \"\\n\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "940b5646-804e-4d13-aefd-0832f9e51354",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metadata.label.tsv \n"
     ]
    }
   ],
   "source": [
    "txt_fdiry = file.path(FD_RES, \"assay_chipseq\", \"encode_chipseq_histone\", \"summary\")\n",
    "vec = dir(txt_fdiry)\n",
    "for (txt in vec){cat(txt, \"\\n\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2e6d7e8e-16bc-49a8-b29e-ac331d2be154",
   "metadata": {},
   "outputs": [],
   "source": [
    "TXT_FDIRY_REGION_LABEL_A = file.path(FD_RES, \"region\", \"summary\")\n",
    "TXT_FNAME_REGION_LABEL_A = \"metadata.label.astarr_macs_merge.tsv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6af4013c-b4c4-4766-8615-b0fa6f4a6d92",
   "metadata": {},
   "outputs": [],
   "source": [
    "TXT_FDIRY_REGION_LABEL_B = file.path(FD_RES, \"assay_chipseq\", \"encode_chipseq_tf_full\", \"summary\")\n",
    "TXT_FNAME_REGION_LABEL_B = \"metadata.label.tsv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b819c043-d6c2-424b-a3f4-f8dd33b53d39",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bda5575f-fe90-46f3-b9bc-de9d1a278ead",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A spec_tbl_df: 2 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Folder</th><th scope=col>FName</th><th scope=col>Label</th><th scope=col>FPath</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>fcc_astarr_macs_merge</td><td>K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz</td><td>fcc_astarr_macs_input_overlap</td><td>/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/fcc_astarr_macs_merge/K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz</td></tr>\n",
       "\t<tr><td>fcc_astarr_macs_merge</td><td>K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz       </td><td>fcc_astarr_macs_input_union  </td><td>/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/fcc_astarr_macs_merge/K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz       </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A spec\\_tbl\\_df: 2 × 4\n",
       "\\begin{tabular}{llll}\n",
       " Folder & FName & Label & FPath\\\\\n",
       " <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t fcc\\_astarr\\_macs\\_merge & K562.hg38.ASTARR.macs.KS91.input.rep\\_all.max\\_overlaps.q5.bed.gz & fcc\\_astarr\\_macs\\_input\\_overlap & /data/reddylab/Kuei/repo/Proj\\_ENCODE\\_FCC/results/region/fcc\\_astarr\\_macs\\_merge/K562.hg38.ASTARR.macs.KS91.input.rep\\_all.max\\_overlaps.q5.bed.gz\\\\\n",
       "\t fcc\\_astarr\\_macs\\_merge & K562.hg38.ASTARR.macs.KS91.input.rep\\_all.union.q5.bed.gz        & fcc\\_astarr\\_macs\\_input\\_union   & /data/reddylab/Kuei/repo/Proj\\_ENCODE\\_FCC/results/region/fcc\\_astarr\\_macs\\_merge/K562.hg38.ASTARR.macs.KS91.input.rep\\_all.union.q5.bed.gz       \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A spec_tbl_df: 2 × 4\n",
       "\n",
       "| Folder &lt;chr&gt; | FName &lt;chr&gt; | Label &lt;chr&gt; | FPath &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| fcc_astarr_macs_merge | K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz | fcc_astarr_macs_input_overlap | /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/fcc_astarr_macs_merge/K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz |\n",
       "| fcc_astarr_macs_merge | K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz        | fcc_astarr_macs_input_union   | /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/fcc_astarr_macs_merge/K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz        |\n",
       "\n"
      ],
      "text/plain": [
       "  Folder               \n",
       "1 fcc_astarr_macs_merge\n",
       "2 fcc_astarr_macs_merge\n",
       "  FName                                                          \n",
       "1 K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz\n",
       "2 K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz       \n",
       "  Label                        \n",
       "1 fcc_astarr_macs_input_overlap\n",
       "2 fcc_astarr_macs_input_union  \n",
       "  FPath                                                                                                                                        \n",
       "1 /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/fcc_astarr_macs_merge/K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz\n",
       "2 /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/fcc_astarr_macs_merge/K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "txt_fdiry = TXT_FDIRY_REGION_LABEL_A\n",
    "txt_fname = TXT_FNAME_REGION_LABEL_A\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "dat_region_label_A = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "dat_region_label_A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "6b47fddb-6e7d-4296-bfc5-e58730ebdedd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Folder</th><th scope=col>FName</th><th scope=col>Label</th><th scope=col>FPath</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>encode_chipseq_tf_full</td><td>ENCFF465ZAR.bigWig</td><td>encode_chipseq_ZNF347_ENCFF465ZAR</td><td>/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF465ZAR.bigWig</td></tr>\n",
       "\t<tr><td>encode_chipseq_tf_full</td><td>ENCFF178FEN.bigWig</td><td>encode_chipseq_ZBTB9_ENCFF178FEN </td><td>/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF178FEN.bigWig</td></tr>\n",
       "\t<tr><td>encode_chipseq_tf_full</td><td>ENCFF040PTP.bigWig</td><td>encode_chipseq_AFF4_ENCFF040PTP  </td><td>/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF040PTP.bigWig</td></tr>\n",
       "\t<tr><td>encode_chipseq_tf_full</td><td>ENCFF193LUC.bigWig</td><td>encode_chipseq_ZFX_ENCFF193LUC   </td><td>/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF193LUC.bigWig</td></tr>\n",
       "\t<tr><td>encode_chipseq_tf_full</td><td>ENCFF922ZTM.bigWig</td><td>encode_chipseq_KAT7_ENCFF922ZTM  </td><td>/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF922ZTM.bigWig</td></tr>\n",
       "\t<tr><td>encode_chipseq_tf_full</td><td>ENCFF118QVV.bigWig</td><td>encode_chipseq_ZNF232_ENCFF118QVV</td><td>/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF118QVV.bigWig</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 4\n",
       "\\begin{tabular}{llll}\n",
       " Folder & FName & Label & FPath\\\\\n",
       " <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t encode\\_chipseq\\_tf\\_full & ENCFF465ZAR.bigWig & encode\\_chipseq\\_ZNF347\\_ENCFF465ZAR & /data/reddylab/Kuei/repo/Proj\\_ENCODE\\_FCC/results/assay\\_chipseq/encode\\_chipseq\\_tf\\_full/signal\\_fold\\_change/ENCFF465ZAR.bigWig\\\\\n",
       "\t encode\\_chipseq\\_tf\\_full & ENCFF178FEN.bigWig & encode\\_chipseq\\_ZBTB9\\_ENCFF178FEN  & /data/reddylab/Kuei/repo/Proj\\_ENCODE\\_FCC/results/assay\\_chipseq/encode\\_chipseq\\_tf\\_full/signal\\_fold\\_change/ENCFF178FEN.bigWig\\\\\n",
       "\t encode\\_chipseq\\_tf\\_full & ENCFF040PTP.bigWig & encode\\_chipseq\\_AFF4\\_ENCFF040PTP   & /data/reddylab/Kuei/repo/Proj\\_ENCODE\\_FCC/results/assay\\_chipseq/encode\\_chipseq\\_tf\\_full/signal\\_fold\\_change/ENCFF040PTP.bigWig\\\\\n",
       "\t encode\\_chipseq\\_tf\\_full & ENCFF193LUC.bigWig & encode\\_chipseq\\_ZFX\\_ENCFF193LUC    & /data/reddylab/Kuei/repo/Proj\\_ENCODE\\_FCC/results/assay\\_chipseq/encode\\_chipseq\\_tf\\_full/signal\\_fold\\_change/ENCFF193LUC.bigWig\\\\\n",
       "\t encode\\_chipseq\\_tf\\_full & ENCFF922ZTM.bigWig & encode\\_chipseq\\_KAT7\\_ENCFF922ZTM   & /data/reddylab/Kuei/repo/Proj\\_ENCODE\\_FCC/results/assay\\_chipseq/encode\\_chipseq\\_tf\\_full/signal\\_fold\\_change/ENCFF922ZTM.bigWig\\\\\n",
       "\t encode\\_chipseq\\_tf\\_full & ENCFF118QVV.bigWig & encode\\_chipseq\\_ZNF232\\_ENCFF118QVV & /data/reddylab/Kuei/repo/Proj\\_ENCODE\\_FCC/results/assay\\_chipseq/encode\\_chipseq\\_tf\\_full/signal\\_fold\\_change/ENCFF118QVV.bigWig\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 4\n",
       "\n",
       "| Folder &lt;chr&gt; | FName &lt;chr&gt; | Label &lt;chr&gt; | FPath &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| encode_chipseq_tf_full | ENCFF465ZAR.bigWig | encode_chipseq_ZNF347_ENCFF465ZAR | /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF465ZAR.bigWig |\n",
       "| encode_chipseq_tf_full | ENCFF178FEN.bigWig | encode_chipseq_ZBTB9_ENCFF178FEN  | /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF178FEN.bigWig |\n",
       "| encode_chipseq_tf_full | ENCFF040PTP.bigWig | encode_chipseq_AFF4_ENCFF040PTP   | /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF040PTP.bigWig |\n",
       "| encode_chipseq_tf_full | ENCFF193LUC.bigWig | encode_chipseq_ZFX_ENCFF193LUC    | /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF193LUC.bigWig |\n",
       "| encode_chipseq_tf_full | ENCFF922ZTM.bigWig | encode_chipseq_KAT7_ENCFF922ZTM   | /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF922ZTM.bigWig |\n",
       "| encode_chipseq_tf_full | ENCFF118QVV.bigWig | encode_chipseq_ZNF232_ENCFF118QVV | /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF118QVV.bigWig |\n",
       "\n"
      ],
      "text/plain": [
       "  Folder                 FName              Label                            \n",
       "1 encode_chipseq_tf_full ENCFF465ZAR.bigWig encode_chipseq_ZNF347_ENCFF465ZAR\n",
       "2 encode_chipseq_tf_full ENCFF178FEN.bigWig encode_chipseq_ZBTB9_ENCFF178FEN \n",
       "3 encode_chipseq_tf_full ENCFF040PTP.bigWig encode_chipseq_AFF4_ENCFF040PTP  \n",
       "4 encode_chipseq_tf_full ENCFF193LUC.bigWig encode_chipseq_ZFX_ENCFF193LUC   \n",
       "5 encode_chipseq_tf_full ENCFF922ZTM.bigWig encode_chipseq_KAT7_ENCFF922ZTM  \n",
       "6 encode_chipseq_tf_full ENCFF118QVV.bigWig encode_chipseq_ZNF232_ENCFF118QVV\n",
       "  FPath                                                                                                                      \n",
       "1 /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF465ZAR.bigWig\n",
       "2 /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF178FEN.bigWig\n",
       "3 /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF040PTP.bigWig\n",
       "4 /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF193LUC.bigWig\n",
       "5 /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF922ZTM.bigWig\n",
       "6 /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_chipseq/encode_chipseq_tf_full/signal_fold_change/ENCFF118QVV.bigWig"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "txt_fdiry = TXT_FDIRY_REGION_LABEL_B\n",
    "txt_fname = TXT_FNAME_REGION_LABEL_B\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "dat_region_label_B = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "dat_region_label_B %>% head"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fd4234d1-ff25-4bbe-9fe5-c82895aee28d",
   "metadata": {},
   "source": [
    "[Document of bigWigAverageOverBed](https://bioweb.pasteur.fr/docs/modules/BigWig_tools/v4/bigWigAverageOverBed.txt)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b7ad3c96-da81-4a2e-9e7b-6d0ebf47902f",
   "metadata": {},
   "source": [
    "```\n",
    "bigWigAverageOverBed - Compute average score of big wig over each bed, which may have introns.\n",
    "usage:\n",
    "   bigWigAverageOverBed in.bw in.bed out.tab\n",
    "The output columns are:\n",
    "   name - name field from bed, which should be unique\n",
    "   size - size of bed (sum of exon sizes\n",
    "   covered - # bases within exons covered by bigWig\n",
    "   sum - sum of values over all bases covered\n",
    "   mean0 - average over bases with non-covered bases counting as zeroes\n",
    "   mean - average over just covered bases\n",
    "Options:\n",
    "   -bedOut=out.bed - Make output bed that is echo of input bed but with mean column appended\n",
    "   -sampleAroundCenter=N - Take sample at region N bases wide centered around bed item, rather\n",
    "                     than the usual sample in the bed item.\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "53bbf81f-5fff-46ac-9dc4-67b91434cc0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "vec_txt_cname = c(\n",
    "    \"Region\",\n",
    "    \"Size\",\n",
    "    \"Covered\",\n",
    "    \"Sum\",\n",
    "    \"Mean0\",\n",
    "    \"Mean\",\n",
    "    \"Min\",\n",
    "    \"Max\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a2dcdeaa-d9eb-4736-967d-59ae7056bfda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$`1`</dt>\n",
       "\t\t<dd><dl>\n",
       "\t<dt>$Folder</dt>\n",
       "\t\t<dd>'fcc_astarr_macs_merge'</dd>\n",
       "\t<dt>$FName</dt>\n",
       "\t\t<dd>'K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz'</dd>\n",
       "\t<dt>$Label</dt>\n",
       "\t\t<dd>'fcc_astarr_macs_input_overlap'</dd>\n",
       "\t<dt>$FPath</dt>\n",
       "\t\t<dd>'/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/fcc_astarr_macs_merge/K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz'</dd>\n",
       "</dl>\n",
       "</dd>\n",
       "\t<dt>$`2`</dt>\n",
       "\t\t<dd><dl>\n",
       "\t<dt>$Folder</dt>\n",
       "\t\t<dd>'fcc_astarr_macs_merge'</dd>\n",
       "\t<dt>$FName</dt>\n",
       "\t\t<dd>'K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz'</dd>\n",
       "\t<dt>$Label</dt>\n",
       "\t\t<dd>'fcc_astarr_macs_input_union'</dd>\n",
       "\t<dt>$FPath</dt>\n",
       "\t\t<dd>'/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/fcc_astarr_macs_merge/K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz'</dd>\n",
       "</dl>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$`1`] \\begin{description}\n",
       "\\item[\\$Folder] 'fcc\\_astarr\\_macs\\_merge'\n",
       "\\item[\\$FName] 'K562.hg38.ASTARR.macs.KS91.input.rep\\_all.max\\_overlaps.q5.bed.gz'\n",
       "\\item[\\$Label] 'fcc\\_astarr\\_macs\\_input\\_overlap'\n",
       "\\item[\\$FPath] '/data/reddylab/Kuei/repo/Proj\\_ENCODE\\_FCC/results/region/fcc\\_astarr\\_macs\\_merge/K562.hg38.ASTARR.macs.KS91.input.rep\\_all.max\\_overlaps.q5.bed.gz'\n",
       "\\end{description}\n",
       "\n",
       "\\item[\\$`2`] \\begin{description}\n",
       "\\item[\\$Folder] 'fcc\\_astarr\\_macs\\_merge'\n",
       "\\item[\\$FName] 'K562.hg38.ASTARR.macs.KS91.input.rep\\_all.union.q5.bed.gz'\n",
       "\\item[\\$Label] 'fcc\\_astarr\\_macs\\_input\\_union'\n",
       "\\item[\\$FPath] '/data/reddylab/Kuei/repo/Proj\\_ENCODE\\_FCC/results/region/fcc\\_astarr\\_macs\\_merge/K562.hg38.ASTARR.macs.KS91.input.rep\\_all.union.q5.bed.gz'\n",
       "\\end{description}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$`1`\n",
       ":   $Folder\n",
       ":   'fcc_astarr_macs_merge'\n",
       "$FName\n",
       ":   'K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz'\n",
       "$Label\n",
       ":   'fcc_astarr_macs_input_overlap'\n",
       "$FPath\n",
       ":   '/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/fcc_astarr_macs_merge/K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz'\n",
       "\n",
       "\n",
       "\n",
       "$`2`\n",
       ":   $Folder\n",
       ":   'fcc_astarr_macs_merge'\n",
       "$FName\n",
       ":   'K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz'\n",
       "$Label\n",
       ":   'fcc_astarr_macs_input_union'\n",
       "$FPath\n",
       ":   '/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/fcc_astarr_macs_merge/K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz'\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$`1`\n",
       "$`1`$Folder\n",
       "[1] \"fcc_astarr_macs_merge\"\n",
       "\n",
       "$`1`$FName\n",
       "[1] \"K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz\"\n",
       "\n",
       "$`1`$Label\n",
       "[1] \"fcc_astarr_macs_input_overlap\"\n",
       "\n",
       "$`1`$FPath\n",
       "[1] \"/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/fcc_astarr_macs_merge/K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz\"\n",
       "\n",
       "\n",
       "$`2`\n",
       "$`2`$Folder\n",
       "[1] \"fcc_astarr_macs_merge\"\n",
       "\n",
       "$`2`$FName\n",
       "[1] \"K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz\"\n",
       "\n",
       "$`2`$Label\n",
       "[1] \"fcc_astarr_macs_input_union\"\n",
       "\n",
       "$`2`$FPath\n",
       "[1] \"/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/fcc_astarr_macs_merge/K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz\"\n",
       "\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_region_label_A\n",
    "lst = split(dat, 1:nrow(dat))\n",
    "lst = lapply(lst, as.list)\n",
    "lst_region_label_A = lst\n",
    "lst"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "9de44148-5b18-4257-91ce-df63839340f7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "encode_chipseq_tf_full \n"
     ]
    }
   ],
   "source": [
    "dat = dat_region_label_B\n",
    "vec = unique(dat$Folder)\n",
    "vec_txt_region_folder_B = vec\n",
    "for(txt in vec){cat(txt, \"\\n\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "979d0b3d-c904-43c2-8b01-69079adc8c18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>1</li><li>2</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1\n",
       "\\item 2\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1\n",
       "2. 2\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1 2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "seq_along(lst_region_label_A)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "01fe7ec6-e684-4896-8039-bad3a6f6cd47",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===========================================\n",
      "Read file... \n",
      "Region A: fcc_astarr_macs_merge | fcc_astarr_macs_input_overlap \n",
      "Region B: encode_chipseq_tf_full \n",
      "FDiry:    \n",
      "/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_chipseq/fcc_astarr_macs_input_overlap/encode_chipseq_tf_full \n",
      "FName:    \n",
      "Count:    720 \n",
      "[1] \"fcc_astarr_macs_input_overlap.encode_chipseq_ADNP_ENCFF852FRV.tsv\"    \n",
      "[2] \"fcc_astarr_macs_input_overlap.encode_chipseq_AFF1_ENCFF003ZRP.tsv\"    \n",
      "[3] \"fcc_astarr_macs_input_overlap.encode_chipseq_AFF1_ENCFF085LMT.tsv\"    \n",
      "[4] \"fcc_astarr_macs_input_overlap.encode_chipseq_AFF4_ENCFF040PTP.tsv\"    \n",
      "[5] \"fcc_astarr_macs_input_overlap.encode_chipseq_AGO1_ENCFF583LSH.tsv\"    \n",
      "[6] \"fcc_astarr_macs_input_overlap.encode_chipseq_ARHGAP35_ENCFF452FAK.tsv\"\n",
      "\n",
      "Summarize number of rows: \n",
      "#Region\n",
      "150042 \n",
      "   720 \n",
      "\n",
      "Check if empty table exist: \n",
      "No filtering. \n",
      "\n",
      "Save file... \n",
      "FDiry: /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_chipseq/fcc_astarr_macs_input_overlap/summary \n",
      "FName: matrix.coverage.encode_chipseq_tf_full.by_index.mean0.tsv \n",
      "\n",
      "===========================================\n",
      "Read file... \n",
      "Region A: fcc_astarr_macs_merge | fcc_astarr_macs_input_union \n",
      "Region B: encode_chipseq_tf_full \n",
      "FDiry:    \n",
      "/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_chipseq/fcc_astarr_macs_input_union/encode_chipseq_tf_full \n",
      "FName:    \n",
      "Count:    720 \n",
      "[1] \"fcc_astarr_macs_input_union.encode_chipseq_ADNP_ENCFF852FRV.tsv\"    \n",
      "[2] \"fcc_astarr_macs_input_union.encode_chipseq_AFF1_ENCFF003ZRP.tsv\"    \n",
      "[3] \"fcc_astarr_macs_input_union.encode_chipseq_AFF1_ENCFF085LMT.tsv\"    \n",
      "[4] \"fcc_astarr_macs_input_union.encode_chipseq_AFF4_ENCFF040PTP.tsv\"    \n",
      "[5] \"fcc_astarr_macs_input_union.encode_chipseq_AGO1_ENCFF583LSH.tsv\"    \n",
      "[6] \"fcc_astarr_macs_input_union.encode_chipseq_ARHGAP35_ENCFF452FAK.tsv\"\n",
      "\n",
      "Summarize number of rows: \n",
      "#Region\n",
      "246852 \n",
      "   720 \n",
      "\n",
      "Check if empty table exist: \n",
      "No filtering. \n",
      "\n",
      "Save file... \n",
      "FDiry: /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_chipseq/fcc_astarr_macs_input_union/summary \n",
      "FName: matrix.coverage.encode_chipseq_tf_full.by_index.mean0.tsv \n",
      "\n"
     ]
    }
   ],
   "source": [
    "### loop through region A\n",
    "for (idx in seq_along(lst_region_label_A)){\n",
    "    \n",
    "    ### get region A\n",
    "    lst = lst_region_label_A[[idx]]\n",
    "    txt_region_folder_A = lst$Folder\n",
    "    txt_region_label_A  = lst$Label\n",
    "\n",
    "    ### loop through region B\n",
    "    for (txt_region_folder_B in vec_txt_region_folder_B){\n",
    "        \n",
    "        ### get directory of annotation table\n",
    "        txt_fdiry = file.path(\n",
    "            FD_RES, \n",
    "            \"region_coverage_chipseq\",\n",
    "            txt_region_label_A, \n",
    "            txt_region_folder_B\n",
    "        )\n",
    "        txt_fname = \"*tsv\"\n",
    "        txt_fglob = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "        ### get annotation table files\n",
    "        vec_txt_fpath = Sys.glob(txt_fglob)\n",
    "        vec_txt_fname = basename(vec_txt_fpath)\n",
    "\n",
    "        ### show progress\n",
    "        cat(\"===========================================\\n\")\n",
    "        cat(\"Read file...\", \"\\n\")\n",
    "        cat(\"Region A:\", txt_region_folder_A, \"|\", txt_region_label_A, \"\\n\")\n",
    "        cat(\"Region B:\", txt_region_folder_B, \"\\n\")\n",
    "        cat(\"FDiry:   \", \"\\n\")\n",
    "        cat(txt_fdiry, \"\\n\")\n",
    "        cat(\"FName:   \", \"\\n\")\n",
    "        cat(\"Count:   \", length(vec_txt_fname), \"\\n\")\n",
    "        print(head(vec_txt_fname))\n",
    "        cat(\"\\n\")\n",
    "        flush.console()\n",
    "\n",
    "        ### import annotation tables\n",
    "        lst = lapply(vec_txt_fname, function(txt_fname){\n",
    "            ### set directory\n",
    "            txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "            \n",
    "            ### get annotation of region A and region B\n",
    "            vec = str_split(txt_fname, \"\\\\.\")[[1]]\n",
    "            txt_annot_A = vec[1]\n",
    "            txt_annot_B = vec[2]\n",
    "\n",
    "            ### get target and file index\n",
    "            ### ex: txt_annot_B: \"encode_chipseq_LARP7_ENCFF689OKC\"\n",
    "            vec = unlist(str_split(txt_annot_B, pattern = \"_\"))\n",
    "            txt_target = vec[3]\n",
    "            txt_index  = vec[4]\n",
    "            txt_label  = paste(txt_target, txt_index, sep = \"_\")\n",
    "            \n",
    "            ### read table and add annotation labels\n",
    "            dat = read_tsv(txt_fpath, col_names = vec_txt_cname, show_col_types = FALSE)\n",
    "            dat = dat %>% dplyr::select(Region, Mean0)\n",
    "            colnames(dat) = c(\"Region\", txt_label)\n",
    "            return(dat)\n",
    "        }) \n",
    "        lst_dat_region_score_import = lst\n",
    "        flush.console()\n",
    "        \n",
    "        ### Check if there is any empty table\n",
    "        cat(\"Summarize number of rows:\", \"\\n\")\n",
    "        lst = lst_dat_region_score_import\n",
    "        lst = lapply(lst, nrow)\n",
    "        vec = unlist(lst)\n",
    "        vec_num_nrow = vec\n",
    "\n",
    "        ### show progress\n",
    "        print(table(vec, dnn=\"#Region\"))\n",
    "        cat(\"\\n\")\n",
    "        \n",
    "        ### filtering out empty table if it exist\n",
    "        cat(\"Check if empty table exist:\", \"\\n\")\n",
    "        vec = vec_num_nrow\n",
    "        idx = which(vec == 0)\n",
    "        \n",
    "        if (length(idx) > 0){\n",
    "            ### filter\n",
    "            lst_dat = lst_dat_region_score_import[-idx]\n",
    "\n",
    "            ### show progress\n",
    "            cat(\"Filter out:\", \"\\n\")\n",
    "            print(vec[idx])\n",
    "            cat(\"\\n\")\n",
    "            \n",
    "            cat(\"After filtering:\", \"\\n\")\n",
    "            lst = lst_dat\n",
    "            lst = lapply(lst, nrow)\n",
    "            vec = unlist(lst)\n",
    "            print(table(vec))\n",
    "            cat(\"\\n\")\n",
    "        } else {\n",
    "            ### no filter\n",
    "            lst_dat = lst_dat_region_score_import\n",
    "\n",
    "            ### show progress\n",
    "            cat(\"No filtering.\", \"\\n\")\n",
    "            cat(\"\\n\")\n",
    "        }\n",
    "        lst_dat_region_score_filter = lst_dat\n",
    "        flush.console()\n",
    "        \n",
    "        ### join columns for each chip-seq file\n",
    "        fun = function(dat1, dat2){\n",
    "            dat = dplyr::full_join(dat1, dat2, by = \"Region\")\n",
    "            return(dat)\n",
    "        }\n",
    "        lst = lst_dat_region_score_filter\n",
    "        dat = Reduce(fun, lst)\n",
    "        dat_region_score_matrix = dat\n",
    "   \n",
    "        ### set file directory\n",
    "        txt_fdiry = file.path(\n",
    "            FD_RES, \n",
    "            \"region_coverage_chipseq\",\n",
    "            txt_region_label_A,\n",
    "            \"summary\"\n",
    "        )\n",
    "        txt_fname = paste(\"matrix\", \"coverage\",  txt_region_folder_B, \"by_index\", \"mean0\", \"tsv\", sep = \".\")\n",
    "        txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "        ### show progress\n",
    "        cat(\"Save file...\",  \"\\n\")\n",
    "        cat(\"FDiry:\", txt_fdiry, \"\\n\")\n",
    "        cat(\"FName:\", txt_fname, \"\\n\")\n",
    "        cat(\"\\n\")\n",
    "        flush.console()\n",
    "        \n",
    "        ### write tables\n",
    "        dir.create(txt_fdiry, showWarnings = FALSE)\n",
    "        dat = dat_region_score_matrix\n",
    "        write_tsv(dat, txt_fpath)\n",
    "    } ### end loop region B\n",
    "} ### end loop region A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ef4756a-3497-41df-b2e7-eb88383a2fdf",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c491550e-2a2e-4799-97e5-4b5f95a66727",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5efe9539-517f-4056-a8e1-2e11c02dbed1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5345ef5f-6143-4b25-8de3-5d392dc6d98a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "04c3ae6f-6039-4b5d-9daa-e9120c28904c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "9c446366-b49a-4a26-8cb8-38a313ef51a1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Region</th><th scope=col>AFF1_ENCFF003ZRP</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1:10038-10405  </td><td>0.116788</td></tr>\n",
       "\t<tr><td>chr1:14282-14614  </td><td>0.000000</td></tr>\n",
       "\t<tr><td>chr1:16025-16338  </td><td>0.330535</td></tr>\n",
       "\t<tr><td>chr1:17288-17689  </td><td>0.000000</td></tr>\n",
       "\t<tr><td>chr1:28934-29499  </td><td>0.000000</td></tr>\n",
       "\t<tr><td>chr1:115429-115969</td><td>2.354290</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Region & AFF1\\_ENCFF003ZRP\\\\\n",
       " <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr1:10038-10405   & 0.116788\\\\\n",
       "\t chr1:14282-14614   & 0.000000\\\\\n",
       "\t chr1:16025-16338   & 0.330535\\\\\n",
       "\t chr1:17288-17689   & 0.000000\\\\\n",
       "\t chr1:28934-29499   & 0.000000\\\\\n",
       "\t chr1:115429-115969 & 2.354290\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| Region &lt;chr&gt; | AFF1_ENCFF003ZRP &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| chr1:10038-10405   | 0.116788 |\n",
       "| chr1:14282-14614   | 0.000000 |\n",
       "| chr1:16025-16338   | 0.330535 |\n",
       "| chr1:17288-17689   | 0.000000 |\n",
       "| chr1:28934-29499   | 0.000000 |\n",
       "| chr1:115429-115969 | 2.354290 |\n",
       "\n"
      ],
      "text/plain": [
       "  Region             AFF1_ENCFF003ZRP\n",
       "1 chr1:10038-10405   0.116788        \n",
       "2 chr1:14282-14614   0.000000        \n",
       "3 chr1:16025-16338   0.330535        \n",
       "4 chr1:17288-17689   0.000000        \n",
       "5 chr1:28934-29499   0.000000        \n",
       "6 chr1:115429-115969 2.354290        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = lst_dat_region_score_filter[[2]]\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "2214540e-56e1-489e-bdaa-a45e6bf1f809",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Region</th><th scope=col>ADNP_ENCFF852FRV</th><th scope=col>AFF1_ENCFF003ZRP</th><th scope=col>AFF1_ENCFF085LMT</th><th scope=col>AFF4_ENCFF040PTP</th><th scope=col>AGO1_ENCFF583LSH</th><th scope=col>ARHGAP35_ENCFF452FAK</th><th scope=col>ARID1B_ENCFF980AHZ</th><th scope=col>ARID2_ENCFF748CGJ</th><th scope=col>ARID3A_ENCFF415DOV</th><th scope=col>ARID3B_ENCFF740OGZ</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1:10038-10405  </td><td>0.972607</td><td>0.116788</td><td>0.0000000</td><td>0.345775</td><td>0.000000</td><td>0.000000</td><td>0.0000000</td><td>0.000000</td><td>0.693779</td><td>0.0000000</td></tr>\n",
       "\t<tr><td>chr1:14282-14614  </td><td>0.000000</td><td>0.000000</td><td>0.0000000</td><td>0.000000</td><td>0.000000</td><td>0.000000</td><td>0.0982520</td><td>0.000000</td><td>0.000000</td><td>0.1251570</td></tr>\n",
       "\t<tr><td>chr1:16025-16338  </td><td>0.189624</td><td>0.330535</td><td>0.0604299</td><td>0.676515</td><td>0.987538</td><td>0.416025</td><td>0.0300948</td><td>0.508451</td><td>0.544713</td><td>0.0706043</td></tr>\n",
       "\t<tr><td>chr1:17288-17689  </td><td>0.190180</td><td>0.000000</td><td>0.5865760</td><td>0.911477</td><td>1.373140</td><td>0.417423</td><td>0.4895330</td><td>0.664600</td><td>0.000000</td><td>1.0071700</td></tr>\n",
       "\t<tr><td>chr1:28934-29499  </td><td>0.000000</td><td>0.000000</td><td>0.0000000</td><td>0.000000</td><td>0.000000</td><td>0.000000</td><td>0.0406541</td><td>0.000000</td><td>0.000000</td><td>0.0000000</td></tr>\n",
       "\t<tr><td>chr1:115429-115969</td><td>1.264990</td><td>2.354290</td><td>3.3698500</td><td>0.839830</td><td>0.000000</td><td>0.510867</td><td>8.4608800</td><td>0.797594</td><td>7.387330</td><td>0.9319340</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 11\n",
       "\\begin{tabular}{lllllllllll}\n",
       " Region & ADNP\\_ENCFF852FRV & AFF1\\_ENCFF003ZRP & AFF1\\_ENCFF085LMT & AFF4\\_ENCFF040PTP & AGO1\\_ENCFF583LSH & ARHGAP35\\_ENCFF452FAK & ARID1B\\_ENCFF980AHZ & ARID2\\_ENCFF748CGJ & ARID3A\\_ENCFF415DOV & ARID3B\\_ENCFF740OGZ\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr1:10038-10405   & 0.972607 & 0.116788 & 0.0000000 & 0.345775 & 0.000000 & 0.000000 & 0.0000000 & 0.000000 & 0.693779 & 0.0000000\\\\\n",
       "\t chr1:14282-14614   & 0.000000 & 0.000000 & 0.0000000 & 0.000000 & 0.000000 & 0.000000 & 0.0982520 & 0.000000 & 0.000000 & 0.1251570\\\\\n",
       "\t chr1:16025-16338   & 0.189624 & 0.330535 & 0.0604299 & 0.676515 & 0.987538 & 0.416025 & 0.0300948 & 0.508451 & 0.544713 & 0.0706043\\\\\n",
       "\t chr1:17288-17689   & 0.190180 & 0.000000 & 0.5865760 & 0.911477 & 1.373140 & 0.417423 & 0.4895330 & 0.664600 & 0.000000 & 1.0071700\\\\\n",
       "\t chr1:28934-29499   & 0.000000 & 0.000000 & 0.0000000 & 0.000000 & 0.000000 & 0.000000 & 0.0406541 & 0.000000 & 0.000000 & 0.0000000\\\\\n",
       "\t chr1:115429-115969 & 1.264990 & 2.354290 & 3.3698500 & 0.839830 & 0.000000 & 0.510867 & 8.4608800 & 0.797594 & 7.387330 & 0.9319340\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 11\n",
       "\n",
       "| Region &lt;chr&gt; | ADNP_ENCFF852FRV &lt;dbl&gt; | AFF1_ENCFF003ZRP &lt;dbl&gt; | AFF1_ENCFF085LMT &lt;dbl&gt; | AFF4_ENCFF040PTP &lt;dbl&gt; | AGO1_ENCFF583LSH &lt;dbl&gt; | ARHGAP35_ENCFF452FAK &lt;dbl&gt; | ARID1B_ENCFF980AHZ &lt;dbl&gt; | ARID2_ENCFF748CGJ &lt;dbl&gt; | ARID3A_ENCFF415DOV &lt;dbl&gt; | ARID3B_ENCFF740OGZ &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1:10038-10405   | 0.972607 | 0.116788 | 0.0000000 | 0.345775 | 0.000000 | 0.000000 | 0.0000000 | 0.000000 | 0.693779 | 0.0000000 |\n",
       "| chr1:14282-14614   | 0.000000 | 0.000000 | 0.0000000 | 0.000000 | 0.000000 | 0.000000 | 0.0982520 | 0.000000 | 0.000000 | 0.1251570 |\n",
       "| chr1:16025-16338   | 0.189624 | 0.330535 | 0.0604299 | 0.676515 | 0.987538 | 0.416025 | 0.0300948 | 0.508451 | 0.544713 | 0.0706043 |\n",
       "| chr1:17288-17689   | 0.190180 | 0.000000 | 0.5865760 | 0.911477 | 1.373140 | 0.417423 | 0.4895330 | 0.664600 | 0.000000 | 1.0071700 |\n",
       "| chr1:28934-29499   | 0.000000 | 0.000000 | 0.0000000 | 0.000000 | 0.000000 | 0.000000 | 0.0406541 | 0.000000 | 0.000000 | 0.0000000 |\n",
       "| chr1:115429-115969 | 1.264990 | 2.354290 | 3.3698500 | 0.839830 | 0.000000 | 0.510867 | 8.4608800 | 0.797594 | 7.387330 | 0.9319340 |\n",
       "\n"
      ],
      "text/plain": [
       "  Region             ADNP_ENCFF852FRV AFF1_ENCFF003ZRP AFF1_ENCFF085LMT\n",
       "1 chr1:10038-10405   0.972607         0.116788         0.0000000       \n",
       "2 chr1:14282-14614   0.000000         0.000000         0.0000000       \n",
       "3 chr1:16025-16338   0.189624         0.330535         0.0604299       \n",
       "4 chr1:17288-17689   0.190180         0.000000         0.5865760       \n",
       "5 chr1:28934-29499   0.000000         0.000000         0.0000000       \n",
       "6 chr1:115429-115969 1.264990         2.354290         3.3698500       \n",
       "  AFF4_ENCFF040PTP AGO1_ENCFF583LSH ARHGAP35_ENCFF452FAK ARID1B_ENCFF980AHZ\n",
       "1 0.345775         0.000000         0.000000             0.0000000         \n",
       "2 0.000000         0.000000         0.000000             0.0982520         \n",
       "3 0.676515         0.987538         0.416025             0.0300948         \n",
       "4 0.911477         1.373140         0.417423             0.4895330         \n",
       "5 0.000000         0.000000         0.000000             0.0406541         \n",
       "6 0.839830         0.000000         0.510867             8.4608800         \n",
       "  ARID2_ENCFF748CGJ ARID3A_ENCFF415DOV ARID3B_ENCFF740OGZ\n",
       "1 0.000000          0.693779           0.0000000         \n",
       "2 0.000000          0.000000           0.1251570         \n",
       "3 0.508451          0.544713           0.0706043         \n",
       "4 0.664600          0.000000           1.0071700         \n",
       "5 0.000000          0.000000           0.0000000         \n",
       "6 0.797594          7.387330           0.9319340         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_region_score_matrix\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8be9154f-5532-4459-b065-77e9930a028a",
   "metadata": {},
   "outputs": [],
   "source": [
    "### get the value I need; here I am taking mean0\n",
    "        lst = lst_region_coverage\n",
    "        lst = lapply(lst, function(dat){\n",
    "            dat = dat %>% \n",
    "                dplyr::mutate(Value = Mean0) %>%\n",
    "                dplyr::select(Chrom, ChromStart, ChromEnd, Region, Value, Target, Index_Label) %>% \n",
    "                dplyr::distinct()\n",
    "            return(dat)\n",
    "        })\n",
    "        dat = bind_rows(lst)\n",
    "        dat_region_coverage_mean0_by_index = dat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "42f2ff31-143c-4dff-8fdc-86047ee7ba84",
   "metadata": {},
   "outputs": [],
   "source": [
    " ### join columns for each chip-seq file\n",
    "        fun = function(dat1, dat2){\n",
    "            dat = dplyr::full_join(dat1, dat2, by = \"Region\")\n",
    "            return(dat)\n",
    "        }\n",
    "        lst = lst_dat_region_score_filter\n",
    "        dat = Reduce(fun, lst)\n",
    "        dat_region_score_matrix = dat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7585cf65-57ed-4849-a619-a555537597de",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07225012-5e0a-4eb9-9f6e-159182f8d94c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1dadb0ec-c6ae-455b-981d-3fc6cc3b7945",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74a60dd0-4748-47ee-92ea-55f92632e493",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===========================================\n",
      "Read file... \n",
      "Region A: fcc_astarr_macs_merge | fcc_astarr_macs_input_overlap \n",
      "Region B: encode_chipseq_tf_full \n",
      "FDiry:    \n",
      "/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_chipseq/fcc_astarr_macs_input_overlap/encode_chipseq_tf_full \n",
      "FName:    \n",
      "Count:    720 \n",
      "[1] \"fcc_astarr_macs_input_overlap.encode_chipseq_ADNP_ENCFF852FRV.tsv\"    \n",
      "[2] \"fcc_astarr_macs_input_overlap.encode_chipseq_AFF1_ENCFF003ZRP.tsv\"    \n",
      "[3] \"fcc_astarr_macs_input_overlap.encode_chipseq_AFF1_ENCFF085LMT.tsv\"    \n",
      "[4] \"fcc_astarr_macs_input_overlap.encode_chipseq_AFF4_ENCFF040PTP.tsv\"    \n",
      "[5] \"fcc_astarr_macs_input_overlap.encode_chipseq_AGO1_ENCFF583LSH.tsv\"    \n",
      "[6] \"fcc_astarr_macs_input_overlap.encode_chipseq_ARHGAP35_ENCFF452FAK.tsv\"\n",
      "\n",
      "Save file... \n",
      "FDiry: /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_chipseq/fcc_astarr_macs_input_overlap/summary \n",
      "FName: region.coverage.encode_chipseq_tf_full.by_index.mean0.tsv \n",
      "\n",
      "===========================================\n",
      "Read file... \n",
      "Region A: fcc_astarr_macs_merge | fcc_astarr_macs_input_union \n",
      "Region B: encode_chipseq_tf_full \n",
      "FDiry:    \n",
      "/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_chipseq/fcc_astarr_macs_input_union/encode_chipseq_tf_full \n",
      "FName:    \n",
      "Count:    720 \n",
      "[1] \"fcc_astarr_macs_input_union.encode_chipseq_ADNP_ENCFF852FRV.tsv\"    \n",
      "[2] \"fcc_astarr_macs_input_union.encode_chipseq_AFF1_ENCFF003ZRP.tsv\"    \n",
      "[3] \"fcc_astarr_macs_input_union.encode_chipseq_AFF1_ENCFF085LMT.tsv\"    \n",
      "[4] \"fcc_astarr_macs_input_union.encode_chipseq_AFF4_ENCFF040PTP.tsv\"    \n",
      "[5] \"fcc_astarr_macs_input_union.encode_chipseq_AGO1_ENCFF583LSH.tsv\"    \n",
      "[6] \"fcc_astarr_macs_input_union.encode_chipseq_ARHGAP35_ENCFF452FAK.tsv\"\n",
      "\n"
     ]
    }
   ],
   "source": [
    "### loop through region A\n",
    "for (idx in seq_along(lst_region_label_A)){\n",
    "    \n",
    "    ### get region A\n",
    "    lst = lst_region_label_A[[idx]]\n",
    "    txt_region_folder_A = lst$Folder\n",
    "    txt_region_label_A  = lst$Label\n",
    "\n",
    "    ### loop through region B\n",
    "    for (txt_region_folder_B in vec_txt_region_folder_B){\n",
    "        \n",
    "        ### get directory of annotation table\n",
    "        txt_fdiry = file.path(\n",
    "            FD_RES, \n",
    "            \"region_coverage_chipseq\",\n",
    "            txt_region_label_A, \n",
    "            txt_region_folder_B\n",
    "        )\n",
    "        txt_fname = \"*tsv\"\n",
    "        txt_fglob = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "        ### get annotation table files\n",
    "        vec_txt_fpath = Sys.glob(txt_fglob)\n",
    "        vec_txt_fname = basename(vec_txt_fpath)\n",
    "\n",
    "        ### show progress\n",
    "        cat(\"===========================================\\n\")\n",
    "        cat(\"Read file...\", \"\\n\")\n",
    "        cat(\"Region A:\", txt_region_folder_A, \"|\", txt_region_label_A, \"\\n\")\n",
    "        cat(\"Region B:\", txt_region_folder_B, \"\\n\")\n",
    "        cat(\"FDiry:   \", \"\\n\")\n",
    "        cat(txt_fdiry, \"\\n\")\n",
    "        cat(\"FName:   \", \"\\n\")\n",
    "        cat(\"Count:   \", length(vec_txt_fname), \"\\n\")\n",
    "        print(head(vec_txt_fname))\n",
    "        cat(\"\\n\")\n",
    "        flush.console()\n",
    "\n",
    "        ### import annotation tables\n",
    "        lst = lapply(vec_txt_fname, function(txt_fname){\n",
    "            ### set directory\n",
    "            txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "            \n",
    "            ### get annotation of region A and region B\n",
    "            vec = str_split(txt_fname, \"\\\\.\")[[1]]\n",
    "            txt_annot_A = vec[1]\n",
    "            txt_annot_B = vec[2]\n",
    "\n",
    "            ### read table and add annotation labels\n",
    "            dat = read_tsv(txt_fpath, col_names = vec_txt_cname, show_col_types = FALSE)\n",
    "            dat = dat %>% \n",
    "                dplyr::mutate(\n",
    "                    Annotation_A = txt_annot_A,\n",
    "                    Annotation_B = txt_annot_B\n",
    "                )\n",
    "            return(dat)\n",
    "        }) \n",
    "\n",
    "        ###\n",
    "        lst = lapply(lst, function(dat){\n",
    "            ### setup location columns\n",
    "            dat = dat %>% \n",
    "                tidyr::separate(\n",
    "                    col  = Region, \n",
    "                    into = c(\"Chrom\", \"ChromStart\", \"ChromEnd\"), \n",
    "                    remove = FALSE)\n",
    "            \n",
    "            ### setup target and label columns\n",
    "            dat = dat %>% \n",
    "                tidyr::separate(\n",
    "                    col  = Annotation_B, \n",
    "                    into = c(\"X1\", \"X2\", \"Target\", \"Index_File\"), \n",
    "                    remove = FALSE) %>%\n",
    "                dplyr::mutate(Index_Label = paste(Target, Index_File, sep = \"_\"))\n",
    "\n",
    "            ### clean up\n",
    "            dat = dat %>%\n",
    "                dplyr::select(-X1, -X2) %>%\n",
    "                dplyr::distinct()\n",
    "            return(dat)\n",
    "        })\n",
    "        lst_region_coverage = lst\n",
    "\n",
    "        ### get the value I need; here I am taking mean0\n",
    "        lst = lst_region_coverage\n",
    "        lst = lapply(lst, function(dat){\n",
    "            dat = dat %>% \n",
    "                dplyr::mutate(Value = Mean0) %>%\n",
    "                dplyr::select(Chrom, ChromStart, ChromEnd, Region, Value, Target, Index_Label) %>% \n",
    "                dplyr::distinct()\n",
    "            return(dat)\n",
    "        })\n",
    "        dat = bind_rows(lst)\n",
    "        dat_region_coverage_mean0_by_index = dat\n",
    "\n",
    "        ### set file directory\n",
    "        txt_fdiry = file.path(\n",
    "            FD_RES, \n",
    "            \"region_coverage_chipseq\",\n",
    "            txt_region_label_A,\n",
    "            \"summary\"\n",
    "        )\n",
    "        txt_fname = paste(\"region\", \"coverage\",  txt_region_folder_B, \"by_index\", \"mean0\", \"tsv\", sep = \".\")\n",
    "        txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "        ### show progress\n",
    "        cat(\"Save file...\",  \"\\n\")\n",
    "        cat(\"FDiry:\", txt_fdiry, \"\\n\")\n",
    "        cat(\"FName:\", txt_fname, \"\\n\")\n",
    "        cat(\"\\n\")\n",
    "        flush.console()\n",
    "        \n",
    "        ### write tables\n",
    "        dir.create(txt_fdiry, showWarnings = FALSE)\n",
    "        dat = dat_region_coverage_mean0_by_index\n",
    "        write_tsv(dat, txt_fpath)\n",
    "    } ### end loop region B\n",
    "} ### end loop region A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7fc3646e-f2d0-471a-9858-2b071476bec4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "62db1404-66dd-4911-bcc6-ed325a8f1e93",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f86c5aa6-f0d2-4c42-a490-35e4a628d5b6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "71f237be-3b7a-4798-a4e0-619a321688db",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 4937040       7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Chrom</th><th scope=col>ChromStart</th><th scope=col>ChromEnd</th><th scope=col>Region</th><th scope=col>Value</th><th scope=col>Target</th><th scope=col>Index_Label</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>10015</td><td>10442</td><td>chr1:10015-10442</td><td>1.056740</td><td>ADNP</td><td>ADNP_ENCFF852FRV</td></tr>\n",
       "\t<tr><td>chr1</td><td>14253</td><td>14645</td><td>chr1:14253-14645</td><td>0.000000</td><td>ADNP</td><td>ADNP_ENCFF852FRV</td></tr>\n",
       "\t<tr><td>chr1</td><td>16015</td><td>16477</td><td>chr1:16015-16477</td><td>0.181395</td><td>ADNP</td><td>ADNP_ENCFF852FRV</td></tr>\n",
       "\t<tr><td>chr1</td><td>17237</td><td>17772</td><td>chr1:17237-17772</td><td>0.177247</td><td>ADNP</td><td>ADNP_ENCFF852FRV</td></tr>\n",
       "\t<tr><td>chr1</td><td>28903</td><td>29613</td><td>chr1:28903-29613</td><td>0.000000</td><td>ADNP</td><td>ADNP_ENCFF852FRV</td></tr>\n",
       "\t<tr><td>chr1</td><td>30803</td><td>31072</td><td>chr1:30803-31072</td><td>0.198925</td><td>ADNP</td><td>ADNP_ENCFF852FRV</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 7\n",
       "\\begin{tabular}{lllllll}\n",
       " Chrom & ChromStart & ChromEnd & Region & Value & Target & Index\\_Label\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t chr1 & 10015 & 10442 & chr1:10015-10442 & 1.056740 & ADNP & ADNP\\_ENCFF852FRV\\\\\n",
       "\t chr1 & 14253 & 14645 & chr1:14253-14645 & 0.000000 & ADNP & ADNP\\_ENCFF852FRV\\\\\n",
       "\t chr1 & 16015 & 16477 & chr1:16015-16477 & 0.181395 & ADNP & ADNP\\_ENCFF852FRV\\\\\n",
       "\t chr1 & 17237 & 17772 & chr1:17237-17772 & 0.177247 & ADNP & ADNP\\_ENCFF852FRV\\\\\n",
       "\t chr1 & 28903 & 29613 & chr1:28903-29613 & 0.000000 & ADNP & ADNP\\_ENCFF852FRV\\\\\n",
       "\t chr1 & 30803 & 31072 & chr1:30803-31072 & 0.198925 & ADNP & ADNP\\_ENCFF852FRV\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 7\n",
       "\n",
       "| Chrom &lt;chr&gt; | ChromStart &lt;chr&gt; | ChromEnd &lt;chr&gt; | Region &lt;chr&gt; | Value &lt;dbl&gt; | Target &lt;chr&gt; | Index_Label &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| chr1 | 10015 | 10442 | chr1:10015-10442 | 1.056740 | ADNP | ADNP_ENCFF852FRV |\n",
       "| chr1 | 14253 | 14645 | chr1:14253-14645 | 0.000000 | ADNP | ADNP_ENCFF852FRV |\n",
       "| chr1 | 16015 | 16477 | chr1:16015-16477 | 0.181395 | ADNP | ADNP_ENCFF852FRV |\n",
       "| chr1 | 17237 | 17772 | chr1:17237-17772 | 0.177247 | ADNP | ADNP_ENCFF852FRV |\n",
       "| chr1 | 28903 | 29613 | chr1:28903-29613 | 0.000000 | ADNP | ADNP_ENCFF852FRV |\n",
       "| chr1 | 30803 | 31072 | chr1:30803-31072 | 0.198925 | ADNP | ADNP_ENCFF852FRV |\n",
       "\n"
      ],
      "text/plain": [
       "  Chrom ChromStart ChromEnd Region           Value    Target Index_Label     \n",
       "1 chr1  10015      10442    chr1:10015-10442 1.056740 ADNP   ADNP_ENCFF852FRV\n",
       "2 chr1  14253      14645    chr1:14253-14645 0.000000 ADNP   ADNP_ENCFF852FRV\n",
       "3 chr1  16015      16477    chr1:16015-16477 0.181395 ADNP   ADNP_ENCFF852FRV\n",
       "4 chr1  17237      17772    chr1:17237-17772 0.177247 ADNP   ADNP_ENCFF852FRV\n",
       "5 chr1  28903      29613    chr1:28903-29613 0.000000 ADNP   ADNP_ENCFF852FRV\n",
       "6 chr1  30803      31072    chr1:30803-31072 0.198925 ADNP   ADNP_ENCFF852FRV"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_region_coverage_mean0_by_index\n",
    "print(dim(dat))\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "24b900c1-0c3a-4770-959b-5e68aeb01cfa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "    ADNP_ENCFF852FRV     AFF1_ENCFF003ZRP     AFF1_ENCFF085LMT \n",
       "              246852               246852               246852 \n",
       "    AFF4_ENCFF040PTP     AGO1_ENCFF583LSH ARHGAP35_ENCFF452FAK \n",
       "              246852               246852               246852 \n",
       "  ARID1B_ENCFF980AHZ    ARID2_ENCFF748CGJ   ARID3A_ENCFF415DOV \n",
       "              246852               246852               246852 \n",
       "  ARID3B_ENCFF740OGZ   ARID4B_ENCFF691FMX   ARID4B_ENCFF831PIW \n",
       "              246852               246852               246852 \n",
       "    ARNT_ENCFF194PVJ     ARNT_ENCFF300RKC     ARNT_ENCFF466GVY \n",
       "              246852               246852               246852 \n",
       "   ASH1L_ENCFF838IQI     ATF1_ENCFF090IHN     ATF1_ENCFF271JYU \n",
       "              246852               246852               246852 \n",
       "    ATF1_ENCFF307NOK     ATF1_ENCFF911VQF \n",
       "              246852               246852 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(dat$Index_Label)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f2f4a5d-e6b1-4734-8206-6e9d253e3748",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "126d3c12-a0f6-4746-baea-a51625adf3a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "        ### average across the same TF\n",
    "        dat = dat_region_coverage_by_index\n",
    "        dat = dat %>%\n",
    "            dplyr::group_by(Chrom, ChromStart, ChromEnd, Region, Target) %>%\n",
    "            dplyr::summarise(\n",
    "                Value = mean(Value),\n",
    "                .groups = \"drop\"\n",
    "            )\n",
    "        dat_region_coverage_by_tfs = dat\n",
    "\n",
    "        ### convert to matrix\n",
    "        dat = dat_region_coverage_by_index\n",
    "        dat = dat %>%\n",
    "            dplyr::select(Chrom, ChromStart, ChromEnd, Region, Index_Label, Value) %>%\n",
    "            tidyr::spread(Index_Label, Value)\n",
    "        mat_region_coverage_by_index = dat\n",
    "\n",
    "        ### convert to matrix\n",
    "        dat = dat_region_coverage_by_tfs\n",
    "        dat = dat %>%\n",
    "            dplyr::select(Chrom, ChromStart, ChromEnd, Region, Target, Value) %>%\n",
    "            tidyr::spread(Target, Value)    \n",
    "        mat_region_coverage_by_tfs = dat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3557f007-6aa4-477f-9e66-37ab4c8bcf5f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "519ee7b8-4d8f-4259-91b2-07dcc29d196d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "35d3841d-3806-4d7d-868a-5f40f424c74d",
   "metadata": {},
   "outputs": [],
   "source": [
    "lst_region_coverage = lst"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "ae91f6b3-d099-48c2-8808-7c8e2636bac8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Region</th><th scope=col>Chrom</th><th scope=col>ChromStart</th><th scope=col>ChromEnd</th><th scope=col>Size</th><th scope=col>Covered</th><th scope=col>Sum</th><th scope=col>Mean0</th><th scope=col>Mean</th><th scope=col>Min</th><th scope=col>Max</th><th scope=col>Annotation_A</th><th scope=col>Annotation_B</th><th scope=col>X2</th><th scope=col>Target</th><th scope=col>Index_File</th><th scope=col>Index_Label</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1:10038-10405  </td><td>chr1</td><td>10038 </td><td>10405 </td><td>367</td><td>367</td><td>157.198</td><td>0.428333</td><td>0.428333</td><td>0</td><td>1.08399</td><td>fcc_astarr_macs_input_overlap</td><td>encode_chipseq_H2AFZ_ENCFF621DJP</td><td>chipseq</td><td>H2AFZ</td><td>ENCFF621DJP</td><td>H2AFZ_ENCFF621DJP</td></tr>\n",
       "\t<tr><td>chr1:14282-14614  </td><td>chr1</td><td>14282 </td><td>14614 </td><td>332</td><td>332</td><td>  0.000</td><td>0.000000</td><td>0.000000</td><td>0</td><td>0.00000</td><td>fcc_astarr_macs_input_overlap</td><td>encode_chipseq_H2AFZ_ENCFF621DJP</td><td>chipseq</td><td>H2AFZ</td><td>ENCFF621DJP</td><td>H2AFZ_ENCFF621DJP</td></tr>\n",
       "\t<tr><td>chr1:16025-16338  </td><td>chr1</td><td>16025 </td><td>16338 </td><td>313</td><td>313</td><td>  0.000</td><td>0.000000</td><td>0.000000</td><td>0</td><td>0.00000</td><td>fcc_astarr_macs_input_overlap</td><td>encode_chipseq_H2AFZ_ENCFF621DJP</td><td>chipseq</td><td>H2AFZ</td><td>ENCFF621DJP</td><td>H2AFZ_ENCFF621DJP</td></tr>\n",
       "\t<tr><td>chr1:17288-17689  </td><td>chr1</td><td>17288 </td><td>17689 </td><td>401</td><td>401</td><td>  0.000</td><td>0.000000</td><td>0.000000</td><td>0</td><td>0.00000</td><td>fcc_astarr_macs_input_overlap</td><td>encode_chipseq_H2AFZ_ENCFF621DJP</td><td>chipseq</td><td>H2AFZ</td><td>ENCFF621DJP</td><td>H2AFZ_ENCFF621DJP</td></tr>\n",
       "\t<tr><td>chr1:28934-29499  </td><td>chr1</td><td>28934 </td><td>29499 </td><td>565</td><td>565</td><td>  0.000</td><td>0.000000</td><td>0.000000</td><td>0</td><td>0.00000</td><td>fcc_astarr_macs_input_overlap</td><td>encode_chipseq_H2AFZ_ENCFF621DJP</td><td>chipseq</td><td>H2AFZ</td><td>ENCFF621DJP</td><td>H2AFZ_ENCFF621DJP</td></tr>\n",
       "\t<tr><td>chr1:115429-115969</td><td>chr1</td><td>115429</td><td>115969</td><td>540</td><td>540</td><td>392.975</td><td>0.727731</td><td>0.727731</td><td>0</td><td>2.71006</td><td>fcc_astarr_macs_input_overlap</td><td>encode_chipseq_H2AFZ_ENCFF621DJP</td><td>chipseq</td><td>H2AFZ</td><td>ENCFF621DJP</td><td>H2AFZ_ENCFF621DJP</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 17\n",
       "\\begin{tabular}{lllllllllllllllll}\n",
       " Region & Chrom & ChromStart & ChromEnd & Size & Covered & Sum & Mean0 & Mean & Min & Max & Annotation\\_A & Annotation\\_B & X2 & Target & Index\\_File & Index\\_Label\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t chr1:10038-10405   & chr1 & 10038  & 10405  & 367 & 367 & 157.198 & 0.428333 & 0.428333 & 0 & 1.08399 & fcc\\_astarr\\_macs\\_input\\_overlap & encode\\_chipseq\\_H2AFZ\\_ENCFF621DJP & chipseq & H2AFZ & ENCFF621DJP & H2AFZ\\_ENCFF621DJP\\\\\n",
       "\t chr1:14282-14614   & chr1 & 14282  & 14614  & 332 & 332 &   0.000 & 0.000000 & 0.000000 & 0 & 0.00000 & fcc\\_astarr\\_macs\\_input\\_overlap & encode\\_chipseq\\_H2AFZ\\_ENCFF621DJP & chipseq & H2AFZ & ENCFF621DJP & H2AFZ\\_ENCFF621DJP\\\\\n",
       "\t chr1:16025-16338   & chr1 & 16025  & 16338  & 313 & 313 &   0.000 & 0.000000 & 0.000000 & 0 & 0.00000 & fcc\\_astarr\\_macs\\_input\\_overlap & encode\\_chipseq\\_H2AFZ\\_ENCFF621DJP & chipseq & H2AFZ & ENCFF621DJP & H2AFZ\\_ENCFF621DJP\\\\\n",
       "\t chr1:17288-17689   & chr1 & 17288  & 17689  & 401 & 401 &   0.000 & 0.000000 & 0.000000 & 0 & 0.00000 & fcc\\_astarr\\_macs\\_input\\_overlap & encode\\_chipseq\\_H2AFZ\\_ENCFF621DJP & chipseq & H2AFZ & ENCFF621DJP & H2AFZ\\_ENCFF621DJP\\\\\n",
       "\t chr1:28934-29499   & chr1 & 28934  & 29499  & 565 & 565 &   0.000 & 0.000000 & 0.000000 & 0 & 0.00000 & fcc\\_astarr\\_macs\\_input\\_overlap & encode\\_chipseq\\_H2AFZ\\_ENCFF621DJP & chipseq & H2AFZ & ENCFF621DJP & H2AFZ\\_ENCFF621DJP\\\\\n",
       "\t chr1:115429-115969 & chr1 & 115429 & 115969 & 540 & 540 & 392.975 & 0.727731 & 0.727731 & 0 & 2.71006 & fcc\\_astarr\\_macs\\_input\\_overlap & encode\\_chipseq\\_H2AFZ\\_ENCFF621DJP & chipseq & H2AFZ & ENCFF621DJP & H2AFZ\\_ENCFF621DJP\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 17\n",
       "\n",
       "| Region &lt;chr&gt; | Chrom &lt;chr&gt; | ChromStart &lt;chr&gt; | ChromEnd &lt;chr&gt; | Size &lt;dbl&gt; | Covered &lt;dbl&gt; | Sum &lt;dbl&gt; | Mean0 &lt;dbl&gt; | Mean &lt;dbl&gt; | Min &lt;dbl&gt; | Max &lt;dbl&gt; | Annotation_A &lt;chr&gt; | Annotation_B &lt;chr&gt; | X2 &lt;chr&gt; | Target &lt;chr&gt; | Index_File &lt;chr&gt; | Index_Label &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1:10038-10405   | chr1 | 10038  | 10405  | 367 | 367 | 157.198 | 0.428333 | 0.428333 | 0 | 1.08399 | fcc_astarr_macs_input_overlap | encode_chipseq_H2AFZ_ENCFF621DJP | chipseq | H2AFZ | ENCFF621DJP | H2AFZ_ENCFF621DJP |\n",
       "| chr1:14282-14614   | chr1 | 14282  | 14614  | 332 | 332 |   0.000 | 0.000000 | 0.000000 | 0 | 0.00000 | fcc_astarr_macs_input_overlap | encode_chipseq_H2AFZ_ENCFF621DJP | chipseq | H2AFZ | ENCFF621DJP | H2AFZ_ENCFF621DJP |\n",
       "| chr1:16025-16338   | chr1 | 16025  | 16338  | 313 | 313 |   0.000 | 0.000000 | 0.000000 | 0 | 0.00000 | fcc_astarr_macs_input_overlap | encode_chipseq_H2AFZ_ENCFF621DJP | chipseq | H2AFZ | ENCFF621DJP | H2AFZ_ENCFF621DJP |\n",
       "| chr1:17288-17689   | chr1 | 17288  | 17689  | 401 | 401 |   0.000 | 0.000000 | 0.000000 | 0 | 0.00000 | fcc_astarr_macs_input_overlap | encode_chipseq_H2AFZ_ENCFF621DJP | chipseq | H2AFZ | ENCFF621DJP | H2AFZ_ENCFF621DJP |\n",
       "| chr1:28934-29499   | chr1 | 28934  | 29499  | 565 | 565 |   0.000 | 0.000000 | 0.000000 | 0 | 0.00000 | fcc_astarr_macs_input_overlap | encode_chipseq_H2AFZ_ENCFF621DJP | chipseq | H2AFZ | ENCFF621DJP | H2AFZ_ENCFF621DJP |\n",
       "| chr1:115429-115969 | chr1 | 115429 | 115969 | 540 | 540 | 392.975 | 0.727731 | 0.727731 | 0 | 2.71006 | fcc_astarr_macs_input_overlap | encode_chipseq_H2AFZ_ENCFF621DJP | chipseq | H2AFZ | ENCFF621DJP | H2AFZ_ENCFF621DJP |\n",
       "\n"
      ],
      "text/plain": [
       "  Region             Chrom ChromStart ChromEnd Size Covered Sum     Mean0   \n",
       "1 chr1:10038-10405   chr1  10038      10405    367  367     157.198 0.428333\n",
       "2 chr1:14282-14614   chr1  14282      14614    332  332       0.000 0.000000\n",
       "3 chr1:16025-16338   chr1  16025      16338    313  313       0.000 0.000000\n",
       "4 chr1:17288-17689   chr1  17288      17689    401  401       0.000 0.000000\n",
       "5 chr1:28934-29499   chr1  28934      29499    565  565       0.000 0.000000\n",
       "6 chr1:115429-115969 chr1  115429     115969   540  540     392.975 0.727731\n",
       "  Mean     Min Max     Annotation_A                 \n",
       "1 0.428333 0   1.08399 fcc_astarr_macs_input_overlap\n",
       "2 0.000000 0   0.00000 fcc_astarr_macs_input_overlap\n",
       "3 0.000000 0   0.00000 fcc_astarr_macs_input_overlap\n",
       "4 0.000000 0   0.00000 fcc_astarr_macs_input_overlap\n",
       "5 0.000000 0   0.00000 fcc_astarr_macs_input_overlap\n",
       "6 0.727731 0   2.71006 fcc_astarr_macs_input_overlap\n",
       "  Annotation_B                     X2      Target Index_File  Index_Label      \n",
       "1 encode_chipseq_H2AFZ_ENCFF621DJP chipseq H2AFZ  ENCFF621DJP H2AFZ_ENCFF621DJP\n",
       "2 encode_chipseq_H2AFZ_ENCFF621DJP chipseq H2AFZ  ENCFF621DJP H2AFZ_ENCFF621DJP\n",
       "3 encode_chipseq_H2AFZ_ENCFF621DJP chipseq H2AFZ  ENCFF621DJP H2AFZ_ENCFF621DJP\n",
       "4 encode_chipseq_H2AFZ_ENCFF621DJP chipseq H2AFZ  ENCFF621DJP H2AFZ_ENCFF621DJP\n",
       "5 encode_chipseq_H2AFZ_ENCFF621DJP chipseq H2AFZ  ENCFF621DJP H2AFZ_ENCFF621DJP\n",
       "6 encode_chipseq_H2AFZ_ENCFF621DJP chipseq H2AFZ  ENCFF621DJP H2AFZ_ENCFF621DJP"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "lst = lst_region_coverage\n",
    "dat = lst[[1]]\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "75231316-071d-4991-a4e1-c59c65a3a039",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 450126      7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Chrom </th>\n",
       "   <th style=\"text-align:left;\"> ChromStart </th>\n",
       "   <th style=\"text-align:left;\"> ChromEnd </th>\n",
       "   <th style=\"text-align:left;\"> Region </th>\n",
       "   <th style=\"text-align:right;\"> Value </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Index_Label </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 10038 </td>\n",
       "   <td style=\"text-align:left;\"> 10405 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:10038-10405 </td>\n",
       "   <td style=\"text-align:right;\"> 0.428333 </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ_ENCFF621DJP </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 14282 </td>\n",
       "   <td style=\"text-align:left;\"> 14614 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:14282-14614 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ_ENCFF621DJP </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 16025 </td>\n",
       "   <td style=\"text-align:left;\"> 16338 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:16025-16338 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ_ENCFF621DJP </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 17288 </td>\n",
       "   <td style=\"text-align:left;\"> 17689 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:17288-17689 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ_ENCFF621DJP </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 28934 </td>\n",
       "   <td style=\"text-align:left;\"> 29499 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:28934-29499 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ_ENCFF621DJP </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 115429 </td>\n",
       "   <td style=\"text-align:left;\"> 115969 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:115429-115969 </td>\n",
       "   <td style=\"text-align:right;\"> 0.727731 </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ_ENCFF621DJP </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "###\n",
    "lst = lst_region_coverage\n",
    "lst = lapply(lst, function(dat){\n",
    "    dat = dat %>% \n",
    "        dplyr::mutate(Value = Mean0) %>%\n",
    "        dplyr::select(Chrom, ChromStart, ChromEnd, Region, Value, Target, Index_Label) %>% \n",
    "        dplyr::distinct()\n",
    "    return(dat)\n",
    "})\n",
    "dat = bind_rows(lst)\n",
    "\n",
    "###\n",
    "dat_region_coverage_by_index = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "f0bea47b-3ce8-4ba9-9ae0-e4693132e1e1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 450126      6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Chrom </th>\n",
       "   <th style=\"text-align:left;\"> ChromStart </th>\n",
       "   <th style=\"text-align:left;\"> ChromEnd </th>\n",
       "   <th style=\"text-align:left;\"> Region </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:right;\"> Value </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 100028014 </td>\n",
       "   <td style=\"text-align:left;\"> 100029653 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:100028014-100029653 </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ </td>\n",
       "   <td style=\"text-align:right;\"> 0.908487 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 100028014 </td>\n",
       "   <td style=\"text-align:left;\"> 100029653 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:100028014-100029653 </td>\n",
       "   <td style=\"text-align:left;\"> H3K27ac </td>\n",
       "   <td style=\"text-align:right;\"> 1.300930 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 100028014 </td>\n",
       "   <td style=\"text-align:left;\"> 100029653 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:100028014-100029653 </td>\n",
       "   <td style=\"text-align:left;\"> H3K27me3 </td>\n",
       "   <td style=\"text-align:right;\"> 0.187109 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 100036895 </td>\n",
       "   <td style=\"text-align:left;\"> 100039189 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:100036895-100039189 </td>\n",
       "   <td style=\"text-align:left;\"> H2AFZ </td>\n",
       "   <td style=\"text-align:right;\"> 6.462060 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 100036895 </td>\n",
       "   <td style=\"text-align:left;\"> 100039189 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:100036895-100039189 </td>\n",
       "   <td style=\"text-align:left;\"> H3K27ac </td>\n",
       "   <td style=\"text-align:right;\"> 14.033400 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 100036895 </td>\n",
       "   <td style=\"text-align:left;\"> 100039189 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:100036895-100039189 </td>\n",
       "   <td style=\"text-align:left;\"> H3K27me3 </td>\n",
       "   <td style=\"text-align:right;\"> 0.193905 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "###\n",
    "dat = dat_region_coverage_by_index\n",
    "dat = dat %>%\n",
    "    dplyr::group_by(Chrom, ChromStart, ChromEnd, Region, Target) %>%\n",
    "    dplyr::summarise(\n",
    "        Value = mean(Value),\n",
    "        .groups = \"drop\"\n",
    "    )\n",
    "\n",
    "###\n",
    "dat_region_coverage_by_tfs = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "f0f33bf4-ba5d-4aa4-8848-5a1cbdc2258e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 150042      7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Chrom </th>\n",
       "   <th style=\"text-align:left;\"> ChromStart </th>\n",
       "   <th style=\"text-align:left;\"> ChromEnd </th>\n",
       "   <th style=\"text-align:left;\"> Region </th>\n",
       "   <th style=\"text-align:right;\"> H2AFZ_ENCFF621DJP </th>\n",
       "   <th style=\"text-align:right;\"> H3K27ac_ENCFF381NDD </th>\n",
       "   <th style=\"text-align:right;\"> H3K27me3_ENCFF139KZL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 10038 </td>\n",
       "   <td style=\"text-align:left;\"> 10405 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:10038-10405 </td>\n",
       "   <td style=\"text-align:right;\"> 0.428333 </td>\n",
       "   <td style=\"text-align:right;\"> 0.817488 </td>\n",
       "   <td style=\"text-align:right;\"> 0.107386 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 14282 </td>\n",
       "   <td style=\"text-align:left;\"> 14614 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:14282-14614 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 16025 </td>\n",
       "   <td style=\"text-align:left;\"> 16338 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:16025-16338 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "   <td style=\"text-align:right;\"> 0.425308 </td>\n",
       "   <td style=\"text-align:right;\"> 0.330878 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 17288 </td>\n",
       "   <td style=\"text-align:left;\"> 17689 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:17288-17689 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "   <td style=\"text-align:right;\"> 0.148424 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 28934 </td>\n",
       "   <td style=\"text-align:left;\"> 29499 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:28934-29499 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 115429 </td>\n",
       "   <td style=\"text-align:left;\"> 115969 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:115429-115969 </td>\n",
       "   <td style=\"text-align:right;\"> 0.727731 </td>\n",
       "   <td style=\"text-align:right;\"> 1.448470 </td>\n",
       "   <td style=\"text-align:right;\"> 0.000000 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_region_coverage_by_index\n",
    "dat = dat %>%\n",
    "    dplyr::select(Chrom, ChromStart, ChromEnd, Region, Index_Label, Value) %>%\n",
    "    tidyr::spread(Index_Label, Value)\n",
    "    \n",
    "mat_region_coverage_by_index = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "308f2226-fa2c-4a0f-85a6-81b93c94821a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 150042      7\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Chrom </th>\n",
       "   <th style=\"text-align:left;\"> ChromStart </th>\n",
       "   <th style=\"text-align:left;\"> ChromEnd </th>\n",
       "   <th style=\"text-align:left;\"> Region </th>\n",
       "   <th style=\"text-align:right;\"> H2AFZ </th>\n",
       "   <th style=\"text-align:right;\"> H3K27ac </th>\n",
       "   <th style=\"text-align:right;\"> H3K27me3 </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 100028014 </td>\n",
       "   <td style=\"text-align:left;\"> 100029653 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:100028014-100029653 </td>\n",
       "   <td style=\"text-align:right;\"> 0.9084870 </td>\n",
       "   <td style=\"text-align:right;\"> 1.30093 </td>\n",
       "   <td style=\"text-align:right;\"> 0.1871090 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 100036895 </td>\n",
       "   <td style=\"text-align:left;\"> 100039189 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:100036895-100039189 </td>\n",
       "   <td style=\"text-align:right;\"> 6.4620600 </td>\n",
       "   <td style=\"text-align:right;\"> 14.03340 </td>\n",
       "   <td style=\"text-align:right;\"> 0.1939050 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 100046252 </td>\n",
       "   <td style=\"text-align:left;\"> 100046533 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:100046252-100046533 </td>\n",
       "   <td style=\"text-align:right;\"> 0.0301893 </td>\n",
       "   <td style=\"text-align:right;\"> 1.13101 </td>\n",
       "   <td style=\"text-align:right;\"> 0.5094600 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 100048290 </td>\n",
       "   <td style=\"text-align:left;\"> 100048918 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:100048290-100048918 </td>\n",
       "   <td style=\"text-align:right;\"> 0.4234280 </td>\n",
       "   <td style=\"text-align:right;\"> 1.79969 </td>\n",
       "   <td style=\"text-align:right;\"> 0.2119060 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 100050225 </td>\n",
       "   <td style=\"text-align:left;\"> 100050529 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:100050225-100050529 </td>\n",
       "   <td style=\"text-align:right;\"> 0.8108740 </td>\n",
       "   <td style=\"text-align:right;\"> 2.30582 </td>\n",
       "   <td style=\"text-align:right;\"> 0.0077113 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> chr1 </td>\n",
       "   <td style=\"text-align:left;\"> 100064757 </td>\n",
       "   <td style=\"text-align:left;\"> 100065161 </td>\n",
       "   <td style=\"text-align:left;\"> chr1:100064757-100065161 </td>\n",
       "   <td style=\"text-align:right;\"> 0.3412110 </td>\n",
       "   <td style=\"text-align:right;\"> 1.36302 </td>\n",
       "   <td style=\"text-align:right;\"> 0.3907490 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_region_coverage_by_tfs\n",
    "dat = dat %>%\n",
    "    dplyr::select(Chrom, ChromStart, ChromEnd, Region, Target, Value) %>%\n",
    "    tidyr::spread(Target, Value)\n",
    "    \n",
    "mat_region_coverage_by_tfs = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4516174f-7938-457a-9c70-f3d1a0a98f5b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed7736ad-afed-434e-8764-85a16d3acb4a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "038939f4-80e6-4c2d-8020-2f054a3c7e03",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Region</th><th scope=col>Size</th><th scope=col>Covered</th><th scope=col>Sum</th><th scope=col>Mean0</th><th scope=col>Mean</th><th scope=col>Min</th><th scope=col>Max</th><th scope=col>Annotation_A</th><th scope=col>Annotation_B</th><th scope=col>Target</th><th scope=col>Index_File</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1:10038-10405  </td><td>367</td><td>367</td><td>157.198</td><td>0.428333</td><td>0.428333</td><td>0</td><td>1.08399</td><td>fcc_astarr_macs_input_overlap</td><td>encode_chipseq_H2AFZ_ENCFF621DJP</td><td>H2AFZ</td><td>ENCFF621DJP</td></tr>\n",
       "\t<tr><td>chr1:14282-14614  </td><td>332</td><td>332</td><td>  0.000</td><td>0.000000</td><td>0.000000</td><td>0</td><td>0.00000</td><td>fcc_astarr_macs_input_overlap</td><td>encode_chipseq_H2AFZ_ENCFF621DJP</td><td>H2AFZ</td><td>ENCFF621DJP</td></tr>\n",
       "\t<tr><td>chr1:16025-16338  </td><td>313</td><td>313</td><td>  0.000</td><td>0.000000</td><td>0.000000</td><td>0</td><td>0.00000</td><td>fcc_astarr_macs_input_overlap</td><td>encode_chipseq_H2AFZ_ENCFF621DJP</td><td>H2AFZ</td><td>ENCFF621DJP</td></tr>\n",
       "\t<tr><td>chr1:17288-17689  </td><td>401</td><td>401</td><td>  0.000</td><td>0.000000</td><td>0.000000</td><td>0</td><td>0.00000</td><td>fcc_astarr_macs_input_overlap</td><td>encode_chipseq_H2AFZ_ENCFF621DJP</td><td>H2AFZ</td><td>ENCFF621DJP</td></tr>\n",
       "\t<tr><td>chr1:28934-29499  </td><td>565</td><td>565</td><td>  0.000</td><td>0.000000</td><td>0.000000</td><td>0</td><td>0.00000</td><td>fcc_astarr_macs_input_overlap</td><td>encode_chipseq_H2AFZ_ENCFF621DJP</td><td>H2AFZ</td><td>ENCFF621DJP</td></tr>\n",
       "\t<tr><td>chr1:115429-115969</td><td>540</td><td>540</td><td>392.975</td><td>0.727731</td><td>0.727731</td><td>0</td><td>2.71006</td><td>fcc_astarr_macs_input_overlap</td><td>encode_chipseq_H2AFZ_ENCFF621DJP</td><td>H2AFZ</td><td>ENCFF621DJP</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 12\n",
       "\\begin{tabular}{llllllllllll}\n",
       " Region & Size & Covered & Sum & Mean0 & Mean & Min & Max & Annotation\\_A & Annotation\\_B & Target & Index\\_File\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t chr1:10038-10405   & 367 & 367 & 157.198 & 0.428333 & 0.428333 & 0 & 1.08399 & fcc\\_astarr\\_macs\\_input\\_overlap & encode\\_chipseq\\_H2AFZ\\_ENCFF621DJP & H2AFZ & ENCFF621DJP\\\\\n",
       "\t chr1:14282-14614   & 332 & 332 &   0.000 & 0.000000 & 0.000000 & 0 & 0.00000 & fcc\\_astarr\\_macs\\_input\\_overlap & encode\\_chipseq\\_H2AFZ\\_ENCFF621DJP & H2AFZ & ENCFF621DJP\\\\\n",
       "\t chr1:16025-16338   & 313 & 313 &   0.000 & 0.000000 & 0.000000 & 0 & 0.00000 & fcc\\_astarr\\_macs\\_input\\_overlap & encode\\_chipseq\\_H2AFZ\\_ENCFF621DJP & H2AFZ & ENCFF621DJP\\\\\n",
       "\t chr1:17288-17689   & 401 & 401 &   0.000 & 0.000000 & 0.000000 & 0 & 0.00000 & fcc\\_astarr\\_macs\\_input\\_overlap & encode\\_chipseq\\_H2AFZ\\_ENCFF621DJP & H2AFZ & ENCFF621DJP\\\\\n",
       "\t chr1:28934-29499   & 565 & 565 &   0.000 & 0.000000 & 0.000000 & 0 & 0.00000 & fcc\\_astarr\\_macs\\_input\\_overlap & encode\\_chipseq\\_H2AFZ\\_ENCFF621DJP & H2AFZ & ENCFF621DJP\\\\\n",
       "\t chr1:115429-115969 & 540 & 540 & 392.975 & 0.727731 & 0.727731 & 0 & 2.71006 & fcc\\_astarr\\_macs\\_input\\_overlap & encode\\_chipseq\\_H2AFZ\\_ENCFF621DJP & H2AFZ & ENCFF621DJP\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 12\n",
       "\n",
       "| Region &lt;chr&gt; | Size &lt;dbl&gt; | Covered &lt;dbl&gt; | Sum &lt;dbl&gt; | Mean0 &lt;dbl&gt; | Mean &lt;dbl&gt; | Min &lt;dbl&gt; | Max &lt;dbl&gt; | Annotation_A &lt;chr&gt; | Annotation_B &lt;chr&gt; | Target &lt;chr&gt; | Index_File &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1:10038-10405   | 367 | 367 | 157.198 | 0.428333 | 0.428333 | 0 | 1.08399 | fcc_astarr_macs_input_overlap | encode_chipseq_H2AFZ_ENCFF621DJP | H2AFZ | ENCFF621DJP |\n",
       "| chr1:14282-14614   | 332 | 332 |   0.000 | 0.000000 | 0.000000 | 0 | 0.00000 | fcc_astarr_macs_input_overlap | encode_chipseq_H2AFZ_ENCFF621DJP | H2AFZ | ENCFF621DJP |\n",
       "| chr1:16025-16338   | 313 | 313 |   0.000 | 0.000000 | 0.000000 | 0 | 0.00000 | fcc_astarr_macs_input_overlap | encode_chipseq_H2AFZ_ENCFF621DJP | H2AFZ | ENCFF621DJP |\n",
       "| chr1:17288-17689   | 401 | 401 |   0.000 | 0.000000 | 0.000000 | 0 | 0.00000 | fcc_astarr_macs_input_overlap | encode_chipseq_H2AFZ_ENCFF621DJP | H2AFZ | ENCFF621DJP |\n",
       "| chr1:28934-29499   | 565 | 565 |   0.000 | 0.000000 | 0.000000 | 0 | 0.00000 | fcc_astarr_macs_input_overlap | encode_chipseq_H2AFZ_ENCFF621DJP | H2AFZ | ENCFF621DJP |\n",
       "| chr1:115429-115969 | 540 | 540 | 392.975 | 0.727731 | 0.727731 | 0 | 2.71006 | fcc_astarr_macs_input_overlap | encode_chipseq_H2AFZ_ENCFF621DJP | H2AFZ | ENCFF621DJP |\n",
       "\n"
      ],
      "text/plain": [
       "  Region             Size Covered Sum     Mean0    Mean     Min Max    \n",
       "1 chr1:10038-10405   367  367     157.198 0.428333 0.428333 0   1.08399\n",
       "2 chr1:14282-14614   332  332       0.000 0.000000 0.000000 0   0.00000\n",
       "3 chr1:16025-16338   313  313       0.000 0.000000 0.000000 0   0.00000\n",
       "4 chr1:17288-17689   401  401       0.000 0.000000 0.000000 0   0.00000\n",
       "5 chr1:28934-29499   565  565       0.000 0.000000 0.000000 0   0.00000\n",
       "6 chr1:115429-115969 540  540     392.975 0.727731 0.727731 0   2.71006\n",
       "  Annotation_A                  Annotation_B                     Target\n",
       "1 fcc_astarr_macs_input_overlap encode_chipseq_H2AFZ_ENCFF621DJP H2AFZ \n",
       "2 fcc_astarr_macs_input_overlap encode_chipseq_H2AFZ_ENCFF621DJP H2AFZ \n",
       "3 fcc_astarr_macs_input_overlap encode_chipseq_H2AFZ_ENCFF621DJP H2AFZ \n",
       "4 fcc_astarr_macs_input_overlap encode_chipseq_H2AFZ_ENCFF621DJP H2AFZ \n",
       "5 fcc_astarr_macs_input_overlap encode_chipseq_H2AFZ_ENCFF621DJP H2AFZ \n",
       "6 fcc_astarr_macs_input_overlap encode_chipseq_H2AFZ_ENCFF621DJP H2AFZ \n",
       "  Index_File \n",
       "1 ENCFF621DJP\n",
       "2 ENCFF621DJP\n",
       "3 ENCFF621DJP\n",
       "4 ENCFF621DJP\n",
       "5 ENCFF621DJP\n",
       "6 ENCFF621DJP"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### concatenate tables and assign\n",
    "#dat = bind_rows(lst)\n",
    "dat = dat %>% \n",
    "    tidyr::separate(\n",
    "        col  = Annotation_B, \n",
    "        into = c(\"X1\", \"X2\", \"Target\", \"Index_File\"), \n",
    "        remove = FALSE) %>%\n",
    "    dplyr::select(-X1, -X2)\n",
    "dat_region_annot_pair = dat\n",
    "\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f6027d5-831c-4a5b-9d47-770f5bc1cdac",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b88034e2-ff07-4c7c-ac77-efb76e13f120",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9124953b-5065-4789-9ae5-6c3f38a2df15",
   "metadata": {},
   "outputs": [],
   "source": [
    "### concatenate tables and assign\n",
    "        dat = bind_rows(lst)\n",
    "        dat = dat %>% \n",
    "            tidyr::separate(\n",
    "                col  = Annotation_B, \n",
    "                into = c(\"X1\", \"X2\", \"Target\", \"Index_File\"), \n",
    "                remove = FALSE) %>%\n",
    "            dplyr::select(-X1, -X2)\n",
    "        dat_region_annot_pair = dat\n",
    "        \n",
    "        ### set directory\n",
    "        txt_fdiry = file.path(\n",
    "            FD_RES, \n",
    "            \"region_coverage_chipseq\",\n",
    "            txt_region_label_A,\n",
    "            \"summary\"\n",
    "        )\n",
    "        txt_fname = paste(\"region\", \"coverage\",  txt_region_folder_B, \"tsv\", sep = \".\")\n",
    "        txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "        ### show progress\n",
    "        cat(\"Save file...\",  \"\\n\")\n",
    "        cat(\"FDiry:\", txt_fdiry, \"\\n\")\n",
    "        cat(\"FName:\", txt_fname, \"\\n\")\n",
    "        cat(\"\\n\")\n",
    "        flush.console()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d1a2d7d5-ce87-488a-bd70-97e5dadfba95",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "===========================================\n",
      "Read file... \n",
      "Region A: fcc_astarr_macs | fcc_astarr_macs_input_overlap \n",
      "Region B: encode_chipseq_histone \n",
      "FDiry:    \n",
      "[1] \"/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_chipseq/fcc_astarr_macs_input_overlap/encode_chipseq_histone\"\n",
      "FName:    \n",
      " [1] \"fcc_astarr_macs_input_overlap.encode_chipseq_H2AFZ_ENCFF621DJP.tsv\"   \n",
      " [2] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K27ac_ENCFF381NDD.tsv\" \n",
      " [3] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K27me3_ENCFF139KZL.tsv\"\n",
      " [4] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K27me3_ENCFF242ENK.tsv\"\n",
      " [5] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K36me3_ENCFF317VHO.tsv\"\n",
      " [6] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K36me3_ENCFF347YYH.tsv\"\n",
      " [7] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K4me1_ENCFF399SGM.tsv\" \n",
      " [8] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K4me1_ENCFF607SUJ.tsv\" \n",
      " [9] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K4me2_ENCFF959YJV.tsv\" \n",
      "[10] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K4me3_ENCFF253TOF.tsv\" \n",
      "[11] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K4me3_ENCFF660WUG.tsv\" \n",
      "[12] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K4me3_ENCFF806YEZ.tsv\" \n",
      "[13] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K4me3_ENCFF911JVK.tsv\" \n",
      "[14] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K79me2_ENCFF544AVW.tsv\"\n",
      "[15] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K9ac_ENCFF286WRJ.tsv\"  \n",
      "[16] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K9ac_ENCFF583BKU.tsv\"  \n",
      "[17] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K9me1_ENCFF654SLZ.tsv\" \n",
      "[18] \"fcc_astarr_macs_input_overlap.encode_chipseq_H3K9me3_ENCFF601JGK.tsv\" \n",
      "[19] \"fcc_astarr_macs_input_overlap.encode_chipseq_H4K20me1_ENCFF605FAF.tsv\"\n",
      "\n",
      "Save file... \n",
      "FDiry: /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_chipseq/fcc_astarr_macs_input_overlap/summary \n",
      "FName: region.coverage.encode_chipseq_histone.tsv \n",
      "\n",
      "===========================================\n",
      "Read file... \n",
      "Region A: fcc_astarr_macs | fcc_astarr_macs_input_union \n",
      "Region B: encode_chipseq_histone \n",
      "FDiry:    \n",
      "[1] \"/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_chipseq/fcc_astarr_macs_input_union/encode_chipseq_histone\"\n",
      "FName:    \n",
      " [1] \"fcc_astarr_macs_input_union.encode_chipseq_H2AFZ_ENCFF621DJP.tsv\"   \n",
      " [2] \"fcc_astarr_macs_input_union.encode_chipseq_H3K27ac_ENCFF381NDD.tsv\" \n",
      " [3] \"fcc_astarr_macs_input_union.encode_chipseq_H3K27me3_ENCFF139KZL.tsv\"\n",
      " [4] \"fcc_astarr_macs_input_union.encode_chipseq_H3K27me3_ENCFF242ENK.tsv\"\n",
      " [5] \"fcc_astarr_macs_input_union.encode_chipseq_H3K36me3_ENCFF317VHO.tsv\"\n",
      " [6] \"fcc_astarr_macs_input_union.encode_chipseq_H3K36me3_ENCFF347YYH.tsv\"\n",
      " [7] \"fcc_astarr_macs_input_union.encode_chipseq_H3K4me1_ENCFF399SGM.tsv\" \n",
      " [8] \"fcc_astarr_macs_input_union.encode_chipseq_H3K4me1_ENCFF607SUJ.tsv\" \n",
      " [9] \"fcc_astarr_macs_input_union.encode_chipseq_H3K4me2_ENCFF959YJV.tsv\" \n",
      "[10] \"fcc_astarr_macs_input_union.encode_chipseq_H3K4me3_ENCFF253TOF.tsv\" \n",
      "[11] \"fcc_astarr_macs_input_union.encode_chipseq_H3K4me3_ENCFF660WUG.tsv\" \n",
      "[12] \"fcc_astarr_macs_input_union.encode_chipseq_H3K4me3_ENCFF806YEZ.tsv\" \n",
      "[13] \"fcc_astarr_macs_input_union.encode_chipseq_H3K4me3_ENCFF911JVK.tsv\" \n",
      "[14] \"fcc_astarr_macs_input_union.encode_chipseq_H3K79me2_ENCFF544AVW.tsv\"\n",
      "[15] \"fcc_astarr_macs_input_union.encode_chipseq_H3K9ac_ENCFF286WRJ.tsv\"  \n",
      "[16] \"fcc_astarr_macs_input_union.encode_chipseq_H3K9ac_ENCFF583BKU.tsv\"  \n",
      "[17] \"fcc_astarr_macs_input_union.encode_chipseq_H3K9me1_ENCFF654SLZ.tsv\" \n",
      "[18] \"fcc_astarr_macs_input_union.encode_chipseq_H3K9me3_ENCFF601JGK.tsv\" \n",
      "[19] \"fcc_astarr_macs_input_union.encode_chipseq_H4K20me1_ENCFF605FAF.tsv\"\n",
      "\n",
      "Save file... \n",
      "FDiry: /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_chipseq/fcc_astarr_macs_input_union/summary \n",
      "FName: region.coverage.encode_chipseq_histone.tsv \n",
      "\n"
     ]
    }
   ],
   "source": [
    "### loop through region A\n",
    "for (idx in seq_along(lst_region_label_A)){\n",
    "    \n",
    "    ### get region A\n",
    "    lst = lst_region_label_A[[idx]]\n",
    "    txt_region_folder_A = lst$Folder\n",
    "    txt_region_label_A  = lst$Label\n",
    "\n",
    "    ### loop through region B\n",
    "    for (txt_region_folder_B in vec_txt_region_folder_B){\n",
    "        \n",
    "        ### get directory of annotation table\n",
    "        txt_fdiry = file.path(\n",
    "            FD_RES, \n",
    "            \"region_coverage_chipseq\",\n",
    "            txt_region_label_A, \n",
    "            txt_region_folder_B\n",
    "        )\n",
    "        txt_fname = \"*tsv\"\n",
    "        txt_fglob = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "        ### get annotation table files\n",
    "        vec_txt_fpath = Sys.glob(txt_fglob)\n",
    "        vec_txt_fname = basename(vec_txt_fpath)\n",
    "\n",
    "        ### show progress\n",
    "        cat(\"===========================================\\n\")\n",
    "        cat(\"Read file...\", \"\\n\")\n",
    "        cat(\"Region A:\", txt_region_folder_A, \"|\", txt_region_label_A, \"\\n\")\n",
    "        cat(\"Region B:\", txt_region_folder_B, \"\\n\")\n",
    "        cat(\"FDiry:   \", \"\\n\")\n",
    "        print(txt_fdiry)\n",
    "        cat(\"FName:   \", \"\\n\")\n",
    "        print(vec_txt_fname)\n",
    "        cat(\"\\n\")\n",
    "        flush.console()\n",
    "\n",
    "        ### import annotation tables\n",
    "        lst = lapply(vec_txt_fname, function(txt_fname){\n",
    "            ### set directory\n",
    "            txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "            \n",
    "            ### get annotation of region A and region B\n",
    "            vec = str_split(txt_fname, \"\\\\.\")[[1]]\n",
    "            txt_annot_A = vec[1]\n",
    "            txt_annot_B = vec[2]\n",
    "\n",
    "            ### read table and add annotation labels\n",
    "            dat = read_tsv(txt_fpath, col_names = vec_txt_cname, show_col_types = FALSE)\n",
    "            dat = dat %>% \n",
    "                dplyr::mutate(\n",
    "                    Annotation_A = txt_annot_A,\n",
    "                    Annotation_B = txt_annot_B\n",
    "                )\n",
    "            return(dat)\n",
    "        }) \n",
    "\n",
    "        ### concatenate tables and assign\n",
    "        dat = bind_rows(lst)\n",
    "        dat = dat %>% \n",
    "            tidyr::separate(\n",
    "                col  = Annotation_B, \n",
    "                into = c(\"X1\", \"X2\", \"Target\", \"Index_File\"), \n",
    "                remove = FALSE) %>%\n",
    "            dplyr::select(-X1, -X2)\n",
    "        dat_region_annot_pair = dat\n",
    "        \n",
    "        ### set directory\n",
    "        txt_fdiry = file.path(\n",
    "            FD_RES, \n",
    "            \"region_coverage_chipseq\",\n",
    "            txt_region_label_A,\n",
    "            \"summary\"\n",
    "        )\n",
    "        txt_fname = paste(\"region\", \"coverage\",  txt_region_folder_B, \"tsv\", sep = \".\")\n",
    "        txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "        ### show progress\n",
    "        cat(\"Save file...\",  \"\\n\")\n",
    "        cat(\"FDiry:\", txt_fdiry, \"\\n\")\n",
    "        cat(\"FName:\", txt_fname, \"\\n\")\n",
    "        cat(\"\\n\")\n",
    "        flush.console()\n",
    "\n",
    "        ### write tables\n",
    "        dir.create(txt_fdiry, showWarnings = FALSE)\n",
    "        write_tsv(dat_region_annot_pair, txt_fpath)\n",
    "    } ### end loop region B\n",
    "} ### end loop region A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "20da8586-4da1-4518-b250-f9aec116c45e",
   "metadata": {},
   "outputs": [],
   "source": [
    "dat = lst[[1]] %>% head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "2f864dce-fd1d-4c85-862a-92614eb4faa7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Region</th><th scope=col>Size</th><th scope=col>Covered</th><th scope=col>Sum</th><th scope=col>Mean0</th><th scope=col>Mean</th><th scope=col>Min</th><th scope=col>Max</th><th scope=col>Annotation_A</th><th scope=col>Annotation_B</th><th scope=col>Target</th><th scope=col>Index_File</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1:10015-10442</td><td>427</td><td>427</td><td> 77.8691</td><td>0.1823630</td><td>0.1823630</td><td>0.07471</td><td>0.25932</td><td>fcc_astarr_macs_input_union</td><td>encode_chipseq_H2AFZ_ENCFF202EVH</td><td>H2AFZ</td><td>ENCFF202EVH</td></tr>\n",
       "\t<tr><td>chr1:14253-14645</td><td>392</td><td>392</td><td> 29.2863</td><td>0.0747100</td><td>0.0747100</td><td>0.07471</td><td>0.07471</td><td>fcc_astarr_macs_input_union</td><td>encode_chipseq_H2AFZ_ENCFF202EVH</td><td>H2AFZ</td><td>ENCFF202EVH</td></tr>\n",
       "\t<tr><td>chr1:16015-16477</td><td>462</td><td>462</td><td> 29.6997</td><td>0.0642851</td><td>0.0642851</td><td>0.02044</td><td>0.07471</td><td>fcc_astarr_macs_input_union</td><td>encode_chipseq_H2AFZ_ENCFF202EVH</td><td>H2AFZ</td><td>ENCFF202EVH</td></tr>\n",
       "\t<tr><td>chr1:17237-17772</td><td>535</td><td>535</td><td> 39.9698</td><td>0.0747100</td><td>0.0747100</td><td>0.07471</td><td>0.07471</td><td>fcc_astarr_macs_input_union</td><td>encode_chipseq_H2AFZ_ENCFF202EVH</td><td>H2AFZ</td><td>ENCFF202EVH</td></tr>\n",
       "\t<tr><td>chr1:28903-29613</td><td>710</td><td>710</td><td> 53.0441</td><td>0.0747100</td><td>0.0747100</td><td>0.07471</td><td>0.07471</td><td>fcc_astarr_macs_input_union</td><td>encode_chipseq_H2AFZ_ENCFF202EVH</td><td>H2AFZ</td><td>ENCFF202EVH</td></tr>\n",
       "\t<tr><td>chr1:30803-31072</td><td>269</td><td>269</td><td>313.7960</td><td>1.1665300</td><td>1.1665300</td><td>0.07471</td><td>3.18188</td><td>fcc_astarr_macs_input_union</td><td>encode_chipseq_H2AFZ_ENCFF202EVH</td><td>H2AFZ</td><td>ENCFF202EVH</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 12\n",
       "\\begin{tabular}{llllllllllll}\n",
       " Region & Size & Covered & Sum & Mean0 & Mean & Min & Max & Annotation\\_A & Annotation\\_B & Target & Index\\_File\\\\\n",
       " <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t chr1:10015-10442 & 427 & 427 &  77.8691 & 0.1823630 & 0.1823630 & 0.07471 & 0.25932 & fcc\\_astarr\\_macs\\_input\\_union & encode\\_chipseq\\_H2AFZ\\_ENCFF202EVH & H2AFZ & ENCFF202EVH\\\\\n",
       "\t chr1:14253-14645 & 392 & 392 &  29.2863 & 0.0747100 & 0.0747100 & 0.07471 & 0.07471 & fcc\\_astarr\\_macs\\_input\\_union & encode\\_chipseq\\_H2AFZ\\_ENCFF202EVH & H2AFZ & ENCFF202EVH\\\\\n",
       "\t chr1:16015-16477 & 462 & 462 &  29.6997 & 0.0642851 & 0.0642851 & 0.02044 & 0.07471 & fcc\\_astarr\\_macs\\_input\\_union & encode\\_chipseq\\_H2AFZ\\_ENCFF202EVH & H2AFZ & ENCFF202EVH\\\\\n",
       "\t chr1:17237-17772 & 535 & 535 &  39.9698 & 0.0747100 & 0.0747100 & 0.07471 & 0.07471 & fcc\\_astarr\\_macs\\_input\\_union & encode\\_chipseq\\_H2AFZ\\_ENCFF202EVH & H2AFZ & ENCFF202EVH\\\\\n",
       "\t chr1:28903-29613 & 710 & 710 &  53.0441 & 0.0747100 & 0.0747100 & 0.07471 & 0.07471 & fcc\\_astarr\\_macs\\_input\\_union & encode\\_chipseq\\_H2AFZ\\_ENCFF202EVH & H2AFZ & ENCFF202EVH\\\\\n",
       "\t chr1:30803-31072 & 269 & 269 & 313.7960 & 1.1665300 & 1.1665300 & 0.07471 & 3.18188 & fcc\\_astarr\\_macs\\_input\\_union & encode\\_chipseq\\_H2AFZ\\_ENCFF202EVH & H2AFZ & ENCFF202EVH\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 12\n",
       "\n",
       "| Region &lt;chr&gt; | Size &lt;dbl&gt; | Covered &lt;dbl&gt; | Sum &lt;dbl&gt; | Mean0 &lt;dbl&gt; | Mean &lt;dbl&gt; | Min &lt;dbl&gt; | Max &lt;dbl&gt; | Annotation_A &lt;chr&gt; | Annotation_B &lt;chr&gt; | Target &lt;chr&gt; | Index_File &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1:10015-10442 | 427 | 427 |  77.8691 | 0.1823630 | 0.1823630 | 0.07471 | 0.25932 | fcc_astarr_macs_input_union | encode_chipseq_H2AFZ_ENCFF202EVH | H2AFZ | ENCFF202EVH |\n",
       "| chr1:14253-14645 | 392 | 392 |  29.2863 | 0.0747100 | 0.0747100 | 0.07471 | 0.07471 | fcc_astarr_macs_input_union | encode_chipseq_H2AFZ_ENCFF202EVH | H2AFZ | ENCFF202EVH |\n",
       "| chr1:16015-16477 | 462 | 462 |  29.6997 | 0.0642851 | 0.0642851 | 0.02044 | 0.07471 | fcc_astarr_macs_input_union | encode_chipseq_H2AFZ_ENCFF202EVH | H2AFZ | ENCFF202EVH |\n",
       "| chr1:17237-17772 | 535 | 535 |  39.9698 | 0.0747100 | 0.0747100 | 0.07471 | 0.07471 | fcc_astarr_macs_input_union | encode_chipseq_H2AFZ_ENCFF202EVH | H2AFZ | ENCFF202EVH |\n",
       "| chr1:28903-29613 | 710 | 710 |  53.0441 | 0.0747100 | 0.0747100 | 0.07471 | 0.07471 | fcc_astarr_macs_input_union | encode_chipseq_H2AFZ_ENCFF202EVH | H2AFZ | ENCFF202EVH |\n",
       "| chr1:30803-31072 | 269 | 269 | 313.7960 | 1.1665300 | 1.1665300 | 0.07471 | 3.18188 | fcc_astarr_macs_input_union | encode_chipseq_H2AFZ_ENCFF202EVH | H2AFZ | ENCFF202EVH |\n",
       "\n"
      ],
      "text/plain": [
       "  Region           Size Covered Sum      Mean0     Mean      Min     Max    \n",
       "1 chr1:10015-10442 427  427      77.8691 0.1823630 0.1823630 0.07471 0.25932\n",
       "2 chr1:14253-14645 392  392      29.2863 0.0747100 0.0747100 0.07471 0.07471\n",
       "3 chr1:16015-16477 462  462      29.6997 0.0642851 0.0642851 0.02044 0.07471\n",
       "4 chr1:17237-17772 535  535      39.9698 0.0747100 0.0747100 0.07471 0.07471\n",
       "5 chr1:28903-29613 710  710      53.0441 0.0747100 0.0747100 0.07471 0.07471\n",
       "6 chr1:30803-31072 269  269     313.7960 1.1665300 1.1665300 0.07471 3.18188\n",
       "  Annotation_A                Annotation_B                     Target\n",
       "1 fcc_astarr_macs_input_union encode_chipseq_H2AFZ_ENCFF202EVH H2AFZ \n",
       "2 fcc_astarr_macs_input_union encode_chipseq_H2AFZ_ENCFF202EVH H2AFZ \n",
       "3 fcc_astarr_macs_input_union encode_chipseq_H2AFZ_ENCFF202EVH H2AFZ \n",
       "4 fcc_astarr_macs_input_union encode_chipseq_H2AFZ_ENCFF202EVH H2AFZ \n",
       "5 fcc_astarr_macs_input_union encode_chipseq_H2AFZ_ENCFF202EVH H2AFZ \n",
       "6 fcc_astarr_macs_input_union encode_chipseq_H2AFZ_ENCFF202EVH H2AFZ \n",
       "  Index_File \n",
       "1 ENCFF202EVH\n",
       "2 ENCFF202EVH\n",
       "3 ENCFF202EVH\n",
       "4 ENCFF202EVH\n",
       "5 ENCFF202EVH\n",
       "6 ENCFF202EVH"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat %>% \n",
    "    tidyr::separate(\n",
    "        col  = Annotation_B, \n",
    "        into = c(\"X1\", \"X2\", \"Target\", \"Index_File\"), \n",
    "        remove = FALSE) %>%\n",
    "    dplyr::select(-X1, -X2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "07b3d48b-00fa-4048-bf7a-276f35eff8f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A spec_tbl_df: 19 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Folder</th><th scope=col>FName</th><th scope=col>Label</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF054RSU.bigWig</td><td>encode_chipseq_H3K4me2_ENCFF054RSU </td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF071GML.bigWig</td><td>encode_chipseq_H3K4me3_ENCFF071GML </td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF178QDA.bigWig</td><td>encode_chipseq_H3K9ac_ENCFF178QDA  </td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF202EVH.bigWig</td><td>encode_chipseq_H2AFZ_ENCFF202EVH   </td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF220RGS.bigWig</td><td>encode_chipseq_H3K9me1_ENCFF220RGS </td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF239EBH.bigWig</td><td>encode_chipseq_H3K9ac_ENCFF239EBH  </td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF287LBI.bigWig</td><td>encode_chipseq_H3K4me1_ENCFF287LBI </td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF334HSS.bigWig</td><td>encode_chipseq_H3K79me2_ENCFF334HSS</td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF405ZDL.bigWig</td><td>encode_chipseq_H3K4me3_ENCFF405ZDL </td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF457URZ.bigWig</td><td>encode_chipseq_H3K4me1_ENCFF457URZ </td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF461RKK.bigWig</td><td>encode_chipseq_H3K4me3_ENCFF461RKK </td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF465GBD.bigWig</td><td>encode_chipseq_H3K27ac_ENCFF465GBD </td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF582IMB.bigWig</td><td>encode_chipseq_H3K27me3_ENCFF582IMB</td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF605EVL.bigWig</td><td>encode_chipseq_H3K36me3_ENCFF605EVL</td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF632NQA.bigWig</td><td>encode_chipseq_H3K9me3_ENCFF632NQA </td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF633OZC.bigWig</td><td>encode_chipseq_H3K36me3_ENCFF633OZC</td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF694ODT.bigWig</td><td>encode_chipseq_H4K20me1_ENCFF694ODT</td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF767UON.bigWig</td><td>encode_chipseq_H3K4me3_ENCFF767UON </td></tr>\n",
       "\t<tr><td>encode_chipseq_histone</td><td>ENCFF847BFA.bigWig</td><td>encode_chipseq_H3K27me3_ENCFF847BFA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A spec\\_tbl\\_df: 19 × 3\n",
       "\\begin{tabular}{lll}\n",
       " Folder & FName & Label\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t encode\\_chipseq\\_histone & ENCFF054RSU.bigWig & encode\\_chipseq\\_H3K4me2\\_ENCFF054RSU \\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF071GML.bigWig & encode\\_chipseq\\_H3K4me3\\_ENCFF071GML \\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF178QDA.bigWig & encode\\_chipseq\\_H3K9ac\\_ENCFF178QDA  \\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF202EVH.bigWig & encode\\_chipseq\\_H2AFZ\\_ENCFF202EVH   \\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF220RGS.bigWig & encode\\_chipseq\\_H3K9me1\\_ENCFF220RGS \\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF239EBH.bigWig & encode\\_chipseq\\_H3K9ac\\_ENCFF239EBH  \\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF287LBI.bigWig & encode\\_chipseq\\_H3K4me1\\_ENCFF287LBI \\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF334HSS.bigWig & encode\\_chipseq\\_H3K79me2\\_ENCFF334HSS\\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF405ZDL.bigWig & encode\\_chipseq\\_H3K4me3\\_ENCFF405ZDL \\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF457URZ.bigWig & encode\\_chipseq\\_H3K4me1\\_ENCFF457URZ \\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF461RKK.bigWig & encode\\_chipseq\\_H3K4me3\\_ENCFF461RKK \\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF465GBD.bigWig & encode\\_chipseq\\_H3K27ac\\_ENCFF465GBD \\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF582IMB.bigWig & encode\\_chipseq\\_H3K27me3\\_ENCFF582IMB\\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF605EVL.bigWig & encode\\_chipseq\\_H3K36me3\\_ENCFF605EVL\\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF632NQA.bigWig & encode\\_chipseq\\_H3K9me3\\_ENCFF632NQA \\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF633OZC.bigWig & encode\\_chipseq\\_H3K36me3\\_ENCFF633OZC\\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF694ODT.bigWig & encode\\_chipseq\\_H4K20me1\\_ENCFF694ODT\\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF767UON.bigWig & encode\\_chipseq\\_H3K4me3\\_ENCFF767UON \\\\\n",
       "\t encode\\_chipseq\\_histone & ENCFF847BFA.bigWig & encode\\_chipseq\\_H3K27me3\\_ENCFF847BFA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A spec_tbl_df: 19 × 3\n",
       "\n",
       "| Folder &lt;chr&gt; | FName &lt;chr&gt; | Label &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| encode_chipseq_histone | ENCFF054RSU.bigWig | encode_chipseq_H3K4me2_ENCFF054RSU  |\n",
       "| encode_chipseq_histone | ENCFF071GML.bigWig | encode_chipseq_H3K4me3_ENCFF071GML  |\n",
       "| encode_chipseq_histone | ENCFF178QDA.bigWig | encode_chipseq_H3K9ac_ENCFF178QDA   |\n",
       "| encode_chipseq_histone | ENCFF202EVH.bigWig | encode_chipseq_H2AFZ_ENCFF202EVH    |\n",
       "| encode_chipseq_histone | ENCFF220RGS.bigWig | encode_chipseq_H3K9me1_ENCFF220RGS  |\n",
       "| encode_chipseq_histone | ENCFF239EBH.bigWig | encode_chipseq_H3K9ac_ENCFF239EBH   |\n",
       "| encode_chipseq_histone | ENCFF287LBI.bigWig | encode_chipseq_H3K4me1_ENCFF287LBI  |\n",
       "| encode_chipseq_histone | ENCFF334HSS.bigWig | encode_chipseq_H3K79me2_ENCFF334HSS |\n",
       "| encode_chipseq_histone | ENCFF405ZDL.bigWig | encode_chipseq_H3K4me3_ENCFF405ZDL  |\n",
       "| encode_chipseq_histone | ENCFF457URZ.bigWig | encode_chipseq_H3K4me1_ENCFF457URZ  |\n",
       "| encode_chipseq_histone | ENCFF461RKK.bigWig | encode_chipseq_H3K4me3_ENCFF461RKK  |\n",
       "| encode_chipseq_histone | ENCFF465GBD.bigWig | encode_chipseq_H3K27ac_ENCFF465GBD  |\n",
       "| encode_chipseq_histone | ENCFF582IMB.bigWig | encode_chipseq_H3K27me3_ENCFF582IMB |\n",
       "| encode_chipseq_histone | ENCFF605EVL.bigWig | encode_chipseq_H3K36me3_ENCFF605EVL |\n",
       "| encode_chipseq_histone | ENCFF632NQA.bigWig | encode_chipseq_H3K9me3_ENCFF632NQA  |\n",
       "| encode_chipseq_histone | ENCFF633OZC.bigWig | encode_chipseq_H3K36me3_ENCFF633OZC |\n",
       "| encode_chipseq_histone | ENCFF694ODT.bigWig | encode_chipseq_H4K20me1_ENCFF694ODT |\n",
       "| encode_chipseq_histone | ENCFF767UON.bigWig | encode_chipseq_H3K4me3_ENCFF767UON  |\n",
       "| encode_chipseq_histone | ENCFF847BFA.bigWig | encode_chipseq_H3K27me3_ENCFF847BFA |\n",
       "\n"
      ],
      "text/plain": [
       "   Folder                 FName             \n",
       "1  encode_chipseq_histone ENCFF054RSU.bigWig\n",
       "2  encode_chipseq_histone ENCFF071GML.bigWig\n",
       "3  encode_chipseq_histone ENCFF178QDA.bigWig\n",
       "4  encode_chipseq_histone ENCFF202EVH.bigWig\n",
       "5  encode_chipseq_histone ENCFF220RGS.bigWig\n",
       "6  encode_chipseq_histone ENCFF239EBH.bigWig\n",
       "7  encode_chipseq_histone ENCFF287LBI.bigWig\n",
       "8  encode_chipseq_histone ENCFF334HSS.bigWig\n",
       "9  encode_chipseq_histone ENCFF405ZDL.bigWig\n",
       "10 encode_chipseq_histone ENCFF457URZ.bigWig\n",
       "11 encode_chipseq_histone ENCFF461RKK.bigWig\n",
       "12 encode_chipseq_histone ENCFF465GBD.bigWig\n",
       "13 encode_chipseq_histone ENCFF582IMB.bigWig\n",
       "14 encode_chipseq_histone ENCFF605EVL.bigWig\n",
       "15 encode_chipseq_histone ENCFF632NQA.bigWig\n",
       "16 encode_chipseq_histone ENCFF633OZC.bigWig\n",
       "17 encode_chipseq_histone ENCFF694ODT.bigWig\n",
       "18 encode_chipseq_histone ENCFF767UON.bigWig\n",
       "19 encode_chipseq_histone ENCFF847BFA.bigWig\n",
       "   Label                              \n",
       "1  encode_chipseq_H3K4me2_ENCFF054RSU \n",
       "2  encode_chipseq_H3K4me3_ENCFF071GML \n",
       "3  encode_chipseq_H3K9ac_ENCFF178QDA  \n",
       "4  encode_chipseq_H2AFZ_ENCFF202EVH   \n",
       "5  encode_chipseq_H3K9me1_ENCFF220RGS \n",
       "6  encode_chipseq_H3K9ac_ENCFF239EBH  \n",
       "7  encode_chipseq_H3K4me1_ENCFF287LBI \n",
       "8  encode_chipseq_H3K79me2_ENCFF334HSS\n",
       "9  encode_chipseq_H3K4me3_ENCFF405ZDL \n",
       "10 encode_chipseq_H3K4me1_ENCFF457URZ \n",
       "11 encode_chipseq_H3K4me3_ENCFF461RKK \n",
       "12 encode_chipseq_H3K27ac_ENCFF465GBD \n",
       "13 encode_chipseq_H3K27me3_ENCFF582IMB\n",
       "14 encode_chipseq_H3K36me3_ENCFF605EVL\n",
       "15 encode_chipseq_H3K9me3_ENCFF632NQA \n",
       "16 encode_chipseq_H3K36me3_ENCFF633OZC\n",
       "17 encode_chipseq_H4K20me1_ENCFF694ODT\n",
       "18 encode_chipseq_H3K4me3_ENCFF767UON \n",
       "19 encode_chipseq_H3K27me3_ENCFF847BFA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat_region_label_B"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a39d4861-9c75-4391-bb94-9a4534fb62c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "### import annotation tables\n",
    "        lst = lapply(vec_txt_fname, function(txt_fname){\n",
    "            ### set directory\n",
    "            txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "            \n",
    "            ### get annotation of region A and region B\n",
    "            vec = str_split(txt_fname, \"\\\\.\")[[1]]\n",
    "            txt_annot_A = vec[1]\n",
    "            txt_annot_B = vec[2]\n",
    "\n",
    "            ### read table and add annotation labels\n",
    "            dat = read_tsv(txt_fpath, col_names = vec_txt_cname, show_col_types = FALSE)\n",
    "            dat = dat %>% \n",
    "                dplyr::mutate(\n",
    "                    Region_A = fun_gen_region(Chrom_A, ChromStart_A, ChromEnd_A),\n",
    "                    Region_B = fun_gen_region(Chrom_B, ChromStart_B, ChromEnd_B),\n",
    "                    Annotation_A = txt_annot_A,\n",
    "                    Annotation_B = txt_annot_B\n",
    "                )\n",
    "            return(dat)\n",
    "        }) \n",
    "\n",
    "        ### concatenate tables and assign\n",
    "        dat = bind_rows(lst)\n",
    "        dat_region_annot_pair = dat\n",
    "        \n",
    "        ### summarize annotation tables\n",
    "        dat = dat_region_annot_pair\n",
    "        dat = dat %>%\n",
    "            dplyr::group_by(\n",
    "                Chrom_A, ChromStart_A, ChromEnd_A, Region_A,\n",
    "                Annotation_A,\n",
    "                Annotation_B,\n",
    "                Group_B,\n",
    "                Label_B\n",
    "            ) %>%\n",
    "            dplyr::summarise(\n",
    "                Region_Annot = paste(Region_B, collapse=\";\"),\n",
    "                Region_Count = n(),\n",
    "                .groups = \"drop\")\n",
    "\n",
    "        ### rename the columns and assign\n",
    "        colnames(dat) = c(\n",
    "            \"Chrom\", \"ChromStart\", \"ChromEnd\", \"Region\", \n",
    "            \"Annotation_A\", \n",
    "            \"Annotation_B\",\n",
    "            \"Group\",\n",
    "            \"Label\",\n",
    "            \"Region_Annot\",\n",
    "            \"Region_Count\"\n",
    "        )\n",
    "        dat_region_annot_summary = dat\n",
    "\n",
    "        ### set directory\n",
    "        txt_fdiry = file.path(\n",
    "            FD_RES, \n",
    "            \"region_annotation\",\n",
    "            txt_region_label_A,\n",
    "            \"summary\"\n",
    "        )\n",
    "        txt_fname1 = paste(\"region\", \"intersect\",  txt_region_folder_B, \"tsv\", sep = \".\")\n",
    "        txt_fpath1 = file.path(txt_fdiry, txt_fname1)\n",
    "        txt_fname2 = paste(\"region\", \"annotation\", txt_region_folder_B, \"tsv\", sep = \".\")\n",
    "        txt_fpath2 = file.path(txt_fdiry, txt_fname2)\n",
    "        \n",
    "        ### show progress\n",
    "        cat(\"Save annotation...\",  \"\\n\")\n",
    "        cat(\"FDiry: \", txt_fdiry, \"\\n\")\n",
    "        cat(\"FName1:\", txt_fname1, \"\\n\")\n",
    "        cat(\"FName2:\", txt_fname2, \"\\n\")\n",
    "        cat(\"\\n\")\n",
    "        flush.console()\n",
    "\n",
    "        ### write tables\n",
    "        dir.create(txt_fdiry, showWarnings = FALSE)\n",
    "        write_tsv(dat_region_annot_pair,    txt_fpath1)\n",
    "        write_tsv(dat_region_annot_summary, txt_fpath2)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
