<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Region Coverage (FCC) 04 â€“ ENCODE FCC Project</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-b8872d4f548cb279fc940b18642fe439.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">ENCODE FCC Project</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../report.html"> 
<span class="menu-text">Report</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../analysis.html"> 
<span class="menu-text">Analysis</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../data.html"> 
<span class="menu-text">Data</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../code.html"> 
<span class="menu-text">Code</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../results.html"> 
<span class="menu-text">Results</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
    <a href="https://github.com/ReddyLab/Proj_ENCODE_FCC" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Region Coverage (FCC) 04</li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">Setup</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../01_setup/work_01_env.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">01 Working Environment</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../01_setup/work_02_dir.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">02 Working Directory</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../01_setup/work_03_ref.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">03 Reference Files</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../01_setup/work_04_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">04 Data Files</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../01_setup/work_11_prep_fcc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11 Prepare: FCC</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../01_setup/work_12_prep_hic.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12 Prepare: Hi-C</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../01_setup/work_13_prep_chipseq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13 Prepare: ChIP-seq</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../01_setup/work_14_prep_region.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14 Prepare: Region</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../01_setup/work_15_prep_others.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15 Prepare: Others</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../01_setup/work_21_results_fcc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12 Results (FCC)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../01_setup/work_21_results_fcc2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12 Results (FCC)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../01_setup/work_22_results_hic.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13 Results (HiC)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../01_setup/work_23_results_region.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14 Results (Region)</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Prepare</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../02_prepare/data_01_fcc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FCC Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../02_prepare/data_02_hic.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Hi-C Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../02_prepare/data_11_chipseq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ChIP-seq TF/Histone</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../02_prepare/data_12_encode_annotation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chromatin States and TF Modules</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../02_prepare/data_21_peak_calling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Peak calling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../02_prepare/data_22_ocr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Open chromatin regions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../02_prepare/data_23_others.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Other Annotations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">Coverage of FCC</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../11_region_coverage_fcc/01_region_coverage_fcc_screen.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FCC coverage at ATAC/DHS regions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../11_region_coverage_fcc/02_region_coverage_fcc_scatter.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Whole genome comparison of FCC assays</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../11_region_coverage_fcc/03_region_coverage_fcc_correlation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Rank normalized FCC effect size</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../11_region_coverage_fcc/04_region_coverage_fcc_comparison.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Comparisons of FCC across genome annotations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">Coverage of ChIP-seq</span></span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Region Annotation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../13_region_annotation/region_annot_01_genome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Intersect/Summarize genomic annotaions by accessible regions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../13_region_annotation/region_annot_02_peak_calls.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Intersect/Summarize FCC peak calls by accessible regions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../13_region_annotation/region_annot_03_starrmpra_vote2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Proximal/Distal-Active/Repressive CREs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../13_region_annotation/region_annot_04_starrmpracrispr_vote2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Full CREs from STARR/MPRA/CRISPR assays</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">FCC correlation</span></span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">FCC comparison</span></span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">FCC enrichment</span></span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">FCC UMAP/Cluster</span></span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">FCC vs HiC</span></span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" role="navigation" aria-expanded="false">
 <span class="menu-text">Test Notebook</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-11" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../notebooks/test/test_notebook.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Test ipynb for documentation</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#prepare" id="toc-prepare" class="nav-link active" data-scroll-target="#prepare">Prepare</a></li>
  <li><a href="#execute" id="toc-execute" class="nav-link" data-scroll-target="#execute">Execute</a></li>
  <li><a href="#execute-1" id="toc-execute-1" class="nav-link" data-scroll-target="#execute-1">Execute</a></li>
  <li><a href="#review" id="toc-review" class="nav-link" data-scroll-target="#review">Review</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Region Coverage (FCC) 04</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li></ul></div></div>
<p class="subtitle lead">Intersect z scores across regions (CRISPRi and Lenti-MPRA)</p>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p><strong>Set environment</strong></p>
<div id="9b801eb9-e308-49c3-9204-f251c0b63c00" class="cell" data-execution_count="1">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> ../run_config_project.sh</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="ex">show_env</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>You are working on             Duke Server: RCC
BASE DIRECTORY (FD_BASE):      /data/reddylab/Kuei
REPO DIRECTORY (FD_REPO):      /data/reddylab/Kuei/repo
WORK DIRECTORY (FD_WORK):      /data/reddylab/Kuei/work
DATA DIRECTORY (FD_DATA):      /data/reddylab/Kuei/data
CONTAINER DIR. (FD_SING):      /data/reddylab/Kuei/container

You are working with           ENCODE FCC
PATH OF PROJECT (FD_PRJ):      /data/reddylab/Kuei/repo/Proj_ENCODE_FCC
PROJECT RESULTS (FD_RES):      /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results
PROJECT SCRIPTS (FD_EXE):      /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/scripts
PROJECT DATA    (FD_DAT):      /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/data
PROJECT NOTE    (FD_NBK):      /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/notebooks
PROJECT DOCS    (FD_DOC):      /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/docs
PROJECT LOG     (FD_LOG):      /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/log
PROJECT REF     (FD_REF):      /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/references
PROJECT IMAGE   (FP_PRJ_SIF):  /data/reddylab/Kuei/container/project/singularity_proj_encode_fcc.sif
PROJECT CONF.   (FP_CNF):      /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/scripts/config_project.sh
</code></pre>
</div>
</div>
<section id="prepare" class="level2">
<h2 class="anchored" data-anchor-id="prepare">Prepare</h2>
<p><strong>Region sets for calculating FCC effect size</strong></p>
<div id="92ec5040-f58b-4cf4-9662-338300bef0f9" class="cell" data-execution_count="2">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">### set directory</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="va">FDIRY</span><span class="op">=</span><span class="va">${FD_RES}</span>/region/summary</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="va">FNAME</span><span class="op">=</span>metadata.label.region_for_analysis.tsv</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="va">FPATH</span><span class="op">=</span><span class="va">${FDIRY}</span>/<span class="va">${FNAME}</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co">### get table</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="va">FP_REGION_LABEL</span><span class="op">=</span><span class="va">${FPATH}</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="va">${FPATH}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Folder  FName   Label   FPath
region_for_analysis K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz fcc_astarr_macs_input_overlap   /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz
region_for_analysis K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz    fcc_astarr_macs_input_union /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz
region_for_analysis K562.hg38.ENCSR000EKS.ENCFF274YGF.DNase.bed.gz  dnase_ENCFF274YGF   /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR000EKS.ENCFF274YGF.DNase.bed.gz
region_for_analysis K562.hg38.ENCSR000EOT.ENCFF185XRG.DNase.bed.gz  dnase_ENCFF185XRG   /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR000EOT.ENCFF185XRG.DNase.bed.gz
region_for_analysis K562.hg38.ENCSR483RKN.ENCFF558BLC.ATAC.bed.gz   atac_ENCFF558BLC    /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR483RKN.ENCFF558BLC.ATAC.bed.gz
region_for_analysis K562.hg38.ENCSR483RKN.ENCFF925CYR.ATAC.bed.gz   atac_ENCFF925CYR    /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR483RKN.ENCFF925CYR.ATAC.bed.gz
region_for_analysis K562.hg38.ENCSR868FGK.ENCFF333TAT.ATAC.bed.gz   atac_ENCFF333TAT    /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR868FGK.ENCFF333TAT.ATAC.bed.gz
region_for_analysis K562.hg38.ENCSR868FGK.ENCFF948AFM.ATAC.bed.gz   atac_ENCFF948AFM    /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR868FGK.ENCFF948AFM.ATAC.bed.gz</code></pre>
</div>
</div>
<div id="c98db639-4238-4fa1-9e86-d45c09ff234b" class="cell" data-execution_count="3">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="va">${FP_REGION_LABEL}</span> <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f</span> 1-3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Folder  FName   Label
region_for_analysis K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz fcc_astarr_macs_input_overlap
region_for_analysis K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz    fcc_astarr_macs_input_union
region_for_analysis K562.hg38.ENCSR000EKS.ENCFF274YGF.DNase.bed.gz  dnase_ENCFF274YGF
region_for_analysis K562.hg38.ENCSR000EOT.ENCFF185XRG.DNase.bed.gz  dnase_ENCFF185XRG
region_for_analysis K562.hg38.ENCSR483RKN.ENCFF558BLC.ATAC.bed.gz   atac_ENCFF558BLC
region_for_analysis K562.hg38.ENCSR483RKN.ENCFF925CYR.ATAC.bed.gz   atac_ENCFF925CYR
region_for_analysis K562.hg38.ENCSR868FGK.ENCFF333TAT.ATAC.bed.gz   atac_ENCFF333TAT
region_for_analysis K562.hg38.ENCSR868FGK.ENCFF948AFM.ATAC.bed.gz   atac_ENCFF948AFM</code></pre>
</div>
</div>
<p><strong>Assay</strong></p>
<div id="c6c3617c-55fd-416a-ba63-2df69b245103" class="cell" data-execution_count="4">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">### Assay</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="va">ASSAY_TYPE</span><span class="op">=</span>assay_fcc</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="va">ASSAY_NAMES</span><span class="op">=</span><span class="va">(</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>    CRISPRi_FlowFISH_K562_Riley_JinWoo</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    CRISPRi_Growth_K562_Gersbach_JinWoo</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    MPRA_Lenti_K562_Nadav_Vikram</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="va">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="execute" class="level2">
<h2 class="anchored" data-anchor-id="execute">Execute</h2>
<p><strong>Test loop: Region</strong></p>
<div id="9e4ebbde-3f5d-4323-9250-011eff4bfe35" class="cell" data-execution_count="5">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">FOLDER_A</span> <span class="va">FNAME_A</span> <span class="va">LABEL_A</span> <span class="va">FPATH_A</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    <span class="co">### show progress</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> ==============================</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"Region folder:"</span> <span class="va">${FOLDER_A}</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"Region label: "</span> <span class="va">${LABEL_A}</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    <span class="co">### Set directory</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    <span class="va">FN_INP_A</span><span class="op">=</span><span class="va">${FNAME_A}</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    <span class="va">FP_INP_A</span><span class="op">=</span><span class="va">${FPATH_A}</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span> <span class="op">&lt;</span> <span class="op">&lt;(</span><span class="fu">cat</span> <span class="va">${FP_REGION_LABEL}</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'NR &gt;=2 {print}'</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_overlap
==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_union
==============================
Region folder: region_for_analysis
Region label:  dnase_ENCFF274YGF
==============================
Region folder: region_for_analysis
Region label:  dnase_ENCFF185XRG
==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF558BLC
==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF925CYR
==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF333TAT
==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF948AFM</code></pre>
</div>
</div>
<div id="51957df2-4779-40ee-a39a-39c229a56d86" class="cell" data-execution_count="6">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="va">${FP_REGION_LABEL}</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'NR &gt;=2 {print}'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>region_for_analysis K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz fcc_astarr_macs_input_overlap   /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz
region_for_analysis K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz    fcc_astarr_macs_input_union /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz
region_for_analysis K562.hg38.ENCSR000EKS.ENCFF274YGF.DNase.bed.gz  dnase_ENCFF274YGF   /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR000EKS.ENCFF274YGF.DNase.bed.gz
region_for_analysis K562.hg38.ENCSR000EOT.ENCFF185XRG.DNase.bed.gz  dnase_ENCFF185XRG   /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR000EOT.ENCFF185XRG.DNase.bed.gz
region_for_analysis K562.hg38.ENCSR483RKN.ENCFF558BLC.ATAC.bed.gz   atac_ENCFF558BLC    /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR483RKN.ENCFF558BLC.ATAC.bed.gz
region_for_analysis K562.hg38.ENCSR483RKN.ENCFF925CYR.ATAC.bed.gz   atac_ENCFF925CYR    /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR483RKN.ENCFF925CYR.ATAC.bed.gz
region_for_analysis K562.hg38.ENCSR868FGK.ENCFF333TAT.ATAC.bed.gz   atac_ENCFF333TAT    /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR868FGK.ENCFF333TAT.ATAC.bed.gz
region_for_analysis K562.hg38.ENCSR868FGK.ENCFF948AFM.ATAC.bed.gz   atac_ENCFF948AFM    /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR868FGK.ENCFF948AFM.ATAC.bed.gz</code></pre>
</div>
</div>
<p><strong>Test loop: Assay</strong></p>
<div id="7b6b41c7-efe0-4127-968a-d7c4c404d092" class="cell" data-execution_count="8">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> ASSAY_NAME <span class="kw">in</span> <span class="va">${ASSAY_NAMES</span><span class="op">[@]</span><span class="va">}</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>    <span class="co">### show progress</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> ==============================</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span> <span class="st">"Assay name: "</span> <span class="va">${ASSAY_NAME}</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    <span class="va">FD_INP</span><span class="op">=</span><span class="va">${FD_RES}</span>/<span class="va">${ASSAY_TYPE}</span>/<span class="va">${ASSAY_NAME}</span>/coverage</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    <span class="co">#FP_INPS=($(ls ${FD_INP}/*hg38*ZScore*bed.gz))</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>    <span class="va">FP_INPS</span><span class="op">=</span><span class="va">($(</span><span class="fu">ls</span> <span class="va">${FD_INP}</span>/<span class="pp">*</span>hg38<span class="pp">*</span>bed.gz<span class="va">))</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>    <span class="co">### loop through each sample</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> FP_INP_B <span class="kw">in</span> <span class="va">${FP_INPS</span><span class="op">[@]</span><span class="va">}</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>        <span class="va">FD_INP_B</span><span class="op">=</span><span class="va">$(</span><span class="fu">dirname</span>  <span class="va">${FP_INP_B})</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>        <span class="va">FN_INP_B</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${FP_INP_B})</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>        <span class="bu">echo</span> <span class="va">${FD_INP_B}</span></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>        <span class="bu">echo</span> <span class="va">${FN_INP_B}</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>    <span class="cf">done</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>    <span class="bu">echo</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>==============================
Assay name:  CRISPRi_FlowFISH_K562_Riley_JinWoo
/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz
/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

==============================
Assay name:  CRISPRi_Growth_K562_Gersbach_JinWoo
/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_fcc/CRISPRi_Growth_K562_Gersbach_JinWoo/coverage
CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

==============================
Assay name:  MPRA_Lenti_K562_Nadav_Vikram
/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
LMPRA_K562.hg38.Log2FC.stranded.bed.gz
/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
LMPRA_K562.hg38.ZScore.stranded.bed.gz
</code></pre>
</div>
</div>
<p><strong>Test loop: Region + Assay</strong></p>
<div id="812ccdda-d46d-4262-a9c2-8cc695d8bd8c" class="cell" data-execution_count="9">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">FOLDER_A</span> <span class="va">FNAME_A</span> <span class="va">LABEL_A</span> <span class="va">FPATH_A</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>    <span class="co">### set input file</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>    <span class="va">FD_INP_A</span><span class="op">=</span><span class="va">$(</span><span class="fu">dirname</span> <span class="va">${FPATH_A})</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    <span class="va">FN_INP_A</span><span class="op">=</span><span class="va">${FNAME_A}</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>    <span class="va">FP_INP_A</span><span class="op">=</span><span class="va">${FPATH_A}</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>    <span class="co">### loop through each assay</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> ASSAY_NAME <span class="kw">in</span> <span class="va">${ASSAY_NAMES</span><span class="op">[@]</span><span class="va">}</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>        <span class="co">### Set input file</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>        <span class="va">FD_INP</span><span class="op">=</span><span class="va">${FD_RES}</span>/<span class="va">${ASSAY_TYPE}</span>/<span class="va">${ASSAY_NAME}</span>/coverage</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>        <span class="va">FP_INPS</span><span class="op">=</span><span class="va">($(</span><span class="fu">ls</span> <span class="va">${FD_INP}</span>/<span class="pp">*</span>hg38<span class="pp">*</span>bed.gz<span class="va">))</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>        <span class="co">#FP_INPS=($(ls ${FD_INP}/*hg38*ZScore*bed.gz))</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>        <span class="co">### loop through each sample</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> FP_INP_B <span class="kw">in</span> <span class="va">${FP_INPS</span><span class="op">[@]</span><span class="va">}</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>            <span class="co">### set input file</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>            <span class="va">FD_INP_B</span><span class="op">=</span><span class="va">$(</span><span class="fu">dirname</span>  <span class="va">${FP_INP_B})</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>            <span class="va">FN_INP_B</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${FP_INP_B})</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>            <span class="co">### setup output file</span></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>            <span class="va">FD_OUT</span><span class="op">=</span><span class="va">${FD_RES}</span>/region_coverage_fcc_score/<span class="va">${LABEL_A}</span>/<span class="va">${ASSAY_NAME}</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>            <span class="va">FN_OUT</span><span class="op">=</span><span class="va">${FN_INP_B}</span></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>            <span class="va">FP_OUT</span><span class="op">=</span><span class="va">${FD_OUT}</span>/<span class="va">${FN_OUT}</span></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>            <span class="co">### setup log file</span></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>            <span class="va">FN_LOG</span><span class="op">=</span>region.coverage.<span class="va">${LABEL_A}</span>.<span class="va">${ASSAY_NAME}</span>.<span class="va">${FN_OUT}</span>.txt</span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>            <span class="va">FP_LOG</span><span class="op">=</span><span class="va">${FD_LOG}</span>/<span class="va">${FN_LOG}</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>            <span class="co">### show progress</span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> ==============================</span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Region folder:"</span> <span class="va">${FOLDER_A}</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Region label: "</span> <span class="va">${LABEL_A}</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Assay  name:  "</span> <span class="va">${ASSAY_NAME}</span></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span></span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Input fdiry:  "</span> <span class="va">${FD_INP_A</span><span class="op">/</span><span class="va">${FD_RES}</span><span class="op">/</span><span class="st">'${FD_RES}'</span><span class="va">}</span></span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Input fname:  "</span> <span class="va">${FN_INP_A}</span></span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Input fdiry:  "</span> <span class="va">${FD_INP_B</span><span class="op">/</span><span class="va">${FD_RES}</span><span class="op">/</span><span class="st">'${FD_RES}'</span><span class="va">}</span></span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Input fname:  "</span> <span class="va">${FN_INP_B}</span></span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span></span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Output fdiry: "</span> <span class="va">${FD_OUT</span><span class="op">/</span><span class="va">${FD_RES}</span><span class="op">/</span><span class="st">'${FD_RES}'</span><span class="va">}</span></span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Output fname: "</span> <span class="va">${FN_OUT}</span></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span></span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Log file:"</span> <span class="st">'${FD_LOG}/'</span><span class="va">${FN_LOG}</span></span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span></span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>        <span class="cf">done</span></span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a>    <span class="cf">done</span></span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span> <span class="op">&lt;</span> <span class="op">&lt;(</span><span class="fu">cat</span> <span class="va">${FP_REGION_LABEL}</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'NR &gt;=2 {print}'</span> <span class="kw">|</span> <span class="fu">head</span> <span class="at">-n</span> 1<span class="op">)</span>    </span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a><span class="co">#done &lt; &lt;(cat ${FP_REGION_LABEL} | awk 'NR &gt;=2 {print}')</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_overlap
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_overlap/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_overlap.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz.txt

==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_overlap
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_overlap/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_overlap.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz.txt

==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_overlap
Assay  name:   CRISPRi_Growth_K562_Gersbach_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_Growth_K562_Gersbach_JinWoo/coverage
Input fname:   CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_overlap/CRISPRi_Growth_K562_Gersbach_JinWoo
Output fname:  CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_overlap.CRISPRi_Growth_K562_Gersbach_JinWoo.CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz.txt

==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_overlap
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_overlap/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_overlap.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.Log2FC.stranded.bed.gz.txt

==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_overlap
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.ZScore.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_overlap/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.ZScore.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_overlap.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.ZScore.stranded.bed.gz.txt
</code></pre>
</div>
</div>
</section>
<section id="execute-1" class="level2">
<h2 class="anchored" data-anchor-id="execute-1">Execute</h2>
<p><strong>Check</strong></p>
<div id="a8adbaa4-8e49-4c90-afad-ffacb8f8e6e1" class="cell" data-execution_count="10">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="va">${FP_CNF}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>/data/reddylab/Kuei/repo/Proj_ENCODE_FCC/scripts/config_project.sh</code></pre>
</div>
</div>
<p><strong>RUN</strong></p>
<div id="5d5427fc-2062-40f2-a0f8-93e19ad4a9e1" class="cell" data-execution_count="11">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">### init: set executable</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="va">FN_EXE</span><span class="op">=</span><span class="st">"run_bedtools_intersect.sh"</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="va">FP_EXE</span><span class="op">=</span><span class="va">${FD_EXE}</span>/<span class="va">${FN_EXE}</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co">### Execute</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="bu">read</span> <span class="va">FOLDER_A</span> <span class="va">FNAME_A</span> <span class="va">LABEL_A</span> <span class="va">FPATH_A</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>    <span class="co">### set input file</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>    <span class="va">FD_INP_A</span><span class="op">=</span><span class="va">$(</span><span class="fu">dirname</span> <span class="va">${FPATH_A})</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>    <span class="va">FN_INP_A</span><span class="op">=</span><span class="va">${FNAME_A}</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    <span class="va">FP_INP_A</span><span class="op">=</span><span class="va">${FPATH_A}</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>    <span class="co">### loop through each assay</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> ASSAY_NAME <span class="kw">in</span> <span class="va">${ASSAY_NAMES</span><span class="op">[@]</span><span class="va">}</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>        <span class="co">### Set input file</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>        <span class="va">FD_INP</span><span class="op">=</span><span class="va">${FD_RES}</span>/<span class="va">${ASSAY_TYPE}</span>/<span class="va">${ASSAY_NAME}</span>/coverage</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>        <span class="va">FP_INPS</span><span class="op">=</span><span class="va">($(</span><span class="fu">ls</span> <span class="va">${FD_INP}</span>/<span class="pp">*</span>hg38<span class="pp">*</span>bed.gz<span class="va">))</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>        <span class="co">#FP_INPS=($(ls ${FD_INP}/*hg38*ZScore*bed.gz))</span></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>        <span class="co">### loop through each sample</span></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> FP_INP_B <span class="kw">in</span> <span class="va">${FP_INPS</span><span class="op">[@]</span><span class="va">}</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>            <span class="co">### set input file</span></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>            <span class="va">FD_INP_B</span><span class="op">=</span><span class="va">$(</span><span class="fu">dirname</span>  <span class="va">${FP_INP_B})</span></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>            <span class="va">FN_INP_B</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">${FP_INP_B})</span></span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>            <span class="co">### setup output file</span></span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>            <span class="va">FD_OUT</span><span class="op">=</span><span class="va">${FD_RES}</span>/region_coverage_fcc_score/<span class="va">${LABEL_A}</span>/<span class="va">${ASSAY_NAME}</span></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>            <span class="va">FN_OUT</span><span class="op">=</span><span class="va">${FN_INP_B}</span></span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>            <span class="va">FP_OUT</span><span class="op">=</span><span class="va">${FD_OUT}</span>/<span class="va">${FN_OUT}</span></span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>            <span class="co">### setup log file</span></span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>            <span class="va">FN_LOG</span><span class="op">=</span>region.coverage.<span class="va">${LABEL_A}</span>.<span class="va">${ASSAY_NAME}</span>.<span class="va">${FN_OUT}</span>.txt</span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>            <span class="va">FP_LOG</span><span class="op">=</span><span class="va">${FD_LOG}</span>/<span class="va">${FN_LOG}</span></span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a>            <span class="co">### show progress</span></span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> ==============================</span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Region folder:"</span> <span class="va">${FOLDER_A}</span></span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Region label: "</span> <span class="va">${LABEL_A}</span></span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Assay  name:  "</span> <span class="va">${ASSAY_NAME}</span></span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span></span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Input fdiry:  "</span> <span class="va">${FD_INP_A</span><span class="op">/</span><span class="va">${FD_RES}</span><span class="op">/</span><span class="st">'${FD_RES}'</span><span class="va">}</span></span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Input fname:  "</span> <span class="va">${FN_INP_A}</span></span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Input fdiry:  "</span> <span class="va">${FD_INP_B</span><span class="op">/</span><span class="va">${FD_RES}</span><span class="op">/</span><span class="st">'${FD_RES}'</span><span class="va">}</span></span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Input fname:  "</span> <span class="va">${FN_INP_B}</span></span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span></span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Output fdiry: "</span> <span class="va">${FD_OUT</span><span class="op">/</span><span class="va">${FD_RES}</span><span class="op">/</span><span class="st">'${FD_RES}'</span><span class="va">}</span></span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Output fname: "</span> <span class="va">${FN_OUT}</span></span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span></span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="st">"Log file:"</span> <span class="st">'${FD_LOG}/'</span><span class="va">${FN_LOG}</span></span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span></span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb18-54"><a href="#cb18-54" aria-hidden="true" tabindex="-1"></a>            <span class="co">### execute</span></span>
<span id="cb18-55"><a href="#cb18-55" aria-hidden="true" tabindex="-1"></a>            <span class="fu">mkdir</span> <span class="at">-p</span> <span class="va">${FD_OUT}</span></span>
<span id="cb18-56"><a href="#cb18-56" aria-hidden="true" tabindex="-1"></a>            <span class="ex">sbatch</span> <span class="dt">\</span></span>
<span id="cb18-57"><a href="#cb18-57" aria-hidden="true" tabindex="-1"></a>                <span class="at">--cpus-per-task</span> 4 <span class="dt">\</span></span>
<span id="cb18-58"><a href="#cb18-58" aria-hidden="true" tabindex="-1"></a>                <span class="at">--mem</span> 10G <span class="dt">\</span></span>
<span id="cb18-59"><a href="#cb18-59" aria-hidden="true" tabindex="-1"></a>                <span class="at">--output</span> <span class="va">${FP_LOG}</span> <span class="dt">\</span></span>
<span id="cb18-60"><a href="#cb18-60" aria-hidden="true" tabindex="-1"></a>                <span class="va">${FP_EXE}</span> <span class="va">${FP_CNF}</span> <span class="va">${FP_INP_A}</span> <span class="va">${FP_INP_B}</span> <span class="va">${FP_OUT}</span></span>
<span id="cb18-61"><a href="#cb18-61" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span></span>
<span id="cb18-62"><a href="#cb18-62" aria-hidden="true" tabindex="-1"></a>        <span class="cf">done</span></span>
<span id="cb18-63"><a href="#cb18-63" aria-hidden="true" tabindex="-1"></a>    <span class="cf">done</span></span>
<span id="cb18-64"><a href="#cb18-64" aria-hidden="true" tabindex="-1"></a><span class="co">#done &lt; &lt;(cat ${FP_REGION_LABEL} | awk 'NR &gt;=2 {print}' | head -n 1)    </span></span>
<span id="cb18-65"><a href="#cb18-65" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span> <span class="op">&lt;</span> <span class="op">&lt;(</span><span class="fu">cat</span> <span class="va">${FP_REGION_LABEL}</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'NR &gt;=2 {print}'</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_overlap
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_overlap/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_overlap.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz.txt

Submitted batch job 310360

==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_overlap
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_overlap/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_overlap.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310361

==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_overlap
Assay  name:   CRISPRi_Growth_K562_Gersbach_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_Growth_K562_Gersbach_JinWoo/coverage
Input fname:   CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_overlap/CRISPRi_Growth_K562_Gersbach_JinWoo
Output fname:  CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_overlap.CRISPRi_Growth_K562_Gersbach_JinWoo.CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310362

==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_overlap
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_overlap/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_overlap.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.Log2FC.stranded.bed.gz.txt

Submitted batch job 310363

==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_overlap
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.ZScore.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_overlap/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.ZScore.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_overlap.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.ZScore.stranded.bed.gz.txt

Submitted batch job 310364

==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_union
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_union/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_union.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz.txt

Submitted batch job 310365

==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_union
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_union/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_union.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310366

==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_union
Assay  name:   CRISPRi_Growth_K562_Gersbach_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_Growth_K562_Gersbach_JinWoo/coverage
Input fname:   CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_union/CRISPRi_Growth_K562_Gersbach_JinWoo
Output fname:  CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_union.CRISPRi_Growth_K562_Gersbach_JinWoo.CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310367

==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_union
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_union/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_union.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.Log2FC.stranded.bed.gz.txt

Submitted batch job 310368

==============================
Region folder: region_for_analysis
Region label:  fcc_astarr_macs_input_union
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.ZScore.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/fcc_astarr_macs_input_union/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.ZScore.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.fcc_astarr_macs_input_union.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.ZScore.stranded.bed.gz.txt

Submitted batch job 310369

==============================
Region folder: region_for_analysis
Region label:  dnase_ENCFF274YGF
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR000EKS.ENCFF274YGF.DNase.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/dnase_ENCFF274YGF/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.dnase_ENCFF274YGF.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz.txt

Submitted batch job 310370

==============================
Region folder: region_for_analysis
Region label:  dnase_ENCFF274YGF
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR000EKS.ENCFF274YGF.DNase.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/dnase_ENCFF274YGF/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.dnase_ENCFF274YGF.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310371

==============================
Region folder: region_for_analysis
Region label:  dnase_ENCFF274YGF
Assay  name:   CRISPRi_Growth_K562_Gersbach_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR000EKS.ENCFF274YGF.DNase.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_Growth_K562_Gersbach_JinWoo/coverage
Input fname:   CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/dnase_ENCFF274YGF/CRISPRi_Growth_K562_Gersbach_JinWoo
Output fname:  CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.dnase_ENCFF274YGF.CRISPRi_Growth_K562_Gersbach_JinWoo.CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310372

==============================
Region folder: region_for_analysis
Region label:  dnase_ENCFF274YGF
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR000EKS.ENCFF274YGF.DNase.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/dnase_ENCFF274YGF/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.dnase_ENCFF274YGF.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.Log2FC.stranded.bed.gz.txt

Submitted batch job 310373

==============================
Region folder: region_for_analysis
Region label:  dnase_ENCFF274YGF
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR000EKS.ENCFF274YGF.DNase.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.ZScore.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/dnase_ENCFF274YGF/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.ZScore.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.dnase_ENCFF274YGF.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.ZScore.stranded.bed.gz.txt

Submitted batch job 310374

==============================
Region folder: region_for_analysis
Region label:  dnase_ENCFF185XRG
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR000EOT.ENCFF185XRG.DNase.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/dnase_ENCFF185XRG/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.dnase_ENCFF185XRG.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz.txt

Submitted batch job 310375

==============================
Region folder: region_for_analysis
Region label:  dnase_ENCFF185XRG
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR000EOT.ENCFF185XRG.DNase.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/dnase_ENCFF185XRG/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.dnase_ENCFF185XRG.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310376

==============================
Region folder: region_for_analysis
Region label:  dnase_ENCFF185XRG
Assay  name:   CRISPRi_Growth_K562_Gersbach_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR000EOT.ENCFF185XRG.DNase.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_Growth_K562_Gersbach_JinWoo/coverage
Input fname:   CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/dnase_ENCFF185XRG/CRISPRi_Growth_K562_Gersbach_JinWoo
Output fname:  CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.dnase_ENCFF185XRG.CRISPRi_Growth_K562_Gersbach_JinWoo.CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310377

==============================
Region folder: region_for_analysis
Region label:  dnase_ENCFF185XRG
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR000EOT.ENCFF185XRG.DNase.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/dnase_ENCFF185XRG/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.dnase_ENCFF185XRG.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.Log2FC.stranded.bed.gz.txt

Submitted batch job 310378

==============================
Region folder: region_for_analysis
Region label:  dnase_ENCFF185XRG
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR000EOT.ENCFF185XRG.DNase.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.ZScore.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/dnase_ENCFF185XRG/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.ZScore.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.dnase_ENCFF185XRG.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.ZScore.stranded.bed.gz.txt

Submitted batch job 310379

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF558BLC
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR483RKN.ENCFF558BLC.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF558BLC/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF558BLC.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz.txt

Submitted batch job 310380

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF558BLC
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR483RKN.ENCFF558BLC.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF558BLC/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF558BLC.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310381

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF558BLC
Assay  name:   CRISPRi_Growth_K562_Gersbach_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR483RKN.ENCFF558BLC.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_Growth_K562_Gersbach_JinWoo/coverage
Input fname:   CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF558BLC/CRISPRi_Growth_K562_Gersbach_JinWoo
Output fname:  CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF558BLC.CRISPRi_Growth_K562_Gersbach_JinWoo.CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310382

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF558BLC
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR483RKN.ENCFF558BLC.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF558BLC/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF558BLC.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.Log2FC.stranded.bed.gz.txt

Submitted batch job 310383

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF558BLC
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR483RKN.ENCFF558BLC.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.ZScore.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF558BLC/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.ZScore.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF558BLC.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.ZScore.stranded.bed.gz.txt

Submitted batch job 310384

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF925CYR
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR483RKN.ENCFF925CYR.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF925CYR/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF925CYR.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz.txt

Submitted batch job 310385

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF925CYR
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR483RKN.ENCFF925CYR.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF925CYR/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF925CYR.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310386

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF925CYR
Assay  name:   CRISPRi_Growth_K562_Gersbach_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR483RKN.ENCFF925CYR.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_Growth_K562_Gersbach_JinWoo/coverage
Input fname:   CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF925CYR/CRISPRi_Growth_K562_Gersbach_JinWoo
Output fname:  CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF925CYR.CRISPRi_Growth_K562_Gersbach_JinWoo.CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310387

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF925CYR
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR483RKN.ENCFF925CYR.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF925CYR/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF925CYR.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.Log2FC.stranded.bed.gz.txt

Submitted batch job 310388

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF925CYR
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR483RKN.ENCFF925CYR.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.ZScore.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF925CYR/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.ZScore.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF925CYR.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.ZScore.stranded.bed.gz.txt

Submitted batch job 310389

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF333TAT
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR868FGK.ENCFF333TAT.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF333TAT/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF333TAT.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz.txt

Submitted batch job 310390

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF333TAT
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR868FGK.ENCFF333TAT.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF333TAT/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF333TAT.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310391

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF333TAT
Assay  name:   CRISPRi_Growth_K562_Gersbach_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR868FGK.ENCFF333TAT.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_Growth_K562_Gersbach_JinWoo/coverage
Input fname:   CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF333TAT/CRISPRi_Growth_K562_Gersbach_JinWoo
Output fname:  CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF333TAT.CRISPRi_Growth_K562_Gersbach_JinWoo.CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310392

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF333TAT
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR868FGK.ENCFF333TAT.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF333TAT/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF333TAT.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.Log2FC.stranded.bed.gz.txt

Submitted batch job 310393

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF333TAT
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR868FGK.ENCFF333TAT.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.ZScore.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF333TAT/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.ZScore.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF333TAT.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.ZScore.stranded.bed.gz.txt

Submitted batch job 310394

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF948AFM
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR868FGK.ENCFF948AFM.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF948AFM/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF948AFM.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz.txt

Submitted batch job 310395

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF948AFM
Assay  name:   CRISPRi_FlowFISH_K562_Riley_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR868FGK.ENCFF948AFM.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage
Input fname:   CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF948AFM/CRISPRi_FlowFISH_K562_Riley_JinWoo
Output fname:  CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF948AFM.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310396

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF948AFM
Assay  name:   CRISPRi_Growth_K562_Gersbach_JinWoo

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR868FGK.ENCFF948AFM.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/CRISPRi_Growth_K562_Gersbach_JinWoo/coverage
Input fname:   CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF948AFM/CRISPRi_Growth_K562_Gersbach_JinWoo
Output fname:  CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF948AFM.CRISPRi_Growth_K562_Gersbach_JinWoo.CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz.txt

Submitted batch job 310397

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF948AFM
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR868FGK.ENCFF948AFM.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF948AFM/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.Log2FC.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF948AFM.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.Log2FC.stranded.bed.gz.txt

Submitted batch job 310398

==============================
Region folder: region_for_analysis
Region label:  atac_ENCFF948AFM
Assay  name:   MPRA_Lenti_K562_Nadav_Vikram

Input fdiry:   ${FD_RES}/region/region_for_analysis
Input fname:   K562.hg38.ENCSR868FGK.ENCFF948AFM.ATAC.bed.gz
Input fdiry:   ${FD_RES}/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage
Input fname:   LMPRA_K562.hg38.ZScore.stranded.bed.gz

Output fdiry:  ${FD_RES}/region_coverage_fcc_score/atac_ENCFF948AFM/MPRA_Lenti_K562_Nadav_Vikram
Output fname:  LMPRA_K562.hg38.ZScore.stranded.bed.gz

Log file: ${FD_LOG}/region.coverage.atac_ENCFF948AFM.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.ZScore.stranded.bed.gz.txt

Submitted batch job 310399
</code></pre>
</div>
</div>
</section>
<section id="review" class="level2">
<h2 class="anchored" data-anchor-id="review">Review</h2>
<div id="0ec6af3e-3fe8-4c0f-86f7-9a38019d7958" class="cell" data-execution_count="11">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="va">${FD_LOG}</span>/region.coverage.fcc_astarr_macs_input_overlap.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Hostname:           plp-rcc-node-24
Slurm Array Index: 
Time Stamp:         07-14-25+21:24:43

Input:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ASTARR.macs.KS91.input.rep_all.max_overlaps.q5.bed.gz

show first few lines of input
chr1    10038   10405   chr1:10038-10405
chr1    14282   14614   chr1:14282-14614
chr1    16025   16338   chr1:16025-16338
chr1    17288   17689   chr1:17288-17689
chr1    28934   29499   chr1:28934-29499
chr1    115429  115969  chr1:115429-115969
chr1    136201  137353  chr1:136201-137353
chr1    137748  138049  chr1:137748-138049
chr1    138321  139517  chr1:138321-139517
chr1    181005  181854  chr1:181005-181854

Input:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage/CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

show first few lines of input
chr11   4091884 4091885 id-99   0.114291    .   HBE1
chr11   4091884 4091885 id-99   0.650791    .   HBG1
chr11   4091884 4091885 id-99   0.694152    .   HBG2
chr11   4091884 4091885 id-99   -0.074553   .   HBS1L
chr11   4091884 4091885 id-99   -0.257127   .   MYB
chr11   4091885 4091886 id-100  -0.546764   .   HBE1
chr11   4091885 4091886 id-100  0.084442    .   HBG1
chr11   4091885 4091886 id-100  -0.119862   .   HBG2
chr11   4091885 4091886 id-100  0.240283    .   HBS1L
chr11   4091885 4091886 id-100  0.212189    .   MYB


Output:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_fcc_score/fcc_astarr_macs_input_overlap/CRISPRi_FlowFISH_K562_Riley_JinWoo/CRISPRi_HCRFF_K562.hg38.ZScore.filtered.unstranded.bed.gz

show first few lines of output:
chr11   4092109 4092511 chr11:4092109-4092511   chr11   4092122 4092123 id-145  -1.10103    .   HBE1    1
chr11   4092109 4092511 chr11:4092109-4092511   chr11   4092122 4092123 id-145  -0.015199   .   HBG1    1
chr11   4092109 4092511 chr11:4092109-4092511   chr11   4092122 4092123 id-145  0.304337    .   HBG2    1
chr11   4092109 4092511 chr11:4092109-4092511   chr11   4092122 4092123 id-145  -0.675545   .   HBS1L   1
chr11   4092109 4092511 chr11:4092109-4092511   chr11   4092122 4092123 id-145  0.134939    .   MYB 1
chr11   4092109 4092511 chr11:4092109-4092511   chr11   4092123 4092124 id-146  -0.061961   .   HBE1    1
chr11   4092109 4092511 chr11:4092109-4092511   chr11   4092123 4092124 id-146  0.666595    .   HBG1    1
chr11   4092109 4092511 chr11:4092109-4092511   chr11   4092123 4092124 id-146  0.267739    .   HBG2    1
chr11   4092109 4092511 chr11:4092109-4092511   chr11   4092123 4092124 id-146  -0.384682   .   HBS1L   1
chr11   4092109 4092511 chr11:4092109-4092511   chr11   4092123 4092124 id-146  -0.100842   .   MYB 1


Done!
Run Time: 1 seconds</code></pre>
</div>
</div>
<div id="ced1f589-c527-41c0-af15-5ddca0e0f040" class="cell" data-execution_count="12">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="va">${FD_LOG}</span>/region.coverage.fcc_astarr_macs_input_union.CRISPRi_FlowFISH_K562_Riley_JinWoo.CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Hostname:           plp-rcc-node-14
Slurm Array Index: 
Time Stamp:         07-14-25+21:24:43

Input:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ASTARR.macs.KS91.input.rep_all.union.q5.bed.gz

show first few lines of input
chr1    10015   10442   chr1:10015-10442
chr1    14253   14645   chr1:14253-14645
chr1    16015   16477   chr1:16015-16477
chr1    17237   17772   chr1:17237-17772
chr1    28903   29613   chr1:28903-29613
chr1    30803   31072   chr1:30803-31072
chr1    101603  101849  chr1:101603-101849
chr1    115411  115986  chr1:115411-115986
chr1    118518  118743  chr1:118518-118743
chr1    136071  137429  chr1:136071-137429

Input:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_fcc/CRISPRi_FlowFISH_K562_Riley_JinWoo/coverage/CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

show first few lines of input
chr1    74582275    74582276    id-1    0.414014    .   CD164
chr1    74582275    74582276    id-1    -0.493091   .   MYC
chr1    74582275    74582276    id-1    0.024514    .   NMU
chr1    74582275    74582276    id-1    0.44089 .   PVT1
chr1    74582309    74582310    id-2    -0.181799   .   CD164
chr1    74582309    74582310    id-2    0.997649    .   MYC
chr1    74582309    74582310    id-2    -0.439687   .   NMU
chr1    74582309    74582310    id-2    1.25627 .   PVT1
chr1    80526338    80526339    id-3    -0.186275   .   CD164
chr1    80526338    80526339    id-3    0.340142    .   MYC


Output:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_fcc_score/fcc_astarr_macs_input_union/CRISPRi_FlowFISH_K562_Riley_JinWoo/CRISPRi_HCRFF_K562.hg38.ZScore.unstranded.bed.gz

show first few lines of output:
chr10   100729027   100729908   chr10:100729027-100729908   chr10   100729052   100729053   id-241  -0.282619   .   CAPRIN1 1
chr10   100729027   100729908   chr10:100729027-100729908   chr10   100729052   100729053   id-241  -1.27947    .   CAT 1
chr10   100729027   100729908   chr10:100729027-100729908   chr10   100729052   100729053   id-340  0.65685 .   CD164   1
chr10   100729027   100729908   chr10:100729027-100729908   chr10   100729052   100729053   id-241  -0.323226   .   ERP29   1
chr10   100729027   100729908   chr10:100729027-100729908   chr10   100729052   100729053   id-241  -0.582477   .   LMO2    1
chr10   100729027   100729908   chr10:100729027-100729908   chr10   100729052   100729053   id-241  -0.499114   .   MEF2C   1
chr10   100729027   100729908   chr10:100729027-100729908   chr10   100729052   100729053   id-340  0.229685    .   NMU 1
chr10   100729027   100729908   chr10:100729027-100729908   chr10   100729112   100729113   id-242  -1.31977    .   CAPRIN1 1
chr10   100729027   100729908   chr10:100729027-100729908   chr10   100729112   100729113   id-242  -0.259811   .   CAT 1
chr10   100729027   100729908   chr10:100729027-100729908   chr10   100729112   100729113   id-341  -0.422878   .   CD164   1


Done!
Run Time: 1 seconds</code></pre>
</div>
</div>
<div id="684249cb-f95e-4c8d-837d-722cfc31257d" class="cell" data-execution_count="13">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="va">${FD_LOG}</span>/region.coverage.atac_ENCFF948AFM.CRISPRi_Growth_K562_Gersbach_JinWoo.CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Hostname:           plp-rcc-node-17
Slurm Array Index: 
Time Stamp:         07-14-25+21:24:43

Input:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR868FGK.ENCFF948AFM.ATAC.bed.gz

show first few lines of input
chr1    42030   42399   chr1:42030-42399
chr1    68963   70035   chr1:68963-70035
chr1    79151   79653   chr1:79151-79653
chr1    778339  779268  chr1:778339-779268
chr1    804424  805054  chr1:804424-805054
chr1    811420  811854  chr1:811420-811854
chr1    812328  812983  chr1:812328-812983
chr1    817227  818384  chr1:817227-818384
chr1    826581  828166  chr1:826581-828166
chr1    842062  843249  chr1:842062-843249

Input:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_fcc/CRISPRi_Growth_K562_Gersbach_JinWoo/coverage/CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

show first few lines of input
chr1    605580  605581  id-1    -0.18066    .
chr1    605581  605582  id-2    0.486646    .
chr1    605589  605590  id-3    3.17124 .
chr1    605591  605592  id-4    -0.364573   .
chr1    605592  605593  id-5    -0.206319   .
chr1    605609  605610  id-6    -0.26675    .
chr1    605617  605618  id-7    -0.7741 .
chr1    605618  605619  id-8    -0.353224   .
chr1    605619  605620  id-9    -0.187707   .
chr1    605620  605621  id-10   -0.420401   .


Output:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_fcc_score/atac_ENCFF948AFM/CRISPRi_Growth_K562_Gersbach_JinWoo/CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

show first few lines of output:
chr1    778339  779268  chr1:778339-779268  chr1    778628  778629  id-11   0.511467    .   1
chr1    778339  779268  chr1:778339-779268  chr1    778629  778630  id-12   0.242298    .   1
chr1    778339  779268  chr1:778339-779268  chr1    778634  778635  id-13   0.537248    .   1
chr1    778339  779268  chr1:778339-779268  chr1    778707  778708  id-14   -0.167125   .   1
chr1    778339  779268  chr1:778339-779268  chr1    778708  778709  id-15   0.666834    .   1
chr1    778339  779268  chr1:778339-779268  chr1    778720  778721  id-16   -0.40553    .   1
chr1    778339  779268  chr1:778339-779268  chr1    778721  778722  id-17   -0.223393   .   1
chr1    778339  779268  chr1:778339-779268  chr1    778745  778746  id-18   -0.173731   .   1
chr1    778339  779268  chr1:778339-779268  chr1    778763  778764  id-19   0.173191    .   1
chr1    778339  779268  chr1:778339-779268  chr1    778877  778878  id-20   8.06e-4 .   1


Done!
Run Time: 3 seconds</code></pre>
</div>
</div>
<div id="ea6666f5-bf9e-4f82-a778-e788a6eaa64a" class="cell" data-execution_count="14">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="va">${FD_LOG}</span>/region.coverage.atac_ENCFF558BLC.CRISPRi_Growth_K562_Gersbach_JinWoo.CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Hostname:           plp-rcc-node-16
Slurm Array Index: 
Time Stamp:         07-14-25+21:24:43

Input:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR483RKN.ENCFF558BLC.ATAC.bed.gz

show first few lines of input
chr1    10509   10674   chr1:10509-10674
chr1    41987   42385   chr1:41987-42385
chr1    46755   47025   chr1:46755-47025
chr1    68969   69964   chr1:68969-69964
chr1    78951   79407   chr1:78951-79407
chr1    778328  779235  chr1:778328-779235
chr1    796042  796413  chr1:796042-796413
chr1    797433  797885  chr1:797433-797885
chr1    804503  805016  chr1:804503-805016
chr1    811468  811809  chr1:811468-811809

Input:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_fcc/CRISPRi_Growth_K562_Gersbach_JinWoo/coverage/CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

show first few lines of input
chr1    605580  605581  id-1    -0.18066    .
chr1    605581  605582  id-2    0.486646    .
chr1    605589  605590  id-3    3.17124 .
chr1    605591  605592  id-4    -0.364573   .
chr1    605592  605593  id-5    -0.206319   .
chr1    605609  605610  id-6    -0.26675    .
chr1    605617  605618  id-7    -0.7741 .
chr1    605618  605619  id-8    -0.353224   .
chr1    605619  605620  id-9    -0.187707   .
chr1    605620  605621  id-10   -0.420401   .


Output:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_fcc_score/atac_ENCFF558BLC/CRISPRi_Growth_K562_Gersbach_JinWoo/CRISPRi_Growth_K562.hg38.ZScore.unstranded.bed.gz

show first few lines of output:
chr1    778328  779235  chr1:778328-779235  chr1    778628  778629  id-11   0.511467    .   1
chr1    778328  779235  chr1:778328-779235  chr1    778629  778630  id-12   0.242298    .   1
chr1    778328  779235  chr1:778328-779235  chr1    778634  778635  id-13   0.537248    .   1
chr1    778328  779235  chr1:778328-779235  chr1    778707  778708  id-14   -0.167125   .   1
chr1    778328  779235  chr1:778328-779235  chr1    778708  778709  id-15   0.666834    .   1
chr1    778328  779235  chr1:778328-779235  chr1    778720  778721  id-16   -0.40553    .   1
chr1    778328  779235  chr1:778328-779235  chr1    778721  778722  id-17   -0.223393   .   1
chr1    778328  779235  chr1:778328-779235  chr1    778745  778746  id-18   -0.173731   .   1
chr1    778328  779235  chr1:778328-779235  chr1    778763  778764  id-19   0.173191    .   1
chr1    778328  779235  chr1:778328-779235  chr1    778877  778878  id-20   8.06e-4 .   1


Done!
Run Time: 3 seconds</code></pre>
</div>
</div>
<div id="3f063cf5-c58c-43e3-ba0a-2348139c933f" class="cell" data-execution_count="15">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="va">${FD_LOG}</span>/region.coverage.atac_ENCFF948AFM.MPRA_Lenti_K562_Nadav_Vikram.LMPRA_K562.hg38.ZScore.stranded.bed.gz.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Hostname:           plp-rcc-node-17
Slurm Array Index: 
Time Stamp:         07-14-25+21:24:43

Input:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region/region_for_analysis/K562.hg38.ENCSR868FGK.ENCFF948AFM.ATAC.bed.gz

show first few lines of input
chr1    42030   42399   chr1:42030-42399
chr1    68963   70035   chr1:68963-70035
chr1    79151   79653   chr1:79151-79653
chr1    778339  779268  chr1:778339-779268
chr1    804424  805054  chr1:804424-805054
chr1    811420  811854  chr1:811420-811854
chr1    812328  812983  chr1:812328-812983
chr1    817227  818384  chr1:817227-818384
chr1    826581  828166  chr1:826581-828166
chr1    842062  843249  chr1:842062-843249

Input:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/assay_fcc/MPRA_Lenti_K562_Nadav_Vikram/coverage/LMPRA_K562.hg38.ZScore.stranded.bed.gz

show first few lines of input
chr1    115631  115831  peak1_Reversed: 2.0511007569226227  -
chr1    138940  139140  peak2_Reversed: 1.5391817792648435  -
chr1    138940  139140  peak2   -0.06555714360564233    +
chr1    203225  203425  peak3   1.2305781402512883  +
chr1    203225  203425  peak3_Reversed: 0.8856681907655505  -
chr1    267911  268111  peak4_Reversed: -1.6049447076261982 -
chr1    267911  268111  peak4   2.7227675006580068  +
chr1    586093  586293  peak5   0.30476722321062366 +
chr1    586093  586293  peak5_Reversed: 1.5137673619343153  -
chr1    629847  630047  peak6_Reversed: -1.4451969415485937 -


Output:  /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results/region_coverage_fcc_score/atac_ENCFF948AFM/MPRA_Lenti_K562_Nadav_Vikram/LMPRA_K562.hg38.ZScore.stranded.bed.gz

show first few lines of output:
chr1    778339  779268  chr1:778339-779268  chr1    778638  778838  peak11  5.376758796174578   +   200
chr1    778339  779268  chr1:778339-779268  chr1    778638  778838  peak11_Reversed:    6.447794955103975   -   200
chr1    842062  843249  chr1:842062-843249  chr1    842870  843070  peak13_Reversed:    3.0059567223410335  -   200
chr1    842062  843249  chr1:842062-843249  chr1    842870  843070  peak13  2.893407159877267   +   200
chr1    898760  899148  chr1:898760-899148  chr1    898800  899000  peak15_Reversed:    0.9764339669460079  -   200
chr1    898760  899148  chr1:898760-899148  chr1    898800  899000  peak15  3.499722544762722   +   200
chr1    904027  905234  chr1:904027-905234  chr1    904350  904550  peak16  3.103983760615928   +   200
chr1    904027  905234  chr1:904027-905234  chr1    904350  904550  peak16_Reversed:    3.358127933921208   -   200
chr1    911019  911495  chr1:911019-911495  chr1    911257  911457  peak18_Reversed:    0.22852397121903947 -   200
chr1    911019  911495  chr1:911019-911495  chr1    911257  911457  peak18  1.771542166286814   +   200


Done!
Run Time: 1 seconds</code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>