{
 "cells": [
  {
   "cell_type": "raw",
   "id": "d871ba1c-8d1d-4e76-9f6c-82e2a85b8ea5",
   "metadata": {},
   "source": [
    "---\n",
    "title: Prepare ENCODE ChIP-seq TF 01\n",
    "subtitle: Generate download script to download the data\n",
    "format:\n",
    "    html:\n",
    "        code-fold:  true\n",
    "        code-tools: true\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28c83ceb-af27-4f51-9862-202422b09294",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "34853b16-1190-4461-acae-162a8949328b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are working on        Singularity: singularity_proj_encode_fcc \n",
      "BASE DIRECTORY (FD_BASE): /data/reddylab/Kuei \n",
      "REPO DIRECTORY (FD_REPO): /data/reddylab/Kuei/repo \n",
      "WORK DIRECTORY (FD_WORK): /data/reddylab/Kuei/work \n",
      "DATA DIRECTORY (FD_DATA): /data/reddylab/Kuei/data \n",
      "\n",
      "You are working with      ENCODE FCC \n",
      "PATH OF PROJECT (FD_PRJ): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC \n",
      "PROJECT RESULTS (FD_RES): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results \n",
      "PROJECT SCRIPTS (FD_EXE): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/scripts \n",
      "PROJECT DATA    (FD_DAT): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/data \n",
      "PROJECT NOTE    (FD_NBK): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/notebooks \n",
      "PROJECT DOCS    (FD_DOC): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/docs \n",
      "PROJECT LOG     (FD_LOG): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/log \n",
      "PROJECT REF     (FD_REF): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/references \n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(source(\"../run_config_project_sing.R\")))\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "925bb3c4-9592-4c3d-82f7-eafb58d81d0c",
   "metadata": {},
   "source": [
    "## Prepare"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4b61fb81-e489-4a6a-8ed6-f2842d8e590c",
   "metadata": {},
   "outputs": [],
   "source": [
    "TXT_FOLDER_INP = \"encode_chipseq_tf\"\n",
    "TXT_FOLDER_OUT = \"encode_chipseq_tf_250120\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4a9e5915-a7e3-4253-89c1-ab861f3120d9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "files.chipseq_tf.default_analysis.250109.txt \n",
      "files.chipseq_tf.default_analysis.250120.txt \n",
      "files.chipseq_tf.default_files.241223.txt \n",
      "files.chipseq_tf.default_files.250108.txt \n",
      "files.chipseq_tf.default_files.250120.txt \n",
      "metadata.chipseq_tf.default_analysis.250109.tsv \n",
      "metadata.chipseq_tf.default_analysis.250120.tsv \n",
      "metadata.chipseq_tf.default_files.241223.tsv \n",
      "metadata.chipseq_tf.default_files.250108.tsv \n",
      "metadata.chipseq_tf.default_files.250120.tsv \n",
      "Readme.md \n"
     ]
    }
   ],
   "source": [
    "txt_folder = TXT_FOLDER_INP\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "\n",
    "vec = dir(txt_fdiry)\n",
    "for (txt in vec){cat(txt, \"\\n\")}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f6dbdc9f-10b9-4bdb-9fed-9b2a51b298bd",
   "metadata": {},
   "source": [
    "## Import metadata from reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bec00756-d133-40d9-a0d9-ca1bff6689ee",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 2406   59\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF447FCS </td>\n",
       "   <td style=\"text-align:left;\"> bigBed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bigBed </td>\n",
       "   <td style=\"text-align:left;\"> narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> CRISPR </td>\n",
       "   <td style=\"text-align:left;\"> insertion </td>\n",
       "   <td style=\"text-align:left;\"> {'schema_version': '14', 'aliases': ['michael-snyder:TS-985'], 'organism': {'schema_version': '6', '@type': ['Organism', 'Item'], 'name': 'human', 'taxon_id': '9606', 'scientific_name': 'Homo sapiens', '@id': '/organisms/human/', 'uuid': '7745b647-ff15-4ff3-9ced-b897d4e2983c', 'status': 'released'}, 'genes': ['/genes/84671/'], '@type': ['Target', 'Item'], 'name': 'ZNF347-human', 'label': 'ZNF347', '@id': '/targets/ZNF347-human/', 'title': 'ZNF347 (Homo sapiens)', 'investigated_as': ['transcription factor'], 'uuid': '7edf642a-7dbc-4d18-b8a7-a36fee801dda', 'status': 'released'} </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> formaldehyde </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2020-08-03 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 450-650 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1, 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF843NBV/ </td>\n",
       "   <td style=\"text-align:right;\"> 83030 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> 76853aa420fdb949461f8032db9e949c </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF447FCS/@@download/ENCFF447FCS.bigBed </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/07/03/352f11df-3c62-4954-8e58-8c0e14adda5e/ENCFF447FCS.bigBed </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/07/03/352f11df-3c62-4954-8e58-8c0e14adda5e/ENCFF447FCS.bigBed?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> moderate library complexity, low read depth, mild to moderate bottlenecking </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_folder = TXT_FOLDER_INP\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "txt_fname  = \"metadata.chipseq_tf.default_files.250120.tsv\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### show and assign\n",
    "dat_metadata_chipseq_v1 = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c29887af-e32f-409d-8aeb-ccc7f64958c9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 17353    59\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:right;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF521GKP </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> CRISPR </td>\n",
       "   <td style=\"text-align:left;\"> insertion </td>\n",
       "   <td style=\"text-align:left;\"> {'schema_version': '14', 'aliases': ['michael-snyder:TS-985'], 'organism': {'schema_version': '6', '@type': ['Organism', 'Item'], 'name': 'human', 'taxon_id': '9606', 'scientific_name': 'Homo sapiens', '@id': '/organisms/human/', 'uuid': '7745b647-ff15-4ff3-9ced-b897d4e2983c', 'status': 'released'}, 'genes': ['/genes/84671/'], '@type': ['Target', 'Item'], 'name': 'ZNF347-human', 'label': 'ZNF347', '@id': '/targets/ZNF347-human/', 'title': 'ZNF347 (Homo sapiens)', 'investigated_as': ['transcription factor'], 'uuid': '7edf642a-7dbc-4d18-b8a7-a36fee801dda', 'status': 'released'} </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> formaldehyde </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2020-08-03 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 450-650 </td>\n",
       "   <td style=\"text-align:left;\"> 2 </td>\n",
       "   <td style=\"text-align:left;\"> 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:right;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF022VGJ/, /files/ENCFF174OKT/ </td>\n",
       "   <td style=\"text-align:right;\"> 921948703 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> afaa174c390f22a9320ec3beeab8e490 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF521GKP/@@download/ENCFF521GKP.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/07/03/a543c551-9440-4331-99de-94621c7a6ee2/ENCFF521GKP.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/07/03/a543c551-9440-4331-99de-94621c7a6ee2/ENCFF521GKP.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> low read depth, mild to moderate bottlenecking, moderate library complexity </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_folder = TXT_FOLDER_INP\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "txt_fname  = \"metadata.chipseq_tf.default_analysis.250120.tsv\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### show and assign\n",
    "dat_metadata_chipseq_v2 = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8aac697b-f35e-4c49-b1a4-0ecf031df9fa",
   "metadata": {},
   "source": [
    "## Explore"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8df8ceda-e355-4db9-bfc3-1f3ffd570c19",
   "metadata": {},
   "source": [
    "**File type**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "52306484-42c4-4e03-b5e4-edb71620a12a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             \n",
       "              bed bigBed bigWig\n",
       "  TF ChIP-seq 802    802    802"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "table(dat$Assay, dat$`File type`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "336311b3-d8e9-4ffd-a4ee-7ddec6b98f58",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             \n",
       "               bam  bed bigBed bigWig\n",
       "  TF ChIP-seq 3208 6125   3260   4760"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "table(dat$Assay, dat$`File type`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8810a2b6-5119-449e-b75c-d54bcb73eddd",
   "metadata": {},
   "source": [
    "**Output type**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6e03305b-628d-41f9-b219-c61a4902def5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        \n",
       "         conservative IDR thresholded peaks IDR thresholded peaks\n",
       "  bed                                    66                   654\n",
       "  bigBed                                 66                   654\n",
       "  bigWig                                  0                     0\n",
       "        \n",
       "         optimal IDR thresholded peaks pseudoreplicated IDR thresholded peaks\n",
       "  bed                               78                                      4\n",
       "  bigBed                            78                                      4\n",
       "  bigWig                             0                                      0\n",
       "        \n",
       "         signal p-value\n",
       "  bed                 0\n",
       "  bigBed              0\n",
       "  bigWig            802"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "table(dat$`File type`, dat$`Output type`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "56f1ded8-2b64-4a60-98a1-72e7d4204440",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        \n",
       "         alignments conservative IDR thresholded peaks fold change over control\n",
       "  bam          1604                                  0                        0\n",
       "  bed             0                                782                        0\n",
       "  bigBed          0                                782                        0\n",
       "  bigWig          0                                  0                     2380\n",
       "        \n",
       "         IDR ranked peaks IDR thresholded peaks optimal IDR thresholded peaks\n",
       "  bam                   0                     0                             0\n",
       "  bed                2865                  2158                            78\n",
       "  bigBed                0                  2158                            78\n",
       "  bigWig                0                     0                             0\n",
       "        \n",
       "         peaks and background as input for IDR\n",
       "  bam                                        0\n",
       "  bed                                      238\n",
       "  bigBed                                   238\n",
       "  bigWig                                     0\n",
       "        \n",
       "         pseudoreplicated IDR thresholded peaks signal p-value\n",
       "  bam                                         0              0\n",
       "  bed                                         4              0\n",
       "  bigBed                                      4              0\n",
       "  bigWig                                      0           2380\n",
       "        \n",
       "         unfiltered alignments\n",
       "  bam                     1604\n",
       "  bed                        0\n",
       "  bigBed                     0\n",
       "  bigWig                     0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "table(dat$`File type`, dat$`Output type`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d438e08-792d-4907-afc4-98cfcae5a9f6",
   "metadata": {},
   "source": [
    "**Genome assembly**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "7b7f0a99-1033-4f6e-8728-84f56e84132a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "GRCh38   hg19 \n",
       "  2283    123 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "table(dat$`File assembly`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6b900c62-6c66-4a22-88d9-78d3594f8905",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "GRCh38   hg19 \n",
       " 16557    796 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "table(dat$`File assembly`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3fbbe4b-4c20-429d-b13f-68b2263a7d82",
   "metadata": {},
   "source": [
    "**Biosample**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "577036df-f576-459f-b03f-e7ded90b92d6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "K562 \n",
       "2406 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "table(dat$`Biosample term name`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "fe5f9632-6859-4197-8b77-d3fb002e6f78",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       " K562 \n",
       "17353 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "table(dat$`Biosample term name`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5243c14f-76cf-4fc1-ae66-341badae3f07",
   "metadata": {},
   "source": [
    "**Target**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "5ec5deb6-4caf-4613-b643-12bb6b1e2552",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Var1</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ADNP-human    </td><td>3</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AFF1-human    </td><td>6</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>AFF4-human    </td><td>3</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AGO1-human    </td><td>9</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ARHGAP35-human</td><td>3</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ARID1B-human  </td><td>3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Var1 & Freq\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & ADNP-human     & 3\\\\\n",
       "\t2 & AFF1-human     & 6\\\\\n",
       "\t3 & AFF4-human     & 3\\\\\n",
       "\t4 & AGO1-human     & 9\\\\\n",
       "\t5 & ARHGAP35-human & 3\\\\\n",
       "\t6 & ARID1B-human   & 3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| <!--/--> | Var1 &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | ADNP-human     | 3 |\n",
       "| 2 | AFF1-human     | 6 |\n",
       "| 3 | AFF4-human     | 3 |\n",
       "| 4 | AGO1-human     | 9 |\n",
       "| 5 | ARHGAP35-human | 3 |\n",
       "| 6 | ARID1B-human   | 3 |\n",
       "\n"
      ],
      "text/plain": [
       "  Var1           Freq\n",
       "1 ADNP-human     3   \n",
       "2 AFF1-human     6   \n",
       "3 AFF4-human     3   \n",
       "4 AGO1-human     9   \n",
       "5 ARHGAP35-human 3   \n",
       "6 ARID1B-human   3   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "dat = as.data.frame(table(dat$`Experiment target`))\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "5818a5a9-743a-4bc5-85fb-07790f073821",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Var1</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>ADNP-human    </td><td>22</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AFF1-human    </td><td>44</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>AFF4-human    </td><td>22</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AGO1-human    </td><td>62</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>ARHGAP35-human</td><td>22</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ARID1B-human  </td><td>22</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Var1 & Freq\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & ADNP-human     & 22\\\\\n",
       "\t2 & AFF1-human     & 44\\\\\n",
       "\t3 & AFF4-human     & 22\\\\\n",
       "\t4 & AGO1-human     & 62\\\\\n",
       "\t5 & ARHGAP35-human & 22\\\\\n",
       "\t6 & ARID1B-human   & 22\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| <!--/--> | Var1 &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | ADNP-human     | 22 |\n",
       "| 2 | AFF1-human     | 44 |\n",
       "| 3 | AFF4-human     | 22 |\n",
       "| 4 | AGO1-human     | 62 |\n",
       "| 5 | ARHGAP35-human | 22 |\n",
       "| 6 | ARID1B-human   | 22 |\n",
       "\n"
      ],
      "text/plain": [
       "  Var1           Freq\n",
       "1 ADNP-human     22  \n",
       "2 AFF1-human     44  \n",
       "3 AFF4-human     22  \n",
       "4 AGO1-human     62  \n",
       "5 ARHGAP35-human 22  \n",
       "6 ARID1B-human   22  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "dat = as.data.frame(table(dat$`Experiment target`))\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4ac99b4-3e67-4a42-ad39-f73f7269aef5",
   "metadata": {},
   "source": [
    "## Arrange metadata tables"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "216aecac-4ab5-4e63-b364-9346ff2be492",
   "metadata": {},
   "source": [
    "**Helper function to arrange and filter the table**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "b1f205a2-9901-442f-b7b3-e24b7377989b",
   "metadata": {},
   "outputs": [],
   "source": [
    "fun = function(dat_input){\n",
    "    ### init\n",
    "    dat = dat_input\n",
    "    \n",
    "    ### get the analysis version\n",
    "    dat = dat %>% \n",
    "        dplyr::rowwise() %>%\n",
    "        dplyr::mutate(Analysis_Version = str_extract(`File analysis title`, \"ENCODE\\\\d\")) %>%\n",
    "        dplyr::ungroup()\n",
    "    \n",
    "    ### rename some columns\n",
    "    dat = dat %>% \n",
    "        dplyr::mutate(\n",
    "            Index_Experiment = `Experiment accession`,\n",
    "            Index_File       = `File accession`,\n",
    "            File_Format      = `File format`,\n",
    "            File_Type        = `File type`,\n",
    "            Output_Type      = `Output type`,\n",
    "            Genome           = `File assembly`,\n",
    "            Target           = str_remove(`Experiment target`, \"-human\"),\n",
    "            Bio_Replicates   = `Biological replicate(s)`,\n",
    "            Analysis         = `File analysis title`,\n",
    "            File_Name\t     = basename(`File download URL`),\n",
    "            File_URL         = `File download URL`\n",
    "        )\n",
    "    \n",
    "    ### select the needed columns\n",
    "    dat = dat %>%\n",
    "        dplyr::select(\n",
    "            Assay,\n",
    "            Index_Experiment,\n",
    "            Index_File,\n",
    "            File_Format,\n",
    "            File_Type,\n",
    "            Output_Type,\n",
    "            Genome,\n",
    "            Target,\n",
    "            Bio_Replicates,\n",
    "            Analysis,\n",
    "            Analysis_Version,\n",
    "            md5sum,\n",
    "            File_Name,\n",
    "            File_URL\n",
    "        )\n",
    "    \n",
    "    ### filter only the hg38 and ENCODE4 analysis pipeline\n",
    "    dat = dat %>% dplyr::filter(\n",
    "        Genome == \"GRCh38\",\n",
    "        Analysis_Version == \"ENCODE4\"\n",
    "    )\n",
    "    \n",
    "    ### return the simplified table\n",
    "    return(dat)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d8ac1bfb-2416-45d5-868e-841c971f1923",
   "metadata": {},
   "source": [
    "**Arrange tables**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "aa30385d-3277-434c-8b3a-50ad22a1e322",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 2160   14\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> Analysis_Version </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF447FCS </td>\n",
       "   <td style=\"text-align:left;\"> bigBed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bigBed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> 76853aa420fdb949461f8032db9e949c </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF447FCS.bigBed </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF447FCS/@@download/ENCFF447FCS.bigBed </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF563XMY </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> d01e3aa935f311f32991f796582632a6 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF563XMY.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF563XMY/@@download/ENCFF563XMY.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF843NBV </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> cab83f2be2220d6d328bd6566fc54d85 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF843NBV.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF843NBV/@@download/ENCFF843NBV.bed.gz </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v1\n",
    "dat = fun(dat)\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_simplify_v1 = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "0ffa51f4-f622-40af-a3a3-af74f8c67352",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 15749    14\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> Analysis_Version </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF521GKP </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> afaa174c390f22a9320ec3beeab8e490 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF521GKP.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF521GKP/@@download/ENCFF521GKP.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF194RBI </td>\n",
       "   <td style=\"text-align:left;\"> bed idr_ranked_peak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> IDR ranked peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> 319a983b0063a6c88ba1b56e20d5238a </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF194RBI.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF194RBI/@@download/ENCFF194RBI.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF914ZHO </td>\n",
       "   <td style=\"text-align:left;\"> bigBed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bigBed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> 8355b8626f20355b3d7efe40259c0b0b </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF914ZHO.bigBed </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF914ZHO/@@download/ENCFF914ZHO.bigBed </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_v2\n",
    "dat = fun(dat)\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_simplify_v2 = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "00f09011-9dcb-47dd-8989-0fcf67e54ba7",
   "metadata": {},
   "source": [
    "## Prepare download files"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd6696c6-5d5e-4cd9-aa39-8b9121c5f30d",
   "metadata": {},
   "source": [
    "**NarrowPeak files**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e095f62e-5f4e-4950-81b0-5cda43e1a004",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 720  14\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> Analysis_Version </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF843NBV </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> cab83f2be2220d6d328bd6566fc54d85 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF843NBV.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF843NBV/@@download/ENCFF843NBV.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR536CBU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF588CXX </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB9 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> c4faaf3e805d2fcfd7c9037fdb768174 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF588CXX.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF588CXX/@@download/ENCFF588CXX.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR436CZV </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF142MJD </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> AFF4 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> c8f76e772192439a7847dbbea9f73f40 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF142MJD.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF142MJD/@@download/ENCFF142MJD.bed.gz </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "###\n",
    "dat = dat_metadata_chipseq_simplify_v1\n",
    "dat = dat %>% dplyr::filter(File_Type == \"bed\", File_Format == \"bed narrowPeak\")\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_bed_narrowpeak = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9be8a33c-d255-416a-af98-979a1a8b5a26",
   "metadata": {},
   "source": [
    "**Signal p-value**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "9a7f50df-de70-4560-a873-d988dc082e1a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 720  14\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> Analysis_Version </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF563XMY </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> d01e3aa935f311f32991f796582632a6 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF563XMY.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF563XMY/@@download/ENCFF563XMY.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR536CBU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF547AGX </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB9 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> c13833244163ed3ab3f74b13924002eb </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF547AGX.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF547AGX/@@download/ENCFF547AGX.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR436CZV </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF798FSS </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> AFF4 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> def72a79a02e4f8ebf0dcd97572eaeb0 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF798FSS.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF798FSS/@@download/ENCFF798FSS.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "###\n",
    "dat = dat_metadata_chipseq_simplify_v1\n",
    "dat = dat %>% dplyr::filter(File_Type == \"bigWig\", Output_Type == \"signal p-value\")\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_bigwig_pvalue = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a0d1358-2d56-4a00-919a-141eb89d6baf",
   "metadata": {},
   "source": [
    "**Fold change**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "7478d560-c07d-4926-ab1e-e934ea46b6ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 2142    4\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:right;\"> Count_Replicates </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF521GKP </td>\n",
       "   <td style=\"text-align:left;\"> 2 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF876INH </td>\n",
       "   <td style=\"text-align:left;\"> 1 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF465ZAR </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:right;\"> 2 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR536CBU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF355MVO </td>\n",
       "   <td style=\"text-align:left;\"> 2 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR536CBU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF178FEN </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:right;\"> 2 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR536CBU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF760YQX </td>\n",
       "   <td style=\"text-align:left;\"> 1 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### helper function\n",
    "fun = function(txt){\n",
    "    lst = str_split(txt, \",\")\n",
    "    lst = lapply(lst, length)\n",
    "    vec = unlist(lst)\n",
    "    return(vec)\n",
    "}\n",
    "\n",
    "### summarize the number of replicates for each file\n",
    "dat = dat_metadata_chipseq_simplify_v2\n",
    "dat = dat %>% dplyr::filter(File_Type == \"bigWig\", Output_Type == \"fold change over control\")\n",
    "dat = dat %>% dplyr::mutate(Count_Replicates = fun(Bio_Replicates))\n",
    "dat = dat %>% dplyr::select(\n",
    "    Index_Experiment,\n",
    "    Index_File,\n",
    "    Bio_Replicates,\n",
    "    Count_Replicates)\n",
    "\n",
    "### assign and show\n",
    "dat_count_replicates = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "1e487ad8-9ffd-42d1-8174-9635441333ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 720  14\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> Analysis_Version </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF465ZAR </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> 9129c700630cd16e31a925d351562f14 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF465ZAR.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF465ZAR/@@download/ENCFF465ZAR.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR536CBU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF178FEN </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB9 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> c3b03e05c986366373390283fe28381f </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF178FEN.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF178FEN/@@download/ENCFF178FEN.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR436CZV </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF040PTP </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> AFF4 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> ec5d483f39e417045f201e447c62bdca </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF040PTP.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF040PTP/@@download/ENCFF040PTP.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### for each experiment, get the one with the most replicates\n",
    "dat = dat_count_replicates\n",
    "dat = dat %>% \n",
    "    dplyr::group_by(Index_Experiment) %>% \n",
    "    dplyr::slice_max(Count_Replicates)\n",
    "vec = dat$Index_File\n",
    "\n",
    "### filter the metadata\n",
    "dat = dat_metadata_chipseq_simplify_v2\n",
    "dat = dat %>% dplyr::filter(Index_File %in% vec)\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_bigwig_fold_change = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "b33b88d3-dae1-4586-a476-f78afc157e87",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "fold change over control \n",
       "                     720 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_chipseq_bigwig_fold_change\n",
    "table(dat$Output_Type)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7822dba6-9173-48b4-86cb-52bbb467feaf",
   "metadata": {},
   "source": [
    "## Save results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08d5c8a6-6b1c-4321-b7a8-22f037ba5c4d",
   "metadata": {},
   "source": [
    "**Check equal size**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "5994636b-1000-4005-9fa6-dafc7a3ab767",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "region_narrowPeak \n",
      "[1] 720  14\n",
      "signal_fold_change \n",
      "[1] 720  14\n",
      "signal_pvalue \n",
      "[1] 720  14\n"
     ]
    }
   ],
   "source": [
    "lst = list(\n",
    "    \"region_narrowPeak\"  = dat_metadata_chipseq_bed_narrowpeak,\n",
    "    \"signal_fold_change\" = dat_metadata_chipseq_bigwig_fold_change,\n",
    "    \"signal_pvalue\"      = dat_metadata_chipseq_bigwig_pvalue\n",
    ")\n",
    "\n",
    "for (idx in names(lst)){\n",
    "    ### show\n",
    "    cat(idx, \"\\n\")\n",
    "\n",
    "    ### extract data\n",
    "    dat_metadata = lst[[idx]]\n",
    "\n",
    "    dat = dat_metadata\n",
    "    print(dim(dat))\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "712b8616-fb2f-4bd0-901c-d87ff108092f",
   "metadata": {},
   "source": [
    "**Check equal experiment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "c6048569-2002-43b9-b093-8c61405e92c2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] TRUE\n",
      "[1] TRUE\n"
     ]
    }
   ],
   "source": [
    "lst = list(\n",
    "    \"region_narrowPeak\"  = dat_metadata_chipseq_bed_narrowpeak,\n",
    "    \"signal_fold_change\" = dat_metadata_chipseq_bigwig_fold_change,\n",
    "    \"signal_pvalue\"      = dat_metadata_chipseq_bigwig_pvalue\n",
    ")\n",
    "\n",
    "lst = lapply(lst, function(dat){\n",
    "    vec = dat$Index_Experiment\n",
    "    vec = sort(vec)\n",
    "})\n",
    "\n",
    "print(all(lst[[1]] == lst[[2]]))\n",
    "print(all(lst[[1]] == lst[[3]]))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b0266cf-fcd8-41ba-9ac3-c489e77ff0d9",
   "metadata": {},
   "source": [
    "**Export merged metadata**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "5891b243-3e92-4d13-bc3a-190eda9bb9c0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 2160   14\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> Analysis_Version </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR590FLL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF843NBV </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF347 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> cab83f2be2220d6d328bd6566fc54d85 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF843NBV.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF843NBV/@@download/ENCFF843NBV.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR536CBU </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF588CXX </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB9 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.4.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> c4faaf3e805d2fcfd7c9037fdb768174 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF588CXX.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF588CXX/@@download/ENCFF588CXX.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR436CZV </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF142MJD </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> AFF4 </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "   <td style=\"text-align:left;\"> c8f76e772192439a7847dbbea9f73f40 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF142MJD.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF142MJD/@@download/ENCFF142MJD.bed.gz </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### combine the metatables\n",
    "lst_dat_metadata = list(\n",
    "    \"region_narrowPeak\"  = dat_metadata_chipseq_bed_narrowpeak,\n",
    "    \"signal_fold_change\" = dat_metadata_chipseq_bigwig_fold_change,\n",
    "    \"signal_pvalue\"      = dat_metadata_chipseq_bigwig_pvalue\n",
    ")\n",
    "\n",
    "### merged table\n",
    "dat = bind_rows(lst_dat_metadata)\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_chipseq_merge = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "66c8c2ee-19c8-4396-be8f-fd3810fd4d47",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Save table: /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/data/external/encode_chipseq_tf_250120/metadata.merged.tsv \n"
     ]
    }
   ],
   "source": [
    "### set directory\n",
    "txt_fdiry = file.path(FD_DAT, \"external\", TXT_FOLDER_OUT)\n",
    "txt_fname = \"metadata.merged.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### write table and show\n",
    "dat = dat_metadata_chipseq_merge\n",
    "write_tsv(dat, txt_fpath)\n",
    "cat(\"Save table:\", txt_fpath, \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "565e08a2-d9b1-413e-bb78-c70f97abc9fc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Save table: ./table.chipseq_tf.metadata.files.tsv \n"
     ]
    }
   ],
   "source": [
    "### set directory\n",
    "txt_fdiry = \".\"\n",
    "txt_fname = \"table.chipseq_tf.metadata.files.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### write table and show\n",
    "dat = dat_metadata_chipseq_merge\n",
    "write_tsv(dat, txt_fpath)\n",
    "cat(\"Save table:\", txt_fpath, \"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc5c9c84-ad5d-46f3-901d-52b67514f421",
   "metadata": {},
   "source": [
    "**Export: metadata table, file list, and checksum: split for parallel execution**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "e5f77501-5c45-48fc-b6f7-fd64b982da39",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "region_narrowPeak \n",
      "Split: 01 ; Shape: 36 14 \n",
      "Split: 02 ; Shape: 36 14 \n",
      "Split: 03 ; Shape: 36 14 \n",
      "Split: 04 ; Shape: 36 14 \n",
      "Split: 05 ; Shape: 36 14 \n",
      "Split: 06 ; Shape: 36 14 \n",
      "Split: 07 ; Shape: 36 14 \n",
      "Split: 08 ; Shape: 36 14 \n",
      "Split: 09 ; Shape: 36 14 \n",
      "Split: 10 ; Shape: 36 14 \n",
      "Split: 11 ; Shape: 36 14 \n",
      "Split: 12 ; Shape: 36 14 \n",
      "Split: 13 ; Shape: 36 14 \n",
      "Split: 14 ; Shape: 36 14 \n",
      "Split: 15 ; Shape: 36 14 \n",
      "Split: 16 ; Shape: 36 14 \n",
      "Split: 17 ; Shape: 36 14 \n",
      "Split: 18 ; Shape: 36 14 \n",
      "Split: 19 ; Shape: 36 14 \n",
      "Split: 20 ; Shape: 36 14 \n",
      "\n",
      "signal_fold_change \n",
      "Split: 01 ; Shape: 36 14 \n",
      "Split: 02 ; Shape: 36 14 \n",
      "Split: 03 ; Shape: 36 14 \n",
      "Split: 04 ; Shape: 36 14 \n",
      "Split: 05 ; Shape: 36 14 \n",
      "Split: 06 ; Shape: 36 14 \n",
      "Split: 07 ; Shape: 36 14 \n",
      "Split: 08 ; Shape: 36 14 \n",
      "Split: 09 ; Shape: 36 14 \n",
      "Split: 10 ; Shape: 36 14 \n",
      "Split: 11 ; Shape: 36 14 \n",
      "Split: 12 ; Shape: 36 14 \n",
      "Split: 13 ; Shape: 36 14 \n",
      "Split: 14 ; Shape: 36 14 \n",
      "Split: 15 ; Shape: 36 14 \n",
      "Split: 16 ; Shape: 36 14 \n",
      "Split: 17 ; Shape: 36 14 \n",
      "Split: 18 ; Shape: 36 14 \n",
      "Split: 19 ; Shape: 36 14 \n",
      "Split: 20 ; Shape: 36 14 \n",
      "\n",
      "signal_pvalue \n",
      "Split: 01 ; Shape: 36 14 \n",
      "Split: 02 ; Shape: 36 14 \n",
      "Split: 03 ; Shape: 36 14 \n",
      "Split: 04 ; Shape: 36 14 \n",
      "Split: 05 ; Shape: 36 14 \n",
      "Split: 06 ; Shape: 36 14 \n",
      "Split: 07 ; Shape: 36 14 \n",
      "Split: 08 ; Shape: 36 14 \n",
      "Split: 09 ; Shape: 36 14 \n",
      "Split: 10 ; Shape: 36 14 \n",
      "Split: 11 ; Shape: 36 14 \n",
      "Split: 12 ; Shape: 36 14 \n",
      "Split: 13 ; Shape: 36 14 \n",
      "Split: 14 ; Shape: 36 14 \n",
      "Split: 15 ; Shape: 36 14 \n",
      "Split: 16 ; Shape: 36 14 \n",
      "Split: 17 ; Shape: 36 14 \n",
      "Split: 18 ; Shape: 36 14 \n",
      "Split: 19 ; Shape: 36 14 \n",
      "Split: 20 ; Shape: 36 14 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "### combine the metatables\n",
    "lst_dat_metadata = list(\n",
    "    \"region_narrowPeak\"  = dat_metadata_chipseq_bed_narrowpeak,\n",
    "    \"signal_fold_change\" = dat_metadata_chipseq_bigwig_fold_change,\n",
    "    \"signal_pvalue\"      = dat_metadata_chipseq_bigwig_pvalue\n",
    ")\n",
    "\n",
    "### split the data for parallelizing the download process\n",
    "NUM_SPLIT = 20\n",
    "\n",
    "for (txt_folder in names(lst_dat_metadata)){\n",
    "    ### show progress\n",
    "    cat(txt_folder, \"\\n\")\n",
    "    \n",
    "    ### extract data\n",
    "    dat_metadata = lst_dat_metadata[[txt_folder]]\n",
    "    \n",
    "    ### create folder if not exist\n",
    "    txt_fdiry = file.path(FD_DAT, \"external\", TXT_FOLDER_OUT, txt_folder)\n",
    "    txt_cmd   = paste(\"mkdir -p\", txt_fdiry)\n",
    "    system(txt_cmd)\n",
    "    \n",
    "    ### write metadata info\n",
    "    txt_fname = \"metadata.tsv\"\n",
    "    txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "    dat = dat_metadata\n",
    "    write_tsv(dat, txt_fpath)  \n",
    "\n",
    "    ### split the table into equal parts for parallel execution\n",
    "    dat = dat_metadata\n",
    "    vec = 1:nrow(dat)\n",
    "    vec = vec %% NUM_SPLIT\n",
    "    lst = split(dat, vec)\n",
    "    lst_dat_metadata_sub = lst\n",
    "\n",
    "    ### for each partition, save the file list\n",
    "    for (idx in 1:length(lst_dat_metadata_sub)){\n",
    "        \n",
    "        ### extract data\n",
    "        dat_metadata_sub = lst_dat_metadata_sub[[idx]]\n",
    "        \n",
    "        ### get file url\n",
    "        dat = dat_metadata_sub\n",
    "        dat = dat %>% dplyr::select(File_URL)\n",
    "        dat_download_furl = dat\n",
    "        \n",
    "        ### get md5sum for each file\n",
    "        dat = dat_metadata_sub\n",
    "        dat = dat %>% dplyr::select(md5sum, File_Name)\n",
    "        dat_download_md5sum = dat\n",
    "\n",
    "        ### set index for file name\n",
    "        txt_index = str_pad(idx, 2, pad=\"0\")\n",
    "        \n",
    "        ### write file list\n",
    "        txt_fname  = paste(\"files\", txt_index, \"txt\", sep=\".\")\n",
    "        txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "        \n",
    "        dat = dat_download_furl\n",
    "        write_tsv(dat, txt_fpath, col_names = FALSE)\n",
    "        \n",
    "        ### write checksum file\n",
    "        txt_fname  = paste(\"checksum_md5sum\", txt_index, \"txt\", sep=\".\")\n",
    "        txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "        \n",
    "        dat = dat_download_md5sum\n",
    "        write_tsv(dat, txt_fpath, col_names = FALSE)\n",
    "\n",
    "        ### show progress\n",
    "        txt = c(\"Split:\", txt_index, \"; Shape:\", dim(dat_metadata_sub))\n",
    "        cat(txt, \"\\n\")\n",
    "    }\n",
    "    cat(\"\\n\")\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f920610b-2fb9-49b7-ab7b-3916e4000382",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
