{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are in Singularity: singularity_proj_encode_fcc \n",
      "BASE DIRECTORY (FD_BASE): /data/reddylab/Kuei \n",
      "WORK DIRECTORY (FD_WORK): /data/reddylab/Kuei/out \n",
      "CODE DIRECTORY (FD_CODE): /data/reddylab/Kuei/code \n",
      "PATH OF PROJECT (FD_PRJ): /data/reddylab/Kuei/code/Proj_CombEffect_ENCODE_FCC \n",
      "PATH OF RESULTS (FD_RES): /data/reddylab/Kuei/out/proj_combeffect_encode_fcc \n",
      "PATH OF LOG     (FD_LOG): /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log \n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(source(\"../config/config_sing.R\")))\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Check data**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'files.txt'</li><li>'metadata.tsv'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'files.txt'\n",
       "\\item 'metadata.tsv'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'files.txt'\n",
       "2. 'metadata.tsv'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"files.txt\"    \"metadata.tsv\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fdiry = file.path(FD_RES, \"source\", \"annotation_chipseq_histone\")\n",
    "dir(fdiry)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1mRows: \u001b[22m\u001b[34m38\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m59\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m────────────────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m  (32): File accession, File format, File type, File format type, Output ...\n",
      "\u001b[32mdbl\u001b[39m   (1): Size\n",
      "\u001b[33mlgl\u001b[39m  (25): Biosample treatments, Biosample treatments amount, Biosample trea...\n",
      "\u001b[34mdate\u001b[39m  (1): Experiment date released\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 38 59\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 59</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>File accession</th><th scope=col>File format</th><th scope=col>File type</th><th scope=col>File format type</th><th scope=col>Output type</th><th scope=col>File assembly</th><th scope=col>Experiment accession</th><th scope=col>Assay</th><th scope=col>Donor(s)</th><th scope=col>Biosample term id</th><th scope=col>⋯</th><th scope=col>Platform</th><th scope=col>Controlled by</th><th scope=col>File Status</th><th scope=col>s3_uri</th><th scope=col>Azure URL</th><th scope=col>File analysis title</th><th scope=col>File analysis status</th><th scope=col>Audit WARNING</th><th scope=col>Audit NOT_COMPLIANT</th><th scope=col>Audit ERROR</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENCFF461RKK</td><td><span style=white-space:pre-wrap>bigWig        </span></td><td>bigWig</td><td><span style=white-space:pre-wrap>NA        </span></td><td><span style=white-space:pre-wrap>signal p-value        </span></td><td>GRCh38</td><td>ENCSR000EWA</td><td>Histone ChIP-seq</td><td>/human-donors/ENCDO000AAD/</td><td>EFO:0002067</td><td>⋯</td><td>NA</td><td>NA</td><td>released</td><td>s3://encode-public/2020/09/29/120fbb5a-db8e-464e-bdeb-9e72fdbb380b/ENCFF461RKK.bigWig</td><td>https://datasetencode.blob.core.windows.net/dataset/2020/09/29/120fbb5a-db8e-464e-bdeb-9e72fdbb380b/ENCFF461RKK.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D</td><td>ENCODE4 v1.5.1 GRCh38</td><td>released</td><td>low read length, low read depth, mild to moderate bottlenecking, moderate library complexity, inconsistent platforms</td><td><span style=white-space:pre-wrap>NA                                                      </span></td><td>NA</td></tr>\n",
       "\t<tr><td>ENCFF122CSI</td><td>bed narrowPeak</td><td><span style=white-space:pre-wrap>bed   </span></td><td>narrowPeak</td><td>pseudoreplicated peaks</td><td>GRCh38</td><td>ENCSR000EWA</td><td>Histone ChIP-seq</td><td>/human-donors/ENCDO000AAD/</td><td>EFO:0002067</td><td>⋯</td><td>NA</td><td>NA</td><td>released</td><td>s3://encode-public/2020/09/29/f70dc6ea-ac1b-4167-a054-e3dfb1f6684f/ENCFF122CSI.bed.gz</td><td>https://datasetencode.blob.core.windows.net/dataset/2020/09/29/f70dc6ea-ac1b-4167-a054-e3dfb1f6684f/ENCFF122CSI.bed.gz?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D</td><td>ENCODE4 v1.5.1 GRCh38</td><td>released</td><td>low read length, low read depth, mild to moderate bottlenecking, moderate library complexity, inconsistent platforms</td><td><span style=white-space:pre-wrap>NA                                                      </span></td><td>NA</td></tr>\n",
       "\t<tr><td>ENCFF801AHF</td><td>bed narrowPeak</td><td><span style=white-space:pre-wrap>bed   </span></td><td>narrowPeak</td><td>pseudoreplicated peaks</td><td>GRCh38</td><td>ENCSR000EWB</td><td>Histone ChIP-seq</td><td>/human-donors/ENCDO000AAD/</td><td>EFO:0002067</td><td>⋯</td><td>NA</td><td>NA</td><td>released</td><td>s3://encode-public/2020/11/01/3c1bac8c-3835-413b-a600-197421de4a1c/ENCFF801AHF.bed.gz</td><td>https://datasetencode.blob.core.windows.net/dataset/2020/11/01/3c1bac8c-3835-413b-a600-197421de4a1c/ENCFF801AHF.bed.gz?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D</td><td>ENCODE4 v1.6.0 GRCh38</td><td>released</td><td><span style=white-space:pre-wrap>low read length, inconsistent platforms                                                                             </span></td><td>insufficient read depth, control insufficient read depth</td><td>NA</td></tr>\n",
       "\t<tr><td>ENCFF847BFA</td><td><span style=white-space:pre-wrap>bigWig        </span></td><td>bigWig</td><td><span style=white-space:pre-wrap>NA        </span></td><td><span style=white-space:pre-wrap>signal p-value        </span></td><td>GRCh38</td><td>ENCSR000EWB</td><td>Histone ChIP-seq</td><td>/human-donors/ENCDO000AAD/</td><td>EFO:0002067</td><td>⋯</td><td>NA</td><td>NA</td><td>released</td><td>s3://encode-public/2020/11/01/57b9e89b-ef43-48ae-b322-65d12770d521/ENCFF847BFA.bigWig</td><td>https://datasetencode.blob.core.windows.net/dataset/2020/11/01/57b9e89b-ef43-48ae-b322-65d12770d521/ENCFF847BFA.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D</td><td>ENCODE4 v1.6.0 GRCh38</td><td>released</td><td><span style=white-space:pre-wrap>low read length, inconsistent platforms                                                                             </span></td><td>insufficient read depth, control insufficient read depth</td><td>NA</td></tr>\n",
       "\t<tr><td>ENCFF193ERO</td><td>bed narrowPeak</td><td><span style=white-space:pre-wrap>bed   </span></td><td>narrowPeak</td><td>pseudoreplicated peaks</td><td>GRCh38</td><td>ENCSR000AKR</td><td>Histone ChIP-seq</td><td>/human-donors/ENCDO000AAD/</td><td>EFO:0002067</td><td>⋯</td><td>NA</td><td>NA</td><td>released</td><td>s3://encode-public/2022/06/14/3e6e3aff-d6a6-4d0e-96c6-1a4d697dae75/ENCFF193ERO.bed.gz</td><td>https://datasetencode.blob.core.windows.net/dataset/2022/06/14/3e6e3aff-d6a6-4d0e-96c6-1a4d697dae75/ENCFF193ERO.bed.gz?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D</td><td>ENCODE4 v1.8.0 GRCh38</td><td>released</td><td><span style=white-space:pre-wrap>mild to moderate bottlenecking, low read length, inconsistent platforms                                             </span></td><td>insufficient read depth, control insufficient read depth</td><td>NA</td></tr>\n",
       "\t<tr><td>ENCFF605EVL</td><td><span style=white-space:pre-wrap>bigWig        </span></td><td>bigWig</td><td><span style=white-space:pre-wrap>NA        </span></td><td><span style=white-space:pre-wrap>signal p-value        </span></td><td>GRCh38</td><td>ENCSR000AKR</td><td>Histone ChIP-seq</td><td>/human-donors/ENCDO000AAD/</td><td>EFO:0002067</td><td>⋯</td><td>NA</td><td>NA</td><td>released</td><td>s3://encode-public/2022/06/14/7329c512-bf93-42ce-9687-ec757c44e4da/ENCFF605EVL.bigWig</td><td>https://datasetencode.blob.core.windows.net/dataset/2022/06/14/7329c512-bf93-42ce-9687-ec757c44e4da/ENCFF605EVL.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D</td><td>ENCODE4 v1.8.0 GRCh38</td><td>released</td><td><span style=white-space:pre-wrap>mild to moderate bottlenecking, low read length, inconsistent platforms                                             </span></td><td>insufficient read depth, control insufficient read depth</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 59\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " File accession & File format & File type & File format type & Output type & File assembly & Experiment accession & Assay & Donor(s) & Biosample term id & ⋯ & Platform & Controlled by & File Status & s3\\_uri & Azure URL & File analysis title & File analysis status & Audit WARNING & Audit NOT\\_COMPLIANT & Audit ERROR\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <lgl> & <lgl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENCFF461RKK & bigWig         & bigWig & NA         & signal p-value         & GRCh38 & ENCSR000EWA & Histone ChIP-seq & /human-donors/ENCDO000AAD/ & EFO:0002067 & ⋯ & NA & NA & released & s3://encode-public/2020/09/29/120fbb5a-db8e-464e-bdeb-9e72fdbb380b/ENCFF461RKK.bigWig & https://datasetencode.blob.core.windows.net/dataset/2020/09/29/120fbb5a-db8e-464e-bdeb-9e72fdbb380b/ENCFF461RKK.bigWig?sv=2019-10-10\\&si=prod\\&sr=c\\&sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c\\%3D & ENCODE4 v1.5.1 GRCh38 & released & low read length, low read depth, mild to moderate bottlenecking, moderate library complexity, inconsistent platforms & NA                                                       & NA\\\\\n",
       "\t ENCFF122CSI & bed narrowPeak & bed    & narrowPeak & pseudoreplicated peaks & GRCh38 & ENCSR000EWA & Histone ChIP-seq & /human-donors/ENCDO000AAD/ & EFO:0002067 & ⋯ & NA & NA & released & s3://encode-public/2020/09/29/f70dc6ea-ac1b-4167-a054-e3dfb1f6684f/ENCFF122CSI.bed.gz & https://datasetencode.blob.core.windows.net/dataset/2020/09/29/f70dc6ea-ac1b-4167-a054-e3dfb1f6684f/ENCFF122CSI.bed.gz?sv=2019-10-10\\&si=prod\\&sr=c\\&sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c\\%3D & ENCODE4 v1.5.1 GRCh38 & released & low read length, low read depth, mild to moderate bottlenecking, moderate library complexity, inconsistent platforms & NA                                                       & NA\\\\\n",
       "\t ENCFF801AHF & bed narrowPeak & bed    & narrowPeak & pseudoreplicated peaks & GRCh38 & ENCSR000EWB & Histone ChIP-seq & /human-donors/ENCDO000AAD/ & EFO:0002067 & ⋯ & NA & NA & released & s3://encode-public/2020/11/01/3c1bac8c-3835-413b-a600-197421de4a1c/ENCFF801AHF.bed.gz & https://datasetencode.blob.core.windows.net/dataset/2020/11/01/3c1bac8c-3835-413b-a600-197421de4a1c/ENCFF801AHF.bed.gz?sv=2019-10-10\\&si=prod\\&sr=c\\&sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c\\%3D & ENCODE4 v1.6.0 GRCh38 & released & low read length, inconsistent platforms                                                                              & insufficient read depth, control insufficient read depth & NA\\\\\n",
       "\t ENCFF847BFA & bigWig         & bigWig & NA         & signal p-value         & GRCh38 & ENCSR000EWB & Histone ChIP-seq & /human-donors/ENCDO000AAD/ & EFO:0002067 & ⋯ & NA & NA & released & s3://encode-public/2020/11/01/57b9e89b-ef43-48ae-b322-65d12770d521/ENCFF847BFA.bigWig & https://datasetencode.blob.core.windows.net/dataset/2020/11/01/57b9e89b-ef43-48ae-b322-65d12770d521/ENCFF847BFA.bigWig?sv=2019-10-10\\&si=prod\\&sr=c\\&sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c\\%3D & ENCODE4 v1.6.0 GRCh38 & released & low read length, inconsistent platforms                                                                              & insufficient read depth, control insufficient read depth & NA\\\\\n",
       "\t ENCFF193ERO & bed narrowPeak & bed    & narrowPeak & pseudoreplicated peaks & GRCh38 & ENCSR000AKR & Histone ChIP-seq & /human-donors/ENCDO000AAD/ & EFO:0002067 & ⋯ & NA & NA & released & s3://encode-public/2022/06/14/3e6e3aff-d6a6-4d0e-96c6-1a4d697dae75/ENCFF193ERO.bed.gz & https://datasetencode.blob.core.windows.net/dataset/2022/06/14/3e6e3aff-d6a6-4d0e-96c6-1a4d697dae75/ENCFF193ERO.bed.gz?sv=2019-10-10\\&si=prod\\&sr=c\\&sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c\\%3D & ENCODE4 v1.8.0 GRCh38 & released & mild to moderate bottlenecking, low read length, inconsistent platforms                                              & insufficient read depth, control insufficient read depth & NA\\\\\n",
       "\t ENCFF605EVL & bigWig         & bigWig & NA         & signal p-value         & GRCh38 & ENCSR000AKR & Histone ChIP-seq & /human-donors/ENCDO000AAD/ & EFO:0002067 & ⋯ & NA & NA & released & s3://encode-public/2022/06/14/7329c512-bf93-42ce-9687-ec757c44e4da/ENCFF605EVL.bigWig & https://datasetencode.blob.core.windows.net/dataset/2022/06/14/7329c512-bf93-42ce-9687-ec757c44e4da/ENCFF605EVL.bigWig?sv=2019-10-10\\&si=prod\\&sr=c\\&sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c\\%3D & ENCODE4 v1.8.0 GRCh38 & released & mild to moderate bottlenecking, low read length, inconsistent platforms                                              & insufficient read depth, control insufficient read depth & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 59\n",
       "\n",
       "| File accession &lt;chr&gt; | File format &lt;chr&gt; | File type &lt;chr&gt; | File format type &lt;chr&gt; | Output type &lt;chr&gt; | File assembly &lt;chr&gt; | Experiment accession &lt;chr&gt; | Assay &lt;chr&gt; | Donor(s) &lt;chr&gt; | Biosample term id &lt;chr&gt; | ⋯ ⋯ | Platform &lt;lgl&gt; | Controlled by &lt;lgl&gt; | File Status &lt;chr&gt; | s3_uri &lt;chr&gt; | Azure URL &lt;chr&gt; | File analysis title &lt;chr&gt; | File analysis status &lt;chr&gt; | Audit WARNING &lt;chr&gt; | Audit NOT_COMPLIANT &lt;chr&gt; | Audit ERROR &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENCFF461RKK | bigWig         | bigWig | NA         | signal p-value         | GRCh38 | ENCSR000EWA | Histone ChIP-seq | /human-donors/ENCDO000AAD/ | EFO:0002067 | ⋯ | NA | NA | released | s3://encode-public/2020/09/29/120fbb5a-db8e-464e-bdeb-9e72fdbb380b/ENCFF461RKK.bigWig | https://datasetencode.blob.core.windows.net/dataset/2020/09/29/120fbb5a-db8e-464e-bdeb-9e72fdbb380b/ENCFF461RKK.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D | ENCODE4 v1.5.1 GRCh38 | released | low read length, low read depth, mild to moderate bottlenecking, moderate library complexity, inconsistent platforms | NA                                                       | NA |\n",
       "| ENCFF122CSI | bed narrowPeak | bed    | narrowPeak | pseudoreplicated peaks | GRCh38 | ENCSR000EWA | Histone ChIP-seq | /human-donors/ENCDO000AAD/ | EFO:0002067 | ⋯ | NA | NA | released | s3://encode-public/2020/09/29/f70dc6ea-ac1b-4167-a054-e3dfb1f6684f/ENCFF122CSI.bed.gz | https://datasetencode.blob.core.windows.net/dataset/2020/09/29/f70dc6ea-ac1b-4167-a054-e3dfb1f6684f/ENCFF122CSI.bed.gz?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D | ENCODE4 v1.5.1 GRCh38 | released | low read length, low read depth, mild to moderate bottlenecking, moderate library complexity, inconsistent platforms | NA                                                       | NA |\n",
       "| ENCFF801AHF | bed narrowPeak | bed    | narrowPeak | pseudoreplicated peaks | GRCh38 | ENCSR000EWB | Histone ChIP-seq | /human-donors/ENCDO000AAD/ | EFO:0002067 | ⋯ | NA | NA | released | s3://encode-public/2020/11/01/3c1bac8c-3835-413b-a600-197421de4a1c/ENCFF801AHF.bed.gz | https://datasetencode.blob.core.windows.net/dataset/2020/11/01/3c1bac8c-3835-413b-a600-197421de4a1c/ENCFF801AHF.bed.gz?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D | ENCODE4 v1.6.0 GRCh38 | released | low read length, inconsistent platforms                                                                              | insufficient read depth, control insufficient read depth | NA |\n",
       "| ENCFF847BFA | bigWig         | bigWig | NA         | signal p-value         | GRCh38 | ENCSR000EWB | Histone ChIP-seq | /human-donors/ENCDO000AAD/ | EFO:0002067 | ⋯ | NA | NA | released | s3://encode-public/2020/11/01/57b9e89b-ef43-48ae-b322-65d12770d521/ENCFF847BFA.bigWig | https://datasetencode.blob.core.windows.net/dataset/2020/11/01/57b9e89b-ef43-48ae-b322-65d12770d521/ENCFF847BFA.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D | ENCODE4 v1.6.0 GRCh38 | released | low read length, inconsistent platforms                                                                              | insufficient read depth, control insufficient read depth | NA |\n",
       "| ENCFF193ERO | bed narrowPeak | bed    | narrowPeak | pseudoreplicated peaks | GRCh38 | ENCSR000AKR | Histone ChIP-seq | /human-donors/ENCDO000AAD/ | EFO:0002067 | ⋯ | NA | NA | released | s3://encode-public/2022/06/14/3e6e3aff-d6a6-4d0e-96c6-1a4d697dae75/ENCFF193ERO.bed.gz | https://datasetencode.blob.core.windows.net/dataset/2022/06/14/3e6e3aff-d6a6-4d0e-96c6-1a4d697dae75/ENCFF193ERO.bed.gz?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D | ENCODE4 v1.8.0 GRCh38 | released | mild to moderate bottlenecking, low read length, inconsistent platforms                                              | insufficient read depth, control insufficient read depth | NA |\n",
       "| ENCFF605EVL | bigWig         | bigWig | NA         | signal p-value         | GRCh38 | ENCSR000AKR | Histone ChIP-seq | /human-donors/ENCDO000AAD/ | EFO:0002067 | ⋯ | NA | NA | released | s3://encode-public/2022/06/14/7329c512-bf93-42ce-9687-ec757c44e4da/ENCFF605EVL.bigWig | https://datasetencode.blob.core.windows.net/dataset/2022/06/14/7329c512-bf93-42ce-9687-ec757c44e4da/ENCFF605EVL.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D | ENCODE4 v1.8.0 GRCh38 | released | mild to moderate bottlenecking, low read length, inconsistent platforms                                              | insufficient read depth, control insufficient read depth | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  File accession File format    File type File format type\n",
       "1 ENCFF461RKK    bigWig         bigWig    NA              \n",
       "2 ENCFF122CSI    bed narrowPeak bed       narrowPeak      \n",
       "3 ENCFF801AHF    bed narrowPeak bed       narrowPeak      \n",
       "4 ENCFF847BFA    bigWig         bigWig    NA              \n",
       "5 ENCFF193ERO    bed narrowPeak bed       narrowPeak      \n",
       "6 ENCFF605EVL    bigWig         bigWig    NA              \n",
       "  Output type            File assembly Experiment accession Assay           \n",
       "1 signal p-value         GRCh38        ENCSR000EWA          Histone ChIP-seq\n",
       "2 pseudoreplicated peaks GRCh38        ENCSR000EWA          Histone ChIP-seq\n",
       "3 pseudoreplicated peaks GRCh38        ENCSR000EWB          Histone ChIP-seq\n",
       "4 signal p-value         GRCh38        ENCSR000EWB          Histone ChIP-seq\n",
       "5 pseudoreplicated peaks GRCh38        ENCSR000AKR          Histone ChIP-seq\n",
       "6 signal p-value         GRCh38        ENCSR000AKR          Histone ChIP-seq\n",
       "  Donor(s)                   Biosample term id ⋯ Platform Controlled by\n",
       "1 /human-donors/ENCDO000AAD/ EFO:0002067       ⋯ NA       NA           \n",
       "2 /human-donors/ENCDO000AAD/ EFO:0002067       ⋯ NA       NA           \n",
       "3 /human-donors/ENCDO000AAD/ EFO:0002067       ⋯ NA       NA           \n",
       "4 /human-donors/ENCDO000AAD/ EFO:0002067       ⋯ NA       NA           \n",
       "5 /human-donors/ENCDO000AAD/ EFO:0002067       ⋯ NA       NA           \n",
       "6 /human-donors/ENCDO000AAD/ EFO:0002067       ⋯ NA       NA           \n",
       "  File Status\n",
       "1 released   \n",
       "2 released   \n",
       "3 released   \n",
       "4 released   \n",
       "5 released   \n",
       "6 released   \n",
       "  s3_uri                                                                               \n",
       "1 s3://encode-public/2020/09/29/120fbb5a-db8e-464e-bdeb-9e72fdbb380b/ENCFF461RKK.bigWig\n",
       "2 s3://encode-public/2020/09/29/f70dc6ea-ac1b-4167-a054-e3dfb1f6684f/ENCFF122CSI.bed.gz\n",
       "3 s3://encode-public/2020/11/01/3c1bac8c-3835-413b-a600-197421de4a1c/ENCFF801AHF.bed.gz\n",
       "4 s3://encode-public/2020/11/01/57b9e89b-ef43-48ae-b322-65d12770d521/ENCFF847BFA.bigWig\n",
       "5 s3://encode-public/2022/06/14/3e6e3aff-d6a6-4d0e-96c6-1a4d697dae75/ENCFF193ERO.bed.gz\n",
       "6 s3://encode-public/2022/06/14/7329c512-bf93-42ce-9687-ec757c44e4da/ENCFF605EVL.bigWig\n",
       "  Azure URL                                                                                                                                                                                           \n",
       "1 https://datasetencode.blob.core.windows.net/dataset/2020/09/29/120fbb5a-db8e-464e-bdeb-9e72fdbb380b/ENCFF461RKK.bigWig?sv=2019-10-10&si=prod&sr=c&sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D\n",
       "2 https://datasetencode.blob.core.windows.net/dataset/2020/09/29/f70dc6ea-ac1b-4167-a054-e3dfb1f6684f/ENCFF122CSI.bed.gz?sv=2019-10-10&si=prod&sr=c&sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D\n",
       "3 https://datasetencode.blob.core.windows.net/dataset/2020/11/01/3c1bac8c-3835-413b-a600-197421de4a1c/ENCFF801AHF.bed.gz?sv=2019-10-10&si=prod&sr=c&sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D\n",
       "4 https://datasetencode.blob.core.windows.net/dataset/2020/11/01/57b9e89b-ef43-48ae-b322-65d12770d521/ENCFF847BFA.bigWig?sv=2019-10-10&si=prod&sr=c&sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D\n",
       "5 https://datasetencode.blob.core.windows.net/dataset/2022/06/14/3e6e3aff-d6a6-4d0e-96c6-1a4d697dae75/ENCFF193ERO.bed.gz?sv=2019-10-10&si=prod&sr=c&sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D\n",
       "6 https://datasetencode.blob.core.windows.net/dataset/2022/06/14/7329c512-bf93-42ce-9687-ec757c44e4da/ENCFF605EVL.bigWig?sv=2019-10-10&si=prod&sr=c&sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D\n",
       "  File analysis title   File analysis status\n",
       "1 ENCODE4 v1.5.1 GRCh38 released            \n",
       "2 ENCODE4 v1.5.1 GRCh38 released            \n",
       "3 ENCODE4 v1.6.0 GRCh38 released            \n",
       "4 ENCODE4 v1.6.0 GRCh38 released            \n",
       "5 ENCODE4 v1.8.0 GRCh38 released            \n",
       "6 ENCODE4 v1.8.0 GRCh38 released            \n",
       "  Audit WARNING                                                                                                       \n",
       "1 low read length, low read depth, mild to moderate bottlenecking, moderate library complexity, inconsistent platforms\n",
       "2 low read length, low read depth, mild to moderate bottlenecking, moderate library complexity, inconsistent platforms\n",
       "3 low read length, inconsistent platforms                                                                             \n",
       "4 low read length, inconsistent platforms                                                                             \n",
       "5 mild to moderate bottlenecking, low read length, inconsistent platforms                                             \n",
       "6 mild to moderate bottlenecking, low read length, inconsistent platforms                                             \n",
       "  Audit NOT_COMPLIANT                                      Audit ERROR\n",
       "1 NA                                                       NA         \n",
       "2 NA                                                       NA         \n",
       "3 insufficient read depth, control insufficient read depth NA         \n",
       "4 insufficient read depth, control insufficient read depth NA         \n",
       "5 insufficient read depth, control insufficient read depth NA         \n",
       "6 insufficient read depth, control insufficient read depth NA         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fdiry = file.path(FD_RES, \"source\", \"annotation_chipseq_histone\")\n",
    "fname = \"metadata.tsv\"\n",
    "fpath = file.path(fdiry, fname)\n",
    "\n",
    "dat = read_tsv(fpath)\n",
    "\n",
    "dat_chipseq_import = dat\n",
    "print(dim(dat))\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'File accession'</li><li>'File format'</li><li>'File type'</li><li>'File format type'</li><li>'Output type'</li><li>'File assembly'</li><li>'Experiment accession'</li><li>'Assay'</li><li>'Donor(s)'</li><li>'Biosample term id'</li><li>'Biosample term name'</li><li>'Biosample type'</li><li>'Biosample organism'</li><li>'Biosample treatments'</li><li>'Biosample treatments amount'</li><li>'Biosample treatments duration'</li><li>'Biosample genetic modifications methods'</li><li>'Biosample genetic modifications categories'</li><li>'Biosample genetic modifications targets'</li><li>'Biosample genetic modifications gene targets'</li><li>'Biosample genetic modifications site coordinates'</li><li>'Biosample genetic modifications zygosity'</li><li>'Experiment target'</li><li>'Library made from'</li><li>'Library depleted in'</li><li>'Library extraction method'</li><li>'Library lysis method'</li><li>'Library crosslinking method'</li><li>'Library strand specific'</li><li>'Experiment date released'</li><li>'Project'</li><li>'RBNS protein concentration'</li><li>'Library fragmentation method'</li><li>'Library size range'</li><li>'Biological replicate(s)'</li><li>'Technical replicate(s)'</li><li>'Read length'</li><li>'Mapped read length'</li><li>'Run type'</li><li>'Paired end'</li><li>'Paired with'</li><li>'Index of'</li><li>'Derived from'</li><li>'Size'</li><li>'Lab'</li><li>'md5sum'</li><li>'dbxrefs'</li><li>'File download URL'</li><li>'Genome annotation'</li><li>'Platform'</li><li>'Controlled by'</li><li>'File Status'</li><li>'s3_uri'</li><li>'Azure URL'</li><li>'File analysis title'</li><li>'File analysis status'</li><li>'Audit WARNING'</li><li>'Audit NOT_COMPLIANT'</li><li>'Audit ERROR'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'File accession'\n",
       "\\item 'File format'\n",
       "\\item 'File type'\n",
       "\\item 'File format type'\n",
       "\\item 'Output type'\n",
       "\\item 'File assembly'\n",
       "\\item 'Experiment accession'\n",
       "\\item 'Assay'\n",
       "\\item 'Donor(s)'\n",
       "\\item 'Biosample term id'\n",
       "\\item 'Biosample term name'\n",
       "\\item 'Biosample type'\n",
       "\\item 'Biosample organism'\n",
       "\\item 'Biosample treatments'\n",
       "\\item 'Biosample treatments amount'\n",
       "\\item 'Biosample treatments duration'\n",
       "\\item 'Biosample genetic modifications methods'\n",
       "\\item 'Biosample genetic modifications categories'\n",
       "\\item 'Biosample genetic modifications targets'\n",
       "\\item 'Biosample genetic modifications gene targets'\n",
       "\\item 'Biosample genetic modifications site coordinates'\n",
       "\\item 'Biosample genetic modifications zygosity'\n",
       "\\item 'Experiment target'\n",
       "\\item 'Library made from'\n",
       "\\item 'Library depleted in'\n",
       "\\item 'Library extraction method'\n",
       "\\item 'Library lysis method'\n",
       "\\item 'Library crosslinking method'\n",
       "\\item 'Library strand specific'\n",
       "\\item 'Experiment date released'\n",
       "\\item 'Project'\n",
       "\\item 'RBNS protein concentration'\n",
       "\\item 'Library fragmentation method'\n",
       "\\item 'Library size range'\n",
       "\\item 'Biological replicate(s)'\n",
       "\\item 'Technical replicate(s)'\n",
       "\\item 'Read length'\n",
       "\\item 'Mapped read length'\n",
       "\\item 'Run type'\n",
       "\\item 'Paired end'\n",
       "\\item 'Paired with'\n",
       "\\item 'Index of'\n",
       "\\item 'Derived from'\n",
       "\\item 'Size'\n",
       "\\item 'Lab'\n",
       "\\item 'md5sum'\n",
       "\\item 'dbxrefs'\n",
       "\\item 'File download URL'\n",
       "\\item 'Genome annotation'\n",
       "\\item 'Platform'\n",
       "\\item 'Controlled by'\n",
       "\\item 'File Status'\n",
       "\\item 's3\\_uri'\n",
       "\\item 'Azure URL'\n",
       "\\item 'File analysis title'\n",
       "\\item 'File analysis status'\n",
       "\\item 'Audit WARNING'\n",
       "\\item 'Audit NOT\\_COMPLIANT'\n",
       "\\item 'Audit ERROR'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'File accession'\n",
       "2. 'File format'\n",
       "3. 'File type'\n",
       "4. 'File format type'\n",
       "5. 'Output type'\n",
       "6. 'File assembly'\n",
       "7. 'Experiment accession'\n",
       "8. 'Assay'\n",
       "9. 'Donor(s)'\n",
       "10. 'Biosample term id'\n",
       "11. 'Biosample term name'\n",
       "12. 'Biosample type'\n",
       "13. 'Biosample organism'\n",
       "14. 'Biosample treatments'\n",
       "15. 'Biosample treatments amount'\n",
       "16. 'Biosample treatments duration'\n",
       "17. 'Biosample genetic modifications methods'\n",
       "18. 'Biosample genetic modifications categories'\n",
       "19. 'Biosample genetic modifications targets'\n",
       "20. 'Biosample genetic modifications gene targets'\n",
       "21. 'Biosample genetic modifications site coordinates'\n",
       "22. 'Biosample genetic modifications zygosity'\n",
       "23. 'Experiment target'\n",
       "24. 'Library made from'\n",
       "25. 'Library depleted in'\n",
       "26. 'Library extraction method'\n",
       "27. 'Library lysis method'\n",
       "28. 'Library crosslinking method'\n",
       "29. 'Library strand specific'\n",
       "30. 'Experiment date released'\n",
       "31. 'Project'\n",
       "32. 'RBNS protein concentration'\n",
       "33. 'Library fragmentation method'\n",
       "34. 'Library size range'\n",
       "35. 'Biological replicate(s)'\n",
       "36. 'Technical replicate(s)'\n",
       "37. 'Read length'\n",
       "38. 'Mapped read length'\n",
       "39. 'Run type'\n",
       "40. 'Paired end'\n",
       "41. 'Paired with'\n",
       "42. 'Index of'\n",
       "43. 'Derived from'\n",
       "44. 'Size'\n",
       "45. 'Lab'\n",
       "46. 'md5sum'\n",
       "47. 'dbxrefs'\n",
       "48. 'File download URL'\n",
       "49. 'Genome annotation'\n",
       "50. 'Platform'\n",
       "51. 'Controlled by'\n",
       "52. 'File Status'\n",
       "53. 's3_uri'\n",
       "54. 'Azure URL'\n",
       "55. 'File analysis title'\n",
       "56. 'File analysis status'\n",
       "57. 'Audit WARNING'\n",
       "58. 'Audit NOT_COMPLIANT'\n",
       "59. 'Audit ERROR'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"File accession\"                                  \n",
       " [2] \"File format\"                                     \n",
       " [3] \"File type\"                                       \n",
       " [4] \"File format type\"                                \n",
       " [5] \"Output type\"                                     \n",
       " [6] \"File assembly\"                                   \n",
       " [7] \"Experiment accession\"                            \n",
       " [8] \"Assay\"                                           \n",
       " [9] \"Donor(s)\"                                        \n",
       "[10] \"Biosample term id\"                               \n",
       "[11] \"Biosample term name\"                             \n",
       "[12] \"Biosample type\"                                  \n",
       "[13] \"Biosample organism\"                              \n",
       "[14] \"Biosample treatments\"                            \n",
       "[15] \"Biosample treatments amount\"                     \n",
       "[16] \"Biosample treatments duration\"                   \n",
       "[17] \"Biosample genetic modifications methods\"         \n",
       "[18] \"Biosample genetic modifications categories\"      \n",
       "[19] \"Biosample genetic modifications targets\"         \n",
       "[20] \"Biosample genetic modifications gene targets\"    \n",
       "[21] \"Biosample genetic modifications site coordinates\"\n",
       "[22] \"Biosample genetic modifications zygosity\"        \n",
       "[23] \"Experiment target\"                               \n",
       "[24] \"Library made from\"                               \n",
       "[25] \"Library depleted in\"                             \n",
       "[26] \"Library extraction method\"                       \n",
       "[27] \"Library lysis method\"                            \n",
       "[28] \"Library crosslinking method\"                     \n",
       "[29] \"Library strand specific\"                         \n",
       "[30] \"Experiment date released\"                        \n",
       "[31] \"Project\"                                         \n",
       "[32] \"RBNS protein concentration\"                      \n",
       "[33] \"Library fragmentation method\"                    \n",
       "[34] \"Library size range\"                              \n",
       "[35] \"Biological replicate(s)\"                         \n",
       "[36] \"Technical replicate(s)\"                          \n",
       "[37] \"Read length\"                                     \n",
       "[38] \"Mapped read length\"                              \n",
       "[39] \"Run type\"                                        \n",
       "[40] \"Paired end\"                                      \n",
       "[41] \"Paired with\"                                     \n",
       "[42] \"Index of\"                                        \n",
       "[43] \"Derived from\"                                    \n",
       "[44] \"Size\"                                            \n",
       "[45] \"Lab\"                                             \n",
       "[46] \"md5sum\"                                          \n",
       "[47] \"dbxrefs\"                                         \n",
       "[48] \"File download URL\"                               \n",
       "[49] \"Genome annotation\"                               \n",
       "[50] \"Platform\"                                        \n",
       "[51] \"Controlled by\"                                   \n",
       "[52] \"File Status\"                                     \n",
       "[53] \"s3_uri\"                                          \n",
       "[54] \"Azure URL\"                                       \n",
       "[55] \"File analysis title\"                             \n",
       "[56] \"File analysis status\"                            \n",
       "[57] \"Audit WARNING\"                                   \n",
       "[58] \"Audit NOT_COMPLIANT\"                             \n",
       "[59] \"Audit ERROR\"                                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_chipseq_import\n",
    "colnames(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## bigwig file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 19 11\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 19 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Experiment accession</th><th scope=col>File accession</th><th scope=col>Experiment target</th><th scope=col>Output type</th><th scope=col>Assay</th><th scope=col>Biosample term name</th><th scope=col>File analysis title</th><th scope=col>File assembly</th><th scope=col>Experiment date released</th><th scope=col>md5sum</th><th scope=col>File download URL</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENCSR000EWA</td><td>ENCFF461RKK</td><td>H3K4me3 </td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-05-19</td><td>46d7c60097cc9a09d3c3bede8302f01b</td><td>https://www.encodeproject.org/files/ENCFF461RKK/@@download/ENCFF461RKK.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000EWB</td><td>ENCFF847BFA</td><td>H3K27me3</td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.0 GRCh38</td><td>GRCh38</td><td>2011-05-19</td><td>5afea5675fa090dad11c5a5ed9568ce4</td><td>https://www.encodeproject.org/files/ENCFF847BFA/@@download/ENCFF847BFA.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000AKR</td><td>ENCFF605EVL</td><td>H3K36me3</td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>c160804206370a082afe6e48ce3ec939</td><td>https://www.encodeproject.org/files/ENCFF605EVL/@@download/ENCFF605EVL.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000APE</td><td>ENCFF632NQA</td><td>H3K9me3 </td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>c695b7694f6c8b02e8bf13c871c387cf</td><td>https://www.encodeproject.org/files/ENCFF632NQA/@@download/ENCFF632NQA.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000EWC</td><td>ENCFF287LBI</td><td>H3K4me1 </td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-10-07</td><td>28df1a757a2e5517209c10d57f0ce03e</td><td>https://www.encodeproject.org/files/ENCFF287LBI/@@download/ENCFF287LBI.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000EVZ</td><td>ENCFF178QDA</td><td>H3K9ac  </td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.0 GRCh38</td><td>GRCh38</td><td>2011-05-19</td><td>16bace7252adafc836204f6fce4016ba</td><td>https://www.encodeproject.org/files/ENCFF178QDA/@@download/ENCFF178QDA.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR668LDD</td><td>ENCFF405ZDL</td><td>H3K4me3 </td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2016-11-08</td><td>eeddd4f2f60f952e3056d0c1bd39c73f</td><td>https://www.encodeproject.org/files/ENCFF405ZDL/@@download/ENCFF405ZDL.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000AKU</td><td>ENCFF767UON</td><td>H3K4me3 </td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>4c102d45be8326062895ed0a03d4ded7</td><td>https://www.encodeproject.org/files/ENCFF767UON/@@download/ENCFF767UON.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000AKX</td><td>ENCFF694ODT</td><td>H4K20me1</td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>197b9c35ecc18077888bf97780ec6f4b</td><td>https://www.encodeproject.org/files/ENCFF694ODT/@@download/ENCFF694ODT.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000AKQ</td><td>ENCFF582IMB</td><td>H3K27me3</td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>2ca48f44075eef7118a387260f2f95b9</td><td>https://www.encodeproject.org/files/ENCFF582IMB/@@download/ENCFF582IMB.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000DWB</td><td>ENCFF633OZC</td><td>H3K36me3</td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-02-11</td><td>4a8dbdc31024795ed8a6f53fba306609</td><td>https://www.encodeproject.org/files/ENCFF633OZC/@@download/ENCFF633OZC.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000AKT</td><td>ENCFF054RSU</td><td>H3K4me2 </td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>5a9a282c89fc45d7a729dcc8e7c71c10</td><td>https://www.encodeproject.org/files/ENCFF054RSU/@@download/ENCFF054RSU.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000AKV</td><td>ENCFF239EBH</td><td>H3K9ac  </td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>123d620e1a4dbc6730d15d90292f72c6</td><td>https://www.encodeproject.org/files/ENCFF239EBH/@@download/ENCFF239EBH.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000AKS</td><td>ENCFF457URZ</td><td>H3K4me1 </td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>c4ee97c60324b7163eabeb1526dc40b6</td><td>https://www.encodeproject.org/files/ENCFF457URZ/@@download/ENCFF457URZ.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000AKW</td><td>ENCFF220RGS</td><td>H3K9me1 </td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>d2b9936c13d8631f7a38d6e8329635bb</td><td>https://www.encodeproject.org/files/ENCFF220RGS/@@download/ENCFF220RGS.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000APC</td><td>ENCFF202EVH</td><td>H2AFZ   </td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>9a8216ae1fd1ae06ecd0a02fed8ced3b</td><td>https://www.encodeproject.org/files/ENCFF202EVH/@@download/ENCFF202EVH.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000APD</td><td>ENCFF334HSS</td><td>H3K79me2</td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>7f0d8f7287f845f0593eb5cb234d1973</td><td>https://www.encodeproject.org/files/ENCFF334HSS/@@download/ENCFF334HSS.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000AKP</td><td>ENCFF465GBD</td><td>H3K27ac </td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>9346587b9a23a2953b6ab49103e6c112</td><td>https://www.encodeproject.org/files/ENCFF465GBD/@@download/ENCFF465GBD.bigWig</td></tr>\n",
       "\t<tr><td>ENCSR000DWD</td><td>ENCFF071GML</td><td>H3K4me3 </td><td>signal p-value</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-02-11</td><td>bf514a6b8cd5fb9e10027629ff5bbf97</td><td>https://www.encodeproject.org/files/ENCFF071GML/@@download/ENCFF071GML.bigWig</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 19 × 11\n",
       "\\begin{tabular}{lllllllllll}\n",
       " Experiment accession & File accession & Experiment target & Output type & Assay & Biosample term name & File analysis title & File assembly & Experiment date released & md5sum & File download URL\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <date> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENCSR000EWA & ENCFF461RKK & H3K4me3  & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-05-19 & 46d7c60097cc9a09d3c3bede8302f01b & https://www.encodeproject.org/files/ENCFF461RKK/@@download/ENCFF461RKK.bigWig\\\\\n",
       "\t ENCSR000EWB & ENCFF847BFA & H3K27me3 & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.6.0 GRCh38 & GRCh38 & 2011-05-19 & 5afea5675fa090dad11c5a5ed9568ce4 & https://www.encodeproject.org/files/ENCFF847BFA/@@download/ENCFF847BFA.bigWig\\\\\n",
       "\t ENCSR000AKR & ENCFF605EVL & H3K36me3 & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & c160804206370a082afe6e48ce3ec939 & https://www.encodeproject.org/files/ENCFF605EVL/@@download/ENCFF605EVL.bigWig\\\\\n",
       "\t ENCSR000APE & ENCFF632NQA & H3K9me3  & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-02-10 & c695b7694f6c8b02e8bf13c871c387cf & https://www.encodeproject.org/files/ENCFF632NQA/@@download/ENCFF632NQA.bigWig\\\\\n",
       "\t ENCSR000EWC & ENCFF287LBI & H3K4me1  & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-10-07 & 28df1a757a2e5517209c10d57f0ce03e & https://www.encodeproject.org/files/ENCFF287LBI/@@download/ENCFF287LBI.bigWig\\\\\n",
       "\t ENCSR000EVZ & ENCFF178QDA & H3K9ac   & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.6.0 GRCh38 & GRCh38 & 2011-05-19 & 16bace7252adafc836204f6fce4016ba & https://www.encodeproject.org/files/ENCFF178QDA/@@download/ENCFF178QDA.bigWig\\\\\n",
       "\t ENCSR668LDD & ENCFF405ZDL & H3K4me3  & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2016-11-08 & eeddd4f2f60f952e3056d0c1bd39c73f & https://www.encodeproject.org/files/ENCFF405ZDL/@@download/ENCFF405ZDL.bigWig\\\\\n",
       "\t ENCSR000AKU & ENCFF767UON & H3K4me3  & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-02-10 & 4c102d45be8326062895ed0a03d4ded7 & https://www.encodeproject.org/files/ENCFF767UON/@@download/ENCFF767UON.bigWig\\\\\n",
       "\t ENCSR000AKX & ENCFF694ODT & H4K20me1 & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.6.1 GRCh38 & GRCh38 & 2011-02-10 & 197b9c35ecc18077888bf97780ec6f4b & https://www.encodeproject.org/files/ENCFF694ODT/@@download/ENCFF694ODT.bigWig\\\\\n",
       "\t ENCSR000AKQ & ENCFF582IMB & H3K27me3 & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & 2ca48f44075eef7118a387260f2f95b9 & https://www.encodeproject.org/files/ENCFF582IMB/@@download/ENCFF582IMB.bigWig\\\\\n",
       "\t ENCSR000DWB & ENCFF633OZC & H3K36me3 & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-02-11 & 4a8dbdc31024795ed8a6f53fba306609 & https://www.encodeproject.org/files/ENCFF633OZC/@@download/ENCFF633OZC.bigWig\\\\\n",
       "\t ENCSR000AKT & ENCFF054RSU & H3K4me2  & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.6.1 GRCh38 & GRCh38 & 2011-02-10 & 5a9a282c89fc45d7a729dcc8e7c71c10 & https://www.encodeproject.org/files/ENCFF054RSU/@@download/ENCFF054RSU.bigWig\\\\\n",
       "\t ENCSR000AKV & ENCFF239EBH & H3K9ac   & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.6.1 GRCh38 & GRCh38 & 2011-02-10 & 123d620e1a4dbc6730d15d90292f72c6 & https://www.encodeproject.org/files/ENCFF239EBH/@@download/ENCFF239EBH.bigWig\\\\\n",
       "\t ENCSR000AKS & ENCFF457URZ & H3K4me1  & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & c4ee97c60324b7163eabeb1526dc40b6 & https://www.encodeproject.org/files/ENCFF457URZ/@@download/ENCFF457URZ.bigWig\\\\\n",
       "\t ENCSR000AKW & ENCFF220RGS & H3K9me1  & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.6.0 GRCh38 & GRCh38 & 2011-02-10 & d2b9936c13d8631f7a38d6e8329635bb & https://www.encodeproject.org/files/ENCFF220RGS/@@download/ENCFF220RGS.bigWig\\\\\n",
       "\t ENCSR000APC & ENCFF202EVH & H2AFZ    & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.6.0 GRCh38 & GRCh38 & 2011-02-10 & 9a8216ae1fd1ae06ecd0a02fed8ced3b & https://www.encodeproject.org/files/ENCFF202EVH/@@download/ENCFF202EVH.bigWig\\\\\n",
       "\t ENCSR000APD & ENCFF334HSS & H3K79me2 & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & 7f0d8f7287f845f0593eb5cb234d1973 & https://www.encodeproject.org/files/ENCFF334HSS/@@download/ENCFF334HSS.bigWig\\\\\n",
       "\t ENCSR000AKP & ENCFF465GBD & H3K27ac  & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & 9346587b9a23a2953b6ab49103e6c112 & https://www.encodeproject.org/files/ENCFF465GBD/@@download/ENCFF465GBD.bigWig\\\\\n",
       "\t ENCSR000DWD & ENCFF071GML & H3K4me3  & signal p-value & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-02-11 & bf514a6b8cd5fb9e10027629ff5bbf97 & https://www.encodeproject.org/files/ENCFF071GML/@@download/ENCFF071GML.bigWig\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 19 × 11\n",
       "\n",
       "| Experiment accession &lt;chr&gt; | File accession &lt;chr&gt; | Experiment target &lt;chr&gt; | Output type &lt;chr&gt; | Assay &lt;chr&gt; | Biosample term name &lt;chr&gt; | File analysis title &lt;chr&gt; | File assembly &lt;chr&gt; | Experiment date released &lt;date&gt; | md5sum &lt;chr&gt; | File download URL &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENCSR000EWA | ENCFF461RKK | H3K4me3  | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-05-19 | 46d7c60097cc9a09d3c3bede8302f01b | https://www.encodeproject.org/files/ENCFF461RKK/@@download/ENCFF461RKK.bigWig |\n",
       "| ENCSR000EWB | ENCFF847BFA | H3K27me3 | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.6.0 GRCh38 | GRCh38 | 2011-05-19 | 5afea5675fa090dad11c5a5ed9568ce4 | https://www.encodeproject.org/files/ENCFF847BFA/@@download/ENCFF847BFA.bigWig |\n",
       "| ENCSR000AKR | ENCFF605EVL | H3K36me3 | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | c160804206370a082afe6e48ce3ec939 | https://www.encodeproject.org/files/ENCFF605EVL/@@download/ENCFF605EVL.bigWig |\n",
       "| ENCSR000APE | ENCFF632NQA | H3K9me3  | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-02-10 | c695b7694f6c8b02e8bf13c871c387cf | https://www.encodeproject.org/files/ENCFF632NQA/@@download/ENCFF632NQA.bigWig |\n",
       "| ENCSR000EWC | ENCFF287LBI | H3K4me1  | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-10-07 | 28df1a757a2e5517209c10d57f0ce03e | https://www.encodeproject.org/files/ENCFF287LBI/@@download/ENCFF287LBI.bigWig |\n",
       "| ENCSR000EVZ | ENCFF178QDA | H3K9ac   | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.6.0 GRCh38 | GRCh38 | 2011-05-19 | 16bace7252adafc836204f6fce4016ba | https://www.encodeproject.org/files/ENCFF178QDA/@@download/ENCFF178QDA.bigWig |\n",
       "| ENCSR668LDD | ENCFF405ZDL | H3K4me3  | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2016-11-08 | eeddd4f2f60f952e3056d0c1bd39c73f | https://www.encodeproject.org/files/ENCFF405ZDL/@@download/ENCFF405ZDL.bigWig |\n",
       "| ENCSR000AKU | ENCFF767UON | H3K4me3  | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-02-10 | 4c102d45be8326062895ed0a03d4ded7 | https://www.encodeproject.org/files/ENCFF767UON/@@download/ENCFF767UON.bigWig |\n",
       "| ENCSR000AKX | ENCFF694ODT | H4K20me1 | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.6.1 GRCh38 | GRCh38 | 2011-02-10 | 197b9c35ecc18077888bf97780ec6f4b | https://www.encodeproject.org/files/ENCFF694ODT/@@download/ENCFF694ODT.bigWig |\n",
       "| ENCSR000AKQ | ENCFF582IMB | H3K27me3 | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | 2ca48f44075eef7118a387260f2f95b9 | https://www.encodeproject.org/files/ENCFF582IMB/@@download/ENCFF582IMB.bigWig |\n",
       "| ENCSR000DWB | ENCFF633OZC | H3K36me3 | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-02-11 | 4a8dbdc31024795ed8a6f53fba306609 | https://www.encodeproject.org/files/ENCFF633OZC/@@download/ENCFF633OZC.bigWig |\n",
       "| ENCSR000AKT | ENCFF054RSU | H3K4me2  | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.6.1 GRCh38 | GRCh38 | 2011-02-10 | 5a9a282c89fc45d7a729dcc8e7c71c10 | https://www.encodeproject.org/files/ENCFF054RSU/@@download/ENCFF054RSU.bigWig |\n",
       "| ENCSR000AKV | ENCFF239EBH | H3K9ac   | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.6.1 GRCh38 | GRCh38 | 2011-02-10 | 123d620e1a4dbc6730d15d90292f72c6 | https://www.encodeproject.org/files/ENCFF239EBH/@@download/ENCFF239EBH.bigWig |\n",
       "| ENCSR000AKS | ENCFF457URZ | H3K4me1  | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | c4ee97c60324b7163eabeb1526dc40b6 | https://www.encodeproject.org/files/ENCFF457URZ/@@download/ENCFF457URZ.bigWig |\n",
       "| ENCSR000AKW | ENCFF220RGS | H3K9me1  | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.6.0 GRCh38 | GRCh38 | 2011-02-10 | d2b9936c13d8631f7a38d6e8329635bb | https://www.encodeproject.org/files/ENCFF220RGS/@@download/ENCFF220RGS.bigWig |\n",
       "| ENCSR000APC | ENCFF202EVH | H2AFZ    | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.6.0 GRCh38 | GRCh38 | 2011-02-10 | 9a8216ae1fd1ae06ecd0a02fed8ced3b | https://www.encodeproject.org/files/ENCFF202EVH/@@download/ENCFF202EVH.bigWig |\n",
       "| ENCSR000APD | ENCFF334HSS | H3K79me2 | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | 7f0d8f7287f845f0593eb5cb234d1973 | https://www.encodeproject.org/files/ENCFF334HSS/@@download/ENCFF334HSS.bigWig |\n",
       "| ENCSR000AKP | ENCFF465GBD | H3K27ac  | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | 9346587b9a23a2953b6ab49103e6c112 | https://www.encodeproject.org/files/ENCFF465GBD/@@download/ENCFF465GBD.bigWig |\n",
       "| ENCSR000DWD | ENCFF071GML | H3K4me3  | signal p-value | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-02-11 | bf514a6b8cd5fb9e10027629ff5bbf97 | https://www.encodeproject.org/files/ENCFF071GML/@@download/ENCFF071GML.bigWig |\n",
       "\n"
      ],
      "text/plain": [
       "   Experiment accession File accession Experiment target Output type   \n",
       "1  ENCSR000EWA          ENCFF461RKK    H3K4me3           signal p-value\n",
       "2  ENCSR000EWB          ENCFF847BFA    H3K27me3          signal p-value\n",
       "3  ENCSR000AKR          ENCFF605EVL    H3K36me3          signal p-value\n",
       "4  ENCSR000APE          ENCFF632NQA    H3K9me3           signal p-value\n",
       "5  ENCSR000EWC          ENCFF287LBI    H3K4me1           signal p-value\n",
       "6  ENCSR000EVZ          ENCFF178QDA    H3K9ac            signal p-value\n",
       "7  ENCSR668LDD          ENCFF405ZDL    H3K4me3           signal p-value\n",
       "8  ENCSR000AKU          ENCFF767UON    H3K4me3           signal p-value\n",
       "9  ENCSR000AKX          ENCFF694ODT    H4K20me1          signal p-value\n",
       "10 ENCSR000AKQ          ENCFF582IMB    H3K27me3          signal p-value\n",
       "11 ENCSR000DWB          ENCFF633OZC    H3K36me3          signal p-value\n",
       "12 ENCSR000AKT          ENCFF054RSU    H3K4me2           signal p-value\n",
       "13 ENCSR000AKV          ENCFF239EBH    H3K9ac            signal p-value\n",
       "14 ENCSR000AKS          ENCFF457URZ    H3K4me1           signal p-value\n",
       "15 ENCSR000AKW          ENCFF220RGS    H3K9me1           signal p-value\n",
       "16 ENCSR000APC          ENCFF202EVH    H2AFZ             signal p-value\n",
       "17 ENCSR000APD          ENCFF334HSS    H3K79me2          signal p-value\n",
       "18 ENCSR000AKP          ENCFF465GBD    H3K27ac           signal p-value\n",
       "19 ENCSR000DWD          ENCFF071GML    H3K4me3           signal p-value\n",
       "   Assay            Biosample term name File analysis title   File assembly\n",
       "1  Histone ChIP-seq K562                ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "2  Histone ChIP-seq K562                ENCODE4 v1.6.0 GRCh38 GRCh38       \n",
       "3  Histone ChIP-seq K562                ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "4  Histone ChIP-seq K562                ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "5  Histone ChIP-seq K562                ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "6  Histone ChIP-seq K562                ENCODE4 v1.6.0 GRCh38 GRCh38       \n",
       "7  Histone ChIP-seq K562                ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "8  Histone ChIP-seq K562                ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "9  Histone ChIP-seq K562                ENCODE4 v1.6.1 GRCh38 GRCh38       \n",
       "10 Histone ChIP-seq K562                ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "11 Histone ChIP-seq K562                ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "12 Histone ChIP-seq K562                ENCODE4 v1.6.1 GRCh38 GRCh38       \n",
       "13 Histone ChIP-seq K562                ENCODE4 v1.6.1 GRCh38 GRCh38       \n",
       "14 Histone ChIP-seq K562                ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "15 Histone ChIP-seq K562                ENCODE4 v1.6.0 GRCh38 GRCh38       \n",
       "16 Histone ChIP-seq K562                ENCODE4 v1.6.0 GRCh38 GRCh38       \n",
       "17 Histone ChIP-seq K562                ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "18 Histone ChIP-seq K562                ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "19 Histone ChIP-seq K562                ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "   Experiment date released md5sum                          \n",
       "1  2011-05-19               46d7c60097cc9a09d3c3bede8302f01b\n",
       "2  2011-05-19               5afea5675fa090dad11c5a5ed9568ce4\n",
       "3  2011-02-10               c160804206370a082afe6e48ce3ec939\n",
       "4  2011-02-10               c695b7694f6c8b02e8bf13c871c387cf\n",
       "5  2011-10-07               28df1a757a2e5517209c10d57f0ce03e\n",
       "6  2011-05-19               16bace7252adafc836204f6fce4016ba\n",
       "7  2016-11-08               eeddd4f2f60f952e3056d0c1bd39c73f\n",
       "8  2011-02-10               4c102d45be8326062895ed0a03d4ded7\n",
       "9  2011-02-10               197b9c35ecc18077888bf97780ec6f4b\n",
       "10 2011-02-10               2ca48f44075eef7118a387260f2f95b9\n",
       "11 2011-02-11               4a8dbdc31024795ed8a6f53fba306609\n",
       "12 2011-02-10               5a9a282c89fc45d7a729dcc8e7c71c10\n",
       "13 2011-02-10               123d620e1a4dbc6730d15d90292f72c6\n",
       "14 2011-02-10               c4ee97c60324b7163eabeb1526dc40b6\n",
       "15 2011-02-10               d2b9936c13d8631f7a38d6e8329635bb\n",
       "16 2011-02-10               9a8216ae1fd1ae06ecd0a02fed8ced3b\n",
       "17 2011-02-10               7f0d8f7287f845f0593eb5cb234d1973\n",
       "18 2011-02-10               9346587b9a23a2953b6ab49103e6c112\n",
       "19 2011-02-11               bf514a6b8cd5fb9e10027629ff5bbf97\n",
       "   File download URL                                                            \n",
       "1  https://www.encodeproject.org/files/ENCFF461RKK/@@download/ENCFF461RKK.bigWig\n",
       "2  https://www.encodeproject.org/files/ENCFF847BFA/@@download/ENCFF847BFA.bigWig\n",
       "3  https://www.encodeproject.org/files/ENCFF605EVL/@@download/ENCFF605EVL.bigWig\n",
       "4  https://www.encodeproject.org/files/ENCFF632NQA/@@download/ENCFF632NQA.bigWig\n",
       "5  https://www.encodeproject.org/files/ENCFF287LBI/@@download/ENCFF287LBI.bigWig\n",
       "6  https://www.encodeproject.org/files/ENCFF178QDA/@@download/ENCFF178QDA.bigWig\n",
       "7  https://www.encodeproject.org/files/ENCFF405ZDL/@@download/ENCFF405ZDL.bigWig\n",
       "8  https://www.encodeproject.org/files/ENCFF767UON/@@download/ENCFF767UON.bigWig\n",
       "9  https://www.encodeproject.org/files/ENCFF694ODT/@@download/ENCFF694ODT.bigWig\n",
       "10 https://www.encodeproject.org/files/ENCFF582IMB/@@download/ENCFF582IMB.bigWig\n",
       "11 https://www.encodeproject.org/files/ENCFF633OZC/@@download/ENCFF633OZC.bigWig\n",
       "12 https://www.encodeproject.org/files/ENCFF054RSU/@@download/ENCFF054RSU.bigWig\n",
       "13 https://www.encodeproject.org/files/ENCFF239EBH/@@download/ENCFF239EBH.bigWig\n",
       "14 https://www.encodeproject.org/files/ENCFF457URZ/@@download/ENCFF457URZ.bigWig\n",
       "15 https://www.encodeproject.org/files/ENCFF220RGS/@@download/ENCFF220RGS.bigWig\n",
       "16 https://www.encodeproject.org/files/ENCFF202EVH/@@download/ENCFF202EVH.bigWig\n",
       "17 https://www.encodeproject.org/files/ENCFF334HSS/@@download/ENCFF334HSS.bigWig\n",
       "18 https://www.encodeproject.org/files/ENCFF465GBD/@@download/ENCFF465GBD.bigWig\n",
       "19 https://www.encodeproject.org/files/ENCFF071GML/@@download/ENCFF071GML.bigWig"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_chipseq_import\n",
    "dat = dat %>% \n",
    "    dplyr::filter(`File format` == \"bigWig\") %>% \n",
    "    dplyr::mutate(`Experiment target` = str_remove(`Experiment target`, \"-human\")) %>%\n",
    "    dplyr::select(\n",
    "        `Experiment accession`,\n",
    "        `File accession`,\n",
    "        `Experiment target`,\n",
    "        `Output type`,\n",
    "        `Assay`,\n",
    "        `Biosample term name`,\n",
    "        `File analysis title`,\n",
    "        `File assembly`,\n",
    "        `Experiment date released`,\n",
    "        `md5sum`,\n",
    "        `File download URL`\n",
    "    ) \n",
    "dat_bw = dat\n",
    "print(dim(dat))\n",
    "dat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 10 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>File accession</th><th scope=col>Output type</th><th scope=col>Experiment accession</th><th scope=col>Experiment target</th><th scope=col>Target</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENCFF461RKK</td><td>signal p-value</td><td>ENCSR000EWA</td><td>H3K4me3-human </td><td>H3K4me3 </td></tr>\n",
       "\t<tr><td>ENCFF847BFA</td><td>signal p-value</td><td>ENCSR000EWB</td><td>H3K27me3-human</td><td>H3K27me3</td></tr>\n",
       "\t<tr><td>ENCFF287LBI</td><td>signal p-value</td><td>ENCSR000EWC</td><td>H3K4me1-human </td><td>H3K4me1 </td></tr>\n",
       "\t<tr><td>ENCFF405ZDL</td><td>signal p-value</td><td>ENCSR668LDD</td><td>H3K4me3-human </td><td>H3K4me3 </td></tr>\n",
       "\t<tr><td>ENCFF767UON</td><td>signal p-value</td><td>ENCSR000AKU</td><td>H3K4me3-human </td><td>H3K4me3 </td></tr>\n",
       "\t<tr><td>ENCFF582IMB</td><td>signal p-value</td><td>ENCSR000AKQ</td><td>H3K27me3-human</td><td>H3K27me3</td></tr>\n",
       "\t<tr><td>ENCFF054RSU</td><td>signal p-value</td><td>ENCSR000AKT</td><td>H3K4me2-human </td><td>H3K4me2 </td></tr>\n",
       "\t<tr><td>ENCFF457URZ</td><td>signal p-value</td><td>ENCSR000AKS</td><td>H3K4me1-human </td><td>H3K4me1 </td></tr>\n",
       "\t<tr><td>ENCFF465GBD</td><td>signal p-value</td><td>ENCSR000AKP</td><td>H3K27ac-human </td><td>H3K27ac </td></tr>\n",
       "\t<tr><td>ENCFF071GML</td><td>signal p-value</td><td>ENCSR000DWD</td><td>H3K4me3-human </td><td>H3K4me3 </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 10 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " File accession & Output type & Experiment accession & Experiment target & Target\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENCFF461RKK & signal p-value & ENCSR000EWA & H3K4me3-human  & H3K4me3 \\\\\n",
       "\t ENCFF847BFA & signal p-value & ENCSR000EWB & H3K27me3-human & H3K27me3\\\\\n",
       "\t ENCFF287LBI & signal p-value & ENCSR000EWC & H3K4me1-human  & H3K4me1 \\\\\n",
       "\t ENCFF405ZDL & signal p-value & ENCSR668LDD & H3K4me3-human  & H3K4me3 \\\\\n",
       "\t ENCFF767UON & signal p-value & ENCSR000AKU & H3K4me3-human  & H3K4me3 \\\\\n",
       "\t ENCFF582IMB & signal p-value & ENCSR000AKQ & H3K27me3-human & H3K27me3\\\\\n",
       "\t ENCFF054RSU & signal p-value & ENCSR000AKT & H3K4me2-human  & H3K4me2 \\\\\n",
       "\t ENCFF457URZ & signal p-value & ENCSR000AKS & H3K4me1-human  & H3K4me1 \\\\\n",
       "\t ENCFF465GBD & signal p-value & ENCSR000AKP & H3K27ac-human  & H3K27ac \\\\\n",
       "\t ENCFF071GML & signal p-value & ENCSR000DWD & H3K4me3-human  & H3K4me3 \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 10 × 5\n",
       "\n",
       "| File accession &lt;chr&gt; | Output type &lt;chr&gt; | Experiment accession &lt;chr&gt; | Experiment target &lt;chr&gt; | Target &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| ENCFF461RKK | signal p-value | ENCSR000EWA | H3K4me3-human  | H3K4me3  |\n",
       "| ENCFF847BFA | signal p-value | ENCSR000EWB | H3K27me3-human | H3K27me3 |\n",
       "| ENCFF287LBI | signal p-value | ENCSR000EWC | H3K4me1-human  | H3K4me1  |\n",
       "| ENCFF405ZDL | signal p-value | ENCSR668LDD | H3K4me3-human  | H3K4me3  |\n",
       "| ENCFF767UON | signal p-value | ENCSR000AKU | H3K4me3-human  | H3K4me3  |\n",
       "| ENCFF582IMB | signal p-value | ENCSR000AKQ | H3K27me3-human | H3K27me3 |\n",
       "| ENCFF054RSU | signal p-value | ENCSR000AKT | H3K4me2-human  | H3K4me2  |\n",
       "| ENCFF457URZ | signal p-value | ENCSR000AKS | H3K4me1-human  | H3K4me1  |\n",
       "| ENCFF465GBD | signal p-value | ENCSR000AKP | H3K27ac-human  | H3K27ac  |\n",
       "| ENCFF071GML | signal p-value | ENCSR000DWD | H3K4me3-human  | H3K4me3  |\n",
       "\n"
      ],
      "text/plain": [
       "   File accession Output type    Experiment accession Experiment target\n",
       "1  ENCFF461RKK    signal p-value ENCSR000EWA          H3K4me3-human    \n",
       "2  ENCFF847BFA    signal p-value ENCSR000EWB          H3K27me3-human   \n",
       "3  ENCFF287LBI    signal p-value ENCSR000EWC          H3K4me1-human    \n",
       "4  ENCFF405ZDL    signal p-value ENCSR668LDD          H3K4me3-human    \n",
       "5  ENCFF767UON    signal p-value ENCSR000AKU          H3K4me3-human    \n",
       "6  ENCFF582IMB    signal p-value ENCSR000AKQ          H3K27me3-human   \n",
       "7  ENCFF054RSU    signal p-value ENCSR000AKT          H3K4me2-human    \n",
       "8  ENCFF457URZ    signal p-value ENCSR000AKS          H3K4me1-human    \n",
       "9  ENCFF465GBD    signal p-value ENCSR000AKP          H3K27ac-human    \n",
       "10 ENCFF071GML    signal p-value ENCSR000DWD          H3K4me3-human    \n",
       "   Target  \n",
       "1  H3K4me3 \n",
       "2  H3K27me3\n",
       "3  H3K4me1 \n",
       "4  H3K4me3 \n",
       "5  H3K4me3 \n",
       "6  H3K27me3\n",
       "7  H3K4me2 \n",
       "8  H3K4me1 \n",
       "9  H3K27ac \n",
       "10 H3K4me3 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "targets = c(\n",
    "    \"H3K27ac\",\n",
    "    \"H3K27me3\",\n",
    "    \"H3K4me1\",\n",
    "    \"H3K4me2\",\n",
    "    \"H3K4me3\"\n",
    ")\n",
    "dat = dat_bw\n",
    "dat = dat %>% dplyr::filter(Target %in% targets)\n",
    "dat"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Bed peak file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 19 11\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 19 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Experiment_Accession</th><th scope=col>File_Accession</th><th scope=col>Target</th><th scope=col>Output</th><th scope=col>Assay</th><th scope=col>Biosample</th><th scope=col>File_Analysis_Title</th><th scope=col>File_Assembly</th><th scope=col>Experiment_Data_Released</th><th scope=col>md5sum</th><th scope=col>URL</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENCSR000EWA</td><td>ENCFF122CSI</td><td>H3K4me3 </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-05-19</td><td>e5694050796ccb5eeace1e6a4efec34d</td><td>https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000EWB</td><td>ENCFF801AHF</td><td>H3K27me3</td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.0 GRCh38</td><td>GRCh38</td><td>2011-05-19</td><td>e2bef555c66ac7cb95dd56dc3339420f</td><td>https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000AKR</td><td>ENCFF193ERO</td><td>H3K36me3</td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>f66a73f2377a38c4ba6c1d4405fd41a5</td><td>https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000APE</td><td>ENCFF963GZJ</td><td>H3K9me3 </td><td>replicated peaks      </td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>8e71cfdb547cdb5057d99d12d7be3ba6</td><td>https://www.encodeproject.org/files/ENCFF963GZJ/@@download/ENCFF963GZJ.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000EWC</td><td>ENCFF540NGG</td><td>H3K4me1 </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-10-07</td><td>63db47e5b9b98dbebff2ce20df066106</td><td>https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000EVZ</td><td>ENCFF148UQI</td><td>H3K9ac  </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.0 GRCh38</td><td>GRCh38</td><td>2011-05-19</td><td>0c4c1141dfc788db377407a9531be112</td><td>https://www.encodeproject.org/files/ENCFF148UQI/@@download/ENCFF148UQI.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR668LDD</td><td>ENCFF885FQN</td><td>H3K4me3 </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2016-11-08</td><td>db483a7fe9eaf56ee507f9852f0e0ca3</td><td>https://www.encodeproject.org/files/ENCFF885FQN/@@download/ENCFF885FQN.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000AKU</td><td>ENCFF689QIJ</td><td>H3K4me3 </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>5dea2993c0831ae344a989d601c09178</td><td>https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000AKX</td><td>ENCFF909RKY</td><td>H4K20me1</td><td>replicated peaks      </td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>fe4c93b6faa0ab4153ef41fe516bced4</td><td>https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000AKQ</td><td>ENCFF323WOT</td><td>H3K27me3</td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>4422969d0b63260e2fcb83e10fdcc02f</td><td>https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000DWB</td><td>ENCFF561OUZ</td><td>H3K36me3</td><td>replicated peaks      </td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-02-11</td><td>7974e4f7e3f7b1db43715efdf2532916</td><td>https://www.encodeproject.org/files/ENCFF561OUZ/@@download/ENCFF561OUZ.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000AKT</td><td>ENCFF749KLQ</td><td>H3K4me2 </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>55c1093b8d98bef1dc7f788fbb44eb7c</td><td>https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000AKV</td><td>ENCFF891CHI</td><td>H3K9ac  </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>0c4d48de710e88b7049fe53eb9b31dbe</td><td>https://www.encodeproject.org/files/ENCFF891CHI/@@download/ENCFF891CHI.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000AKS</td><td>ENCFF135ZLM</td><td>H3K4me1 </td><td>replicated peaks      </td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>b2db31175e0856a2b27c064da8e4bbb4</td><td>https://www.encodeproject.org/files/ENCFF135ZLM/@@download/ENCFF135ZLM.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000AKW</td><td>ENCFF462AVD</td><td>H3K9me1 </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>5ddbfd027332e8d38e801dd1eaa7575c</td><td>https://www.encodeproject.org/files/ENCFF462AVD/@@download/ENCFF462AVD.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000APC</td><td>ENCFF213OTI</td><td>H2AFZ   </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>a5934e60f1ebf8e393aa111a26f74074</td><td>https://www.encodeproject.org/files/ENCFF213OTI/@@download/ENCFF213OTI.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000APD</td><td>ENCFF209OQD</td><td>H3K79me2</td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>b0e6f4594be0ec60fd34e218ebc1dcaf</td><td>https://www.encodeproject.org/files/ENCFF209OQD/@@download/ENCFF209OQD.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000AKP</td><td>ENCFF544LXB</td><td>H3K27ac </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>81a2d09ef690a3317f6b0e793690cfdc</td><td>https://www.encodeproject.org/files/ENCFF544LXB/@@download/ENCFF544LXB.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000DWD</td><td>ENCFF706WUF</td><td>H3K4me3 </td><td>replicated peaks      </td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-02-11</td><td>f7616f8937225476151f4af2262dcfeb</td><td>https://www.encodeproject.org/files/ENCFF706WUF/@@download/ENCFF706WUF.bed.gz</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 19 × 11\n",
       "\\begin{tabular}{lllllllllll}\n",
       " Experiment\\_Accession & File\\_Accession & Target & Output & Assay & Biosample & File\\_Analysis\\_Title & File\\_Assembly & Experiment\\_Data\\_Released & md5sum & URL\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <date> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENCSR000EWA & ENCFF122CSI & H3K4me3  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-05-19 & e5694050796ccb5eeace1e6a4efec34d & https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz\\\\\n",
       "\t ENCSR000EWB & ENCFF801AHF & H3K27me3 & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.6.0 GRCh38 & GRCh38 & 2011-05-19 & e2bef555c66ac7cb95dd56dc3339420f & https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz\\\\\n",
       "\t ENCSR000AKR & ENCFF193ERO & H3K36me3 & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & f66a73f2377a38c4ba6c1d4405fd41a5 & https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz\\\\\n",
       "\t ENCSR000APE & ENCFF963GZJ & H3K9me3  & replicated peaks       & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-02-10 & 8e71cfdb547cdb5057d99d12d7be3ba6 & https://www.encodeproject.org/files/ENCFF963GZJ/@@download/ENCFF963GZJ.bed.gz\\\\\n",
       "\t ENCSR000EWC & ENCFF540NGG & H3K4me1  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-10-07 & 63db47e5b9b98dbebff2ce20df066106 & https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz\\\\\n",
       "\t ENCSR000EVZ & ENCFF148UQI & H3K9ac   & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.6.0 GRCh38 & GRCh38 & 2011-05-19 & 0c4c1141dfc788db377407a9531be112 & https://www.encodeproject.org/files/ENCFF148UQI/@@download/ENCFF148UQI.bed.gz\\\\\n",
       "\t ENCSR668LDD & ENCFF885FQN & H3K4me3  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2016-11-08 & db483a7fe9eaf56ee507f9852f0e0ca3 & https://www.encodeproject.org/files/ENCFF885FQN/@@download/ENCFF885FQN.bed.gz\\\\\n",
       "\t ENCSR000AKU & ENCFF689QIJ & H3K4me3  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-02-10 & 5dea2993c0831ae344a989d601c09178 & https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz\\\\\n",
       "\t ENCSR000AKX & ENCFF909RKY & H4K20me1 & replicated peaks       & Histone ChIP-seq & K562 & ENCODE4 v1.6.1 GRCh38 & GRCh38 & 2011-02-10 & fe4c93b6faa0ab4153ef41fe516bced4 & https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz\\\\\n",
       "\t ENCSR000AKQ & ENCFF323WOT & H3K27me3 & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & 4422969d0b63260e2fcb83e10fdcc02f & https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz\\\\\n",
       "\t ENCSR000DWB & ENCFF561OUZ & H3K36me3 & replicated peaks       & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-02-11 & 7974e4f7e3f7b1db43715efdf2532916 & https://www.encodeproject.org/files/ENCFF561OUZ/@@download/ENCFF561OUZ.bed.gz\\\\\n",
       "\t ENCSR000AKT & ENCFF749KLQ & H3K4me2  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.6.1 GRCh38 & GRCh38 & 2011-02-10 & 55c1093b8d98bef1dc7f788fbb44eb7c & https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz\\\\\n",
       "\t ENCSR000AKV & ENCFF891CHI & H3K9ac   & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.6.1 GRCh38 & GRCh38 & 2011-02-10 & 0c4d48de710e88b7049fe53eb9b31dbe & https://www.encodeproject.org/files/ENCFF891CHI/@@download/ENCFF891CHI.bed.gz\\\\\n",
       "\t ENCSR000AKS & ENCFF135ZLM & H3K4me1  & replicated peaks       & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & b2db31175e0856a2b27c064da8e4bbb4 & https://www.encodeproject.org/files/ENCFF135ZLM/@@download/ENCFF135ZLM.bed.gz\\\\\n",
       "\t ENCSR000AKW & ENCFF462AVD & H3K9me1  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.6.0 GRCh38 & GRCh38 & 2011-02-10 & 5ddbfd027332e8d38e801dd1eaa7575c & https://www.encodeproject.org/files/ENCFF462AVD/@@download/ENCFF462AVD.bed.gz\\\\\n",
       "\t ENCSR000APC & ENCFF213OTI & H2AFZ    & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.6.0 GRCh38 & GRCh38 & 2011-02-10 & a5934e60f1ebf8e393aa111a26f74074 & https://www.encodeproject.org/files/ENCFF213OTI/@@download/ENCFF213OTI.bed.gz\\\\\n",
       "\t ENCSR000APD & ENCFF209OQD & H3K79me2 & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & b0e6f4594be0ec60fd34e218ebc1dcaf & https://www.encodeproject.org/files/ENCFF209OQD/@@download/ENCFF209OQD.bed.gz\\\\\n",
       "\t ENCSR000AKP & ENCFF544LXB & H3K27ac  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & 81a2d09ef690a3317f6b0e793690cfdc & https://www.encodeproject.org/files/ENCFF544LXB/@@download/ENCFF544LXB.bed.gz\\\\\n",
       "\t ENCSR000DWD & ENCFF706WUF & H3K4me3  & replicated peaks       & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-02-11 & f7616f8937225476151f4af2262dcfeb & https://www.encodeproject.org/files/ENCFF706WUF/@@download/ENCFF706WUF.bed.gz\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 19 × 11\n",
       "\n",
       "| Experiment_Accession &lt;chr&gt; | File_Accession &lt;chr&gt; | Target &lt;chr&gt; | Output &lt;chr&gt; | Assay &lt;chr&gt; | Biosample &lt;chr&gt; | File_Analysis_Title &lt;chr&gt; | File_Assembly &lt;chr&gt; | Experiment_Data_Released &lt;date&gt; | md5sum &lt;chr&gt; | URL &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENCSR000EWA | ENCFF122CSI | H3K4me3  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-05-19 | e5694050796ccb5eeace1e6a4efec34d | https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz |\n",
       "| ENCSR000EWB | ENCFF801AHF | H3K27me3 | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.6.0 GRCh38 | GRCh38 | 2011-05-19 | e2bef555c66ac7cb95dd56dc3339420f | https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz |\n",
       "| ENCSR000AKR | ENCFF193ERO | H3K36me3 | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | f66a73f2377a38c4ba6c1d4405fd41a5 | https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz |\n",
       "| ENCSR000APE | ENCFF963GZJ | H3K9me3  | replicated peaks       | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-02-10 | 8e71cfdb547cdb5057d99d12d7be3ba6 | https://www.encodeproject.org/files/ENCFF963GZJ/@@download/ENCFF963GZJ.bed.gz |\n",
       "| ENCSR000EWC | ENCFF540NGG | H3K4me1  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-10-07 | 63db47e5b9b98dbebff2ce20df066106 | https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz |\n",
       "| ENCSR000EVZ | ENCFF148UQI | H3K9ac   | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.6.0 GRCh38 | GRCh38 | 2011-05-19 | 0c4c1141dfc788db377407a9531be112 | https://www.encodeproject.org/files/ENCFF148UQI/@@download/ENCFF148UQI.bed.gz |\n",
       "| ENCSR668LDD | ENCFF885FQN | H3K4me3  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2016-11-08 | db483a7fe9eaf56ee507f9852f0e0ca3 | https://www.encodeproject.org/files/ENCFF885FQN/@@download/ENCFF885FQN.bed.gz |\n",
       "| ENCSR000AKU | ENCFF689QIJ | H3K4me3  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-02-10 | 5dea2993c0831ae344a989d601c09178 | https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz |\n",
       "| ENCSR000AKX | ENCFF909RKY | H4K20me1 | replicated peaks       | Histone ChIP-seq | K562 | ENCODE4 v1.6.1 GRCh38 | GRCh38 | 2011-02-10 | fe4c93b6faa0ab4153ef41fe516bced4 | https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz |\n",
       "| ENCSR000AKQ | ENCFF323WOT | H3K27me3 | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | 4422969d0b63260e2fcb83e10fdcc02f | https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz |\n",
       "| ENCSR000DWB | ENCFF561OUZ | H3K36me3 | replicated peaks       | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-02-11 | 7974e4f7e3f7b1db43715efdf2532916 | https://www.encodeproject.org/files/ENCFF561OUZ/@@download/ENCFF561OUZ.bed.gz |\n",
       "| ENCSR000AKT | ENCFF749KLQ | H3K4me2  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.6.1 GRCh38 | GRCh38 | 2011-02-10 | 55c1093b8d98bef1dc7f788fbb44eb7c | https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz |\n",
       "| ENCSR000AKV | ENCFF891CHI | H3K9ac   | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.6.1 GRCh38 | GRCh38 | 2011-02-10 | 0c4d48de710e88b7049fe53eb9b31dbe | https://www.encodeproject.org/files/ENCFF891CHI/@@download/ENCFF891CHI.bed.gz |\n",
       "| ENCSR000AKS | ENCFF135ZLM | H3K4me1  | replicated peaks       | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | b2db31175e0856a2b27c064da8e4bbb4 | https://www.encodeproject.org/files/ENCFF135ZLM/@@download/ENCFF135ZLM.bed.gz |\n",
       "| ENCSR000AKW | ENCFF462AVD | H3K9me1  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.6.0 GRCh38 | GRCh38 | 2011-02-10 | 5ddbfd027332e8d38e801dd1eaa7575c | https://www.encodeproject.org/files/ENCFF462AVD/@@download/ENCFF462AVD.bed.gz |\n",
       "| ENCSR000APC | ENCFF213OTI | H2AFZ    | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.6.0 GRCh38 | GRCh38 | 2011-02-10 | a5934e60f1ebf8e393aa111a26f74074 | https://www.encodeproject.org/files/ENCFF213OTI/@@download/ENCFF213OTI.bed.gz |\n",
       "| ENCSR000APD | ENCFF209OQD | H3K79me2 | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | b0e6f4594be0ec60fd34e218ebc1dcaf | https://www.encodeproject.org/files/ENCFF209OQD/@@download/ENCFF209OQD.bed.gz |\n",
       "| ENCSR000AKP | ENCFF544LXB | H3K27ac  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | 81a2d09ef690a3317f6b0e793690cfdc | https://www.encodeproject.org/files/ENCFF544LXB/@@download/ENCFF544LXB.bed.gz |\n",
       "| ENCSR000DWD | ENCFF706WUF | H3K4me3  | replicated peaks       | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-02-11 | f7616f8937225476151f4af2262dcfeb | https://www.encodeproject.org/files/ENCFF706WUF/@@download/ENCFF706WUF.bed.gz |\n",
       "\n"
      ],
      "text/plain": [
       "   Experiment_Accession File_Accession Target   Output                \n",
       "1  ENCSR000EWA          ENCFF122CSI    H3K4me3  pseudoreplicated peaks\n",
       "2  ENCSR000EWB          ENCFF801AHF    H3K27me3 pseudoreplicated peaks\n",
       "3  ENCSR000AKR          ENCFF193ERO    H3K36me3 pseudoreplicated peaks\n",
       "4  ENCSR000APE          ENCFF963GZJ    H3K9me3  replicated peaks      \n",
       "5  ENCSR000EWC          ENCFF540NGG    H3K4me1  pseudoreplicated peaks\n",
       "6  ENCSR000EVZ          ENCFF148UQI    H3K9ac   pseudoreplicated peaks\n",
       "7  ENCSR668LDD          ENCFF885FQN    H3K4me3  pseudoreplicated peaks\n",
       "8  ENCSR000AKU          ENCFF689QIJ    H3K4me3  pseudoreplicated peaks\n",
       "9  ENCSR000AKX          ENCFF909RKY    H4K20me1 replicated peaks      \n",
       "10 ENCSR000AKQ          ENCFF323WOT    H3K27me3 pseudoreplicated peaks\n",
       "11 ENCSR000DWB          ENCFF561OUZ    H3K36me3 replicated peaks      \n",
       "12 ENCSR000AKT          ENCFF749KLQ    H3K4me2  pseudoreplicated peaks\n",
       "13 ENCSR000AKV          ENCFF891CHI    H3K9ac   pseudoreplicated peaks\n",
       "14 ENCSR000AKS          ENCFF135ZLM    H3K4me1  replicated peaks      \n",
       "15 ENCSR000AKW          ENCFF462AVD    H3K9me1  pseudoreplicated peaks\n",
       "16 ENCSR000APC          ENCFF213OTI    H2AFZ    pseudoreplicated peaks\n",
       "17 ENCSR000APD          ENCFF209OQD    H3K79me2 pseudoreplicated peaks\n",
       "18 ENCSR000AKP          ENCFF544LXB    H3K27ac  pseudoreplicated peaks\n",
       "19 ENCSR000DWD          ENCFF706WUF    H3K4me3  replicated peaks      \n",
       "   Assay            Biosample File_Analysis_Title   File_Assembly\n",
       "1  Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "2  Histone ChIP-seq K562      ENCODE4 v1.6.0 GRCh38 GRCh38       \n",
       "3  Histone ChIP-seq K562      ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "4  Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "5  Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "6  Histone ChIP-seq K562      ENCODE4 v1.6.0 GRCh38 GRCh38       \n",
       "7  Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "8  Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "9  Histone ChIP-seq K562      ENCODE4 v1.6.1 GRCh38 GRCh38       \n",
       "10 Histone ChIP-seq K562      ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "11 Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "12 Histone ChIP-seq K562      ENCODE4 v1.6.1 GRCh38 GRCh38       \n",
       "13 Histone ChIP-seq K562      ENCODE4 v1.6.1 GRCh38 GRCh38       \n",
       "14 Histone ChIP-seq K562      ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "15 Histone ChIP-seq K562      ENCODE4 v1.6.0 GRCh38 GRCh38       \n",
       "16 Histone ChIP-seq K562      ENCODE4 v1.6.0 GRCh38 GRCh38       \n",
       "17 Histone ChIP-seq K562      ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "18 Histone ChIP-seq K562      ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "19 Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "   Experiment_Data_Released md5sum                          \n",
       "1  2011-05-19               e5694050796ccb5eeace1e6a4efec34d\n",
       "2  2011-05-19               e2bef555c66ac7cb95dd56dc3339420f\n",
       "3  2011-02-10               f66a73f2377a38c4ba6c1d4405fd41a5\n",
       "4  2011-02-10               8e71cfdb547cdb5057d99d12d7be3ba6\n",
       "5  2011-10-07               63db47e5b9b98dbebff2ce20df066106\n",
       "6  2011-05-19               0c4c1141dfc788db377407a9531be112\n",
       "7  2016-11-08               db483a7fe9eaf56ee507f9852f0e0ca3\n",
       "8  2011-02-10               5dea2993c0831ae344a989d601c09178\n",
       "9  2011-02-10               fe4c93b6faa0ab4153ef41fe516bced4\n",
       "10 2011-02-10               4422969d0b63260e2fcb83e10fdcc02f\n",
       "11 2011-02-11               7974e4f7e3f7b1db43715efdf2532916\n",
       "12 2011-02-10               55c1093b8d98bef1dc7f788fbb44eb7c\n",
       "13 2011-02-10               0c4d48de710e88b7049fe53eb9b31dbe\n",
       "14 2011-02-10               b2db31175e0856a2b27c064da8e4bbb4\n",
       "15 2011-02-10               5ddbfd027332e8d38e801dd1eaa7575c\n",
       "16 2011-02-10               a5934e60f1ebf8e393aa111a26f74074\n",
       "17 2011-02-10               b0e6f4594be0ec60fd34e218ebc1dcaf\n",
       "18 2011-02-10               81a2d09ef690a3317f6b0e793690cfdc\n",
       "19 2011-02-11               f7616f8937225476151f4af2262dcfeb\n",
       "   URL                                                                          \n",
       "1  https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz\n",
       "2  https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz\n",
       "3  https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz\n",
       "4  https://www.encodeproject.org/files/ENCFF963GZJ/@@download/ENCFF963GZJ.bed.gz\n",
       "5  https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz\n",
       "6  https://www.encodeproject.org/files/ENCFF148UQI/@@download/ENCFF148UQI.bed.gz\n",
       "7  https://www.encodeproject.org/files/ENCFF885FQN/@@download/ENCFF885FQN.bed.gz\n",
       "8  https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz\n",
       "9  https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz\n",
       "10 https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz\n",
       "11 https://www.encodeproject.org/files/ENCFF561OUZ/@@download/ENCFF561OUZ.bed.gz\n",
       "12 https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz\n",
       "13 https://www.encodeproject.org/files/ENCFF891CHI/@@download/ENCFF891CHI.bed.gz\n",
       "14 https://www.encodeproject.org/files/ENCFF135ZLM/@@download/ENCFF135ZLM.bed.gz\n",
       "15 https://www.encodeproject.org/files/ENCFF462AVD/@@download/ENCFF462AVD.bed.gz\n",
       "16 https://www.encodeproject.org/files/ENCFF213OTI/@@download/ENCFF213OTI.bed.gz\n",
       "17 https://www.encodeproject.org/files/ENCFF209OQD/@@download/ENCFF209OQD.bed.gz\n",
       "18 https://www.encodeproject.org/files/ENCFF544LXB/@@download/ENCFF544LXB.bed.gz\n",
       "19 https://www.encodeproject.org/files/ENCFF706WUF/@@download/ENCFF706WUF.bed.gz"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_chipseq_import\n",
    "dat = dat %>% \n",
    "    dplyr::filter(`File format` == \"bed narrowPeak\") %>% \n",
    "    dplyr::mutate(`Experiment target` = str_remove(`Experiment target`, \"-human\")) %>%\n",
    "    dplyr::select(\n",
    "        `Experiment accession`,\n",
    "        `File accession`,\n",
    "        `Experiment target`,\n",
    "        `Output type`,\n",
    "        `Assay`,\n",
    "        `Biosample term name`,\n",
    "        `File analysis title`,\n",
    "        `File assembly`,\n",
    "        `Experiment date released`,\n",
    "        `md5sum`,\n",
    "        `File download URL`\n",
    "    ) \n",
    "\n",
    "colnames(dat) = c(\n",
    "    \"Experiment_Accession\",\n",
    "    \"File_Accession\",\n",
    "    \"Target\",\n",
    "    \"Output\",\n",
    "    \"Assay\",\n",
    "    \"Biosample\",\n",
    "    \"File_Analysis_Title\",\n",
    "    \"File_Assembly\",\n",
    "    \"Experiment_Data_Released\",\n",
    "    \"md5sum\",\n",
    "    \"URL\"\n",
    ")\n",
    "\n",
    "dat_chipseq_bed = dat\n",
    "print(dim(dat))\n",
    "dat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 19 13\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 19 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Experiment_Accession</th><th scope=col>File_Accession</th><th scope=col>Target</th><th scope=col>Output</th><th scope=col>Assay</th><th scope=col>Biosample</th><th scope=col>File_Analysis_Title</th><th scope=col>File_Assembly</th><th scope=col>Experiment_Data_Released</th><th scope=col>md5sum</th><th scope=col>URL</th><th scope=col>FN_OUT</th><th scope=col>CMD</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;date&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENCSR000EWA</td><td>ENCFF122CSI</td><td>H3K4me3 </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-05-19</td><td>e5694050796ccb5eeace1e6a4efec34d</td><td>https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz</td><td>K562.ENCSR000EWA.ENCFF122CSI.H3K4me3.bed.gz </td><td>wget https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz -O K562.ENCSR000EWA.ENCFF122CSI.H3K4me3.bed.gz </td></tr>\n",
       "\t<tr><td>ENCSR000EWB</td><td>ENCFF801AHF</td><td>H3K27me3</td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.0 GRCh38</td><td>GRCh38</td><td>2011-05-19</td><td>e2bef555c66ac7cb95dd56dc3339420f</td><td>https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz</td><td>K562.ENCSR000EWB.ENCFF801AHF.H3K27me3.bed.gz</td><td>wget https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz -O K562.ENCSR000EWB.ENCFF801AHF.H3K27me3.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000AKR</td><td>ENCFF193ERO</td><td>H3K36me3</td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>f66a73f2377a38c4ba6c1d4405fd41a5</td><td>https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz</td><td>K562.ENCSR000AKR.ENCFF193ERO.H3K36me3.bed.gz</td><td>wget https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz -O K562.ENCSR000AKR.ENCFF193ERO.H3K36me3.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000APE</td><td>ENCFF963GZJ</td><td>H3K9me3 </td><td>replicated peaks      </td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>8e71cfdb547cdb5057d99d12d7be3ba6</td><td>https://www.encodeproject.org/files/ENCFF963GZJ/@@download/ENCFF963GZJ.bed.gz</td><td>K562.ENCSR000APE.ENCFF963GZJ.H3K9me3.bed.gz </td><td>wget https://www.encodeproject.org/files/ENCFF963GZJ/@@download/ENCFF963GZJ.bed.gz -O K562.ENCSR000APE.ENCFF963GZJ.H3K9me3.bed.gz </td></tr>\n",
       "\t<tr><td>ENCSR000EWC</td><td>ENCFF540NGG</td><td>H3K4me1 </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-10-07</td><td>63db47e5b9b98dbebff2ce20df066106</td><td>https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz</td><td>K562.ENCSR000EWC.ENCFF540NGG.H3K4me1.bed.gz </td><td>wget https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz -O K562.ENCSR000EWC.ENCFF540NGG.H3K4me1.bed.gz </td></tr>\n",
       "\t<tr><td>ENCSR000EVZ</td><td>ENCFF148UQI</td><td>H3K9ac  </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.0 GRCh38</td><td>GRCh38</td><td>2011-05-19</td><td>0c4c1141dfc788db377407a9531be112</td><td>https://www.encodeproject.org/files/ENCFF148UQI/@@download/ENCFF148UQI.bed.gz</td><td>K562.ENCSR000EVZ.ENCFF148UQI.H3K9ac.bed.gz  </td><td>wget https://www.encodeproject.org/files/ENCFF148UQI/@@download/ENCFF148UQI.bed.gz -O K562.ENCSR000EVZ.ENCFF148UQI.H3K9ac.bed.gz  </td></tr>\n",
       "\t<tr><td>ENCSR668LDD</td><td>ENCFF885FQN</td><td>H3K4me3 </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2016-11-08</td><td>db483a7fe9eaf56ee507f9852f0e0ca3</td><td>https://www.encodeproject.org/files/ENCFF885FQN/@@download/ENCFF885FQN.bed.gz</td><td>K562.ENCSR668LDD.ENCFF885FQN.H3K4me3.bed.gz </td><td>wget https://www.encodeproject.org/files/ENCFF885FQN/@@download/ENCFF885FQN.bed.gz -O K562.ENCSR668LDD.ENCFF885FQN.H3K4me3.bed.gz </td></tr>\n",
       "\t<tr><td>ENCSR000AKU</td><td>ENCFF689QIJ</td><td>H3K4me3 </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>5dea2993c0831ae344a989d601c09178</td><td>https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz</td><td>K562.ENCSR000AKU.ENCFF689QIJ.H3K4me3.bed.gz </td><td>wget https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz -O K562.ENCSR000AKU.ENCFF689QIJ.H3K4me3.bed.gz </td></tr>\n",
       "\t<tr><td>ENCSR000AKX</td><td>ENCFF909RKY</td><td>H4K20me1</td><td>replicated peaks      </td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>fe4c93b6faa0ab4153ef41fe516bced4</td><td>https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz</td><td>K562.ENCSR000AKX.ENCFF909RKY.H4K20me1.bed.gz</td><td>wget https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz -O K562.ENCSR000AKX.ENCFF909RKY.H4K20me1.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000AKQ</td><td>ENCFF323WOT</td><td>H3K27me3</td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>4422969d0b63260e2fcb83e10fdcc02f</td><td>https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz</td><td>K562.ENCSR000AKQ.ENCFF323WOT.H3K27me3.bed.gz</td><td>wget https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz -O K562.ENCSR000AKQ.ENCFF323WOT.H3K27me3.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000DWB</td><td>ENCFF561OUZ</td><td>H3K36me3</td><td>replicated peaks      </td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-02-11</td><td>7974e4f7e3f7b1db43715efdf2532916</td><td>https://www.encodeproject.org/files/ENCFF561OUZ/@@download/ENCFF561OUZ.bed.gz</td><td>K562.ENCSR000DWB.ENCFF561OUZ.H3K36me3.bed.gz</td><td>wget https://www.encodeproject.org/files/ENCFF561OUZ/@@download/ENCFF561OUZ.bed.gz -O K562.ENCSR000DWB.ENCFF561OUZ.H3K36me3.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000AKT</td><td>ENCFF749KLQ</td><td>H3K4me2 </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>55c1093b8d98bef1dc7f788fbb44eb7c</td><td>https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz</td><td>K562.ENCSR000AKT.ENCFF749KLQ.H3K4me2.bed.gz </td><td>wget https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz -O K562.ENCSR000AKT.ENCFF749KLQ.H3K4me2.bed.gz </td></tr>\n",
       "\t<tr><td>ENCSR000AKV</td><td>ENCFF891CHI</td><td>H3K9ac  </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.1 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>0c4d48de710e88b7049fe53eb9b31dbe</td><td>https://www.encodeproject.org/files/ENCFF891CHI/@@download/ENCFF891CHI.bed.gz</td><td>K562.ENCSR000AKV.ENCFF891CHI.H3K9ac.bed.gz  </td><td>wget https://www.encodeproject.org/files/ENCFF891CHI/@@download/ENCFF891CHI.bed.gz -O K562.ENCSR000AKV.ENCFF891CHI.H3K9ac.bed.gz  </td></tr>\n",
       "\t<tr><td>ENCSR000AKS</td><td>ENCFF135ZLM</td><td>H3K4me1 </td><td>replicated peaks      </td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>b2db31175e0856a2b27c064da8e4bbb4</td><td>https://www.encodeproject.org/files/ENCFF135ZLM/@@download/ENCFF135ZLM.bed.gz</td><td>K562.ENCSR000AKS.ENCFF135ZLM.H3K4me1.bed.gz </td><td>wget https://www.encodeproject.org/files/ENCFF135ZLM/@@download/ENCFF135ZLM.bed.gz -O K562.ENCSR000AKS.ENCFF135ZLM.H3K4me1.bed.gz </td></tr>\n",
       "\t<tr><td>ENCSR000AKW</td><td>ENCFF462AVD</td><td>H3K9me1 </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>5ddbfd027332e8d38e801dd1eaa7575c</td><td>https://www.encodeproject.org/files/ENCFF462AVD/@@download/ENCFF462AVD.bed.gz</td><td>K562.ENCSR000AKW.ENCFF462AVD.H3K9me1.bed.gz </td><td>wget https://www.encodeproject.org/files/ENCFF462AVD/@@download/ENCFF462AVD.bed.gz -O K562.ENCSR000AKW.ENCFF462AVD.H3K9me1.bed.gz </td></tr>\n",
       "\t<tr><td>ENCSR000APC</td><td>ENCFF213OTI</td><td>H2AFZ   </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.6.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>a5934e60f1ebf8e393aa111a26f74074</td><td>https://www.encodeproject.org/files/ENCFF213OTI/@@download/ENCFF213OTI.bed.gz</td><td>K562.ENCSR000APC.ENCFF213OTI.H2AFZ.bed.gz   </td><td>wget https://www.encodeproject.org/files/ENCFF213OTI/@@download/ENCFF213OTI.bed.gz -O K562.ENCSR000APC.ENCFF213OTI.H2AFZ.bed.gz   </td></tr>\n",
       "\t<tr><td>ENCSR000APD</td><td>ENCFF209OQD</td><td>H3K79me2</td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>b0e6f4594be0ec60fd34e218ebc1dcaf</td><td>https://www.encodeproject.org/files/ENCFF209OQD/@@download/ENCFF209OQD.bed.gz</td><td>K562.ENCSR000APD.ENCFF209OQD.H3K79me2.bed.gz</td><td>wget https://www.encodeproject.org/files/ENCFF209OQD/@@download/ENCFF209OQD.bed.gz -O K562.ENCSR000APD.ENCFF209OQD.H3K79me2.bed.gz</td></tr>\n",
       "\t<tr><td>ENCSR000AKP</td><td>ENCFF544LXB</td><td>H3K27ac </td><td>pseudoreplicated peaks</td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.8.0 GRCh38</td><td>GRCh38</td><td>2011-02-10</td><td>81a2d09ef690a3317f6b0e793690cfdc</td><td>https://www.encodeproject.org/files/ENCFF544LXB/@@download/ENCFF544LXB.bed.gz</td><td>K562.ENCSR000AKP.ENCFF544LXB.H3K27ac.bed.gz </td><td>wget https://www.encodeproject.org/files/ENCFF544LXB/@@download/ENCFF544LXB.bed.gz -O K562.ENCSR000AKP.ENCFF544LXB.H3K27ac.bed.gz </td></tr>\n",
       "\t<tr><td>ENCSR000DWD</td><td>ENCFF706WUF</td><td>H3K4me3 </td><td>replicated peaks      </td><td>Histone ChIP-seq</td><td>K562</td><td>ENCODE4 v1.5.1 GRCh38</td><td>GRCh38</td><td>2011-02-11</td><td>f7616f8937225476151f4af2262dcfeb</td><td>https://www.encodeproject.org/files/ENCFF706WUF/@@download/ENCFF706WUF.bed.gz</td><td>K562.ENCSR000DWD.ENCFF706WUF.H3K4me3.bed.gz </td><td>wget https://www.encodeproject.org/files/ENCFF706WUF/@@download/ENCFF706WUF.bed.gz -O K562.ENCSR000DWD.ENCFF706WUF.H3K4me3.bed.gz </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 19 × 13\n",
       "\\begin{tabular}{lllllllllllll}\n",
       " Experiment\\_Accession & File\\_Accession & Target & Output & Assay & Biosample & File\\_Analysis\\_Title & File\\_Assembly & Experiment\\_Data\\_Released & md5sum & URL & FN\\_OUT & CMD\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <date> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENCSR000EWA & ENCFF122CSI & H3K4me3  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-05-19 & e5694050796ccb5eeace1e6a4efec34d & https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz & K562.ENCSR000EWA.ENCFF122CSI.H3K4me3.bed.gz  & wget https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz -O K562.ENCSR000EWA.ENCFF122CSI.H3K4me3.bed.gz \\\\\n",
       "\t ENCSR000EWB & ENCFF801AHF & H3K27me3 & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.6.0 GRCh38 & GRCh38 & 2011-05-19 & e2bef555c66ac7cb95dd56dc3339420f & https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz & K562.ENCSR000EWB.ENCFF801AHF.H3K27me3.bed.gz & wget https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz -O K562.ENCSR000EWB.ENCFF801AHF.H3K27me3.bed.gz\\\\\n",
       "\t ENCSR000AKR & ENCFF193ERO & H3K36me3 & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & f66a73f2377a38c4ba6c1d4405fd41a5 & https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz & K562.ENCSR000AKR.ENCFF193ERO.H3K36me3.bed.gz & wget https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz -O K562.ENCSR000AKR.ENCFF193ERO.H3K36me3.bed.gz\\\\\n",
       "\t ENCSR000APE & ENCFF963GZJ & H3K9me3  & replicated peaks       & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-02-10 & 8e71cfdb547cdb5057d99d12d7be3ba6 & https://www.encodeproject.org/files/ENCFF963GZJ/@@download/ENCFF963GZJ.bed.gz & K562.ENCSR000APE.ENCFF963GZJ.H3K9me3.bed.gz  & wget https://www.encodeproject.org/files/ENCFF963GZJ/@@download/ENCFF963GZJ.bed.gz -O K562.ENCSR000APE.ENCFF963GZJ.H3K9me3.bed.gz \\\\\n",
       "\t ENCSR000EWC & ENCFF540NGG & H3K4me1  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-10-07 & 63db47e5b9b98dbebff2ce20df066106 & https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz & K562.ENCSR000EWC.ENCFF540NGG.H3K4me1.bed.gz  & wget https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz -O K562.ENCSR000EWC.ENCFF540NGG.H3K4me1.bed.gz \\\\\n",
       "\t ENCSR000EVZ & ENCFF148UQI & H3K9ac   & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.6.0 GRCh38 & GRCh38 & 2011-05-19 & 0c4c1141dfc788db377407a9531be112 & https://www.encodeproject.org/files/ENCFF148UQI/@@download/ENCFF148UQI.bed.gz & K562.ENCSR000EVZ.ENCFF148UQI.H3K9ac.bed.gz   & wget https://www.encodeproject.org/files/ENCFF148UQI/@@download/ENCFF148UQI.bed.gz -O K562.ENCSR000EVZ.ENCFF148UQI.H3K9ac.bed.gz  \\\\\n",
       "\t ENCSR668LDD & ENCFF885FQN & H3K4me3  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2016-11-08 & db483a7fe9eaf56ee507f9852f0e0ca3 & https://www.encodeproject.org/files/ENCFF885FQN/@@download/ENCFF885FQN.bed.gz & K562.ENCSR668LDD.ENCFF885FQN.H3K4me3.bed.gz  & wget https://www.encodeproject.org/files/ENCFF885FQN/@@download/ENCFF885FQN.bed.gz -O K562.ENCSR668LDD.ENCFF885FQN.H3K4me3.bed.gz \\\\\n",
       "\t ENCSR000AKU & ENCFF689QIJ & H3K4me3  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-02-10 & 5dea2993c0831ae344a989d601c09178 & https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz & K562.ENCSR000AKU.ENCFF689QIJ.H3K4me3.bed.gz  & wget https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz -O K562.ENCSR000AKU.ENCFF689QIJ.H3K4me3.bed.gz \\\\\n",
       "\t ENCSR000AKX & ENCFF909RKY & H4K20me1 & replicated peaks       & Histone ChIP-seq & K562 & ENCODE4 v1.6.1 GRCh38 & GRCh38 & 2011-02-10 & fe4c93b6faa0ab4153ef41fe516bced4 & https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz & K562.ENCSR000AKX.ENCFF909RKY.H4K20me1.bed.gz & wget https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz -O K562.ENCSR000AKX.ENCFF909RKY.H4K20me1.bed.gz\\\\\n",
       "\t ENCSR000AKQ & ENCFF323WOT & H3K27me3 & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & 4422969d0b63260e2fcb83e10fdcc02f & https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz & K562.ENCSR000AKQ.ENCFF323WOT.H3K27me3.bed.gz & wget https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz -O K562.ENCSR000AKQ.ENCFF323WOT.H3K27me3.bed.gz\\\\\n",
       "\t ENCSR000DWB & ENCFF561OUZ & H3K36me3 & replicated peaks       & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-02-11 & 7974e4f7e3f7b1db43715efdf2532916 & https://www.encodeproject.org/files/ENCFF561OUZ/@@download/ENCFF561OUZ.bed.gz & K562.ENCSR000DWB.ENCFF561OUZ.H3K36me3.bed.gz & wget https://www.encodeproject.org/files/ENCFF561OUZ/@@download/ENCFF561OUZ.bed.gz -O K562.ENCSR000DWB.ENCFF561OUZ.H3K36me3.bed.gz\\\\\n",
       "\t ENCSR000AKT & ENCFF749KLQ & H3K4me2  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.6.1 GRCh38 & GRCh38 & 2011-02-10 & 55c1093b8d98bef1dc7f788fbb44eb7c & https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz & K562.ENCSR000AKT.ENCFF749KLQ.H3K4me2.bed.gz  & wget https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz -O K562.ENCSR000AKT.ENCFF749KLQ.H3K4me2.bed.gz \\\\\n",
       "\t ENCSR000AKV & ENCFF891CHI & H3K9ac   & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.6.1 GRCh38 & GRCh38 & 2011-02-10 & 0c4d48de710e88b7049fe53eb9b31dbe & https://www.encodeproject.org/files/ENCFF891CHI/@@download/ENCFF891CHI.bed.gz & K562.ENCSR000AKV.ENCFF891CHI.H3K9ac.bed.gz   & wget https://www.encodeproject.org/files/ENCFF891CHI/@@download/ENCFF891CHI.bed.gz -O K562.ENCSR000AKV.ENCFF891CHI.H3K9ac.bed.gz  \\\\\n",
       "\t ENCSR000AKS & ENCFF135ZLM & H3K4me1  & replicated peaks       & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & b2db31175e0856a2b27c064da8e4bbb4 & https://www.encodeproject.org/files/ENCFF135ZLM/@@download/ENCFF135ZLM.bed.gz & K562.ENCSR000AKS.ENCFF135ZLM.H3K4me1.bed.gz  & wget https://www.encodeproject.org/files/ENCFF135ZLM/@@download/ENCFF135ZLM.bed.gz -O K562.ENCSR000AKS.ENCFF135ZLM.H3K4me1.bed.gz \\\\\n",
       "\t ENCSR000AKW & ENCFF462AVD & H3K9me1  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.6.0 GRCh38 & GRCh38 & 2011-02-10 & 5ddbfd027332e8d38e801dd1eaa7575c & https://www.encodeproject.org/files/ENCFF462AVD/@@download/ENCFF462AVD.bed.gz & K562.ENCSR000AKW.ENCFF462AVD.H3K9me1.bed.gz  & wget https://www.encodeproject.org/files/ENCFF462AVD/@@download/ENCFF462AVD.bed.gz -O K562.ENCSR000AKW.ENCFF462AVD.H3K9me1.bed.gz \\\\\n",
       "\t ENCSR000APC & ENCFF213OTI & H2AFZ    & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.6.0 GRCh38 & GRCh38 & 2011-02-10 & a5934e60f1ebf8e393aa111a26f74074 & https://www.encodeproject.org/files/ENCFF213OTI/@@download/ENCFF213OTI.bed.gz & K562.ENCSR000APC.ENCFF213OTI.H2AFZ.bed.gz    & wget https://www.encodeproject.org/files/ENCFF213OTI/@@download/ENCFF213OTI.bed.gz -O K562.ENCSR000APC.ENCFF213OTI.H2AFZ.bed.gz   \\\\\n",
       "\t ENCSR000APD & ENCFF209OQD & H3K79me2 & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & b0e6f4594be0ec60fd34e218ebc1dcaf & https://www.encodeproject.org/files/ENCFF209OQD/@@download/ENCFF209OQD.bed.gz & K562.ENCSR000APD.ENCFF209OQD.H3K79me2.bed.gz & wget https://www.encodeproject.org/files/ENCFF209OQD/@@download/ENCFF209OQD.bed.gz -O K562.ENCSR000APD.ENCFF209OQD.H3K79me2.bed.gz\\\\\n",
       "\t ENCSR000AKP & ENCFF544LXB & H3K27ac  & pseudoreplicated peaks & Histone ChIP-seq & K562 & ENCODE4 v1.8.0 GRCh38 & GRCh38 & 2011-02-10 & 81a2d09ef690a3317f6b0e793690cfdc & https://www.encodeproject.org/files/ENCFF544LXB/@@download/ENCFF544LXB.bed.gz & K562.ENCSR000AKP.ENCFF544LXB.H3K27ac.bed.gz  & wget https://www.encodeproject.org/files/ENCFF544LXB/@@download/ENCFF544LXB.bed.gz -O K562.ENCSR000AKP.ENCFF544LXB.H3K27ac.bed.gz \\\\\n",
       "\t ENCSR000DWD & ENCFF706WUF & H3K4me3  & replicated peaks       & Histone ChIP-seq & K562 & ENCODE4 v1.5.1 GRCh38 & GRCh38 & 2011-02-11 & f7616f8937225476151f4af2262dcfeb & https://www.encodeproject.org/files/ENCFF706WUF/@@download/ENCFF706WUF.bed.gz & K562.ENCSR000DWD.ENCFF706WUF.H3K4me3.bed.gz  & wget https://www.encodeproject.org/files/ENCFF706WUF/@@download/ENCFF706WUF.bed.gz -O K562.ENCSR000DWD.ENCFF706WUF.H3K4me3.bed.gz \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 19 × 13\n",
       "\n",
       "| Experiment_Accession &lt;chr&gt; | File_Accession &lt;chr&gt; | Target &lt;chr&gt; | Output &lt;chr&gt; | Assay &lt;chr&gt; | Biosample &lt;chr&gt; | File_Analysis_Title &lt;chr&gt; | File_Assembly &lt;chr&gt; | Experiment_Data_Released &lt;date&gt; | md5sum &lt;chr&gt; | URL &lt;chr&gt; | FN_OUT &lt;chr&gt; | CMD &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENCSR000EWA | ENCFF122CSI | H3K4me3  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-05-19 | e5694050796ccb5eeace1e6a4efec34d | https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz | K562.ENCSR000EWA.ENCFF122CSI.H3K4me3.bed.gz  | wget https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz -O K562.ENCSR000EWA.ENCFF122CSI.H3K4me3.bed.gz  |\n",
       "| ENCSR000EWB | ENCFF801AHF | H3K27me3 | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.6.0 GRCh38 | GRCh38 | 2011-05-19 | e2bef555c66ac7cb95dd56dc3339420f | https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz | K562.ENCSR000EWB.ENCFF801AHF.H3K27me3.bed.gz | wget https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz -O K562.ENCSR000EWB.ENCFF801AHF.H3K27me3.bed.gz |\n",
       "| ENCSR000AKR | ENCFF193ERO | H3K36me3 | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | f66a73f2377a38c4ba6c1d4405fd41a5 | https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz | K562.ENCSR000AKR.ENCFF193ERO.H3K36me3.bed.gz | wget https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz -O K562.ENCSR000AKR.ENCFF193ERO.H3K36me3.bed.gz |\n",
       "| ENCSR000APE | ENCFF963GZJ | H3K9me3  | replicated peaks       | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-02-10 | 8e71cfdb547cdb5057d99d12d7be3ba6 | https://www.encodeproject.org/files/ENCFF963GZJ/@@download/ENCFF963GZJ.bed.gz | K562.ENCSR000APE.ENCFF963GZJ.H3K9me3.bed.gz  | wget https://www.encodeproject.org/files/ENCFF963GZJ/@@download/ENCFF963GZJ.bed.gz -O K562.ENCSR000APE.ENCFF963GZJ.H3K9me3.bed.gz  |\n",
       "| ENCSR000EWC | ENCFF540NGG | H3K4me1  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-10-07 | 63db47e5b9b98dbebff2ce20df066106 | https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz | K562.ENCSR000EWC.ENCFF540NGG.H3K4me1.bed.gz  | wget https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz -O K562.ENCSR000EWC.ENCFF540NGG.H3K4me1.bed.gz  |\n",
       "| ENCSR000EVZ | ENCFF148UQI | H3K9ac   | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.6.0 GRCh38 | GRCh38 | 2011-05-19 | 0c4c1141dfc788db377407a9531be112 | https://www.encodeproject.org/files/ENCFF148UQI/@@download/ENCFF148UQI.bed.gz | K562.ENCSR000EVZ.ENCFF148UQI.H3K9ac.bed.gz   | wget https://www.encodeproject.org/files/ENCFF148UQI/@@download/ENCFF148UQI.bed.gz -O K562.ENCSR000EVZ.ENCFF148UQI.H3K9ac.bed.gz   |\n",
       "| ENCSR668LDD | ENCFF885FQN | H3K4me3  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2016-11-08 | db483a7fe9eaf56ee507f9852f0e0ca3 | https://www.encodeproject.org/files/ENCFF885FQN/@@download/ENCFF885FQN.bed.gz | K562.ENCSR668LDD.ENCFF885FQN.H3K4me3.bed.gz  | wget https://www.encodeproject.org/files/ENCFF885FQN/@@download/ENCFF885FQN.bed.gz -O K562.ENCSR668LDD.ENCFF885FQN.H3K4me3.bed.gz  |\n",
       "| ENCSR000AKU | ENCFF689QIJ | H3K4me3  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-02-10 | 5dea2993c0831ae344a989d601c09178 | https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz | K562.ENCSR000AKU.ENCFF689QIJ.H3K4me3.bed.gz  | wget https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz -O K562.ENCSR000AKU.ENCFF689QIJ.H3K4me3.bed.gz  |\n",
       "| ENCSR000AKX | ENCFF909RKY | H4K20me1 | replicated peaks       | Histone ChIP-seq | K562 | ENCODE4 v1.6.1 GRCh38 | GRCh38 | 2011-02-10 | fe4c93b6faa0ab4153ef41fe516bced4 | https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz | K562.ENCSR000AKX.ENCFF909RKY.H4K20me1.bed.gz | wget https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz -O K562.ENCSR000AKX.ENCFF909RKY.H4K20me1.bed.gz |\n",
       "| ENCSR000AKQ | ENCFF323WOT | H3K27me3 | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | 4422969d0b63260e2fcb83e10fdcc02f | https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz | K562.ENCSR000AKQ.ENCFF323WOT.H3K27me3.bed.gz | wget https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz -O K562.ENCSR000AKQ.ENCFF323WOT.H3K27me3.bed.gz |\n",
       "| ENCSR000DWB | ENCFF561OUZ | H3K36me3 | replicated peaks       | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-02-11 | 7974e4f7e3f7b1db43715efdf2532916 | https://www.encodeproject.org/files/ENCFF561OUZ/@@download/ENCFF561OUZ.bed.gz | K562.ENCSR000DWB.ENCFF561OUZ.H3K36me3.bed.gz | wget https://www.encodeproject.org/files/ENCFF561OUZ/@@download/ENCFF561OUZ.bed.gz -O K562.ENCSR000DWB.ENCFF561OUZ.H3K36me3.bed.gz |\n",
       "| ENCSR000AKT | ENCFF749KLQ | H3K4me2  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.6.1 GRCh38 | GRCh38 | 2011-02-10 | 55c1093b8d98bef1dc7f788fbb44eb7c | https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz | K562.ENCSR000AKT.ENCFF749KLQ.H3K4me2.bed.gz  | wget https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz -O K562.ENCSR000AKT.ENCFF749KLQ.H3K4me2.bed.gz  |\n",
       "| ENCSR000AKV | ENCFF891CHI | H3K9ac   | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.6.1 GRCh38 | GRCh38 | 2011-02-10 | 0c4d48de710e88b7049fe53eb9b31dbe | https://www.encodeproject.org/files/ENCFF891CHI/@@download/ENCFF891CHI.bed.gz | K562.ENCSR000AKV.ENCFF891CHI.H3K9ac.bed.gz   | wget https://www.encodeproject.org/files/ENCFF891CHI/@@download/ENCFF891CHI.bed.gz -O K562.ENCSR000AKV.ENCFF891CHI.H3K9ac.bed.gz   |\n",
       "| ENCSR000AKS | ENCFF135ZLM | H3K4me1  | replicated peaks       | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | b2db31175e0856a2b27c064da8e4bbb4 | https://www.encodeproject.org/files/ENCFF135ZLM/@@download/ENCFF135ZLM.bed.gz | K562.ENCSR000AKS.ENCFF135ZLM.H3K4me1.bed.gz  | wget https://www.encodeproject.org/files/ENCFF135ZLM/@@download/ENCFF135ZLM.bed.gz -O K562.ENCSR000AKS.ENCFF135ZLM.H3K4me1.bed.gz  |\n",
       "| ENCSR000AKW | ENCFF462AVD | H3K9me1  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.6.0 GRCh38 | GRCh38 | 2011-02-10 | 5ddbfd027332e8d38e801dd1eaa7575c | https://www.encodeproject.org/files/ENCFF462AVD/@@download/ENCFF462AVD.bed.gz | K562.ENCSR000AKW.ENCFF462AVD.H3K9me1.bed.gz  | wget https://www.encodeproject.org/files/ENCFF462AVD/@@download/ENCFF462AVD.bed.gz -O K562.ENCSR000AKW.ENCFF462AVD.H3K9me1.bed.gz  |\n",
       "| ENCSR000APC | ENCFF213OTI | H2AFZ    | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.6.0 GRCh38 | GRCh38 | 2011-02-10 | a5934e60f1ebf8e393aa111a26f74074 | https://www.encodeproject.org/files/ENCFF213OTI/@@download/ENCFF213OTI.bed.gz | K562.ENCSR000APC.ENCFF213OTI.H2AFZ.bed.gz    | wget https://www.encodeproject.org/files/ENCFF213OTI/@@download/ENCFF213OTI.bed.gz -O K562.ENCSR000APC.ENCFF213OTI.H2AFZ.bed.gz    |\n",
       "| ENCSR000APD | ENCFF209OQD | H3K79me2 | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | b0e6f4594be0ec60fd34e218ebc1dcaf | https://www.encodeproject.org/files/ENCFF209OQD/@@download/ENCFF209OQD.bed.gz | K562.ENCSR000APD.ENCFF209OQD.H3K79me2.bed.gz | wget https://www.encodeproject.org/files/ENCFF209OQD/@@download/ENCFF209OQD.bed.gz -O K562.ENCSR000APD.ENCFF209OQD.H3K79me2.bed.gz |\n",
       "| ENCSR000AKP | ENCFF544LXB | H3K27ac  | pseudoreplicated peaks | Histone ChIP-seq | K562 | ENCODE4 v1.8.0 GRCh38 | GRCh38 | 2011-02-10 | 81a2d09ef690a3317f6b0e793690cfdc | https://www.encodeproject.org/files/ENCFF544LXB/@@download/ENCFF544LXB.bed.gz | K562.ENCSR000AKP.ENCFF544LXB.H3K27ac.bed.gz  | wget https://www.encodeproject.org/files/ENCFF544LXB/@@download/ENCFF544LXB.bed.gz -O K562.ENCSR000AKP.ENCFF544LXB.H3K27ac.bed.gz  |\n",
       "| ENCSR000DWD | ENCFF706WUF | H3K4me3  | replicated peaks       | Histone ChIP-seq | K562 | ENCODE4 v1.5.1 GRCh38 | GRCh38 | 2011-02-11 | f7616f8937225476151f4af2262dcfeb | https://www.encodeproject.org/files/ENCFF706WUF/@@download/ENCFF706WUF.bed.gz | K562.ENCSR000DWD.ENCFF706WUF.H3K4me3.bed.gz  | wget https://www.encodeproject.org/files/ENCFF706WUF/@@download/ENCFF706WUF.bed.gz -O K562.ENCSR000DWD.ENCFF706WUF.H3K4me3.bed.gz  |\n",
       "\n"
      ],
      "text/plain": [
       "   Experiment_Accession File_Accession Target   Output                \n",
       "1  ENCSR000EWA          ENCFF122CSI    H3K4me3  pseudoreplicated peaks\n",
       "2  ENCSR000EWB          ENCFF801AHF    H3K27me3 pseudoreplicated peaks\n",
       "3  ENCSR000AKR          ENCFF193ERO    H3K36me3 pseudoreplicated peaks\n",
       "4  ENCSR000APE          ENCFF963GZJ    H3K9me3  replicated peaks      \n",
       "5  ENCSR000EWC          ENCFF540NGG    H3K4me1  pseudoreplicated peaks\n",
       "6  ENCSR000EVZ          ENCFF148UQI    H3K9ac   pseudoreplicated peaks\n",
       "7  ENCSR668LDD          ENCFF885FQN    H3K4me3  pseudoreplicated peaks\n",
       "8  ENCSR000AKU          ENCFF689QIJ    H3K4me3  pseudoreplicated peaks\n",
       "9  ENCSR000AKX          ENCFF909RKY    H4K20me1 replicated peaks      \n",
       "10 ENCSR000AKQ          ENCFF323WOT    H3K27me3 pseudoreplicated peaks\n",
       "11 ENCSR000DWB          ENCFF561OUZ    H3K36me3 replicated peaks      \n",
       "12 ENCSR000AKT          ENCFF749KLQ    H3K4me2  pseudoreplicated peaks\n",
       "13 ENCSR000AKV          ENCFF891CHI    H3K9ac   pseudoreplicated peaks\n",
       "14 ENCSR000AKS          ENCFF135ZLM    H3K4me1  replicated peaks      \n",
       "15 ENCSR000AKW          ENCFF462AVD    H3K9me1  pseudoreplicated peaks\n",
       "16 ENCSR000APC          ENCFF213OTI    H2AFZ    pseudoreplicated peaks\n",
       "17 ENCSR000APD          ENCFF209OQD    H3K79me2 pseudoreplicated peaks\n",
       "18 ENCSR000AKP          ENCFF544LXB    H3K27ac  pseudoreplicated peaks\n",
       "19 ENCSR000DWD          ENCFF706WUF    H3K4me3  replicated peaks      \n",
       "   Assay            Biosample File_Analysis_Title   File_Assembly\n",
       "1  Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "2  Histone ChIP-seq K562      ENCODE4 v1.6.0 GRCh38 GRCh38       \n",
       "3  Histone ChIP-seq K562      ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "4  Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "5  Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "6  Histone ChIP-seq K562      ENCODE4 v1.6.0 GRCh38 GRCh38       \n",
       "7  Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "8  Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "9  Histone ChIP-seq K562      ENCODE4 v1.6.1 GRCh38 GRCh38       \n",
       "10 Histone ChIP-seq K562      ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "11 Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "12 Histone ChIP-seq K562      ENCODE4 v1.6.1 GRCh38 GRCh38       \n",
       "13 Histone ChIP-seq K562      ENCODE4 v1.6.1 GRCh38 GRCh38       \n",
       "14 Histone ChIP-seq K562      ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "15 Histone ChIP-seq K562      ENCODE4 v1.6.0 GRCh38 GRCh38       \n",
       "16 Histone ChIP-seq K562      ENCODE4 v1.6.0 GRCh38 GRCh38       \n",
       "17 Histone ChIP-seq K562      ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "18 Histone ChIP-seq K562      ENCODE4 v1.8.0 GRCh38 GRCh38       \n",
       "19 Histone ChIP-seq K562      ENCODE4 v1.5.1 GRCh38 GRCh38       \n",
       "   Experiment_Data_Released md5sum                          \n",
       "1  2011-05-19               e5694050796ccb5eeace1e6a4efec34d\n",
       "2  2011-05-19               e2bef555c66ac7cb95dd56dc3339420f\n",
       "3  2011-02-10               f66a73f2377a38c4ba6c1d4405fd41a5\n",
       "4  2011-02-10               8e71cfdb547cdb5057d99d12d7be3ba6\n",
       "5  2011-10-07               63db47e5b9b98dbebff2ce20df066106\n",
       "6  2011-05-19               0c4c1141dfc788db377407a9531be112\n",
       "7  2016-11-08               db483a7fe9eaf56ee507f9852f0e0ca3\n",
       "8  2011-02-10               5dea2993c0831ae344a989d601c09178\n",
       "9  2011-02-10               fe4c93b6faa0ab4153ef41fe516bced4\n",
       "10 2011-02-10               4422969d0b63260e2fcb83e10fdcc02f\n",
       "11 2011-02-11               7974e4f7e3f7b1db43715efdf2532916\n",
       "12 2011-02-10               55c1093b8d98bef1dc7f788fbb44eb7c\n",
       "13 2011-02-10               0c4d48de710e88b7049fe53eb9b31dbe\n",
       "14 2011-02-10               b2db31175e0856a2b27c064da8e4bbb4\n",
       "15 2011-02-10               5ddbfd027332e8d38e801dd1eaa7575c\n",
       "16 2011-02-10               a5934e60f1ebf8e393aa111a26f74074\n",
       "17 2011-02-10               b0e6f4594be0ec60fd34e218ebc1dcaf\n",
       "18 2011-02-10               81a2d09ef690a3317f6b0e793690cfdc\n",
       "19 2011-02-11               f7616f8937225476151f4af2262dcfeb\n",
       "   URL                                                                          \n",
       "1  https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz\n",
       "2  https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz\n",
       "3  https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz\n",
       "4  https://www.encodeproject.org/files/ENCFF963GZJ/@@download/ENCFF963GZJ.bed.gz\n",
       "5  https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz\n",
       "6  https://www.encodeproject.org/files/ENCFF148UQI/@@download/ENCFF148UQI.bed.gz\n",
       "7  https://www.encodeproject.org/files/ENCFF885FQN/@@download/ENCFF885FQN.bed.gz\n",
       "8  https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz\n",
       "9  https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz\n",
       "10 https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz\n",
       "11 https://www.encodeproject.org/files/ENCFF561OUZ/@@download/ENCFF561OUZ.bed.gz\n",
       "12 https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz\n",
       "13 https://www.encodeproject.org/files/ENCFF891CHI/@@download/ENCFF891CHI.bed.gz\n",
       "14 https://www.encodeproject.org/files/ENCFF135ZLM/@@download/ENCFF135ZLM.bed.gz\n",
       "15 https://www.encodeproject.org/files/ENCFF462AVD/@@download/ENCFF462AVD.bed.gz\n",
       "16 https://www.encodeproject.org/files/ENCFF213OTI/@@download/ENCFF213OTI.bed.gz\n",
       "17 https://www.encodeproject.org/files/ENCFF209OQD/@@download/ENCFF209OQD.bed.gz\n",
       "18 https://www.encodeproject.org/files/ENCFF544LXB/@@download/ENCFF544LXB.bed.gz\n",
       "19 https://www.encodeproject.org/files/ENCFF706WUF/@@download/ENCFF706WUF.bed.gz\n",
       "   FN_OUT                                      \n",
       "1  K562.ENCSR000EWA.ENCFF122CSI.H3K4me3.bed.gz \n",
       "2  K562.ENCSR000EWB.ENCFF801AHF.H3K27me3.bed.gz\n",
       "3  K562.ENCSR000AKR.ENCFF193ERO.H3K36me3.bed.gz\n",
       "4  K562.ENCSR000APE.ENCFF963GZJ.H3K9me3.bed.gz \n",
       "5  K562.ENCSR000EWC.ENCFF540NGG.H3K4me1.bed.gz \n",
       "6  K562.ENCSR000EVZ.ENCFF148UQI.H3K9ac.bed.gz  \n",
       "7  K562.ENCSR668LDD.ENCFF885FQN.H3K4me3.bed.gz \n",
       "8  K562.ENCSR000AKU.ENCFF689QIJ.H3K4me3.bed.gz \n",
       "9  K562.ENCSR000AKX.ENCFF909RKY.H4K20me1.bed.gz\n",
       "10 K562.ENCSR000AKQ.ENCFF323WOT.H3K27me3.bed.gz\n",
       "11 K562.ENCSR000DWB.ENCFF561OUZ.H3K36me3.bed.gz\n",
       "12 K562.ENCSR000AKT.ENCFF749KLQ.H3K4me2.bed.gz \n",
       "13 K562.ENCSR000AKV.ENCFF891CHI.H3K9ac.bed.gz  \n",
       "14 K562.ENCSR000AKS.ENCFF135ZLM.H3K4me1.bed.gz \n",
       "15 K562.ENCSR000AKW.ENCFF462AVD.H3K9me1.bed.gz \n",
       "16 K562.ENCSR000APC.ENCFF213OTI.H2AFZ.bed.gz   \n",
       "17 K562.ENCSR000APD.ENCFF209OQD.H3K79me2.bed.gz\n",
       "18 K562.ENCSR000AKP.ENCFF544LXB.H3K27ac.bed.gz \n",
       "19 K562.ENCSR000DWD.ENCFF706WUF.H3K4me3.bed.gz \n",
       "   CMD                                                                                                                               \n",
       "1  wget https://www.encodeproject.org/files/ENCFF122CSI/@@download/ENCFF122CSI.bed.gz -O K562.ENCSR000EWA.ENCFF122CSI.H3K4me3.bed.gz \n",
       "2  wget https://www.encodeproject.org/files/ENCFF801AHF/@@download/ENCFF801AHF.bed.gz -O K562.ENCSR000EWB.ENCFF801AHF.H3K27me3.bed.gz\n",
       "3  wget https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz -O K562.ENCSR000AKR.ENCFF193ERO.H3K36me3.bed.gz\n",
       "4  wget https://www.encodeproject.org/files/ENCFF963GZJ/@@download/ENCFF963GZJ.bed.gz -O K562.ENCSR000APE.ENCFF963GZJ.H3K9me3.bed.gz \n",
       "5  wget https://www.encodeproject.org/files/ENCFF540NGG/@@download/ENCFF540NGG.bed.gz -O K562.ENCSR000EWC.ENCFF540NGG.H3K4me1.bed.gz \n",
       "6  wget https://www.encodeproject.org/files/ENCFF148UQI/@@download/ENCFF148UQI.bed.gz -O K562.ENCSR000EVZ.ENCFF148UQI.H3K9ac.bed.gz  \n",
       "7  wget https://www.encodeproject.org/files/ENCFF885FQN/@@download/ENCFF885FQN.bed.gz -O K562.ENCSR668LDD.ENCFF885FQN.H3K4me3.bed.gz \n",
       "8  wget https://www.encodeproject.org/files/ENCFF689QIJ/@@download/ENCFF689QIJ.bed.gz -O K562.ENCSR000AKU.ENCFF689QIJ.H3K4me3.bed.gz \n",
       "9  wget https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz -O K562.ENCSR000AKX.ENCFF909RKY.H4K20me1.bed.gz\n",
       "10 wget https://www.encodeproject.org/files/ENCFF323WOT/@@download/ENCFF323WOT.bed.gz -O K562.ENCSR000AKQ.ENCFF323WOT.H3K27me3.bed.gz\n",
       "11 wget https://www.encodeproject.org/files/ENCFF561OUZ/@@download/ENCFF561OUZ.bed.gz -O K562.ENCSR000DWB.ENCFF561OUZ.H3K36me3.bed.gz\n",
       "12 wget https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz -O K562.ENCSR000AKT.ENCFF749KLQ.H3K4me2.bed.gz \n",
       "13 wget https://www.encodeproject.org/files/ENCFF891CHI/@@download/ENCFF891CHI.bed.gz -O K562.ENCSR000AKV.ENCFF891CHI.H3K9ac.bed.gz  \n",
       "14 wget https://www.encodeproject.org/files/ENCFF135ZLM/@@download/ENCFF135ZLM.bed.gz -O K562.ENCSR000AKS.ENCFF135ZLM.H3K4me1.bed.gz \n",
       "15 wget https://www.encodeproject.org/files/ENCFF462AVD/@@download/ENCFF462AVD.bed.gz -O K562.ENCSR000AKW.ENCFF462AVD.H3K9me1.bed.gz \n",
       "16 wget https://www.encodeproject.org/files/ENCFF213OTI/@@download/ENCFF213OTI.bed.gz -O K562.ENCSR000APC.ENCFF213OTI.H2AFZ.bed.gz   \n",
       "17 wget https://www.encodeproject.org/files/ENCFF209OQD/@@download/ENCFF209OQD.bed.gz -O K562.ENCSR000APD.ENCFF209OQD.H3K79me2.bed.gz\n",
       "18 wget https://www.encodeproject.org/files/ENCFF544LXB/@@download/ENCFF544LXB.bed.gz -O K562.ENCSR000AKP.ENCFF544LXB.H3K27ac.bed.gz \n",
       "19 wget https://www.encodeproject.org/files/ENCFF706WUF/@@download/ENCFF706WUF.bed.gz -O K562.ENCSR000DWD.ENCFF706WUF.H3K4me3.bed.gz "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_chipseq_bed\n",
    "dat = dat %>%\n",
    "    dplyr::mutate(\n",
    "        FN_OUT = paste(\n",
    "            \"K562\", \n",
    "            Experiment_Accession, \n",
    "            File_Accession,\n",
    "            Target,\n",
    "            \"bed.gz\",\n",
    "            sep = \".\")) %>%\n",
    "    dplyr::mutate(\n",
    "        CMD = paste(\n",
    "            \"wget\",\n",
    "            URL,\n",
    "            \"-O\",\n",
    "            FN_OUT))\n",
    "\n",
    "dat_chipseq_bed_cmd = dat\n",
    "print(dim(dat))\n",
    "dat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "fdiry = file.path(FD_RES, \"results\", \"region\", \"annotation_chipseq_histone\")\n",
    "fname = \"accession_info_peaks.tsv\"\n",
    "fpath = file.path(fdiry, fname)\n",
    "\n",
    "dat = dat_chipseq_bed_cmd\n",
    "write_tsv(dat, fpath)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
