{
 "cells": [
  {
   "cell_type": "raw",
   "id": "23e50943-a8a4-4e44-8132-9c300fb512e1",
   "metadata": {},
   "source": [
    "---\n",
    "title: Prepare ENCODE RNA-seq 01\n",
    "subtitle: Generate download script to download the data\n",
    "format:\n",
    "    html:\n",
    "        code-fold:  true\n",
    "        code-tools: true\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "515b5303-7e3f-4c54-8317-87dbb9d96151",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a00a4b22-6ee0-414b-90b9-26945bacf0a6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are working on        Singularity: singularity_proj_encode_fcc \n",
      "BASE DIRECTORY (FD_BASE): /data/reddylab/Kuei \n",
      "REPO DIRECTORY (FD_REPO): /data/reddylab/Kuei/repo \n",
      "WORK DIRECTORY (FD_WORK): /data/reddylab/Kuei/work \n",
      "DATA DIRECTORY (FD_DATA): /data/reddylab/Kuei/data \n",
      "\n",
      "You are working with      ENCODE FCC \n",
      "PATH OF PROJECT (FD_PRJ): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC \n",
      "PROJECT RESULTS (FD_RES): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/results \n",
      "PROJECT SCRIPTS (FD_EXE): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/scripts \n",
      "PROJECT DATA    (FD_DAT): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/data \n",
      "PROJECT NOTE    (FD_NBK): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/notebooks \n",
      "PROJECT DOCS    (FD_DOC): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/docs \n",
      "PROJECT LOG     (FD_LOG): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/log \n",
      "PROJECT REF     (FD_REF): /data/reddylab/Kuei/repo/Proj_ENCODE_FCC/references \n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(source(\"../run_config_project_sing.R\")))\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef475fc7-2794-4f5e-a730-4583260268ce",
   "metadata": {},
   "source": [
    "**Set global variables**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b9567b45-b5dd-4410-a912-6c89c83f46ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "TXT_FOLDER_REF = \"encode_rnaseq\"\n",
    "TXT_FOLDER_OUT = \"encode_rnaseq\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b1d023a3-eb80-4d14-9a2a-689baf8baa7e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'ENCODE_K562_hg38_RNAseq_tmp.tsv'</li><li>'ENCODE_K562_hg38_RNAseq.tsv'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'ENCODE\\_K562\\_hg38\\_RNAseq\\_tmp.tsv'\n",
       "\\item 'ENCODE\\_K562\\_hg38\\_RNAseq.tsv'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'ENCODE_K562_hg38_RNAseq_tmp.tsv'\n",
       "2. 'ENCODE_K562_hg38_RNAseq.tsv'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"ENCODE_K562_hg38_RNAseq_tmp.tsv\" \"ENCODE_K562_hg38_RNAseq.tsv\"    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "txt_fdiry  = file.path(FD_REF, TXT_FOLDER_REF)\n",
    "dir(txt_fdiry)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e404876-ca86-47fb-a55c-226ff97c9fd8",
   "metadata": {},
   "source": [
    "## Import metadata from reference file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ca669306-0582-4cfd-9b77-a0bb3719ff3d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:right;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF421TJX </td>\n",
       "   <td style=\"text-align:left;\"> tsv </td>\n",
       "   <td style=\"text-align:left;\"> tsv </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> gene quantifications </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR615EEK </td>\n",
       "   <td style=\"text-align:left;\"> total RNA-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> RNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> reverse </td>\n",
       "   <td style=\"text-align:left;\"> 1/4/2021 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF285DRD/, /files/ENCFF820EYJ/ </td>\n",
       "   <td style=\"text-align:right;\"> 11061127 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> a7875635bd3e8a70902a43d4b814832c </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF421TJX/@@download/ENCFF421TJX.tsv </td>\n",
       "   <td style=\"text-align:left;\"> V29 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/10/30/2033273a-286f-4c94-a652-9d75098cdfb5/ENCFF421TJX.tsv </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/10/30/2033273a-286f-4c94-a652-9d75098cdfb5/ENCFF421TJX.tsv?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.2.1 GRCh38 V29 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> unreplicated experiment </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF585HTZ </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> plus strand signal of unique reads </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR615EEK </td>\n",
       "   <td style=\"text-align:left;\"> total RNA-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> RNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> reverse </td>\n",
       "   <td style=\"text-align:left;\"> 1/4/2021 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF833WFD/, /files/GRCh38_EBV.chrom.sizes/ </td>\n",
       "   <td style=\"text-align:right;\"> 112512459 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> cc22df00ccf3af13fd208828480a7f04 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF585HTZ/@@download/ENCFF585HTZ.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> V29 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/10/30/2b5767d4-57fc-4a6f-b50b-660d164f77b4/ENCFF585HTZ.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/10/30/2b5767d4-57fc-4a6f-b50b-660d164f77b4/ENCFF585HTZ.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.2.1 GRCh38 V29 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> unreplicated experiment </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF876JOV </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> minus strand signal of unique reads </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR615EEK </td>\n",
       "   <td style=\"text-align:left;\"> total RNA-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> RNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> reverse </td>\n",
       "   <td style=\"text-align:left;\"> 1/4/2021 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF833WFD/, /files/GRCh38_EBV.chrom.sizes/ </td>\n",
       "   <td style=\"text-align:right;\"> 109644032 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> cd9f7432892b5d9355ff405220508915 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF876JOV/@@download/ENCFF876JOV.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> V29 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/10/30/92b9cac8-9587-417b-b3bc-abc7059bf075/ENCFF876JOV.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/10/30/92b9cac8-9587-417b-b3bc-abc7059bf075/ENCFF876JOV.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.2.1 GRCh38 V29 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> unreplicated experiment </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_folder = TXT_FOLDER_REF\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "txt_fname = \"ENCODE_K562_hg38_RNAseq.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### show and assign\n",
    "dat_metadata_import = dat\n",
    "fun_display_table(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9b0da380-af55-4b58-8797-13d8bed2b3e5",
   "metadata": {},
   "source": [
    "## Explore table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef9e2022-d41a-4ef0-9519-eb5386e418bb",
   "metadata": {},
   "source": [
    "**Check genome assembly**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "bbfb2e83-047a-4377-86aa-b4bece6e84b2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "GRCh38 \n",
       "     3 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_import\n",
    "table(dat$`File assembly`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "17c71f0e-b5b5-4754-b448-63144fb28c2f",
   "metadata": {},
   "source": [
    "**Check biosample**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "54abc00b-1527-4476-a2e0-7d0f6d079d9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "K562 \n",
       "   3 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_import\n",
    "table(dat$`Biosample term name`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "95ab42a2-1222-4c5a-8470-651b4df8fd25",
   "metadata": {},
   "source": [
    "## Arrange metadata tables"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35d46d03-8a7c-4eb5-bd47-8a007aee689a",
   "metadata": {},
   "source": [
    "**Helper function**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "84eb0c12-fa93-4f6b-8d51-92d06df85758",
   "metadata": {},
   "outputs": [],
   "source": [
    "fun_simplify_table = function(dat){\n",
    "    \n",
    "    ### rename some columns\n",
    "    dat = dat %>% \n",
    "        dplyr::mutate(\n",
    "            Index_Experiment = `Experiment accession`,\n",
    "            Index_File       = `File accession`,\n",
    "            File_Format      = `File format`,\n",
    "            File_Type        = `File type`,\n",
    "            Output_Type      = `Output type`,\n",
    "            Genome           = `File assembly`,\n",
    "            #Target           = str_remove(`Experiment target`, \"-human\"),\n",
    "            Bio_Replicates   = `Biological replicate(s)`,\n",
    "            Analysis         = `File analysis title`,\n",
    "            File_Name\t     = basename(`File download URL`),\n",
    "            File_URL         = `File download URL`\n",
    "        )\n",
    "    \n",
    "    ### select the needed columns\n",
    "    dat = dat %>%\n",
    "        dplyr::select(\n",
    "            Assay,\n",
    "            Index_Experiment,\n",
    "            Index_File,\n",
    "            File_Format,\n",
    "            File_Type,\n",
    "            Output_Type,\n",
    "            Genome,\n",
    "            #Target,\n",
    "            Bio_Replicates,\n",
    "            Analysis,\n",
    "            md5sum,\n",
    "            File_Name,\n",
    "            File_URL\n",
    "        )\n",
    "\n",
    "    ### return the simplified table\n",
    "    return(dat)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "696b294c-f81a-4eb1-9049-7e886a716f79",
   "metadata": {},
   "source": [
    "**Simplify the metatable**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a235ef27-152c-44b2-966a-46bbe735ee6a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:right;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> total RNA-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR615EEK </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF421TJX </td>\n",
       "   <td style=\"text-align:left;\"> tsv </td>\n",
       "   <td style=\"text-align:left;\"> tsv </td>\n",
       "   <td style=\"text-align:left;\"> gene quantifications </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.2.1 GRCh38 V29 </td>\n",
       "   <td style=\"text-align:left;\"> a7875635bd3e8a70902a43d4b814832c </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF421TJX.tsv </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF421TJX/@@download/ENCFF421TJX.tsv </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> total RNA-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR615EEK </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF585HTZ </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> plus strand signal of unique reads </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.2.1 GRCh38 V29 </td>\n",
       "   <td style=\"text-align:left;\"> cc22df00ccf3af13fd208828480a7f04 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF585HTZ.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF585HTZ/@@download/ENCFF585HTZ.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> total RNA-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR615EEK </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF876JOV </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> minus strand signal of unique reads </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.2.1 GRCh38 V29 </td>\n",
       "   <td style=\"text-align:left;\"> cd9f7432892b5d9355ff405220508915 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF876JOV.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF876JOV/@@download/ENCFF876JOV.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### arrange and simplify the table\n",
    "dat = dat_metadata_import\n",
    "dat = fun_simplify_table(dat)\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_simplify = dat\n",
    "fun_display_table(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f53e97cd-6f6d-438d-847e-79ce0658f22e",
   "metadata": {},
   "source": [
    "## Prepare download files"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b725cef4-baf8-45b1-b28a-33316623f876",
   "metadata": {},
   "source": [
    "**Helper function**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5ada19ce-f863-4bfd-93bf-171133173e02",
   "metadata": {},
   "outputs": [],
   "source": [
    "fun_map_file_label = function(txt){\n",
    "    vec1 = c(\"quantifications\", \"plus strand\",   \"minus strand\")\n",
    "    vec2 = c(\"tsv\",             \"strand_pos.bw\", \"strand_neg.bw\")\n",
    "    res  = fun_str_map_detect(txt, vec1, vec2, .default=txt)\n",
    "    return(res)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "489daafd-d00f-480d-8ae7-dd2de5607034",
   "metadata": {},
   "source": [
    "**Rename filename**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6f9acbe7-4bea-4374-b60a-4b30665302bf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:right;\"> Bio_Replicates </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "   <th style=\"text-align:left;\"> File_URL </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> total RNA-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR615EEK </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF421TJX </td>\n",
       "   <td style=\"text-align:left;\"> tsv </td>\n",
       "   <td style=\"text-align:left;\"> tsv </td>\n",
       "   <td style=\"text-align:left;\"> gene quantifications </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.2.1 GRCh38 V29 </td>\n",
       "   <td style=\"text-align:left;\"> a7875635bd3e8a70902a43d4b814832c </td>\n",
       "   <td style=\"text-align:left;\"> K562.hg38.ENCSR615EEK.ENCFF421TJX.RNAseq_total.tsv </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF421TJX/@@download/ENCFF421TJX.tsv </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> total RNA-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR615EEK </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF585HTZ </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> plus strand signal of unique reads </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.2.1 GRCh38 V29 </td>\n",
       "   <td style=\"text-align:left;\"> cc22df00ccf3af13fd208828480a7f04 </td>\n",
       "   <td style=\"text-align:left;\"> K562.hg38.ENCSR615EEK.ENCFF585HTZ.RNAseq_total.strand_pos.bw </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF585HTZ/@@download/ENCFF585HTZ.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> total RNA-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR615EEK </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF876JOV </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> minus strand signal of unique reads </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.2.1 GRCh38 V29 </td>\n",
       "   <td style=\"text-align:left;\"> cd9f7432892b5d9355ff405220508915 </td>\n",
       "   <td style=\"text-align:left;\"> K562.hg38.ENCSR615EEK.ENCFF876JOV.RNAseq_total.strand_neg.bw </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF876JOV/@@download/ENCFF876JOV.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### rename filename\n",
    "dat = dat_metadata_simplify\n",
    "dat = dat %>% dplyr::mutate(\n",
    "    File_Name = paste(\n",
    "        \"K562\",\n",
    "        \"hg38\",\n",
    "        Index_Experiment,\n",
    "        Index_File,\n",
    "        \"RNAseq_total\",\n",
    "        fun_map_file_label(Output_Type),\n",
    "    sep = \".\")\n",
    ")\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_arrange = dat\n",
    "fun_display_table(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "11398fb0-01df-4542-9720-29fafd35c687",
   "metadata": {},
   "source": [
    "**Check results**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3f099a7a-4e61-4be8-91c7-08d07b58ed97",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Format </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR615EEK </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF421TJX </td>\n",
       "   <td style=\"text-align:left;\"> tsv </td>\n",
       "   <td style=\"text-align:left;\"> K562.hg38.ENCSR615EEK.ENCFF421TJX.RNAseq_total.tsv </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR615EEK </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF585HTZ </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> K562.hg38.ENCSR615EEK.ENCFF585HTZ.RNAseq_total.strand_pos.bw </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR615EEK </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF876JOV </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> K562.hg38.ENCSR615EEK.ENCFF876JOV.RNAseq_total.strand_neg.bw </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_arrange\n",
    "dat = dat %>% dplyr::select(Index_Experiment, Index_File, File_Format, File_Name)\n",
    "fun_display_table(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18bab986-3c84-4790-b514-0bc8e119c2ef",
   "metadata": {},
   "source": [
    "**Checksum table**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "63823f94-4155-4e80-ae95-c3aade4811ee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> File_Name </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> a7875635bd3e8a70902a43d4b814832c </td>\n",
       "   <td style=\"text-align:left;\"> K562.hg38.ENCSR615EEK.ENCFF421TJX.RNAseq_total.tsv </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> cc22df00ccf3af13fd208828480a7f04 </td>\n",
       "   <td style=\"text-align:left;\"> K562.hg38.ENCSR615EEK.ENCFF585HTZ.RNAseq_total.strand_pos.bw </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> cd9f7432892b5d9355ff405220508915 </td>\n",
       "   <td style=\"text-align:left;\"> K562.hg38.ENCSR615EEK.ENCFF876JOV.RNAseq_total.strand_neg.bw </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### get md5sum for each file\n",
    "dat = dat_metadata_arrange\n",
    "dat = dat %>% dplyr::select(md5sum, File_Name)\n",
    "\n",
    "### assign and show\n",
    "dat_download_checksum = dat\n",
    "fun_display_table(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f73f54ba-8b3b-4e4c-8639-d70a1eb31474",
   "metadata": {},
   "source": [
    "## Generate download scripts\n",
    "```\n",
    "wget -O FILE URL\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "bea72beb-ac50-4bfa-92e1-9abf456a5578",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> #!/bin/bash </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> echo -n \"\" &gt; run_download.log.txt </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR615EEK.ENCFF421TJX.RNAseq_total.tsv https://www.encodeproject.org/files/ENCFF421TJX/@@download/ENCFF421TJX.tsv </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR615EEK.ENCFF585HTZ.RNAseq_total.strand_pos.bw https://www.encodeproject.org/files/ENCFF585HTZ/@@download/ENCFF585HTZ.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> wget --append-output=run_download.log.txt -O K562.hg38.ENCSR615EEK.ENCFF876JOV.RNAseq_total.strand_neg.bw https://www.encodeproject.org/files/ENCFF876JOV/@@download/ENCFF876JOV.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### setup download file wget command\n",
    "dat = dat_metadata_arrange\n",
    "dat = dat %>% dplyr::mutate(\n",
    "        CMD = paste(\n",
    "            \"wget\", \"--append-output=run_download.log.txt\", \"-O\", File_Name, File_URL\n",
    "        )\n",
    "    )\n",
    "\n",
    "### add Shebang and initial commands\n",
    "dat = dat %>% dplyr::select(CMD)\n",
    "dat = rbind('echo -n \"\" > run_download.log.txt', dat)\n",
    "colnames(dat) = \"#!/bin/bash\"\n",
    "\n",
    "### assign and show\n",
    "dat_download_script = dat\n",
    "fun_display_table(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5760f801-9513-4dd9-b24f-8296d30b127f",
   "metadata": {},
   "source": [
    "## Save results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ff6e277c-490c-4dc1-abd3-e2ce884e2d0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set output path\n",
    "txt_folder = TXT_FOLDER_OUT\n",
    "txt_fdiry  = file.path(FD_DAT, \"external\", txt_folder)\n",
    "\n",
    "### create directory if not exist\n",
    "dir.create(txt_fdiry, showWarnings = FALSE)\n",
    "\n",
    "### write checksum file\n",
    "txt_fname  = \"run_download_files.sh\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "dat = dat_download_script\n",
    "write_tsv(dat, txt_fpath)\n",
    "\n",
    "### save table\n",
    "txt_fname  = \"checksum_md5sum.txt\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "dat = dat_download_checksum\n",
    "write_tsv(dat, txt_fpath, col_names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e4dda40-b0c3-45cb-b956-ed6204173ee1",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
