{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3b309689-78fd-4d93-933a-72dc96a22777",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2904b635-b416-41e6-a6d6-ba4ec92d52d0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are on Duke Server: HARDAC\n",
      "BASE DIRECTORY:     /data/reddylab/Kuei\n",
      "WORK DIRECTORY:     /data/reddylab/Kuei/out\n",
      "CODE DIRECTORY:     /data/reddylab/Kuei/code\n",
      "SING DIRECTORY:     /data/reddylab/Kuei/singularity\n",
      "PATH OF SOURCE:     /data/reddylab/Kuei/source\n",
      "PATH OF EXECUTABLE: /data/reddylab/Kuei/bin\n",
      "PATH OF ANNOTATION: /data/reddylab/Kuei/annotation\n",
      "PATH OF PROJECT:    /data/reddylab/Kuei/code/Proj_CombEffect_ENCODE_FCC\n",
      "PATH OF RESULTS:    /data/reddylab/Kuei/out/proj_combeffect_encode_fcc\n",
      "\n"
     ]
    }
   ],
   "source": [
    "source ../config/config_duke.sh\n",
    "source ../config/config_load_module_bedtools.sh\n",
    "show_env"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "69b6d89e-e1f0-4de0-9375-2c50fab3bfdc",
   "metadata": {},
   "source": [
    "## Calculate coverage perbase for input and output"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "072dbeba-f7f5-4a00-ae9b-998b6d6d9659",
   "metadata": {},
   "source": [
    "**TEST**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3e2daf24-416b-488e-9c58-7aa8d27230e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep1.FADS.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep1.GATA1.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep1.MYC.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep1.WGS.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep2.FADS.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep2.GATA1.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep2.MYC.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep2.WGS.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep3.FADS.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep3.GATA1.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep3.MYC.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep3.WGS.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep4.FADS.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep4.GATA1.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep4.MYC.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Input.rep4.WGS.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Output.rep1.FADS.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Output.rep1.GATA1.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Output.rep1.MYC.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Output.rep1.WGS.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Output.rep2.FADS.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Output.rep2.GATA1.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Output.rep2.MYC.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Output.rep2.WGS.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Output.rep3.FADS.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Output.rep3.GATA1.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Output.rep3.MYC.unstranded.bed.gz\u001b[0m\n",
      "\u001b[38;5;9mA001_K562_WSTARRseq.Output.rep3.WGS.unstranded.bed.gz\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ASSAY=A001_K562_WSTARRseq\n",
    "PREFIX=A001_K562_WSTARRseq\n",
    "FD_INP=${FD_RES}/results/${ASSAY}/fragment_count\n",
    "ls ${FD_INP}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "72fa235f-14e0-4901-a6eb-603d06d073b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total:  7\n",
      "A001_K562_WSTARRseq.Input.rep1.GATA1.unstranded.bed.gz\n",
      "A001_K562_WSTARRseq.Input.rep2.GATA1.unstranded.bed.gz\n",
      "A001_K562_WSTARRseq.Input.rep3.GATA1.unstranded.bed.gz\n",
      "A001_K562_WSTARRseq.Input.rep4.GATA1.unstranded.bed.gz\n",
      "A001_K562_WSTARRseq.Output.rep1.GATA1.unstranded.bed.gz\n",
      "A001_K562_WSTARRseq.Output.rep2.GATA1.unstranded.bed.gz\n",
      "A001_K562_WSTARRseq.Output.rep3.GATA1.unstranded.bed.gz\n"
     ]
    }
   ],
   "source": [
    "ASSAY=A001_K562_WSTARRseq\n",
    "PREFIX=A001_K562_WSTARRseq\n",
    "REGION=GATA1\n",
    "\n",
    "FD_INP=${FD_RES}/results/${ASSAY}/fragment_count\n",
    "FD_OUT=${FD_RES}/results/${ASSAY}/coverage\n",
    "\n",
    "FP_INPS=($(ls ${FD_INP}/*${REGION}*))\n",
    "echo \"Total:  ${#FP_INPS[@]}\"\n",
    "\n",
    "for FP_INP in ${FP_INPS[@]}; do\n",
    "    FN_INP=$(basename ${FP_INP})\n",
    "    echo ${FN_INP}\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "499e342e-5ebc-43fe-abca-43387c226c52",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Region: GATA1\n",
      "/data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/region_GATA1_perbase.bed\n",
      "A001_K562_WSTARRseq.Input.rep1.GATA1.unstranded.bed.gz\n",
      "\n",
      "Region: MYC\n",
      "/data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/region_MYC_perbase.bed\n",
      "A001_K562_WSTARRseq.Input.rep1.MYC.unstranded.bed.gz\n",
      "\n",
      "Region: FADS\n",
      "/data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/region_FADS_perbase.bed\n",
      "A001_K562_WSTARRseq.Input.rep1.FADS.unstranded.bed.gz\n",
      "\n"
     ]
    }
   ],
   "source": [
    "ASSAY=A001_K562_WSTARRseq\n",
    "PREFIX=A001_K562_WSTARRseq\n",
    "\n",
    "FD_INP=${FD_RES}/results/${ASSAY}/fragment_count\n",
    "FD_OUT=${FD_RES}/results/${ASSAY}/coverage\n",
    "IDX=0\n",
    "\n",
    "REGIONS=(GATA1 MYC FADS)\n",
    "for REGION in ${REGIONS[@]}; do\n",
    "    FP_REG=${FD_OUT}/region_${REGION}_perbase.bed\n",
    "    FP_INPS=($(ls ${FD_INP}/*${REGION}*))\n",
    "    FP_INP=${FP_INPS[${IDX}]}\n",
    "    FN_INP=$(basename ${FP_INP})\n",
    "    \n",
    "    echo \"Region: ${REGION}\"\n",
    "    echo ${FP_REG}\n",
    "    echo ${FN_INP}\n",
    "    echo\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "99b15ee5-d84b-4a3e-90a4-119868442b93",
   "metadata": {},
   "source": [
    "**TEST**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "272ba87c-16df-4211-a0c6-8ea217cd6446",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 29422846\n"
     ]
    }
   ],
   "source": [
    "sbatch -p ${NODE} \\\n",
    "    --mem 8G \\\n",
    "    --array 0 \\\n",
    "    -o ${FD_LOG}/test_bedtools.txt \\\n",
    "    <<'EOF'\n",
    "#!/bin/bash\n",
    "### set directories & global variables\n",
    "source ../config/config_duke.sh\n",
    "\n",
    "### print start message\n",
    "timer_start=`date +%s`\n",
    "echo \"Hostname:          \" $(hostname)\n",
    "echo \"Slurm Array Index: \" ${SLURM_ARRAY_TASK_ID}\n",
    "echo \"Time Stamp:        \" $(date +\"%m-%d-%y+%T\")\n",
    "echo\n",
    "\n",
    "### test bedtools\n",
    "bedtools --help\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4141c822-a14d-473c-b441-ec0a582a165d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Hostname:           dl-01.genome.duke.edu\n",
      "Slurm Array Index:  0\n",
      "Time Stamp:         12-09-22+12:47:06\n",
      "\n",
      "bedtools: flexible tools for genome arithmetic and DNA sequence analysis.\n",
      "usage:    bedtools <subcommand> [options]\n",
      "\n",
      "The bedtools sub-commands include:\n",
      "\n",
      "[ Genome arithmetic ]\n",
      "    intersect     Find overlapping intervals in various ways.\n",
      "    window        Find overlapping intervals within a window around an interval.\n",
      "    closest       Find the closest, potentially non-overlapping interval.\n",
      "    coverage      Compute the coverage over defined intervals.\n",
      "    map           Apply a function to a column for each overlapping interval.\n",
      "    genomecov     Compute the coverage over an entire genome.\n",
      "    merge         Combine overlapping/nearby intervals into a single interval.\n",
      "    cluster       Cluster (but don't merge) overlapping/nearby intervals.\n",
      "    complement    Extract intervals _not_ represented by an interval file.\n",
      "    subtract      Remove intervals based on overlaps b/w two files.\n",
      "    slop          Adjust the size of intervals.\n",
      "    flank         Create new intervals from the flanks of existing intervals.\n",
      "    sort          Order the intervals in a file.\n",
      "    random        Generate random intervals in a genome.\n",
      "    shuffle       Randomly redistrubute intervals in a genome.\n",
      "    sample        Sample random records from file using reservoir sampling.\n",
      "    spacing       Report the gap lengths between intervals in a file.\n",
      "    annotate      Annotate coverage of features from multiple files.\n",
      "\n",
      "[ Multi-way file comparisons ]\n",
      "    multiinter    Identifies common intervals among multiple interval files.\n",
      "    unionbedg     Combines coverage intervals from multiple BEDGRAPH files.\n",
      "\n",
      "[ Paired-end manipulation ]\n",
      "    pairtobed     Find pairs that overlap intervals in various ways.\n",
      "    pairtopair    Find pairs that overlap other pairs in various ways.\n",
      "\n",
      "[ Format conversion ]\n",
      "    bamtobed      Convert BAM alignments to BED (& other) formats.\n",
      "    bedtobam      Convert intervals to BAM records.\n",
      "    bamtofastq    Convert BAM records to FASTQ records.\n",
      "    bedpetobam    Convert BEDPE intervals to BAM records.\n",
      "    bed12tobed6   Breaks BED12 intervals into discrete BED6 intervals.\n",
      "\n",
      "[ Fasta manipulation ]\n",
      "    getfasta      Use intervals to extract sequences from a FASTA file.\n",
      "    maskfasta     Use intervals to mask sequences from a FASTA file.\n",
      "    nuc           Profile the nucleotide content of intervals in a FASTA file.\n",
      "\n",
      "[ BAM focused tools ]\n",
      "    multicov      Counts coverage from multiple BAMs at specific intervals.\n",
      "    tag           Tag BAM alignments based on overlaps with interval files.\n",
      "\n",
      "[ Statistical relationships ]\n",
      "    jaccard       Calculate the Jaccard statistic b/w two sets of intervals.\n",
      "    reldist       Calculate the distribution of relative distances b/w two files.\n",
      "    fisher        Calculate Fisher statistic b/w two feature files.\n",
      "\n",
      "[ Miscellaneous tools ]\n",
      "    overlap       Computes the amount of overlap from two intervals.\n",
      "    igv           Create an IGV snapshot batch script.\n",
      "    links         Create a HTML page of links to UCSC locations.\n",
      "    makewindows   Make interval \"windows\" across a genome.\n",
      "    groupby       Group by common cols. & summarize oth. cols. (~ SQL \"groupBy\")\n",
      "    expand        Replicate lines based on lists of values in columns.\n",
      "    split         Split a file into multiple files with equal records or base pairs.\n",
      "\n",
      "[ General help ]\n",
      "    --help        Print this help menu.\n",
      "    --version     What version of bedtools are you using?.\n",
      "    --contact     Feature requests, bugs, mailing lists, etc.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cat ${FD_LOG}/test_bedtools.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5f8f058a-b693-4abe-97a2-45877b6c36b2",
   "metadata": {},
   "source": [
    "**RUN**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "386e1974-4294-4b44-a115-9d264362edb6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 29422857\n"
     ]
    }
   ],
   "source": [
    "sbatch -p ${NODE} \\\n",
    "    --mem 20G \\\n",
    "    --array 0-6 \\\n",
    "    -o ${FD_LOG}/coverage_perbase_WSTARR.%a.txt \\\n",
    "    <<'EOF'\n",
    "#!/bin/bash\n",
    "### set directories & global variables\n",
    "source ../config/config_duke.sh\n",
    "ASSAY=A001_K562_WSTARRseq\n",
    "PREFIX=A001_K562_WSTARRseq\n",
    "REGIONS=(GATA1 MYC FADS)\n",
    "\n",
    "### set input and output directory\n",
    "FD_INP=${FD_RES}/results/${ASSAY}/fragment_count\n",
    "FD_OUT=${FD_RES}/results/${ASSAY}/coverage\n",
    "\n",
    "### print start message\n",
    "timer_start=`date +%s`\n",
    "echo \"Hostname:          \" $(hostname)\n",
    "echo \"Slurm Array Index: \" ${SLURM_ARRAY_TASK_ID}\n",
    "echo \"Time Stamp:        \" $(date +\"%m-%d-%y+%T\")\n",
    "echo\n",
    "\n",
    "### +++++++++++++++++++++++++++++++++++++++++++++++\n",
    "\n",
    "for REGION in ${REGIONS[@]}; do\n",
    "\n",
    "    ### set input and output file\n",
    "    FP_REG=${FD_OUT}/region_${REGION}_perbase.bed\n",
    "    \n",
    "    FP_INPS=($(ls ${FD_INP}/*${REGION}*))\n",
    "    FP_INP=${FP_INPS[${SLURM_ARRAY_TASK_ID}]}\n",
    "    FN_INP=$(basename ${FP_INP})\n",
    "    \n",
    "    FN_OUT=${FN_INP}\n",
    "    FP_OUT=${FD_OUT}/${FN_OUT}\n",
    "    \n",
    "    ### show input file\n",
    "    echo ++++++++++++++++++++++++++++++++++++++\n",
    "    echo\n",
    "    echo \"Input: \" ${FP_INP}\n",
    "    echo\n",
    "    echo \"show first few lines of input\"\n",
    "    zcat ${FP_INP} | head -n 5\n",
    "    echo\n",
    "    echo \"show last few lines of output\"\n",
    "    zcat ${FP_INP} | tail -n 5\n",
    "    echo\n",
    "    echo \"Input: \" ${FP_REG}\n",
    "    echo\n",
    "    echo \"show first few lines of input\"\n",
    "    cat  ${FP_REG} | head -n 5\n",
    "    echo\n",
    "    echo \"show last few lines of output\"\n",
    "    cat  ${FP_REG} | tail -n 5\n",
    "    echo\n",
    "    echo\n",
    "    \n",
    "    ### execute\n",
    "    bedtools map \\\n",
    "        -a ${FP_REG} \\\n",
    "        -b ${FP_INP} \\\n",
    "        -o sum |\\\n",
    "        gzip -c > ${FP_OUT}\n",
    "    \n",
    "    ### show output file\n",
    "    echo \"Output: \" ${FP_OUT}\n",
    "    echo\n",
    "    echo \"show first few lines of output:\"\n",
    "    zcat ${FP_OUT} | head -n 5\n",
    "    echo\n",
    "    echo \"show last few lines of output\"\n",
    "    zcat ${FP_OUT} | tail -n 5\n",
    "    echo\n",
    "done\n",
    "\n",
    "### print end message\n",
    "timer=`date +%s`\n",
    "runtime=$(echo \"${timer} - ${timer_start}\" | bc -l)\n",
    "echo\n",
    "echo 'Done!'\n",
    "echo \"Run Time: $(displaytime ${runtime})\"\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "06035dcf-d9f0-49f2-ba76-df0a9e116237",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.0.txt\n",
      "/data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.1.txt\n",
      "/data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.2.txt\n",
      "/data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.3.txt\n",
      "/data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.4.txt\n",
      "/data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.5.txt\n",
      "/data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.6.txt\n"
     ]
    }
   ],
   "source": [
    "ls ${FD_LOG}/coverage_perbase_WSTARR.*.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "4e3e5177-2f5b-4155-a559-5e995b7688f5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.0.txt <==\n",
      "show last few lines of output\n",
      "chr11\t61665816\t61665817\t7\n",
      "chr11\t61665817\t61665818\t7\n",
      "chr11\t61665818\t61665819\t7\n",
      "chr11\t61665819\t61665820\t7\n",
      "chr11\t61665820\t61665821\t7\n",
      "\n",
      "\n",
      "Done!\n",
      "Run Time: 20 seconds\n",
      "\n",
      "==> /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.1.txt <==\n",
      "show last few lines of output\n",
      "chr11\t61665816\t61665817\t25\n",
      "chr11\t61665817\t61665818\t25\n",
      "chr11\t61665818\t61665819\t25\n",
      "chr11\t61665819\t61665820\t25\n",
      "chr11\t61665820\t61665821\t25\n",
      "\n",
      "\n",
      "Done!\n",
      "Run Time: 26 seconds\n",
      "\n",
      "==> /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.2.txt <==\n",
      "show last few lines of output\n",
      "chr11\t61665816\t61665817\t20\n",
      "chr11\t61665817\t61665818\t20\n",
      "chr11\t61665818\t61665819\t20\n",
      "chr11\t61665819\t61665820\t19\n",
      "chr11\t61665820\t61665821\t19\n",
      "\n",
      "\n",
      "Done!\n",
      "Run Time: 26 seconds\n",
      "\n",
      "==> /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.3.txt <==\n",
      "show last few lines of output\n",
      "chr11\t61665816\t61665817\t20\n",
      "chr11\t61665817\t61665818\t20\n",
      "chr11\t61665818\t61665819\t20\n",
      "chr11\t61665819\t61665820\t20\n",
      "chr11\t61665820\t61665821\t20\n",
      "\n",
      "\n",
      "Done!\n",
      "Run Time: 25 seconds\n",
      "\n",
      "==> /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.4.txt <==\n",
      "show last few lines of output\n",
      "chr11\t61665816\t61665817\t19\n",
      "chr11\t61665817\t61665818\t18\n",
      "chr11\t61665818\t61665819\t18\n",
      "chr11\t61665819\t61665820\t18\n",
      "chr11\t61665820\t61665821\t18\n",
      "\n",
      "\n",
      "Done!\n",
      "Run Time: 29 seconds\n",
      "\n",
      "==> /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.5.txt <==\n",
      "show last few lines of output\n",
      "chr11\t61665816\t61665817\t19\n",
      "chr11\t61665817\t61665818\t19\n",
      "chr11\t61665818\t61665819\t19\n",
      "chr11\t61665819\t61665820\t19\n",
      "chr11\t61665820\t61665821\t19\n",
      "\n",
      "\n",
      "Done!\n",
      "Run Time: 29 seconds\n",
      "\n",
      "==> /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/log/coverage_perbase_WSTARR.6.txt <==\n",
      "show last few lines of output\n",
      "chr11\t61665816\t61665817\t41\n",
      "chr11\t61665817\t61665818\t41\n",
      "chr11\t61665818\t61665819\t41\n",
      "chr11\t61665819\t61665820\t41\n",
      "chr11\t61665820\t61665821\t41\n",
      "\n",
      "\n",
      "Done!\n",
      "Run Time: 37 seconds\n"
     ]
    }
   ],
   "source": [
    "tail -n 10 ${FD_LOG}/coverage_perbase_WSTARR.*.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "067377fc-3646-4a30-8897-182a1f5dfefd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Hostname:           dl-01.genome.duke.edu\n",
      "Slurm Array Index:  0\n",
      "Time Stamp:         12-09-22+12:50:36\n",
      "\n",
      "++++++++++++++++++++++++++++++++++++++\n",
      "\n",
      "Input:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/fragment_count/A001_K562_WSTARRseq.Input.rep1.GATA1.unstranded.bed.gz\n",
      "\n",
      "show first few lines of input\n",
      "chrX\t47785906\t47786328\tchrX_47785906_47786328\t1\t.\n",
      "chrX\t47787533\t47787772\tchrX_47787533_47787772\t1\t.\n",
      "chrX\t47787569\t47787735\tchrX_47787569_47787735\t1\t.\n",
      "chrX\t47787714\t47788125\tchrX_47787714_47788125\t1\t.\n",
      "chrX\t47787767\t47788062\tchrX_47787767_47788062\t1\t.\n",
      "\n",
      "show last few lines of output\n",
      "chrX\t49880213\t49880761\tchrX_49880213_49880761\t1\t.\n",
      "chrX\t49880242\t49880404\tchrX_49880242_49880404\t1\t.\n",
      "chrX\t49880288\t49880772\tchrX_49880288_49880772\t1\t.\n",
      "chrX\t49880289\t49880612\tchrX_49880289_49880612\t1\t.\n",
      "chrX\t49880412\t49880731\tchrX_49880412_49880731\t1\t.\n",
      "\n",
      "Input:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/region_GATA1_perbase.bed\n",
      "\n",
      "show first few lines of input\n",
      "chrX\t47785501\t47785502\n",
      "chrX\t47785502\t47785503\n",
      "chrX\t47785503\t47785504\n",
      "chrX\t47785504\t47785505\n",
      "chrX\t47785505\t47785506\n",
      "\n",
      "show last few lines of output\n",
      "chrX\t49880646\t49880647\n",
      "chrX\t49880647\t49880648\n",
      "chrX\t49880648\t49880649\n",
      "chrX\t49880649\t49880650\n",
      "chrX\t49880650\t49880651\n",
      "\n",
      "\n",
      "Output:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/A001_K562_WSTARRseq.Input.rep1.GATA1.unstranded.bed.gz\n",
      "\n",
      "show first few lines of output:\n",
      "chrX\t47785501\t47785502\t.\n",
      "chrX\t47785502\t47785503\t.\n",
      "chrX\t47785503\t47785504\t.\n",
      "chrX\t47785504\t47785505\t.\n",
      "chrX\t47785505\t47785506\t.\n",
      "\n",
      "show last few lines of output\n",
      "chrX\t49880646\t49880647\t4\n",
      "chrX\t49880647\t49880648\t4\n",
      "chrX\t49880648\t49880649\t4\n",
      "chrX\t49880649\t49880650\t4\n",
      "chrX\t49880650\t49880651\t4\n",
      "\n",
      "++++++++++++++++++++++++++++++++++++++\n",
      "\n",
      "Input:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/fragment_count/A001_K562_WSTARRseq.Input.rep1.MYC.unstranded.bed.gz\n",
      "\n",
      "show first few lines of input\n",
      "chr8\t126735578\t126735929\tchr8_126735578_126735929\t1\t.\n",
      "chr8\t126735592\t126736074\tchr8_126735592_126736074\t1\t.\n",
      "chr8\t126735615\t126735927\tchr8_126735615_126735927\t1\t.\n",
      "chr8\t126735669\t126735990\tchr8_126735669_126735990\t1\t.\n",
      "chr8\t126735731\t126736258\tchr8_126735731_126736258\t1\t.\n",
      "\n",
      "show last few lines of output\n",
      "chr8\t128735973\t128736390\tchr8_128735973_128736390\t1\t.\n",
      "chr8\t128736009\t128736504\tchr8_128736009_128736504\t1\t.\n",
      "chr8\t128736188\t128736550\tchr8_128736188_128736550\t1\t.\n",
      "chr8\t128736220\t128736734\tchr8_128736220_128736734\t1\t.\n",
      "chr8\t128736265\t128736860\tchr8_128736265_128736860\t1\t.\n",
      "\n",
      "Input:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/region_MYC_perbase.bed\n",
      "\n",
      "show first few lines of input\n",
      "chr8\t126735901\t126735902\n",
      "chr8\t126735902\t126735903\n",
      "chr8\t126735903\t126735904\n",
      "chr8\t126735904\t126735905\n",
      "chr8\t126735905\t126735906\n",
      "\n",
      "show last few lines of output\n",
      "chr8\t128736546\t128736547\n",
      "chr8\t128736547\t128736548\n",
      "chr8\t128736548\t128736549\n",
      "chr8\t128736549\t128736550\n",
      "chr8\t128736550\t128736551\n",
      "\n",
      "\n",
      "Output:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/A001_K562_WSTARRseq.Input.rep1.MYC.unstranded.bed.gz\n",
      "\n",
      "show first few lines of output:\n",
      "chr8\t126735901\t126735902\t10\n",
      "chr8\t126735902\t126735903\t10\n",
      "chr8\t126735903\t126735904\t10\n",
      "chr8\t126735904\t126735905\t10\n",
      "chr8\t126735905\t126735906\t10\n",
      "\n",
      "show last few lines of output\n",
      "chr8\t128736546\t128736547\t3\n",
      "chr8\t128736547\t128736548\t3\n",
      "chr8\t128736548\t128736549\t3\n",
      "chr8\t128736549\t128736550\t3\n",
      "chr8\t128736550\t128736551\t2\n",
      "\n",
      "++++++++++++++++++++++++++++++++++++++\n",
      "\n",
      "Input:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/fragment_count/A001_K562_WSTARRseq.Input.rep1.FADS.unstranded.bed.gz\n",
      "\n",
      "show first few lines of input\n",
      "chr11\t61554443\t61554815\tchr11_61554443_61554815\t1\t.\n",
      "chr11\t61554444\t61554815\tchr11_61554444_61554815\t1\t.\n",
      "chr11\t61554449\t61554957\tchr11_61554449_61554957\t1\t.\n",
      "chr11\t61554467\t61554882\tchr11_61554467_61554882\t1\t.\n",
      "chr11\t61554504\t61554909\tchr11_61554504_61554909\t1\t.\n",
      "\n",
      "show last few lines of output\n",
      "chr11\t61665376\t61665892\tchr11_61665376_61665892\t1\t.\n",
      "chr11\t61665384\t61666050\tchr11_61665384_61666050\t1\t.\n",
      "chr11\t61665618\t61666097\tchr11_61665618_61666097\t1\t.\n",
      "chr11\t61665655\t61666003\tchr11_61665655_61666003\t1\t.\n",
      "chr11\t61665785\t61666147\tchr11_61665785_61666147\t1\t.\n",
      "\n",
      "Input:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/region_FADS_perbase.bed\n",
      "\n",
      "show first few lines of input\n",
      "chr11\t61554801\t61554802\n",
      "chr11\t61554802\t61554803\n",
      "chr11\t61554803\t61554804\n",
      "chr11\t61554804\t61554805\n",
      "chr11\t61554805\t61554806\n",
      "\n",
      "show last few lines of output\n",
      "chr11\t61665816\t61665817\n",
      "chr11\t61665817\t61665818\n",
      "chr11\t61665818\t61665819\n",
      "chr11\t61665819\t61665820\n",
      "chr11\t61665820\t61665821\n",
      "\n",
      "\n",
      "Output:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/A001_K562_WSTARRseq.Input.rep1.FADS.unstranded.bed.gz\n",
      "\n",
      "show first few lines of output:\n",
      "chr11\t61554801\t61554802\t10\n",
      "chr11\t61554802\t61554803\t10\n",
      "chr11\t61554803\t61554804\t10\n",
      "chr11\t61554804\t61554805\t10\n",
      "chr11\t61554805\t61554806\t10\n",
      "\n",
      "show last few lines of output\n",
      "chr11\t61665816\t61665817\t7\n",
      "chr11\t61665817\t61665818\t7\n",
      "chr11\t61665818\t61665819\t7\n",
      "chr11\t61665819\t61665820\t7\n",
      "chr11\t61665820\t61665821\t7\n",
      "\n",
      "\n",
      "Done!\n",
      "Run Time: 20 seconds\n"
     ]
    }
   ],
   "source": [
    "cat ${FD_LOG}/coverage_perbase_WSTARR.0.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "2855426f-89ff-46a2-a8ab-d9203f103eb7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Hostname:           dl-01.genome.duke.edu\n",
      "Slurm Array Index:  6\n",
      "Time Stamp:         12-09-22+12:50:37\n",
      "\n",
      "++++++++++++++++++++++++++++++++++++++\n",
      "\n",
      "Input:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/fragment_count/A001_K562_WSTARRseq.Output.rep3.GATA1.unstranded.bed.gz\n",
      "\n",
      "show first few lines of input\n",
      "chrX\t47785062\t47785649\tchrX_47785062_47785649\t1\t.\n",
      "chrX\t47785094\t47785628\tchrX_47785094_47785628\t1\t.\n",
      "chrX\t47785156\t47785640\tchrX_47785156_47785640\t1\t.\n",
      "chrX\t47785290\t47785672\tchrX_47785290_47785672\t1\t.\n",
      "chrX\t47785381\t47785723\tchrX_47785381_47785723\t1\t.\n",
      "\n",
      "show last few lines of output\n",
      "chrX\t49880608\t49881056\tchrX_49880608_49881056\t1\t.\n",
      "chrX\t49880626\t49881029\tchrX_49880626_49881029\t1\t.\n",
      "chrX\t49880629\t49881260\tchrX_49880629_49881260\t1\t.\n",
      "chrX\t49880644\t49881032\tchrX_49880644_49881032\t1\t.\n",
      "chrX\t49880649\t49880957\tchrX_49880649_49880957\t1\t.\n",
      "\n",
      "Input:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/region_GATA1_perbase.bed\n",
      "\n",
      "show first few lines of input\n",
      "chrX\t47785501\t47785502\n",
      "chrX\t47785502\t47785503\n",
      "chrX\t47785503\t47785504\n",
      "chrX\t47785504\t47785505\n",
      "chrX\t47785505\t47785506\n",
      "\n",
      "show last few lines of output\n",
      "chrX\t49880646\t49880647\n",
      "chrX\t49880647\t49880648\n",
      "chrX\t49880648\t49880649\n",
      "chrX\t49880649\t49880650\n",
      "chrX\t49880650\t49880651\n",
      "\n",
      "\n",
      "Output:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/A001_K562_WSTARRseq.Output.rep3.GATA1.unstranded.bed.gz\n",
      "\n",
      "show first few lines of output:\n",
      "chrX\t47785501\t47785502\t8\n",
      "chrX\t47785502\t47785503\t8\n",
      "chrX\t47785503\t47785504\t8\n",
      "chrX\t47785504\t47785505\t8\n",
      "chrX\t47785505\t47785506\t8\n",
      "\n",
      "show last few lines of output\n",
      "chrX\t49880646\t49880647\t87\n",
      "chrX\t49880647\t49880648\t86\n",
      "chrX\t49880648\t49880649\t86\n",
      "chrX\t49880649\t49880650\t86\n",
      "chrX\t49880650\t49880651\t86\n",
      "\n",
      "++++++++++++++++++++++++++++++++++++++\n",
      "\n",
      "Input:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/fragment_count/A001_K562_WSTARRseq.Output.rep3.MYC.unstranded.bed.gz\n",
      "\n",
      "show first few lines of input\n",
      "chr8\t126735490\t126735925\tchr8_126735490_126735925\t1\t.\n",
      "chr8\t126735503\t126735986\tchr8_126735503_126735986\t1\t.\n",
      "chr8\t126735540\t126735902\tchr8_126735540_126735902\t1\t.\n",
      "chr8\t126735572\t126735932\tchr8_126735572_126735932\t1\t.\n",
      "chr8\t126735591\t126736031\tchr8_126735591_126736031\t1\t.\n",
      "\n",
      "show last few lines of output\n",
      "chr8\t128736357\t128736486\tchr8_128736357_128736486\t1\t.\n",
      "chr8\t128736382\t128736845\tchr8_128736382_128736845\t1\t.\n",
      "chr8\t128736414\t128736699\tchr8_128736414_128736699\t1\t.\n",
      "chr8\t128736476\t128736935\tchr8_128736476_128736935\t1\t.\n",
      "chr8\t128736530\t128736909\tchr8_128736530_128736909\t1\t.\n",
      "\n",
      "Input:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/region_MYC_perbase.bed\n",
      "\n",
      "show first few lines of input\n",
      "chr8\t126735901\t126735902\n",
      "chr8\t126735902\t126735903\n",
      "chr8\t126735903\t126735904\n",
      "chr8\t126735904\t126735905\n",
      "chr8\t126735905\t126735906\n",
      "\n",
      "show last few lines of output\n",
      "chr8\t128736546\t128736547\n",
      "chr8\t128736547\t128736548\n",
      "chr8\t128736548\t128736549\n",
      "chr8\t128736549\t128736550\n",
      "chr8\t128736550\t128736551\n",
      "\n",
      "\n",
      "Output:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/A001_K562_WSTARRseq.Output.rep3.MYC.unstranded.bed.gz\n",
      "\n",
      "show first few lines of output:\n",
      "chr8\t126735901\t126735902\t52\n",
      "chr8\t126735902\t126735903\t52\n",
      "chr8\t126735903\t126735904\t52\n",
      "chr8\t126735904\t126735905\t52\n",
      "chr8\t126735905\t126735906\t52\n",
      "\n",
      "show last few lines of output\n",
      "chr8\t128736546\t128736547\t18\n",
      "chr8\t128736547\t128736548\t18\n",
      "chr8\t128736548\t128736549\t18\n",
      "chr8\t128736549\t128736550\t18\n",
      "chr8\t128736550\t128736551\t18\n",
      "\n",
      "++++++++++++++++++++++++++++++++++++++\n",
      "\n",
      "Input:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/fragment_count/A001_K562_WSTARRseq.Output.rep3.FADS.unstranded.bed.gz\n",
      "\n",
      "show first few lines of input\n",
      "chr11\t61554286\t61554911\tchr11_61554286_61554911\t1\t.\n",
      "chr11\t61554306\t61555169\tchr11_61554306_61555169\t1\t.\n",
      "chr11\t61554312\t61554854\tchr11_61554312_61554854\t1\t.\n",
      "chr11\t61554353\t61554821\tchr11_61554353_61554821\t1\t.\n",
      "chr11\t61554359\t61554810\tchr11_61554359_61554810\t1\t.\n",
      "\n",
      "show last few lines of output\n",
      "chr11\t61665770\t61666477\tchr11_61665770_61666477\t1\t.\n",
      "chr11\t61665771\t61665889\tchr11_61665771_61665889\t1\t.\n",
      "chr11\t61665771\t61666114\tchr11_61665771_61666114\t1\t.\n",
      "chr11\t61665778\t61666152\tchr11_61665778_61666152\t1\t.\n",
      "chr11\t61665779\t61665928\tchr11_61665779_61665928\t1\t.\n",
      "\n",
      "Input:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/region_FADS_perbase.bed\n",
      "\n",
      "show first few lines of input\n",
      "chr11\t61554801\t61554802\n",
      "chr11\t61554802\t61554803\n",
      "chr11\t61554803\t61554804\n",
      "chr11\t61554804\t61554805\n",
      "chr11\t61554805\t61554806\n",
      "\n",
      "show last few lines of output\n",
      "chr11\t61665816\t61665817\n",
      "chr11\t61665817\t61665818\n",
      "chr11\t61665818\t61665819\n",
      "chr11\t61665819\t61665820\n",
      "chr11\t61665820\t61665821\n",
      "\n",
      "\n",
      "Output:  /data/reddylab/Kuei/out/proj_combeffect_encode_fcc/results/A001_K562_WSTARRseq/coverage/A001_K562_WSTARRseq.Output.rep3.FADS.unstranded.bed.gz\n",
      "\n",
      "show first few lines of output:\n",
      "chr11\t61554801\t61554802\t117\n",
      "chr11\t61554802\t61554803\t117\n",
      "chr11\t61554803\t61554804\t119\n",
      "chr11\t61554804\t61554805\t120\n",
      "chr11\t61554805\t61554806\t120\n",
      "\n",
      "show last few lines of output\n",
      "chr11\t61665816\t61665817\t41\n",
      "chr11\t61665817\t61665818\t41\n",
      "chr11\t61665818\t61665819\t41\n",
      "chr11\t61665819\t61665820\t41\n",
      "chr11\t61665820\t61665821\t41\n",
      "\n",
      "\n",
      "Done!\n",
      "Run Time: 37 seconds\n"
     ]
    }
   ],
   "source": [
    "cat ${FD_LOG}/coverage_perbase_WSTARR.6.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b54b80bd-2f93-4ea1-8702-25d27e83b5a1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6fb2b62b-f49c-4fe7-b76b-b47620b658b2",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
