<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Data dictionary – ENCODE FCC Project</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">ENCODE FCC Project</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./report.html"> 
<span class="menu-text">Report</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./analysis.html"> 
<span class="menu-text">Analysis</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./data.html" aria-current="page"> 
<span class="menu-text">Data</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./code.html"> 
<span class="menu-text">Code</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./results.html"> 
<span class="menu-text">Results</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
    <a href="https://github.com/ReddyLab/Proj_ENCODE_FCC" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Data dictionary</li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">Setup</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01_setup/work_01_env.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">01 Working Environment</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01_setup/work_02_dir.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">02 Working Directory</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01_setup/work_03_ref.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">03 Reference Files</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01_setup/work_04_data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">04 Data Files</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01_setup/work_11_prep_fcc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11 Prepare: FCC</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01_setup/work_12_prep_hic.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12 Prepare: Hi-C</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01_setup/work_13_prep_chipseq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13 Prepare: ChIP-seq</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01_setup/work_14_prep_region.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14 Prepare: Region</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01_setup/work_15_prep_others.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15 Prepare: Others</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01_setup/work_21_results_fcc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12 Results (FCC)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01_setup/work_21_results_fcc2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12 Results (FCC)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01_setup/work_22_results_hic.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13 Results (HiC)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01_setup/work_23_results_region.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14 Results (Region)</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Prepare</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02_prepare/data_01_fcc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FCC Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02_prepare/data_02_hic.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Hi-C Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02_prepare/data_11_chipseq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ChIP-seq TF/Histone</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02_prepare/data_12_encode_annotation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Chromatin States and TF Modules</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02_prepare/data_21_peak_calling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Peak calling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02_prepare/data_22_ocr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Open chromatin regions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02_prepare/data_23_others.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Other Annotations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">FCC Summary</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./11_fcc_summary/01_fcc_astarr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ATAC-STARR-seq (ASTARR)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./11_fcc_summary/02_fcc_wstarr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">WHG-STARR-seq (WSTARR)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./11_fcc_summary/03_fcc_tmpra.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Tiling MPRA (TMPRA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./11_fcc_summary/04_fcc_lmpra.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lenti-MPRA (LMPRA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./11_fcc_summary/05_fcc_crispri_hcrff.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CRISPRi HCR-FlowFISH (CRISPRi-HCRFF)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./11_fcc_summary/06_fcc_crispri_growth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CRISPRi Growth screen (CRISPRi-Growth)</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">FCC Coverage</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./12_fcc_coverage/011_region_ocr_astarr_KS91.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ATAC/DHS coverage (ASTARR KS91)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./12_fcc_coverage/012_region_ocr_astarr_KS274.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ATAC/DHS coverage (ASTARR KS274)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./12_fcc_coverage/02_region_ocr_wstarr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ATAC/DHS coverage (WSTARR)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./12_fcc_coverage/03_region_ocr_tmpra.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ATAC/DHS coverage (TMPRA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./12_fcc_coverage/04_region_ocr_lmpra.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ATAC/DHS coverage (LMPRA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./12_fcc_coverage/05_region_ocr_crispri_hcrff.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ATAC/DHS coverage (CRISPRi-HCRFF)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./12_fcc_coverage/06_region_ocr_crispri_growth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ATAC/DHS coverage (CRISPRi-Growth)</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">FCC Comparison</span></span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">HiC Coverage</span></span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <span class="sidebar-item-text sidebar-link text-start">
 <span class="menu-text">FCC vs HiC</span></span>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="false">
 <span class="menu-text">Test Notebook</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./notebooks/test/test_notebook.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Test ipynb for documentation</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#summary-fcc-assays-done-in-k562" id="toc-summary-fcc-assays-done-in-k562" class="nav-link active" data-scroll-target="#summary-fcc-assays-done-in-k562">Summary: FCC assays done in K562</a>
  <ul class="collapse">
  <li><a href="#atac-starr-seq-astarr" id="toc-atac-starr-seq-astarr" class="nav-link" data-scroll-target="#atac-starr-seq-astarr">ATAC-STARR-seq (ASTARR)</a></li>
  <li><a href="#whg-starr-seq-wstarr" id="toc-whg-starr-seq-wstarr" class="nav-link" data-scroll-target="#whg-starr-seq-wstarr">WHG-STARR-seq (WSTARR)</a></li>
  <li><a href="#tiling-mpra-tmpra" id="toc-tiling-mpra-tmpra" class="nav-link" data-scroll-target="#tiling-mpra-tmpra">Tiling MPRA (TMPRA)</a></li>
  <li><a href="#lenti-mpra-lmpra" id="toc-lenti-mpra-lmpra" class="nav-link" data-scroll-target="#lenti-mpra-lmpra">Lenti-MPRA (LMPRA)</a></li>
  <li><a href="#crispri-hcr-flowfish-crispri-hcrff" id="toc-crispri-hcr-flowfish-crispri-hcrff" class="nav-link" data-scroll-target="#crispri-hcr-flowfish-crispri-hcrff">CRISPRi-HCR FlowFISH (CRISPRi-HCRFF)</a></li>
  <li><a href="#crispri-growth-screen-crispri-growth" id="toc-crispri-growth-screen-crispri-growth" class="nav-link" data-scroll-target="#crispri-growth-screen-crispri-growth">CRISPRi Growth screen (CRISPRi-Growth)</a></li>
  </ul></li>
  <li><a href="#summary-crespeaks-identified-in-k562" id="toc-summary-crespeaks-identified-in-k562" class="nav-link" data-scroll-target="#summary-crespeaks-identified-in-k562">Summary: CREs/Peaks identified in K562</a>
  <ul class="collapse">
  <li><a href="#cres-calls-for-reporter-assays-starrmpra" id="toc-cres-calls-for-reporter-assays-starrmpra" class="nav-link" data-scroll-target="#cres-calls-for-reporter-assays-starrmpra">CREs calls for reporter assays (STARR/MPRA)</a></li>
  <li><a href="#cres-calls-for-crispri-hcr-flowfish" id="toc-cres-calls-for-crispri-hcr-flowfish" class="nav-link" data-scroll-target="#cres-calls-for-crispri-hcr-flowfish">CREs calls for CRISPRi-HCR FlowFISH</a></li>
  <li><a href="#cres-calls-for-crispri-growth-screen" id="toc-cres-calls-for-crispri-growth-screen" class="nav-link" data-scroll-target="#cres-calls-for-crispri-growth-screen">CREs calls for CRISPRi-Growth screen</a></li>
  <li><a href="#encode-re2g-predictions-of-enhancer-gene-regulatory-interactions-crispr-e2g" id="toc-encode-re2g-predictions-of-enhancer-gene-regulatory-interactions-crispr-e2g" class="nav-link" data-scroll-target="#encode-re2g-predictions-of-enhancer-gene-regulatory-interactions-crispr-e2g">ENCODE-rE2G predictions of enhancer-gene regulatory interactions (CRISPR-E2G)</a></li>
  <li><a href="#atac-peaks" id="toc-atac-peaks" class="nav-link" data-scroll-target="#atac-peaks">ATAC peaks</a></li>
  </ul></li>
  <li><a href="#summary-hi-c-assays-done-in-k562" id="toc-summary-hi-c-assays-done-in-k562" class="nav-link" data-scroll-target="#summary-hi-c-assays-done-in-k562">Summary: Hi-C assays done in K562</a>
  <ul class="collapse">
  <li><a href="#in-situ-hi-c" id="toc-in-situ-hi-c" class="nav-link" data-scroll-target="#in-situ-hi-c">in-situ Hi-C</a></li>
  </ul></li>
  <li><a href="#summary-genomictranscriptomic-information-of-k562" id="toc-summary-genomictranscriptomic-information-of-k562" class="nav-link" data-scroll-target="#summary-genomictranscriptomic-information-of-k562">Summary: Genomic/Transcriptomic information of K562</a>
  <ul class="collapse">
  <li><a href="#chromatin-states" id="toc-chromatin-states" class="nav-link" data-scroll-target="#chromatin-states">Chromatin states</a></li>
  <li><a href="#chip-seq-tfhistone-data" id="toc-chip-seq-tfhistone-data" class="nav-link" data-scroll-target="#chip-seq-tfhistone-data">ChIP-seq (TF/Histone) data</a></li>
  <li><a href="#tf-binding-modules" id="toc-tf-binding-modules" class="nav-link" data-scroll-target="#tf-binding-modules">TF binding modules</a></li>
  <li><a href="#accessible-regions-atacdhs-regions" id="toc-accessible-regions-atacdhs-regions" class="nav-link" data-scroll-target="#accessible-regions-atacdhs-regions">Accessible regions (ATAC/DHS regions)</a></li>
  <li><a href="#tss-annotation" id="toc-tss-annotation" class="nav-link" data-scroll-target="#tss-annotation">TSS annotation</a></li>
  <li><a href="#gene-expression-rna-seq" id="toc-gene-expression-rna-seq" class="nav-link" data-scroll-target="#gene-expression-rna-seq">Gene expression (RNA-seq)</a></li>
  <li><a href="#gene-annotation" id="toc-gene-annotation" class="nav-link" data-scroll-target="#gene-annotation">Gene annotation</a></li>
  </ul></li>
  <li><a href="#more-information" id="toc-more-information" class="nav-link" data-scroll-target="#more-information">More Information</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Data dictionary</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>Below are the main data used in the analyses. We express our sincere appreciation to the research teams and individuals involved in conducting, processing, and sharing the experiments and data that underpin our analyses.</p>
<section id="summary-fcc-assays-done-in-k562" class="level1">
<h1>Summary: FCC assays done in K562</h1>
<p><a href="./02_prepare/data_01_fcc.html">Read more about the FCC data</a></p>
<ul>
<li>ATAC-STARR (ASTARR): sequenced fragments (proccessed by Alex and Revathy; experiment done by Keith)</li>
<li>WHG-STARR (WSTARR): sequenced fragments (proccessed by Alex; experiment done by Kari)</li>
<li>Tiling MPRA (TMPRA): raw barcode count table (proccessed by Hannah)</li>
<li>Lenti-MPRA (LMPRA): log2 fold change of each element (shared by Vikram from the Lenti-MPRA study)</li>
<li>CRISPRi-HCR FlowFISH (CRISPRi-HCRFF): z scores (proccessed by Jin Woo)</li>
<li>CRISPRi-Growth screen (CRISPRi-Growth): z scores (proccessed by Jin Woo)</li>
</ul>
<section id="atac-starr-seq-astarr" class="level2">
<h2 class="anchored" data-anchor-id="atac-starr-seq-astarr">ATAC-STARR-seq (ASTARR)</h2>
<p>https://www.encodeproject.org/functional-characterization-experiments/ENCSR312UQM/</p>
</section>
<section id="whg-starr-seq-wstarr" class="level2">
<h2 class="anchored" data-anchor-id="whg-starr-seq-wstarr">WHG-STARR-seq (WSTARR)</h2>
<ul>
<li>More information
<ul>
<li>https://www.encodeproject.org/functional-characterization-experiments/ENCSR661FOW/</li>
</ul></li>
</ul>
</section>
<section id="tiling-mpra-tmpra" class="level2">
<h2 class="anchored" data-anchor-id="tiling-mpra-tmpra">Tiling MPRA (TMPRA)</h2>
<p><a href="https://www.encodeproject.org/search/?type=FunctionalCharacterizationExperiment&amp;type=FunctionalCharacterizationSeries&amp;type=TransgenicEnhancerExperiment&amp;config=FunctionalCharacterization&amp;datapoint=false&amp;control_type!=*&amp;status=released&amp;biosamples.donor.organism.scientific_name=Homo+sapiens&amp;biosample_ontology.term_name=K562&amp;assay_title=MPRA&amp;perturbation_type!=*&amp;lab.title=Ryan+Tewhey%2C+JAX">ENCODE FCC K562 Tiling MPRA from Tewhey’s lab</a></p>
<ul>
<li><a href="https://www.encodeproject.org/functional-characterization-experiments/ENCSR917SFD/">ENCODE MPRA K562 OL43-GATA/MYC region</a>
<ul>
<li>Homo sapiens K562, 24 hours post-nucleic acid delivery time genetically modified (episome) using transient transfection for multiple loci</li>
</ul></li>
<li>[ENCODE MPRA]</li>
</ul>
<p>https://www.encodeproject.org/search/?type=File&amp;searchTerm=OL43&amp;file_format=bed</p>
<p>MPRA of multiple loci in K562 Homo sapiens K562, 48 hours post-nucleic acid delivery time genetically modified (episome) using transient transfection for multiple loci Loci: FEN1, FADS1, FADS2, FADS3 Lab: Ryan Tewhey, JAX Project: ENCODE</p>
<p>Functional Characterization Experiment ENCSR394HXI</p>
<p>ryan-tewhey:mpra-OL13-fads-k562</p>
<p>MPRA of multiple loci in K562 Homo sapiens K562, 24 hours post-nucleic acid delivery time genetically modified (episome) using transient transfection for multiple loci Loci: GATA1, MYC Lab: Ryan Tewhey, JAX Project: ENCODE</p>
<p>Functional Characterization Experiment ENCSR917SFD</p>
<p>ryan-tewhey:mpra-OL43-gata_myc_ctrl-k562</p>
<p>MPRA of multiple loci in K562 Homo sapiens K562, 24 hours post-nucleic acid delivery time genetically modified (episome) using transient transfection for multiple loci Loci: LMO2, HBE1, RBM38, HBA2, BCL11A Lab: Ryan Tewhey, JAX Project: ENCODE</p>
<p>Functional Characterization Experiment ENCSR363XER</p>
<p>ryan-tewhey:mpra-OL45-common_ctrl-k562</p>
<p>In this analysis, we are using the unnormalized and normalized data processed and shared by Hannah from the Tewhey lab.</p>
</section>
<section id="lenti-mpra-lmpra" class="level2">
<h2 class="anchored" data-anchor-id="lenti-mpra-lmpra">Lenti-MPRA (LMPRA)</h2>
<p>Lentivirus-based MPRA (lentiMPRA) produces “in-genome” readouts <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7550205/">(M Grace Gordon et al., 2020)</a></p>
<ul>
<li>Benefits
<ul>
<li>enables the use of MPRA in hard-to-transfect cells.</li>
</ul></li>
<li>Reference
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7550205/">M Grace Gordon et al.&nbsp;(2020) lentiMPRA and MPRAflow for high-throughput functional characterization of gene regulatory elements. Nature protocols, 15, 2387–2412.</a></li>
</ul></li>
</ul>
</section>
<section id="crispri-hcr-flowfish-crispri-hcrff" class="level2">
<h2 class="anchored" data-anchor-id="crispri-hcr-flowfish-crispri-hcrff">CRISPRi-HCR FlowFISH (CRISPRi-HCRFF)</h2>
</section>
<section id="crispri-growth-screen-crispri-growth" class="level2">
<h2 class="anchored" data-anchor-id="crispri-growth-screen-crispri-growth">CRISPRi Growth screen (CRISPRi-Growth)</h2>
</section>
</section>
<section id="summary-crespeaks-identified-in-k562" class="level1">
<h1>Summary: CREs/Peaks identified in K562</h1>
<ul>
<li>CREs calls for reporter assays (STARR/MPRA)
<ul>
<li>Unified processing pipeline for peak calling in high-throughput reporter assays</li>
</ul></li>
<li>CREs calls for CRISPRi-HCR FlowFISH
<ul>
<li>CRISPR activity screen analysis (CASA) on CRISPRi-HCR Flow-FISH data</li>
</ul></li>
<li>CREs calls for CRISPRi-Growth screen
<ul>
<li>Calling DHS regions using DESeq analysis for CRISPRi-Growth</li>
</ul></li>
<li>CRISPR Enhancer-Gene (E2G) benchmark/prediction
<ul>
<li>the benchmark data and prediction results of enhancer-gene linking model</li>
</ul></li>
<li>ATAC peaks
<ul>
<li>MACS peaks called of the ASTARR input libraries</li>
</ul></li>
</ul>
<p>https://github.com/ENCODE-DCC/encValData/blob/master/as/mpra_starr.as</p>
<section id="cres-calls-for-reporter-assays-starrmpra" class="level2">
<h2 class="anchored" data-anchor-id="cres-calls-for-reporter-assays-starrmpra">CREs calls for reporter assays (STARR/MPRA)</h2>
<p>An unified processing pipeline has been developed by Junke from the <a href="https://www.yulab.org/lab_member/">Yu Lab</a> to standardize the enhancer calling process for high-throughput reporter assays.</p>
</section>
<section id="cres-calls-for-crispri-hcr-flowfish" class="level2">
<h2 class="anchored" data-anchor-id="cres-calls-for-crispri-hcr-flowfish">CREs calls for CRISPRi-HCR FlowFISH</h2>
<p>CRISPR activity screen analysis (CASA) on CRISPRi-HCR Flow-FISH data</p>
<p>The <a href="https://github.com/sjgosai/casa">CASA</a> analysis pipeline, developed by the Sabeti Lab, has been applied to CRISPRi-HCR Flow-FISH data to identify regulatory elements. The results of significant regions in K562 FlowFISH CRISPRi screens from Reilly lab can be downloaded from the ENCODE portal (<a href="https://www.encodeproject.org/search/?type=FunctionalCharacterizationExperiment&amp;type=FunctionalCharacterizationSeries&amp;type=TransgenicEnhancerExperiment&amp;config=FunctionalCharacterization&amp;datapoint=false&amp;control_type!=*&amp;status=released&amp;assay_title=Flow-FISH+CRISPR+screen&amp;lab.title=Pardis+Sabeti%2C+Broad&amp;biosample_ontology.term_name=K562">Link</a>)</p>
</section>
<section id="cres-calls-for-crispri-growth-screen" class="level2">
<h2 class="anchored" data-anchor-id="cres-calls-for-crispri-growth-screen">CREs calls for CRISPRi-Growth screen</h2>
<p>Calling DHS regions using DESeq analysis for CRISPRi-Growth</p>
<p>For the analysis of CRISPRi-Growth data, DHS (DNase I hypersensitive sites) regions with significant effect on cell fittness have been identified using DESeq analysis, performed by Alex.</p>
</section>
<section id="encode-re2g-predictions-of-enhancer-gene-regulatory-interactions-crispr-e2g" class="level2">
<h2 class="anchored" data-anchor-id="encode-re2g-predictions-of-enhancer-gene-regulatory-interactions-crispr-e2g">ENCODE-rE2G predictions of enhancer-gene regulatory interactions (CRISPR-E2G)</h2>
<ul>
<li><a href="https://www.encodeproject.org/software/distal-regulation-encode_re2g/">Distal regulation ENCODE-rE2G</a>
<ul>
<li><a href="https://github.com/karbalayghareh/ENCODE-E2G/">ENCODE-E2G models</a></li>
</ul></li>
</ul>
<p><strong>CRISPR E2G Benchmark</strong></p>
<ul>
<li><a href="https://github.com/EngreitzLab/CRISPR_comparison">Benchmark enhancer-gene prediction models against CRISPR data</a></li>
</ul>
<p><strong>CRISPR E2G Prediction</strong></p>
<ul>
<li><a href="https://www.encodeproject.org/annotations/ENCSR328LMT/">ENCODE: ENCSR328LMT</a>
<ul>
<li>Description:
<ul>
<li>ENCODE-rE2G (extended) predictions of enhancer-gene regulatory interactions for K562</li>
</ul></li>
<li>Annotation type:
<ul>
<li>element gene regulatory interaction predictions</li>
</ul></li>
<li>Experimental input (Assay: DNase-seq, ChIP-seq, HiC, ChIA-PET):
<ul>
<li>ENCSR000EOT (K562 DNase-seq)</li>
<li>ENCSR668LDD (K562 ChIP-seq: H3K4me3)</li>
<li>ENCSR000AKP (K562 ChIP-seq: H3K27ac)</li>
<li>ENCSR000EGE (K562 ChIP-seq: EP300)</li>
<li>ENCSR000DWE (K562 ChIP-seq: CTCF)</li>
<li>ENCSR597AKG (K562 ChIA-PET: CTCF)</li>
<li>ENCSR545YBD (K562 in situ Hi-C)</li>
<li>ENCSR479XDG (K562 intact Hi-C)</li>
</ul></li>
</ul></li>
</ul>
</section>
<section id="atac-peaks" class="level2">
<h2 class="anchored" data-anchor-id="atac-peaks">ATAC peaks</h2>
<p>MACS peaks called of the ASTARR input libraries</p>
<p>processed by Alex</p>
</section>
</section>
<section id="summary-hi-c-assays-done-in-k562" class="level1">
<h1>Summary: Hi-C assays done in K562</h1>
<p><a href="./02_prepare/data_02_hic.html">Read more about the Hi-C data</a></p>
<ul>
<li>in-situ Hi-C</li>
<li>intact Hi-C</li>
<li>Deep intact Hi-C</li>
</ul>
<section id="in-situ-hi-c" class="level2">
<h2 class="anchored" data-anchor-id="in-situ-hi-c">in-situ Hi-C</h2>
<p><a href="https://www.encodeproject.org/experiments/ENCSR545YBD/">Experiment summary for ENCSR545YBD</a> - HiC (in situ Hi-C) - Homo sapiens K562 - K562 in situ Hi-C experiment</p>
</section>
</section>
<section id="summary-genomictranscriptomic-information-of-k562" class="level1">
<h1>Summary: Genomic/Transcriptomic information of K562</h1>
<ul>
<li>Chromatin states (cCREs / ChromHMM)</li>
<li>TSS annotation</li>
<li>ChIP-seq (TF/Histone) data</li>
<li>TF binding modules</li>
<li>Accessible regions (ATAC/DHS regions)</li>
<li>TSS annotation</li>
<li>Gene expression (RNA-seq)</li>
</ul>
<section id="chromatin-states" class="level2">
<h2 class="anchored" data-anchor-id="chromatin-states">Chromatin states</h2>
<p><a href="./02_prepare/data_12_encode_annotation.html">Read more about the chromatin states data</a></p>
<ul>
<li>ENCODE candidate cis-regulatory elements (cCREs)
<ul>
<li>ENCODE Link: <a href="https://www.encodeproject.org/annotations/ENCSR913HQX/">ENCODE v4 cCREs (ENCSR913HQX)</a></li>
<li>Description: candidate regulatory elements for GRCh38 in K562</li>
</ul></li>
<li>ENCODE ChromHMM chromatin states (ChromHMM)
<ul>
<li>ENCODE Link: <a href="https://www.encodeproject.org/annotations/ENCSR365YNI/">ENCODE v4 ChromHMM (ENCSR365YNI)</a></li>
<li>Description: ChromHMM 15-state model of K562</li>
</ul></li>
</ul>
</section>
<section id="chip-seq-tfhistone-data" class="level2">
<h2 class="anchored" data-anchor-id="chip-seq-tfhistone-data">ChIP-seq (TF/Histone) data</h2>
<ul>
<li><p>K562 ChIP-seq table from the ENCODE flagship</p></li>
<li><p>ENCODE ChIP-seq (Histone)</p>
<ul>
<li>Link:
<ul>
<li>https://www.encodeproject.org/search/?type=Experiment&amp;searchTerm=chipseq&amp;assay_title=Histone+ChIP-seq&amp;biosample_ontology.term_name=K562&amp;perturbed=false&amp;assembly=GRCh38&amp;files.file_type=bed+narrowPeak&amp;files.file_type=bigWig</li>
</ul></li>
</ul></li>
<li><p>ENCODE ChIP-seq (Transcription Factors)</p>
<ul>
<li>Link:</li>
</ul></li>
</ul>
</section>
<section id="tf-binding-modules" class="level2">
<h2 class="anchored" data-anchor-id="tf-binding-modules">TF binding modules</h2>
<ul>
<li>Shannon et al.&nbsp;look at TF binding patterns across the genome enrichment by cluster the TFs into different modules/groups.</li>
</ul>
</section>
<section id="accessible-regions-atacdhs-regions" class="level2">
<h2 class="anchored" data-anchor-id="accessible-regions-atacdhs-regions">Accessible regions (ATAC/DHS regions)</h2>
<ul>
<li>K562 Accessible regions (ATAC/DHS regions) used for testing assay coverage</li>
</ul>
</section>
<section id="tss-annotation" class="level2">
<h2 class="anchored" data-anchor-id="tss-annotation">TSS annotation</h2>
<p><a href="./02_prepare/data_23_others.html">Read more about the TSS annotation used in this documentation</a></p>
</section>
<section id="gene-expression-rna-seq" class="level2">
<h2 class="anchored" data-anchor-id="gene-expression-rna-seq">Gene expression (RNA-seq)</h2>
<ul>
<li><a href="https://www.encodeproject.org/experiments/ENCSR615EEK/">ENCODE: ENCSR615EEK</a>
<ul>
<li>Assay: RNA-seq (total RNA-seq)</li>
<li>Biosample: K562</li>
<li>External resources: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE174954">GEO:GSE174954</a></li>
</ul></li>
</ul>
</section>
<section id="gene-annotation" class="level2">
<h2 class="anchored" data-anchor-id="gene-annotation">Gene annotation</h2>
<ul>
<li>Essential gene <a href="https://depmap.org/portal/data_page/?tab=allData&amp;releasename=DepMap+Public+24Q2&amp;filename=AchillesCommonEssentialControls.csv">DeMap: AchillesCommonEssentialControls.csv</a></li>
</ul>
<p>Description - List of genes used as positive controls, intersection of Biomen (2014) and Hart (2015) essentials. Each entry is separated by a newline. - The scores of these genes are used as the dependent distribution for inferring dependency probability</p>
<p>Release Citation - Current DepMap Release data, including CRISPR Screens, PRISM Drug Screens, Copy Number, Mutation, Expression, and Fusions - DepMap, Broad (2024). DepMap 24Q2 Public. Figshare+. Dataset. https://doi.org/10.25452/figshare.plus.25880521.v1</p>
</section>
</section>
<section id="more-information" class="level1">
<h1>More Information</h1>
<ul>
<li>ENCODE4 Flagship</li>
<li>ENCODE4 FCC study
<ul>
<li>Functional Characterization of the Human Genome</li>
</ul></li>
<li>ENCODE4 STARR/MPRA study
<ul>
<li>Unified processing of MPRAs</li>
</ul></li>
<li>ENCODE4 Lenti-MPRA study
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10028905/">Massively parallel characterization of transcriptional regulatory elements in three diverse human cell types</a>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10028905/bin/media-3.xlsx">Supplementary Table 3.</a> Genomic coordinates, element categorization, and sequences for designed elements for the three large-scale lentiMPRA assays.</li>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10028905/bin/media-4.xlsx">Supplementary Table 4.</a> Activity scores computed for each element for the three large-scale lentiMPRA assays.</li>
</ul></li>
</ul></li>
<li>ENCODE4 CRISPR study
<ul>
<li><a href="https://www.biorxiv.org/content/10.1101/2022.12.21.520137v1">Multi-center integrated analysis of non-coding CRISPR screens</a></li>
</ul></li>
<li>ENCODE4 cCREs study
<ul>
<li>An Expanded Registry of candidate cis-Regulatory Elements for Studying Transcriptional Regulation</li>
</ul></li>
<li>ENCODE4 Distal Regulation (E2G) study
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10680627/">An encyclopedia of enhancer-gene regulatory interactions in the human genome</a></li>
</ul></li>
<li>ENCODE4 Nuclear Architecture study</li>
<li>TF binding modules study</li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>