{
 "cells": [
  {
   "cell_type": "raw",
   "id": "ab2db43a-a87d-4b82-9095-f434d29a762e",
   "metadata": {},
   "source": [
    "---\n",
    "title: Prepare ENCODE ChIP-seq 21 (TF; Flagship)\n",
    "subtitle: Generate download script to download the data\n",
    "format:\n",
    "    html:\n",
    "        code-fold:  true\n",
    "        code-tools: true\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7cd87cbf-55cd-440f-bdbe-a6e38b43c158",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "07082407-194a-41fc-8680-3e9fbb04733c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are working on        Singularity \n",
      "BASE DIRECTORY (FD_BASE): /mount \n",
      "REPO DIRECTORY (FD_REPO): /mount/repo \n",
      "WORK DIRECTORY (FD_WORK): /mount/work \n",
      "DATA DIRECTORY (FD_DATA): /mount/data \n",
      "\n",
      "You are working with      ENCODE FCC \n",
      "PATH OF PROJECT (FD_PRJ): /mount/repo/Proj_ENCODE_FCC \n",
      "PROJECT RESULTS (FD_RES): /mount/repo/Proj_ENCODE_FCC/results \n",
      "PROJECT SCRIPTS (FD_EXE): /mount/repo/Proj_ENCODE_FCC/scripts \n",
      "PROJECT DATA    (FD_DAT): /mount/repo/Proj_ENCODE_FCC/data \n",
      "PROJECT NOTE    (FD_NBK): /mount/repo/Proj_ENCODE_FCC/notebooks \n",
      "PROJECT DOCS    (FD_DOC): /mount/repo/Proj_ENCODE_FCC/docs \n",
      "PROJECT LOG     (FD_LOG): /mount/repo/Proj_ENCODE_FCC/log \n",
      "PROJECT APP     (FD_APP): /mount/repo/Proj_ENCODE_FCC/app \n",
      "PROJECT REF     (FD_REF): /mount/repo/Proj_ENCODE_FCC/references \n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(source(\"../run_config_project_sing.R\")))\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bc87a9cf-a731-4a2d-a63f-90d5fbb2971d",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Import metadata from reference file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "041e524f-385e-4a7a-a2ac-c2ef81746082",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "encode_chipseq_agarwal2023 \n",
      "encode_chipseq_encode4 \n",
      "encode_chipseq_flagship \n",
      "encode_chipseq_histone \n",
      "encode_chipseq_latest \n",
      "encode_chipseq_subset \n",
      "encode_chromatin_states \n",
      "encode_crispri_hcrff \n",
      "encode_e2g_benchmark \n",
      "encode_e2g_prediction \n",
      "encode_hic \n",
      "encode_open_chromatin \n",
      "encode_rnaseq \n",
      "fcc_peak_call_atac \n",
      "fcc_peak_call_cres \n",
      "fcc_trackhub_JinWoo \n",
      "genome_cres \n",
      "genome_gene \n",
      "genome_tss \n",
      "TF_modules_Shannon \n"
     ]
    }
   ],
   "source": [
    "txt_fdiry = FD_REF\n",
    "vec = dir(txt_fdiry)\n",
    "for (txt in vec){cat(txt, \"\\n\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8f37776c-a7ed-4bd7-9ff9-63f7bf0a2718",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'ENCODE4 TF Accessions for Flagship_updated221025.tsv'"
      ],
      "text/latex": [
       "'ENCODE4 TF Accessions for Flagship\\_updated221025.tsv'"
      ],
      "text/markdown": [
       "'ENCODE4 TF Accessions for Flagship_updated221025.tsv'"
      ],
      "text/plain": [
       "[1] \"ENCODE4 TF Accessions for Flagship_updated221025.tsv\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "txt_fdiry = file.path(FD_REF, \"encode_chipseq_flagship\")\n",
    "dir(txt_fdiry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "01472aec-921f-4a28-be35-117f8fee714b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "files_240620.txt \n",
      "files_default_files_240701.txt \n",
      "files_default_selected_240701.txt \n",
      "metadata_240620.tsv \n",
      "metadata_default_files_240701.tsv \n",
      "metadata_default_selected_240701.tsv \n"
     ]
    }
   ],
   "source": [
    "txt_fdiry = file.path(FD_REF, \"encode_chipseq_latest\")\n",
    "vec = dir(txt_fdiry)\n",
    "for (txt in vec){cat(txt, \"\\n\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "34ce7c07-2304-4210-a297-ec8acd792967",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 734   6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Experiment Accession </th>\n",
       "   <th style=\"text-align:left;\"> Peak Accession </th>\n",
       "   <th style=\"text-align:left;\"> Biosample </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> RFA </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR805SIA </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF500BWO </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens K562 genetically modified (insertion) using CRISPR targeting H. sapiens PURB </td>\n",
       "   <td style=\"text-align:left;\"> PURB </td>\n",
       "   <td style=\"text-align:left;\"> Richard Myers, HAIB </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR125RFR </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF863ZFH </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens K562 genetically modified (insertion) using CRISPR targeting H. sapiens ATF6 </td>\n",
       "   <td style=\"text-align:left;\"> ATF6 </td>\n",
       "   <td style=\"text-align:left;\"> Richard Myers, HAIB </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCSR841GLE </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF515LWL </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens K562 genetically modified (insertion) using CRISPR targeting H. sapiens ZNF217 </td>\n",
       "   <td style=\"text-align:left;\"> ZNF217 </td>\n",
       "   <td style=\"text-align:left;\"> Richard Myers, HAIB </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_fdiry = file.path(FD_REF, \"encode_chipseq_flagship\")\n",
    "txt_fname = \"ENCODE4 TF Accessions for Flagship_updated221025.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "dat = dat %>% dplyr::filter(str_detect(Biosample, \"K562\"))\n",
    "dat = dat %>% na.omit\n",
    "\n",
    "### show and assign\n",
    "dat_metadata_k562_flagship = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "691147e2-9709-48d1-b97a-5fc08c133a56",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "files_240620.txt \n",
      "files_default_files_240701.txt \n",
      "files_default_selected_240701.txt \n",
      "metadata_240620.tsv \n",
      "metadata_default_files_240701.tsv \n",
      "metadata_default_selected_240701.tsv \n"
     ]
    }
   ],
   "source": [
    "txt_fdiry = file.path(FD_REF, \"encode_chipseq_latest\")\n",
    "vec = dir(txt_fdiry)\n",
    "for (txt in vec){cat(txt, \"\\n\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2465cfa2-90a9-42ab-8c92-841d715cc039",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 2544   59\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF214SNH </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR228ELU </td>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> CRISPR </td>\n",
       "   <td style=\"text-align:left;\"> insertion </td>\n",
       "   <td style=\"text-align:left;\"> {'schema_version': '14', 'aliases': [], 'organism': {'schema_version': '6', '@type': ['Organism', 'Item'], 'name': 'human', 'taxon_id': '9606', 'scientific_name': 'Homo sapiens', '@id': '/organisms/human/', 'uuid': '7745b647-ff15-4ff3-9ced-b897d4e2983c', 'status': 'released'}, 'genes': ['/genes/7003/'], '@type': ['Target', 'Item'], 'name': 'TEAD1-human', 'label': 'TEAD1', '@id': '/targets/TEAD1-human/', 'title': 'TEAD1 (Homo sapiens)', 'investigated_as': ['transcription factor'], 'uuid': '8a929742-012b-441c-a1b0-3dfaa7d78ef7', 'status': 'released'} </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> TEAD1-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> formaldehyde </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2022-01-13 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> sonication (generic) </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1, 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF604PVE/, /files/ENCFF932FOK/, /files/ENCFF872HPF/, /files/ENCFF356LFX/ </td>\n",
       "   <td style=\"text-align:right;\"> 330511 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> a864ccf68ff3ca9e7539f90ef7208260 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF214SNH/@@download/ENCFF214SNH.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2022/01/07/970c51e3-6569-4fe1-a7e7-c12344cfe827/ENCFF214SNH.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2022/01/07/970c51e3-6569-4fe1-a7e7-c12344cfe827/ENCFF214SNH.bed.gz?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> mild to moderate bottlenecking, missing genetic modification reagents </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_fdiry = file.path(FD_REF, \"encode_chipseq_latest\")\n",
    "#txt_fname = \"metadata_240620.tsv\"\n",
    "txt_fname = \"metadata_default_files_240701.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### show and assign\n",
    "dat_metadata_k562_latest = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f12f8eb5-86fa-4e63-8134-31a56d2483ff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "vec\n",
       "    conservative IDR thresholded peaks                  IDR thresholded peaks \n",
       "                                   136                                   1358 \n",
       "         optimal IDR thresholded peaks pseudoreplicated IDR thresholded peaks \n",
       "                                   156                                      8 \n",
       "                pseudoreplicated peaks                       replicated peaks \n",
       "                                    28                                     10 \n",
       "                        signal p-value \n",
       "                                   848 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_k562_latest\n",
    "vec = dat$`Output type`\n",
    "table(vec)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ed4e05b9-3c1e-4ea7-9e41-912671dcbd96",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "733"
      ],
      "text/latex": [
       "733"
      ],
      "text/markdown": [
       "733"
      ],
      "text/plain": [
       "[1] 733"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat  = dat_metadata_k562_flagship\n",
    "vec1 = dat$`Peak Accession`\n",
    "\n",
    "dat  = dat_metadata_k562_latest\n",
    "vec2 = dat$`File accession`\n",
    "\n",
    "sum(vec1 %in% vec2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e88d941f-064f-4982-b316-5288c0959d58",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1 6\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 1 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Experiment Accession</th><th scope=col>Peak Accession</th><th scope=col>Biosample</th><th scope=col>Target</th><th scope=col>Lab</th><th scope=col>RFA</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENCSR000EGP</td><td>ENCFF540TWL</td><td>Homo sapiens K562</td><td>ZNF143</td><td>Michael Snyder, Stanford</td><td>ENCODE2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 1 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " Experiment Accession & Peak Accession & Biosample & Target & Lab & RFA\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENCSR000EGP & ENCFF540TWL & Homo sapiens K562 & ZNF143 & Michael Snyder, Stanford & ENCODE2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 1 × 6\n",
       "\n",
       "| Experiment Accession &lt;chr&gt; | Peak Accession &lt;chr&gt; | Biosample &lt;chr&gt; | Target &lt;chr&gt; | Lab &lt;chr&gt; | RFA &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| ENCSR000EGP | ENCFF540TWL | Homo sapiens K562 | ZNF143 | Michael Snyder, Stanford | ENCODE2 |\n",
       "\n"
      ],
      "text/plain": [
       "  Experiment Accession Peak Accession Biosample         Target\n",
       "1 ENCSR000EGP          ENCFF540TWL    Homo sapiens K562 ZNF143\n",
       "  Lab                      RFA    \n",
       "1 Michael Snyder, Stanford ENCODE2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_k562_flagship\n",
    "dat = dat %>% dplyr::filter(!(`Peak Accession` %in% vec2))\n",
    "print(dim(dat))\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "e691988d-ec0c-4c55-ad19-7ad6c1534bb5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Index_File</th><th scope=col>md5sum</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENCFF214SNH</td><td>a864ccf68ff3ca9e7539f90ef7208260</td></tr>\n",
       "\t<tr><td>ENCFF038QYW</td><td>b6b93bfb75a2f06e3922589a96e31d10</td></tr>\n",
       "\t<tr><td>ENCFF254RJL</td><td>2d3508612116afe400aa02151d9e9f7b</td></tr>\n",
       "\t<tr><td>ENCFF319JRY</td><td>8028cdfb3a9d68da19a7550279701540</td></tr>\n",
       "\t<tr><td>ENCFF830LLA</td><td>39755551ff0aed7bf13e862715b6d767</td></tr>\n",
       "\t<tr><td>ENCFF165ZEP</td><td>eb34fc26945d28a7df54b336aa7904e9</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Index\\_File & md5sum\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENCFF214SNH & a864ccf68ff3ca9e7539f90ef7208260\\\\\n",
       "\t ENCFF038QYW & b6b93bfb75a2f06e3922589a96e31d10\\\\\n",
       "\t ENCFF254RJL & 2d3508612116afe400aa02151d9e9f7b\\\\\n",
       "\t ENCFF319JRY & 8028cdfb3a9d68da19a7550279701540\\\\\n",
       "\t ENCFF830LLA & 39755551ff0aed7bf13e862715b6d767\\\\\n",
       "\t ENCFF165ZEP & eb34fc26945d28a7df54b336aa7904e9\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| Index_File &lt;chr&gt; | md5sum &lt;chr&gt; |\n",
       "|---|---|\n",
       "| ENCFF214SNH | a864ccf68ff3ca9e7539f90ef7208260 |\n",
       "| ENCFF038QYW | b6b93bfb75a2f06e3922589a96e31d10 |\n",
       "| ENCFF254RJL | 2d3508612116afe400aa02151d9e9f7b |\n",
       "| ENCFF319JRY | 8028cdfb3a9d68da19a7550279701540 |\n",
       "| ENCFF830LLA | 39755551ff0aed7bf13e862715b6d767 |\n",
       "| ENCFF165ZEP | eb34fc26945d28a7df54b336aa7904e9 |\n",
       "\n"
      ],
      "text/plain": [
       "  Index_File  md5sum                          \n",
       "1 ENCFF214SNH a864ccf68ff3ca9e7539f90ef7208260\n",
       "2 ENCFF038QYW b6b93bfb75a2f06e3922589a96e31d10\n",
       "3 ENCFF254RJL 2d3508612116afe400aa02151d9e9f7b\n",
       "4 ENCFF319JRY 8028cdfb3a9d68da19a7550279701540\n",
       "5 ENCFF830LLA 39755551ff0aed7bf13e862715b6d767\n",
       "6 ENCFF165ZEP eb34fc26945d28a7df54b336aa7904e9"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_k562_latest\n",
    "dat = dat %>%\n",
    "    dplyr::mutate(Index_File = `File accession`) %>%\n",
    "    dplyr::select(Index_File, md5sum)\n",
    "\n",
    "dat2 = dat\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "76a0ca73-aa40-4556-8f73-5923193ce757",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Index_File</th><th scope=col>File_Name</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENCFF500BWO</td><td>K562.hg38.ENCSR805SIA.ENCFF500BWO.ChIPseq.PURB.bed.gz  </td></tr>\n",
       "\t<tr><td>ENCFF863ZFH</td><td>K562.hg38.ENCSR125RFR.ENCFF863ZFH.ChIPseq.ATF6.bed.gz  </td></tr>\n",
       "\t<tr><td>ENCFF515LWL</td><td>K562.hg38.ENCSR841GLE.ENCFF515LWL.ChIPseq.ZNF217.bed.gz</td></tr>\n",
       "\t<tr><td>ENCFF121HYT</td><td>K562.hg38.ENCSR014ARU.ENCFF121HYT.ChIPseq.ATF2.bed.gz  </td></tr>\n",
       "\t<tr><td>ENCFF169QYL</td><td>K562.hg38.ENCSR172XJS.ENCFF169QYL.ChIPseq.ZNF165.bed.gz</td></tr>\n",
       "\t<tr><td>ENCFF528IDR</td><td>K562.hg38.ENCSR676KDJ.ENCFF528IDR.ChIPseq.ARID4B.bed.gz</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Index\\_File & File\\_Name\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENCFF500BWO & K562.hg38.ENCSR805SIA.ENCFF500BWO.ChIPseq.PURB.bed.gz  \\\\\n",
       "\t ENCFF863ZFH & K562.hg38.ENCSR125RFR.ENCFF863ZFH.ChIPseq.ATF6.bed.gz  \\\\\n",
       "\t ENCFF515LWL & K562.hg38.ENCSR841GLE.ENCFF515LWL.ChIPseq.ZNF217.bed.gz\\\\\n",
       "\t ENCFF121HYT & K562.hg38.ENCSR014ARU.ENCFF121HYT.ChIPseq.ATF2.bed.gz  \\\\\n",
       "\t ENCFF169QYL & K562.hg38.ENCSR172XJS.ENCFF169QYL.ChIPseq.ZNF165.bed.gz\\\\\n",
       "\t ENCFF528IDR & K562.hg38.ENCSR676KDJ.ENCFF528IDR.ChIPseq.ARID4B.bed.gz\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| Index_File &lt;chr&gt; | File_Name &lt;chr&gt; |\n",
       "|---|---|\n",
       "| ENCFF500BWO | K562.hg38.ENCSR805SIA.ENCFF500BWO.ChIPseq.PURB.bed.gz   |\n",
       "| ENCFF863ZFH | K562.hg38.ENCSR125RFR.ENCFF863ZFH.ChIPseq.ATF6.bed.gz   |\n",
       "| ENCFF515LWL | K562.hg38.ENCSR841GLE.ENCFF515LWL.ChIPseq.ZNF217.bed.gz |\n",
       "| ENCFF121HYT | K562.hg38.ENCSR014ARU.ENCFF121HYT.ChIPseq.ATF2.bed.gz   |\n",
       "| ENCFF169QYL | K562.hg38.ENCSR172XJS.ENCFF169QYL.ChIPseq.ZNF165.bed.gz |\n",
       "| ENCFF528IDR | K562.hg38.ENCSR676KDJ.ENCFF528IDR.ChIPseq.ARID4B.bed.gz |\n",
       "\n"
      ],
      "text/plain": [
       "  Index_File  File_Name                                              \n",
       "1 ENCFF500BWO K562.hg38.ENCSR805SIA.ENCFF500BWO.ChIPseq.PURB.bed.gz  \n",
       "2 ENCFF863ZFH K562.hg38.ENCSR125RFR.ENCFF863ZFH.ChIPseq.ATF6.bed.gz  \n",
       "3 ENCFF515LWL K562.hg38.ENCSR841GLE.ENCFF515LWL.ChIPseq.ZNF217.bed.gz\n",
       "4 ENCFF121HYT K562.hg38.ENCSR014ARU.ENCFF121HYT.ChIPseq.ATF2.bed.gz  \n",
       "5 ENCFF169QYL K562.hg38.ENCSR172XJS.ENCFF169QYL.ChIPseq.ZNF165.bed.gz\n",
       "6 ENCFF528IDR K562.hg38.ENCSR676KDJ.ENCFF528IDR.ChIPseq.ARID4B.bed.gz"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### init\n",
    "dat = dat_metadata_k562_flagship\n",
    "\n",
    "### setup download file name and wget command\n",
    "dat = dat %>%\n",
    "    dplyr::mutate(\n",
    "        File_Name = paste(\n",
    "            \"K562\", \n",
    "            \"hg38\", \n",
    "            `Experiment Accession`, \n",
    "            `Peak Accession`,\n",
    "            \"ChIPseq\",\n",
    "            Target,\n",
    "            \"bed.gz\", \n",
    "            sep=\".\")\n",
    "    ) %>%\n",
    "    dplyr::mutate(\n",
    "        File_URL_Download = file.path(\n",
    "            \"https://www.encodeproject.org/files\",\n",
    "            `Peak Accession`,\n",
    "            \"@@download\",\n",
    "            paste(`Peak Accession`, \"bed.gz\", sep = \".\")\n",
    "        )\n",
    "    ) %>%\n",
    "    dplyr::mutate(\n",
    "        CMD = paste(\"wget\", \"--append-output=run_download.log.txt\", \"-O\", File_Name, File_URL_Download)\n",
    "    ) \n",
    "\n",
    "dat = dat %>% \n",
    "    dplyr::mutate(Index_File = `Peak Accession`) %>%\n",
    "    dplyr::select(Index_File, File_Name) \n",
    "\n",
    "dat1 = dat\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "4d91de7c-7139-48f4-a5e9-4d655aa01902",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>md5sum</th><th scope=col>File_Name</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>b08bfdf72a39b00c730fb5707cc295cd</td><td>K562.hg38.ENCSR805SIA.ENCFF500BWO.ChIPseq.PURB.bed.gz  </td></tr>\n",
       "\t<tr><td>7dba139384a66c92d2ee77041f4fc9d2</td><td>K562.hg38.ENCSR125RFR.ENCFF863ZFH.ChIPseq.ATF6.bed.gz  </td></tr>\n",
       "\t<tr><td>3d33786ffc6dc671531cf38bde441b6a</td><td>K562.hg38.ENCSR841GLE.ENCFF515LWL.ChIPseq.ZNF217.bed.gz</td></tr>\n",
       "\t<tr><td>bb407abe86df5439cef62a5fa5246ec7</td><td>K562.hg38.ENCSR014ARU.ENCFF121HYT.ChIPseq.ATF2.bed.gz  </td></tr>\n",
       "\t<tr><td>4c482aed84e47f4d3383d190e33458e0</td><td>K562.hg38.ENCSR172XJS.ENCFF169QYL.ChIPseq.ZNF165.bed.gz</td></tr>\n",
       "\t<tr><td>6ef0239549730fa777af55dc284d48b9</td><td>K562.hg38.ENCSR676KDJ.ENCFF528IDR.ChIPseq.ARID4B.bed.gz</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " md5sum & File\\_Name\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t b08bfdf72a39b00c730fb5707cc295cd & K562.hg38.ENCSR805SIA.ENCFF500BWO.ChIPseq.PURB.bed.gz  \\\\\n",
       "\t 7dba139384a66c92d2ee77041f4fc9d2 & K562.hg38.ENCSR125RFR.ENCFF863ZFH.ChIPseq.ATF6.bed.gz  \\\\\n",
       "\t 3d33786ffc6dc671531cf38bde441b6a & K562.hg38.ENCSR841GLE.ENCFF515LWL.ChIPseq.ZNF217.bed.gz\\\\\n",
       "\t bb407abe86df5439cef62a5fa5246ec7 & K562.hg38.ENCSR014ARU.ENCFF121HYT.ChIPseq.ATF2.bed.gz  \\\\\n",
       "\t 4c482aed84e47f4d3383d190e33458e0 & K562.hg38.ENCSR172XJS.ENCFF169QYL.ChIPseq.ZNF165.bed.gz\\\\\n",
       "\t 6ef0239549730fa777af55dc284d48b9 & K562.hg38.ENCSR676KDJ.ENCFF528IDR.ChIPseq.ARID4B.bed.gz\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| md5sum &lt;chr&gt; | File_Name &lt;chr&gt; |\n",
       "|---|---|\n",
       "| b08bfdf72a39b00c730fb5707cc295cd | K562.hg38.ENCSR805SIA.ENCFF500BWO.ChIPseq.PURB.bed.gz   |\n",
       "| 7dba139384a66c92d2ee77041f4fc9d2 | K562.hg38.ENCSR125RFR.ENCFF863ZFH.ChIPseq.ATF6.bed.gz   |\n",
       "| 3d33786ffc6dc671531cf38bde441b6a | K562.hg38.ENCSR841GLE.ENCFF515LWL.ChIPseq.ZNF217.bed.gz |\n",
       "| bb407abe86df5439cef62a5fa5246ec7 | K562.hg38.ENCSR014ARU.ENCFF121HYT.ChIPseq.ATF2.bed.gz   |\n",
       "| 4c482aed84e47f4d3383d190e33458e0 | K562.hg38.ENCSR172XJS.ENCFF169QYL.ChIPseq.ZNF165.bed.gz |\n",
       "| 6ef0239549730fa777af55dc284d48b9 | K562.hg38.ENCSR676KDJ.ENCFF528IDR.ChIPseq.ARID4B.bed.gz |\n",
       "\n"
      ],
      "text/plain": [
       "  md5sum                          \n",
       "1 b08bfdf72a39b00c730fb5707cc295cd\n",
       "2 7dba139384a66c92d2ee77041f4fc9d2\n",
       "3 3d33786ffc6dc671531cf38bde441b6a\n",
       "4 bb407abe86df5439cef62a5fa5246ec7\n",
       "5 4c482aed84e47f4d3383d190e33458e0\n",
       "6 6ef0239549730fa777af55dc284d48b9\n",
       "  File_Name                                              \n",
       "1 K562.hg38.ENCSR805SIA.ENCFF500BWO.ChIPseq.PURB.bed.gz  \n",
       "2 K562.hg38.ENCSR125RFR.ENCFF863ZFH.ChIPseq.ATF6.bed.gz  \n",
       "3 K562.hg38.ENCSR841GLE.ENCFF515LWL.ChIPseq.ZNF217.bed.gz\n",
       "4 K562.hg38.ENCSR014ARU.ENCFF121HYT.ChIPseq.ATF2.bed.gz  \n",
       "5 K562.hg38.ENCSR172XJS.ENCFF169QYL.ChIPseq.ZNF165.bed.gz\n",
       "6 K562.hg38.ENCSR676KDJ.ENCFF528IDR.ChIPseq.ARID4B.bed.gz"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dplyr::left_join(dat1, dat2, by = \"Index_File\")\n",
    "dat = dat %>% dplyr::select(md5sum, File_Name)\n",
    "\n",
    "dat_md5sum = dat\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "cc53162c-a61e-4866-92e3-cc3b67c56259",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K562.hg38.ENCSR000AKO.ENCFF769AUF.ChIPseq.CTCF.bed.gz \n",
      "K562.hg38.ENCSR000AQA.ENCFF807FNB.ChIPseq.KDM5B.bed.gz \n",
      "K562.hg38.ENCSR000AQD.ENCFF408NUX.ChIPseq.CHD1.bed.gz \n",
      "K562.hg38.ENCSR000AQE.ENCFF804RVA.ChIPseq.EZH2.bed.gz \n",
      "K562.hg38.ENCSR000AQF.ENCFF669MJX.ChIPseq.HDAC1.bed.gz \n",
      "K562.hg38.ENCSR000AQG.ENCFF182MPT.ChIPseq.HDAC2.bed.gz \n"
     ]
    }
   ],
   "source": [
    "txt_fdiry = file.path(FD_DAT, \"external\", \"encode_chipseq_flagship\")\n",
    "vec = dir(txt_fdiry)\n",
    "vec = head(vec)\n",
    "for (txt in vec){cat(txt, \"\\n\")}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "ad4aa7ff-86db-4a0d-ad0e-ee773b612a23",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set directory\n",
    "txt_fdiry = file.path(FD_DAT, \"external\", \"encode_chipseq_flagship\")\n",
    "txt_fname = \"checksum_md5sum.txt\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "###\n",
    "dat = dat_md5sum\n",
    "dat = head(dat)\n",
    "write_delim(dat, txt_fpath, delim = \" \", col_names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66eff94c-3ac8-4a41-9f6c-b90a721b8ba4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R (alex_py3)",
   "language": "R",
   "name": "ir_alex_py3"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
