{
 "cells": [
  {
   "cell_type": "raw",
   "id": "6386e4f9-e10e-4242-9ce3-40f6e2fb7194",
   "metadata": {},
   "source": [
    "---\n",
    "title: Prepare ENCODE ChIP-seq 11 (Histone)\n",
    "subtitle: Generate download script to download the data\n",
    "format:\n",
    "    html:\n",
    "        code-fold:  true\n",
    "        code-tools: true\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "202bd433-f08c-416a-bd2d-81800db506ea",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b40f3608-9d98-44bb-88ca-36bca0fe4db4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are working on        Singularity \n",
      "BASE DIRECTORY (FD_BASE): /mount \n",
      "REPO DIRECTORY (FD_REPO): /mount/repo \n",
      "WORK DIRECTORY (FD_WORK): /mount/work \n",
      "DATA DIRECTORY (FD_DATA): /mount/data \n",
      "\n",
      "You are working with      ENCODE FCC \n",
      "PATH OF PROJECT (FD_PRJ): /mount/repo/Proj_ENCODE_FCC \n",
      "PROJECT RESULTS (FD_RES): /mount/repo/Proj_ENCODE_FCC/results \n",
      "PROJECT SCRIPTS (FD_EXE): /mount/repo/Proj_ENCODE_FCC/scripts \n",
      "PROJECT DATA    (FD_DAT): /mount/repo/Proj_ENCODE_FCC/data \n",
      "PROJECT NOTE    (FD_NBK): /mount/repo/Proj_ENCODE_FCC/notebooks \n",
      "PROJECT DOCS    (FD_DOC): /mount/repo/Proj_ENCODE_FCC/docs \n",
      "PROJECT LOG     (FD_LOG): /mount/repo/Proj_ENCODE_FCC/log \n",
      "PROJECT APP     (FD_APP): /mount/repo/Proj_ENCODE_FCC/app \n",
      "PROJECT REF     (FD_REF): /mount/repo/Proj_ENCODE_FCC/references \n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(source(\"../run_config_project_sing.R\")))\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0f772baa-7cd6-4746-8162-30985b3a1819",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Import data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6395f9ad-ff92-4a0d-be3f-fc9360327e9e",
   "metadata": {},
   "source": [
    "**Read metadata from reference file**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7fbbfbc6-75d3-4ae9-87fa-539725c6e6b1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 2544   59\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF214SNH </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> bed </td>\n",
       "   <td style=\"text-align:left;\"> narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> IDR thresholded peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR228ELU </td>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> CRISPR </td>\n",
       "   <td style=\"text-align:left;\"> insertion </td>\n",
       "   <td style=\"text-align:left;\"> {'schema_version': '14', 'aliases': [], 'organism': {'schema_version': '6', '@type': ['Organism', 'Item'], 'name': 'human', 'taxon_id': '9606', 'scientific_name': 'Homo sapiens', '@id': '/organisms/human/', 'uuid': '7745b647-ff15-4ff3-9ced-b897d4e2983c', 'status': 'released'}, 'genes': ['/genes/7003/'], '@type': ['Target', 'Item'], 'name': 'TEAD1-human', 'label': 'TEAD1', '@id': '/targets/TEAD1-human/', 'title': 'TEAD1 (Homo sapiens)', 'investigated_as': ['transcription factor'], 'uuid': '8a929742-012b-441c-a1b0-3dfaa7d78ef7', 'status': 'released'} </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> TEAD1-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> formaldehyde </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2022-01-13 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> sonication (generic) </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1, 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF604PVE/, /files/ENCFF932FOK/, /files/ENCFF872HPF/, /files/ENCFF356LFX/ </td>\n",
       "   <td style=\"text-align:right;\"> 330511 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> a864ccf68ff3ca9e7539f90ef7208260 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF214SNH/@@download/ENCFF214SNH.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2022/01/07/970c51e3-6569-4fe1-a7e7-c12344cfe827/ENCFF214SNH.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2022/01/07/970c51e3-6569-4fe1-a7e7-c12344cfe827/ENCFF214SNH.bed.gz?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> mild to moderate bottlenecking, missing genetic modification reagents </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_fdiry = file.path(FD_REF, \"encode_chipseq_latest\")\n",
    "txt_fname = \"metadata_default_files_240701.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### show and assign\n",
    "dat_metadata_import = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "aa0ec10a-93c1-444e-86c1-836770de2362",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 38 59\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF461RKK </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000EWA </td>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me3-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2011-05-19 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> see document </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 1, 2 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1, 2_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF181ANT/, /files/ENCFF747HEB/, /files/ENCFF355LXQ/ </td>\n",
       "   <td style=\"text-align:right;\"> 510104216 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> 46d7c60097cc9a09d3c3bede8302f01b </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF461RKK/@@download/ENCFF461RKK.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/09/29/120fbb5a-db8e-464e-bdeb-9e72fdbb380b/ENCFF461RKK.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2020/09/29/120fbb5a-db8e-464e-bdeb-9e72fdbb380b/ENCFF461RKK.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.5.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> low read length, low read depth, mild to moderate bottlenecking, moderate library complexity, inconsistent platforms </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set file path\n",
    "txt_fdiry = file.path(FD_REF, \"encode_chipseq_histone\")\n",
    "txt_fname = \"metadata.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### show and assign\n",
    "dat_metadata_check = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8b6a5a0-40d8-471d-b340-135211acb078",
   "metadata": {},
   "source": [
    "**Check: Genome**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b71b5fa7-d3f6-4abb-8046-9852229bd230",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "GRCh38   hg19 \n",
       "  2421    123 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_import\n",
    "table(dat$`File assembly`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "225847fd-2c60-4e1f-b4fb-6d3051288b97",
   "metadata": {},
   "source": [
    "**Check: Assay**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1c56475f-8de6-44f8-b9b9-4680c67f570b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Histone ChIP-seq      TF ChIP-seq \n",
       "              57             2487 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_import\n",
    "table(dat$Assay)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f43d0775-3438-45f6-8a3f-1f44c41f7f71",
   "metadata": {},
   "source": [
    "**Check: Biosample**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f630173c-8bcf-42cf-9ebd-db325c59986d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "K562 \n",
       "2544 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_import\n",
    "table(dat$`Biosample term name`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "666a869d-ecd3-4a02-b188-a6f77bcb1532",
   "metadata": {},
   "source": [
    "**Check: File format**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2ab401fc-a0b7-4ee5-9ccc-2be6ba6d65af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "   bed narrowPeak bigBed narrowPeak            bigWig \n",
       "              848               848               848 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_import\n",
    "table(dat$`File format`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "7e17ee4d-9680-4c68-a8b6-fd9abf49abbb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                   \n",
       "                    conservative IDR thresholded peaks IDR thresholded peaks\n",
       "  bed narrowPeak                                    68                   679\n",
       "  bigBed narrowPeak                                 68                   679\n",
       "  bigWig                                             0                     0\n",
       "                   \n",
       "                    optimal IDR thresholded peaks\n",
       "  bed narrowPeak                               78\n",
       "  bigBed narrowPeak                            78\n",
       "  bigWig                                        0\n",
       "                   \n",
       "                    pseudoreplicated IDR thresholded peaks\n",
       "  bed narrowPeak                                         4\n",
       "  bigBed narrowPeak                                      4\n",
       "  bigWig                                                 0\n",
       "                   \n",
       "                    pseudoreplicated peaks replicated peaks signal p-value\n",
       "  bed narrowPeak                        14                5              0\n",
       "  bigBed narrowPeak                     14                5              0\n",
       "  bigWig                                 0                0            848"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_metadata_import\n",
    "table(dat$`File format`, dat$`Output type`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0f3a28f5-6724-4455-9f75-f1ad828fc331",
   "metadata": {},
   "source": [
    "## Get Histone ChIP-seq files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "74623577-0b70-4b2b-9540-c4d68c666e02",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 38 11\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:left;\"> Analysis </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> FName </th>\n",
       "   <th style=\"text-align:left;\"> FUrl </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKR </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF193ERO </td>\n",
       "   <td style=\"text-align:left;\"> bed narrowPeak </td>\n",
       "   <td style=\"text-align:left;\"> pseudoreplicated peaks </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K36me3 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> f66a73f2377a38c4ba6c1d4405fd41a5 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF193ERO.bed.gz </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKR </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF605EVL </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K36me3 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> c160804206370a082afe6e48ce3ec939 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF605EVL.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF605EVL/@@download/ENCFF605EVL.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> Histone ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR000AKT </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF054RSU </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> signal p-value </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> H3K4me2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.6.1 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> 5a9a282c89fc45d7a729dcc8e7c71c10 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF054RSU.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF054RSU/@@download/ENCFF054RSU.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### init\n",
    "txt_genome    = \"GRCh38\"\n",
    "txt_assay     = \"Histone ChIP-seq\"\n",
    "vec_txt_ftype = c(\"bed narrowPeak\", \"bigWig\")\n",
    "\n",
    "### filter and arrange\n",
    "dat = dat_metadata_import\n",
    "dat = dat %>% \n",
    "    dplyr::mutate(\n",
    "        Biosample        = `Biosample term name`,\n",
    "        Index_Experiment = `Experiment accession`,\n",
    "        Index_File       = `File accession`,\n",
    "        File_Type        = `File format`,\n",
    "        Output_Type      = `Output type`,\n",
    "        Genome           = `File assembly`,\n",
    "        Target           = str_remove(`Experiment target`, \"-human\"),\n",
    "        FName            = basename(`File download URL`),\n",
    "        FUrl             = `File download URL`,\n",
    "        Analysis         = `File analysis title`\n",
    "    )  %>%\n",
    "    dplyr::select(\n",
    "        Assay, \n",
    "        Index_Experiment, \n",
    "        Index_File,\n",
    "        File_Type,\n",
    "        Output_Type,\n",
    "        Genome,\n",
    "        Target,\n",
    "        Analysis,\n",
    "        md5sum,\n",
    "        FName,\n",
    "        FUrl\n",
    "    ) %>% dplyr::filter(\n",
    "        Genome    == txt_genome,\n",
    "        Assay     == txt_assay,\n",
    "        File_Type %in% vec_txt_ftype\n",
    "    )\n",
    "\n",
    "### assign and show\n",
    "dat_metadata_histone = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d7e0f2c5-9285-4c7b-ac4c-6e558353f122",
   "metadata": {},
   "source": [
    "**Check**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "af22dc41-7bee-4191-afb8-1018a7620c07",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat  = dat_metadata_histone\n",
    "vec  = dat$Index_File\n",
    "vec1 = sort(vec)\n",
    "\n",
    "dat  = dat_metadata_check\n",
    "vec  = dat$`File accession`\n",
    "vec2 = sort(vec)\n",
    "\n",
    "all(vec1 == vec2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "08408fc2-7f02-4dad-a73a-b3636e5d9a60",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "             \n",
       "              bed narrowPeak bigWig\n",
       "  ENCSR000AKP              1      1\n",
       "  ENCSR000AKQ              1      1\n",
       "  ENCSR000AKR              1      1\n",
       "  ENCSR000AKS              1      1\n",
       "  ENCSR000AKT              1      1\n",
       "  ENCSR000AKU              1      1\n",
       "  ENCSR000AKV              1      1\n",
       "  ENCSR000AKW              1      1\n",
       "  ENCSR000AKX              1      1\n",
       "  ENCSR000APC              1      1\n",
       "  ENCSR000APD              1      1\n",
       "  ENCSR000APE              1      1\n",
       "  ENCSR000DWB              1      1\n",
       "  ENCSR000DWD              1      1\n",
       "  ENCSR000EVZ              1      1\n",
       "  ENCSR000EWA              1      1\n",
       "  ENCSR000EWB              1      1\n",
       "  ENCSR000EWC              1      1\n",
       "  ENCSR668LDD              1      1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat  = dat_metadata_histone\n",
    "table(dat$Index_Experiment, dat$File_Type)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "65841717-8835-4e15-b37f-0026b330c981",
   "metadata": {},
   "source": [
    "## Prepare download file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ad4a58ae-9237-478f-bd11-f9d49a38fa14",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 19  1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> FUrl </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF193ERO/@@download/ENCFF193ERO.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF749KLQ/@@download/ENCFF749KLQ.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF909RKY/@@download/ENCFF909RKY.bed.gz </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### init\n",
    "dat = dat_metadata_histone\n",
    "\n",
    "### select file urls\n",
    "dat = dat %>% dplyr::filter(File_Type == \"bed narrowPeak\") %>% dplyr::select(FUrl) \n",
    "\n",
    "### assign and show\n",
    "dat_download_file = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "497ad350-3208-4dc0-9d0f-f002b1a2f616",
   "metadata": {},
   "source": [
    "## Prepare checksum file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "55e77fc9-b62f-4dba-8f26-62856f6a1f19",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 19  2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> FName </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> f66a73f2377a38c4ba6c1d4405fd41a5 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF193ERO.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> 55c1093b8d98bef1dc7f788fbb44eb7c </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF749KLQ.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> fe4c93b6faa0ab4153ef41fe516bced4 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF909RKY.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> 8e71cfdb547cdb5057d99d12d7be3ba6 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF963GZJ.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> 7974e4f7e3f7b1db43715efdf2532916 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF561OUZ.bed.gz </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> 5ddbfd027332e8d38e801dd1eaa7575c </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF462AVD.bed.gz </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### init\n",
    "dat = dat_metadata_histone\n",
    "\n",
    "### select md5sum for each file\n",
    "dat = dat %>% dplyr::filter(File_Type == \"bed narrowPeak\") %>% dplyr::select(md5sum, FName)\n",
    "\n",
    "### assing and show\n",
    "dat_download_md5sum = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ceedae1-adbe-4964-afc7-ac54da3db9e0",
   "metadata": {},
   "source": [
    "## Save results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "16727088-1ef5-4f5f-bad5-33b2a36d08b1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/mount/repo/Proj_ENCODE_FCC/data/external/encode_chipseq_histone\"\n",
      "[1] \"/mount/repo/Proj_ENCODE_FCC/results/region/encode_chipseq_histone/summary\"\n"
     ]
    }
   ],
   "source": [
    "### set directory\n",
    "txt_folder = \"encode_chipseq_histone\"\n",
    "\n",
    "txt_fdiry = file.path(FD_DAT, \"external\", txt_folder)\n",
    "txt_cmd   = paste(\"mkdir -p\", txt_fdiry)\n",
    "system(txt_cmd)\n",
    "print(txt_fdiry)\n",
    "\n",
    "txt_fdiry = file.path(FD_RES, \"region\", txt_folder, \"summary\")\n",
    "txt_cmd   = paste(\"mkdir -p\", txt_fdiry)\n",
    "system(txt_cmd)\n",
    "print(txt_fdiry)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e298750a-a29d-4809-a457-42821396911b",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set directory\n",
    "txt_fdiry = file.path(FD_DAT, \"external\", txt_folder)\n",
    "txt_fname = \"files.txt\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### write table\n",
    "dat = dat_download_file\n",
    "write_tsv(dat, txt_fpath, col_names = FALSE)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "eb676d25-d177-4932-8f59-06d2958d20a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set directory\n",
    "txt_fdiry = file.path(FD_DAT, \"external\", txt_folder)\n",
    "txt_fname = \"checksum_md5sum.txt\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### write table\n",
    "dat = dat_download_md5sum\n",
    "write_delim(dat, txt_fpath, delim = \" \", col_names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "242c59de-7dd9-415b-b9b1-69dc221884d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set output path\n",
    "txt_fdiry = file.path(FD_RES, \"region\", txt_folder, \"summary\")\n",
    "txt_fname = \"metadata.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### save table\n",
    "dat = dat_metadata_histone\n",
    "write_tsv(dat, txt_fpath)  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0b5372f-74d9-4d7e-974f-9c0b8a8b1962",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f2b8f85f-ab46-4ea1-878f-b5d9983ed421",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9721463c-7d17-4f3b-8afb-149c70e3dd5a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80d927ab-5db8-428d-80c5-079c7f636245",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R (alex_py3)",
   "language": "R",
   "name": "ir_alex_py3"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
