{
 "cells": [
  {
   "cell_type": "raw",
   "id": "584949de-85b6-41d0-9b63-af592549c511",
   "metadata": {},
   "source": [
    "---\n",
    "title: Prepare ENCODE samples 41 (agarwal2023)\n",
    "subtitle: ...\n",
    "format:\n",
    "    html:\n",
    "        code-fold:  true\n",
    "        code-tools: true\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0db0714d-51a1-4a59-bdcc-a9f9b1b2bb05",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "39ac5fd6-1c4b-4126-b0d2-eff8089be8de",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You are working on        Singularity \n",
      "BASE DIRECTORY (FD_BASE): /mount \n",
      "REPO DIRECTORY (FD_REPO): /mount/repo \n",
      "WORK DIRECTORY (FD_WORK): /mount/work \n",
      "DATA DIRECTORY (FD_DATA): /mount/data \n",
      "\n",
      "You are working with      ENCODE FCC \n",
      "PATH OF PROJECT (FD_PRJ): /mount/repo/Proj_ENCODE_FCC \n",
      "PROJECT RESULTS (FD_RES): /mount/repo/Proj_ENCODE_FCC/results \n",
      "PROJECT SCRIPTS (FD_EXE): /mount/repo/Proj_ENCODE_FCC/scripts \n",
      "PROJECT DATA    (FD_DAT): /mount/repo/Proj_ENCODE_FCC/data \n",
      "PROJECT NOTE    (FD_NBK): /mount/repo/Proj_ENCODE_FCC/notebooks \n",
      "PROJECT DOCS    (FD_DOC): /mount/repo/Proj_ENCODE_FCC/docs \n",
      "PROJECT LOG     (FD_LOG): /mount/repo/Proj_ENCODE_FCC/log \n",
      "PROJECT APP     (FD_APP): /mount/repo/Proj_ENCODE_FCC/app \n",
      "PROJECT REF     (FD_REF): /mount/repo/Proj_ENCODE_FCC/references \n",
      "\n"
     ]
    }
   ],
   "source": [
    "suppressMessages(suppressWarnings(source(\"../run_config_project_sing.R\")))\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3cbb7768-7358-4229-b323-da0e657e27a0",
   "metadata": {},
   "source": [
    "**Set global variables**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ff0eee77-f1b1-4883-8823-34434446e9ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "TXT_FOLDER_REGION = \"encode_chipseq_agarwal2023\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fa17f56c-abe9-4727-8f27-eedce4e15e56",
   "metadata": {},
   "source": [
    "## Import data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c07c2a6a-e4b7-49cf-bcfc-e91149820aff",
   "metadata": {},
   "source": [
    "**Read Agarwal2023 TableS5**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "631f9a53-baf0-499d-8a48-5390d6356ad5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1206   13\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:right;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF254FCX </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR440VKE </td>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> ADNP-human </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF254FCX/@@download/ENCFF254FCX.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2020/12/04/ba7cfb89-7802-41db-aada-bef45e7e97a7/ENCFF254FCX.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set directory\n",
    "txt_folder = TXT_FOLDER_REGION\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "txt_fname  = \"Agarwal2023_TableS5.tsv\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "dat = dat %>% dplyr::filter(`Biosample term name` == \"K562\")\n",
    "\n",
    "### assing and show\n",
    "dat_meta_k562_agarwal2023 = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f9c6f025-198f-4316-b300-a764c914049a",
   "metadata": {},
   "source": [
    "**Read metadata from ENCODE portal**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0fdf85db-c315-434b-a8d8-ab967a0c49dd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 18598    59\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> File accession </th>\n",
       "   <th style=\"text-align:left;\"> File format </th>\n",
       "   <th style=\"text-align:left;\"> File type </th>\n",
       "   <th style=\"text-align:left;\"> File format type </th>\n",
       "   <th style=\"text-align:left;\"> Output type </th>\n",
       "   <th style=\"text-align:left;\"> File assembly </th>\n",
       "   <th style=\"text-align:left;\"> Experiment accession </th>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Donor(s) </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term id </th>\n",
       "   <th style=\"text-align:left;\"> Biosample term name </th>\n",
       "   <th style=\"text-align:left;\"> Biosample type </th>\n",
       "   <th style=\"text-align:left;\"> Biosample organism </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments amount </th>\n",
       "   <th style=\"text-align:left;\"> Biosample treatments duration </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications methods </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications categories </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications gene targets </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications site coordinates </th>\n",
       "   <th style=\"text-align:left;\"> Biosample genetic modifications zygosity </th>\n",
       "   <th style=\"text-align:left;\"> Experiment target </th>\n",
       "   <th style=\"text-align:left;\"> Library made from </th>\n",
       "   <th style=\"text-align:left;\"> Library depleted in </th>\n",
       "   <th style=\"text-align:left;\"> Library extraction method </th>\n",
       "   <th style=\"text-align:left;\"> Library lysis method </th>\n",
       "   <th style=\"text-align:left;\"> Library crosslinking method </th>\n",
       "   <th style=\"text-align:left;\"> Library strand specific </th>\n",
       "   <th style=\"text-align:left;\"> Experiment date released </th>\n",
       "   <th style=\"text-align:left;\"> Project </th>\n",
       "   <th style=\"text-align:left;\"> RBNS protein concentration </th>\n",
       "   <th style=\"text-align:left;\"> Library fragmentation method </th>\n",
       "   <th style=\"text-align:left;\"> Library size range </th>\n",
       "   <th style=\"text-align:left;\"> Biological replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Technical replicate(s) </th>\n",
       "   <th style=\"text-align:left;\"> Read length </th>\n",
       "   <th style=\"text-align:left;\"> Mapped read length </th>\n",
       "   <th style=\"text-align:left;\"> Run type </th>\n",
       "   <th style=\"text-align:left;\"> Paired end </th>\n",
       "   <th style=\"text-align:left;\"> Paired with </th>\n",
       "   <th style=\"text-align:left;\"> Index of </th>\n",
       "   <th style=\"text-align:left;\"> Derived from </th>\n",
       "   <th style=\"text-align:right;\"> Size </th>\n",
       "   <th style=\"text-align:left;\"> Lab </th>\n",
       "   <th style=\"text-align:left;\"> md5sum </th>\n",
       "   <th style=\"text-align:left;\"> dbxrefs </th>\n",
       "   <th style=\"text-align:left;\"> File download URL </th>\n",
       "   <th style=\"text-align:left;\"> Genome annotation </th>\n",
       "   <th style=\"text-align:left;\"> Platform </th>\n",
       "   <th style=\"text-align:left;\"> Controlled by </th>\n",
       "   <th style=\"text-align:left;\"> File Status </th>\n",
       "   <th style=\"text-align:left;\"> s3_uri </th>\n",
       "   <th style=\"text-align:left;\"> Azure URL </th>\n",
       "   <th style=\"text-align:left;\"> File analysis title </th>\n",
       "   <th style=\"text-align:left;\"> File analysis status </th>\n",
       "   <th style=\"text-align:left;\"> Audit WARNING </th>\n",
       "   <th style=\"text-align:left;\"> Audit NOT_COMPLIANT </th>\n",
       "   <th style=\"text-align:left;\"> Audit ERROR </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> ENCFF076LYF </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR800KMQ </td>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> /human-donors/ENCDO000AAD/ </td>\n",
       "   <td style=\"text-align:left;\"> EFO:0002067 </td>\n",
       "   <td style=\"text-align:left;\"> K562 </td>\n",
       "   <td style=\"text-align:left;\"> cell line </td>\n",
       "   <td style=\"text-align:left;\"> Homo sapiens </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> CRISPR </td>\n",
       "   <td style=\"text-align:left;\"> insertion </td>\n",
       "   <td style=\"text-align:left;\"> {'schema_version': '14', 'aliases': [], 'organism': {'schema_version': '6', '@type': ['Organism', 'Item'], 'name': 'human', 'taxon_id': '9606', 'scientific_name': 'Homo sapiens', '@id': '/organisms/human/', 'uuid': '7745b647-ff15-4ff3-9ced-b897d4e2983c', 'status': 'released'}, 'genes': ['/genes/23099/'], '@type': ['Target', 'Item'], 'name': 'ZBTB43-human', 'label': 'ZBTB43', '@id': '/targets/ZBTB43-human/', 'title': 'ZBTB43 (Homo sapiens)', 'investigated_as': ['transcription factor'], 'uuid': 'c4fd1273-5058-4d6b-90b2-4fa94d61d714', 'status': 'released'} </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> ZBTB43-human </td>\n",
       "   <td style=\"text-align:left;\"> DNA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> formaldehyde </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 2023-01-10 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> sonication (generic) </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> 1_1 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> /files/ENCFF441EKA/, /files/ENCFF837JNH/ </td>\n",
       "   <td style=\"text-align:right;\"> 508674667 </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE Processing Pipeline </td>\n",
       "   <td style=\"text-align:left;\"> 128444a048c5894ea1fdffda07a00a16 </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF076LYF/@@download/ENCFF076LYF.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> s3://encode-public/2022/07/23/7fafea14-e96a-47d6-8945-706b0f1e0800/ENCFF076LYF.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://datasetencode.blob.core.windows.net/dataset/2022/07/23/7fafea14-e96a-47d6-8945-706b0f1e0800/ENCFF076LYF.bigWig?sv=2019-10-10&amp;si=prod&amp;sr=c&amp;sig=9qSQZo4ggrCNpybBExU8SypuUZV33igI11xw0P7rB3c%3D </td>\n",
       "   <td style=\"text-align:left;\"> ENCODE4 v1.8.0 GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> released </td>\n",
       "   <td style=\"text-align:left;\"> mild to moderate bottlenecking, missing genetic modification reagents </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "   <td style=\"text-align:left;\"> NA </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### set directory\n",
    "txt_folder = \"encode_chipseq_latest\"\n",
    "txt_fdiry  = file.path(FD_REF, txt_folder)\n",
    "txt_fname  = \"metadata_240620.tsv\"\n",
    "txt_fpath  = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = read_tsv(txt_fpath, show_col_types = FALSE)\n",
    "\n",
    "### assing and show\n",
    "dat_meta_k562_encode_latest = dat\n",
    "print(dim(dat))\n",
    "fun_display_table(head(dat, 1))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "658f503a-f1a0-43ab-bfd4-2932eabb1eb9",
   "metadata": {},
   "source": [
    "## Explore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b35fbf00-ab73-4d12-9a8e-0e954a9d6da9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                              \n",
       "                               ATAC-seq DNase-seq Histone ChIP-seq TF ChIP-seq\n",
       "  fold change over control            7         0               13        1181\n",
       "  read-depth normalized signal        0         5                0           0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_meta_k562_agarwal2023\n",
    "table(dat$`Output type`, dat$Assay)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ffa71c41-fd15-4c60-a8f5-e414e4a96103",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        \n",
       "         ATAC-seq DNase-seq Histone ChIP-seq TF ChIP-seq\n",
       "  bigWig        7         5               13        1181"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_meta_k562_agarwal2023\n",
    "table(dat$`File type`, dat$Assay)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "95bb29fd-16fe-4889-99a7-975fc55edec8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "        \n",
       "         ATAC-seq DNase-seq Histone ChIP-seq TF ChIP-seq\n",
       "  bed          38        27              266        7719\n",
       "  bigWig       26        19              414       10089"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_meta_k562_encode_latest\n",
    "table(dat$`File type`, dat$Assay)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d7c44054-5380-4ef7-bf44-e675d22a1eff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                                        \n",
       "                                         ATAC-seq DNase-seq Histone ChIP-seq\n",
       "  base overlap signal                           0         3                0\n",
       "  conservative IDR thresholded peaks            4         0                0\n",
       "  fold change over control                     13         0              197\n",
       "  IDR thresholded peaks                        15         0                0\n",
       "  optimal IDR thresholded peaks                 0         0                0\n",
       "  peaks                                         0        27              120\n",
       "  peaks and background as input for IDR         0         0                0\n",
       "  pseudoreplicated IDR thresholded peaks        0         0                0\n",
       "  pseudoreplicated peaks                       13         0               77\n",
       "  raw signal                                    0         0                4\n",
       "  read-depth normalized signal                  0        13                0\n",
       "  replicated peaks                              6         0               69\n",
       "  signal                                        0         3               16\n",
       "  signal p-value                               13         0              197\n",
       "                                        \n",
       "                                         TF ChIP-seq\n",
       "  base overlap signal                              3\n",
       "  conservative IDR thresholded peaks            1599\n",
       "  fold change over control                      4946\n",
       "  IDR thresholded peaks                         2248\n",
       "  optimal IDR thresholded peaks                 1028\n",
       "  peaks                                           89\n",
       "  peaks and background as input for IDR         2737\n",
       "  pseudoreplicated IDR thresholded peaks          18\n",
       "  pseudoreplicated peaks                           0\n",
       "  raw signal                                      84\n",
       "  read-depth normalized signal                     0\n",
       "  replicated peaks                                 0\n",
       "  signal                                         110\n",
       "  signal p-value                                4946"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_meta_k562_encode_latest\n",
    "table(dat$`Output type`, dat$Assay)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d26afe3f-debd-47b7-a40d-b0d44061317b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                  \n",
       "                   FALSE TRUE\n",
       "  ATAC-seq             0    7\n",
       "  DNase-seq            0    5\n",
       "  Histone ChIP-seq     0   13\n",
       "  TF ChIP-seq          2 1179"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_meta_k562_encode_latest\n",
    "vec = dat$`File accession`\n",
    "\n",
    "dat = dat_meta_k562_agarwal2023\n",
    "dat = dat %>% dplyr::select(`File accession`, Assay)\n",
    "\n",
    "dat = dat %>% dplyr::mutate(Match = `File accession` %in% vec)\n",
    "table(dat$Assay, dat$Match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6132a67c-575f-4e13-9d9d-8617f9471ed7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 2 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>File accession</th><th scope=col>Assay</th><th scope=col>Match</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENCFF576BIF</td><td>TF ChIP-seq</td><td>FALSE</td></tr>\n",
       "\t<tr><td>ENCFF543ZZV</td><td>TF ChIP-seq</td><td>FALSE</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 2 × 3\n",
       "\\begin{tabular}{lll}\n",
       " File accession & Assay & Match\\\\\n",
       " <chr> & <chr> & <lgl>\\\\\n",
       "\\hline\n",
       "\t ENCFF576BIF & TF ChIP-seq & FALSE\\\\\n",
       "\t ENCFF543ZZV & TF ChIP-seq & FALSE\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 2 × 3\n",
       "\n",
       "| File accession &lt;chr&gt; | Assay &lt;chr&gt; | Match &lt;lgl&gt; |\n",
       "|---|---|---|\n",
       "| ENCFF576BIF | TF ChIP-seq | FALSE |\n",
       "| ENCFF543ZZV | TF ChIP-seq | FALSE |\n",
       "\n"
      ],
      "text/plain": [
       "  File accession Assay       Match\n",
       "1 ENCFF576BIF    TF ChIP-seq FALSE\n",
       "2 ENCFF543ZZV    TF ChIP-seq FALSE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat %>% dplyr::filter(!Match)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b26d355d-86de-4e71-a6c9-8aa56b376ddb",
   "metadata": {},
   "source": [
    "**Both `ENCFF576BIF` and `ENCFF543ZZV` belong to the experiment `ENCSR000EGP`, which has been revoked.**\n",
    "- [ENCFF543ZZV – ENCODE](https://www.encodeproject.org/files/ENCFF543ZZV/)\n",
    "- [ENCFF576BIF – ENCODE](https://www.encodeproject.org/files/ENCFF576BIF/)\n",
    "- [ENCSR000EGP – ENCODE](https://www.encodeproject.org/experiments/ENCSR000EGP/)\n",
    "    - Status: revoked\n",
    "    - Assay: ChIP-seq (TF ChIP-seq)\n",
    "    - Target: ZNF143\n",
    "    - Biosample summary: Homo sapiens K562"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e77edc32-bcb8-419e-b105-15813057dd39",
   "metadata": {},
   "source": [
    "## Filter and arrange"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "730d04ea-36a9-4dc3-a0cd-c4f2aa6bc163",
   "metadata": {},
   "source": [
    "**Fitler and arrange table**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "83d045c6-009e-49c1-8830-ffb8eff8b791",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Before filter: 1206 \n",
      "After filter:  1204 \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       " <thead>\n",
       "  <tr>\n",
       "   <th style=\"text-align:left;\"> Assay </th>\n",
       "   <th style=\"text-align:left;\"> Index_Experiment </th>\n",
       "   <th style=\"text-align:left;\"> Index_File </th>\n",
       "   <th style=\"text-align:left;\"> File_Type </th>\n",
       "   <th style=\"text-align:left;\"> Output_Type </th>\n",
       "   <th style=\"text-align:left;\"> Genome </th>\n",
       "   <th style=\"text-align:left;\"> Target </th>\n",
       "   <th style=\"text-align:right;\"> Replicate </th>\n",
       "   <th style=\"text-align:left;\"> FName </th>\n",
       "   <th style=\"text-align:left;\"> FUrl </th>\n",
       "  </tr>\n",
       " </thead>\n",
       "<tbody>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR440VKE </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF254FCX </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ADNP </td>\n",
       "   <td style=\"text-align:right;\"> 1 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF254FCX.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF254FCX/@@download/ENCFF254FCX.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR440VKE </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF580YKQ </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> ADNP </td>\n",
       "   <td style=\"text-align:right;\"> 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF580YKQ.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF580YKQ/@@download/ENCFF580YKQ.bigWig </td>\n",
       "  </tr>\n",
       "  <tr>\n",
       "   <td style=\"text-align:left;\"> TF ChIP-seq </td>\n",
       "   <td style=\"text-align:left;\"> ENCSR641BSL </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF057BCB </td>\n",
       "   <td style=\"text-align:left;\"> bigWig </td>\n",
       "   <td style=\"text-align:left;\"> fold change over control </td>\n",
       "   <td style=\"text-align:left;\"> GRCh38 </td>\n",
       "   <td style=\"text-align:left;\"> AGO1 </td>\n",
       "   <td style=\"text-align:right;\"> 2 </td>\n",
       "   <td style=\"text-align:left;\"> ENCFF057BCB.bigWig </td>\n",
       "   <td style=\"text-align:left;\"> https://www.encodeproject.org/files/ENCFF057BCB/@@download/ENCFF057BCB.bigWig </td>\n",
       "  </tr>\n",
       "</tbody>\n",
       "</table>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### init: get file accession numbers\n",
    "dat = dat_meta_k562_encode_latest\n",
    "vec_txt_index = dat$`File accession`\n",
    "\n",
    "### filter\n",
    "dat = dat_meta_k562_agarwal2023\n",
    "dat = dat %>% dplyr::filter(`File accession` %in% vec_txt_index)\n",
    "cat(\"Before filter:\", nrow(dat_meta_k562_agarwal2023), \"\\n\")\n",
    "\n",
    "### arrange\n",
    "dat = dat %>% \n",
    "    dplyr::mutate(\n",
    "        Biosample        = `Biosample term name`,\n",
    "        Index_Experiment = `Experiment accession`,\n",
    "        Index_File       = `File accession`,\n",
    "        File_Type        = `File type`,\n",
    "        Output_Type      = `Output type`,\n",
    "        Genome           = `File assembly`,\n",
    "        Target           = str_remove(`Experiment target`, \"-human\"),\n",
    "        Replicate        = `Biological replicate(s)`,\n",
    "        FName            = basename(`File download URL`),\n",
    "        FUrl             = `File download URL`\n",
    "    ) %>%\n",
    "    dplyr::select(\n",
    "        Assay, \n",
    "        Index_Experiment, \n",
    "        Index_File,\n",
    "        File_Type,\n",
    "        Output_Type,\n",
    "        Genome,\n",
    "        Target,\n",
    "        Replicate,\n",
    "        FName,\n",
    "        FUrl)\n",
    "\n",
    "### assing and show\n",
    "dat_meta_results = dat\n",
    "cat(\"After filter: \", nrow(dat), \"\\n\")\n",
    "fun_display_table(head(dat, 3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1276d1df-0585-4dee-b7b5-5f316281ab7b",
   "metadata": {},
   "source": [
    "**Get File URL**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "caf6d821-6a41-49d5-b304-a1a0b69ed9b7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1204    1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>FUrl</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>https://www.encodeproject.org/files/ENCFF254FCX/@@download/ENCFF254FCX.bigWig</td></tr>\n",
       "\t<tr><td>https://www.encodeproject.org/files/ENCFF580YKQ/@@download/ENCFF580YKQ.bigWig</td></tr>\n",
       "\t<tr><td>https://www.encodeproject.org/files/ENCFF057BCB/@@download/ENCFF057BCB.bigWig</td></tr>\n",
       "\t<tr><td>https://www.encodeproject.org/files/ENCFF593VAW/@@download/ENCFF593VAW.bigWig</td></tr>\n",
       "\t<tr><td>https://www.encodeproject.org/files/ENCFF020OLK/@@download/ENCFF020OLK.bigWig</td></tr>\n",
       "\t<tr><td>https://www.encodeproject.org/files/ENCFF521JBC/@@download/ENCFF521JBC.bigWig</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 1\n",
       "\\begin{tabular}{l}\n",
       " FUrl\\\\\n",
       " <chr>\\\\\n",
       "\\hline\n",
       "\t https://www.encodeproject.org/files/ENCFF254FCX/@@download/ENCFF254FCX.bigWig\\\\\n",
       "\t https://www.encodeproject.org/files/ENCFF580YKQ/@@download/ENCFF580YKQ.bigWig\\\\\n",
       "\t https://www.encodeproject.org/files/ENCFF057BCB/@@download/ENCFF057BCB.bigWig\\\\\n",
       "\t https://www.encodeproject.org/files/ENCFF593VAW/@@download/ENCFF593VAW.bigWig\\\\\n",
       "\t https://www.encodeproject.org/files/ENCFF020OLK/@@download/ENCFF020OLK.bigWig\\\\\n",
       "\t https://www.encodeproject.org/files/ENCFF521JBC/@@download/ENCFF521JBC.bigWig\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 1\n",
       "\n",
       "| FUrl &lt;chr&gt; |\n",
       "|---|\n",
       "| https://www.encodeproject.org/files/ENCFF254FCX/@@download/ENCFF254FCX.bigWig |\n",
       "| https://www.encodeproject.org/files/ENCFF580YKQ/@@download/ENCFF580YKQ.bigWig |\n",
       "| https://www.encodeproject.org/files/ENCFF057BCB/@@download/ENCFF057BCB.bigWig |\n",
       "| https://www.encodeproject.org/files/ENCFF593VAW/@@download/ENCFF593VAW.bigWig |\n",
       "| https://www.encodeproject.org/files/ENCFF020OLK/@@download/ENCFF020OLK.bigWig |\n",
       "| https://www.encodeproject.org/files/ENCFF521JBC/@@download/ENCFF521JBC.bigWig |\n",
       "\n"
      ],
      "text/plain": [
       "  FUrl                                                                         \n",
       "1 https://www.encodeproject.org/files/ENCFF254FCX/@@download/ENCFF254FCX.bigWig\n",
       "2 https://www.encodeproject.org/files/ENCFF580YKQ/@@download/ENCFF580YKQ.bigWig\n",
       "3 https://www.encodeproject.org/files/ENCFF057BCB/@@download/ENCFF057BCB.bigWig\n",
       "4 https://www.encodeproject.org/files/ENCFF593VAW/@@download/ENCFF593VAW.bigWig\n",
       "5 https://www.encodeproject.org/files/ENCFF020OLK/@@download/ENCFF020OLK.bigWig\n",
       "6 https://www.encodeproject.org/files/ENCFF521JBC/@@download/ENCFF521JBC.bigWig"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat = dat_meta_results\n",
    "dat = dat %>% dplyr::select(FUrl)\n",
    "\n",
    "dat_meta_furl = dat\n",
    "print(dim(dat))\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e92a984-7514-49e6-8a9e-452494de16dd",
   "metadata": {},
   "source": [
    "**Get File Checksum**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "a94102bc-d7d4-4546-9871-2da7181a265f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 1204    2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>md5sum</th><th scope=col>FName</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>2a8eebea81e867a124a4d76d142c69e6</td><td>ENCFF403XOV.bigWig</td></tr>\n",
       "\t<tr><td>7663da654f6de358e935f977633446ca</td><td>ENCFF216UCG.bigWig</td></tr>\n",
       "\t<tr><td>5b270babaf701dd7ca2b615395e35198</td><td>ENCFF100VDV.bigWig</td></tr>\n",
       "\t<tr><td>ce9fc55ad9517fd2b40625e5a01d3983</td><td>ENCFF961LTP.bigWig</td></tr>\n",
       "\t<tr><td>f29a0fc38384d20345b512536a4f9b64</td><td>ENCFF702BPM.bigWig</td></tr>\n",
       "\t<tr><td>04a1dbe8b21ecb46fd41d5a64b1b07aa</td><td>ENCFF738KPN.bigWig</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " md5sum & FName\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 2a8eebea81e867a124a4d76d142c69e6 & ENCFF403XOV.bigWig\\\\\n",
       "\t 7663da654f6de358e935f977633446ca & ENCFF216UCG.bigWig\\\\\n",
       "\t 5b270babaf701dd7ca2b615395e35198 & ENCFF100VDV.bigWig\\\\\n",
       "\t ce9fc55ad9517fd2b40625e5a01d3983 & ENCFF961LTP.bigWig\\\\\n",
       "\t f29a0fc38384d20345b512536a4f9b64 & ENCFF702BPM.bigWig\\\\\n",
       "\t 04a1dbe8b21ecb46fd41d5a64b1b07aa & ENCFF738KPN.bigWig\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| md5sum &lt;chr&gt; | FName &lt;chr&gt; |\n",
       "|---|---|\n",
       "| 2a8eebea81e867a124a4d76d142c69e6 | ENCFF403XOV.bigWig |\n",
       "| 7663da654f6de358e935f977633446ca | ENCFF216UCG.bigWig |\n",
       "| 5b270babaf701dd7ca2b615395e35198 | ENCFF100VDV.bigWig |\n",
       "| ce9fc55ad9517fd2b40625e5a01d3983 | ENCFF961LTP.bigWig |\n",
       "| f29a0fc38384d20345b512536a4f9b64 | ENCFF702BPM.bigWig |\n",
       "| 04a1dbe8b21ecb46fd41d5a64b1b07aa | ENCFF738KPN.bigWig |\n",
       "\n"
      ],
      "text/plain": [
       "  md5sum                           FName             \n",
       "1 2a8eebea81e867a124a4d76d142c69e6 ENCFF403XOV.bigWig\n",
       "2 7663da654f6de358e935f977633446ca ENCFF216UCG.bigWig\n",
       "3 5b270babaf701dd7ca2b615395e35198 ENCFF100VDV.bigWig\n",
       "4 ce9fc55ad9517fd2b40625e5a01d3983 ENCFF961LTP.bigWig\n",
       "5 f29a0fc38384d20345b512536a4f9b64 ENCFF702BPM.bigWig\n",
       "6 04a1dbe8b21ecb46fd41d5a64b1b07aa ENCFF738KPN.bigWig"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "###\n",
    "dat = dat_meta_results\n",
    "vec = dat$FName\n",
    "\n",
    "###\n",
    "dat = dat_meta_k562_encode_latest\n",
    "dat = dat %>% \n",
    "    dplyr::mutate(FName = basename(`File download URL`)) %>%\n",
    "    dplyr::filter(FName %in% vec) %>%\n",
    "    dplyr::select(md5sum, FName)\n",
    "\n",
    "### assign and show\n",
    "dat_meta_md5sum = dat\n",
    "print(dim(dat))\n",
    "head(dat)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "66de82b8-ff1f-4582-b2a8-39fa26aa21e1",
   "metadata": {},
   "source": [
    "## Save results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "dd487ddb-8355-4241-a821-6a31855ccb7c",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set directory\n",
    "txt_fdiry = file.path(FD_DAT, \"external\", TXT_FOLDER_REGION)\n",
    "txt_fname = \"files.txt\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "###\n",
    "dir.create(txt_fdiry, showWarnings = FALSE)\n",
    "\n",
    "###\n",
    "dat = dat_meta_furl\n",
    "write_delim(dat, txt_fpath, col_names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "94b4bb73-683d-4245-8000-1dcdec965ee0",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set directory\n",
    "txt_fdiry = file.path(FD_DAT, \"external\", TXT_FOLDER_REGION)\n",
    "txt_fname = \"checksum_md5sum.txt\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "###\n",
    "dat = dat_meta_md5sum\n",
    "write_delim(dat, txt_fpath, delim = \" \", col_names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "9b644f96-b218-434c-8ffa-cf6eb6999a73",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set directory\n",
    "txt_fdiry = file.path(FD_DAT, \"external\", TXT_FOLDER_REGION, \"summary\")\n",
    "txt_fname = \"metadata.tsv\"\n",
    "txt_fpath = file.path(txt_fdiry, txt_fname)\n",
    "\n",
    "###\n",
    "dir.create(txt_fdiry, showWarnings = FALSE)\n",
    "\n",
    "###\n",
    "dat = dat_meta_results\n",
    "write_delim(dat, txt_fpath, col_names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "26cf9feb-560a-432e-871a-182fae5df619",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0b0d5451-93c3-420f-b13c-452ced381b5d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R (alex_py3)",
   "language": "R",
   "name": "ir_alex_py3"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
